---
output: github_document
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# CNAqc <a href="caravagnalab.github.io/CNAqc"><img src="logo.png" align="right" height="69" /></a>

<!-- badges: start -->
[![R-CMD-check](https://github.com/caravagnalab/CNAqc/workflows/R-CMD-check/badge.svg)](https://github.com/caravagnalab/CNAqc/actions)
[![pkgdown](https://github.com/caravagnalab/CNAqc/actions/workflows/pkgdown.yaml/badge.svg)](https://github.com/caravagnalab/CNAqc/actions/workflows/pkgdown.yaml)
[![Lifecycle: stable](https://img.shields.io/badge/lifecycle-stable-green.svg)](https://www.tidyverse.org/lifecycle/#stable)
<!-- badges: end -->

CNAqc is a package to manipulate, visualise and quality control (QC) variants data from bulk cancer sequencing assays. The tool processes i) somatic single-nucleotide variants and insertion-deletions, ii) allele-specific Copy Number Alterations (CNAs) and iii) tumour purity estimates. QC procedures can be used to validate copy number segmentations against variant allele frequencies of somatic mutations, and to rank alternative tumour segmentations and purity/ ploidy estimates. The tool also provides an automatic copy number calling pipeline that uses the Sequenza caller algorithm and performs QC-driven copy numbner calling. CNAqc provides also algorithms to i) estimate Cancer Cell Fractions (CCFs) per mutation, ii) identify patterns of over-fragmentation of chromosome arms, iii) annotate driver mutationts and iv) perform various manipulation tasks for somatic tumour data.

#### Citation

[![](https://img.shields.io/badge/doi-10.1101/2021.02.13.429885-red.svg)](https://doi.org/10.1101/2021.02.13.429885)

If you use `CNAqc`, please cite:

* _Integrated quality control of allele-specific copy numbers, mutations and tumour purity from cancer whole genome sequencing assays_
  Jacob Househam, Riccardo Bergamin, Salvatore Milite, Nicola Calonaci, Alice Antonello, Marc J Williams, William CH Cross, Giulio Caravagna. [biorXiv 2022](https://www.biorxiv.org/content/10.1101/2021.02.13.429885v3).


#### Help and support

[![](https://img.shields.io/badge/GitHub%20Pages-https://caravagnalab.github.io/CNAqc/-yellow.svg)](https://caravagnalab.github.io/CNAqc)
-----

### Installation

``` r
# install.packages("devtools")
devtools::install_github("caravagnalab/CNAqc")
```

-----

#### Copyright and contacts

Cancer Data Science (CDS) Laboratory, University of Trieste, Italy.

[![](https://img.shields.io/badge/CDS%20Lab%20Github-caravagnalab-seagreen.svg)](https://github.com/caravagnalab)
[![](https://img.shields.io/badge/CDS%20Lab%20webpage-https://www.caravagnalab.org/-red.svg)](https://www.caravagnalab.org/)
