---
title: "1. Introduction"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{1. Introduction}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

options(crayon.enabled=F)
```


```{r setup}
library(CNAqc)
```

We discuss the input format and references used in `CNAqc`.

# Input format

_CNAqc notation:_

- `Major:minor` denotes a CNA segments with `Major`/`minor` copies of the major/minor allele. We sometimes call `"1:1"` the karyotype or the copy state of a segment.

- Clonal denotes something (a CNA, or a mutaiton) that is found at clonality 100% or, so to say, in all tumour cells; subclonal is something that is found in a subset of the tumour cells.

## Somatic mutations

You need to provide data for _somatic mutations_, _copy number segments_ and _tumour purity_; CNAqc comes with a template dataset.

```{r}
# Load template data
data('example_dataset_CNAqc', package = 'CNAqc')
```

These fields are required for somatic mutations:

* the mutation location, as `chr`, `from`, `to`. 
* the mutation reference and alternative alleles `ref` and `alt`;
* the total number of reads covering the mutated base(s), `DP` (depth);
* the total number of reads covering only the mutant allele(s), `NV` (number of reads with variant);
* the Variant Allele Frequency, `VAF`, defined as `NV/DP`.

Chromosome names and alleles should be in character format; chromosomes must be in the format `chr1`, `chr2`, etc..

```{r}
# Example input SNVs
print(
  example_dataset_CNAqc$mutations %>% 
        select(
          chr, from, to, # Genomic coordinates
          ref, alt,      # Alleles (reference and alternative)
          DP, NV, VAF    # Read counts (depth, number of variant reads, tumour VAF)
          )
  )
```

Optionally, you can annotate also _driver mutations_ by adding the following columns to your data:

* `is_driver`: whether the mutation is a driver, or not;
* `driver_label`:  the label to be shown in the plots that report also drivers (e.g., `BRAF V600E` could be a label).

```{r}
print(
  example_dataset_CNAqc$mutations %>% 
        select(
          chr, from, to, ref, alt, is_driver, driver_label, everything()
        )
  )
```

## Copy number segments

CNAqc distinguishes between 3 types of CNAs:

* clonal simple CNAs:
  * any clonal `"1:0"` (loss of heterzygosity, LOH), `"2:0"` (copy neutral LOH), `"1:1"` (diploid heterozygous), `"2:1"` (trisomy) and `"2:2"` (tetraploidy);
* clonal complex CNAs:
  * any other clonal CNA that is not simple;
* subclonal CNAs:
  * any mixture of 2 subclones, where each one of the subclones is defined by a simple CNA.

A fixed set of colours is used to identify the simple CNAs, and these fields are required for all types of  **copy number segments**:

* the segment location, as `chr`, `from` and `to`;
* the segment absolute number of copies of the major and minor alleles, as `Major` and `minor`;

Optionally, you can annotate also annotate the Cancer Cell Fraction (CCF) `CCF` for each segment (the clonality of the segment, which is returned only by callers that support subclonal segments):

* segments with `CCF = 1` are _clonal_, otherwise _subclonal_;
* by default, if missing, all segments are assumed to be clonal and assigned `CCF = 1`. Some computations might subset only clonal CNAs (see the manual of each function).

```{r}
# Example input CNA
print(
  example_dataset_CNAqc$cna %>% 
        select(
          chr, from, to, # Genomic coordinates
          Major, minor  # Number of copies of major/ and minor allele (B-allele)
        )
  )
```

If you wish to use subclonal CNAs, further columns are required.

* `Major_2` and `minor_2` reporting the e major and minor alleles for the second clone.

The model captures a mixture of two subclones, one with segment `Major:minor` and CCF `CCF` (which is compulsory), and another with segment `Major_2:minor_2` and CCF `1 - CCF`.

The values of `Major_2` and `minor_2` for clonal segments (`CCF = 1`) can be `NA` and will not be used.


## Tumour purity

Tumour _purity_, defined as the percentage of reads coming from tumour cells must be a value in $[0, 1]$.

```{r}
# Example purity
print(example_dataset_CNAqc$purity)
```

## Initialisation of a new dataset

To use `CNAqc`, you need to initialize a `cnaqc` S3 object with the initialisation function `init`.

This function will check input formats, and will map mutations to CNA segments. This function does not subset the data and retains all and only the mutations that map on top of a CNA segment.

When you create a dataset it is required to explicit the reference genome for the assembly (see below).

```{r}
# Use SNVs, CNAs and tumour purity (hg19 reference, see below)
x = init(
  mutations = example_dataset_CNAqc$mutations, 
  cna = example_dataset_CNAqc$cna,
  purity = example_dataset_CNAqc$purity,
  ref = 'hg19'
  )
```

The summary of `x` can be `print` to provide a number of usefull information.
```{r}
print(x)
```


## Subsetting data

You can _subset randomly_ the data; if drivers are annotated, they can be forced to stay in.

```{r, fig.width=4, fig.height=4}
y_5000 = subsample(x, N = 5000, keep_drivers = TRUE)

# 5000 + the ranomd entries that we sampled before
print(y_5000)
```

You can also subset data by _karyotype of the segments_, and by _total copy number_ of the segment. 

Both subset functions do not keep drivers that map off from the selected segments.

```{r, fig.width=4, fig.height=4}
# Triploid and copy-neutral LOH segments 
y_tripl_cnloh = subset_by_segment_karyotype(x, karyotypes = c('2:1', '2:0'))

print(y_tripl_cnloh)

# Two and four copies
y_2_4 = subset_by_segment_totalcn(x, totalcn = c(2, 4))

print(y_2_4)
```


## Reference genome coordinates

`CNAqc` contains two coordinates for two reference genomes, `hg19` and `GRCh38` (default). For each reference genome the package stores the location of the chromosomes, and their centromers; you can access these data.

```{r, fig.width=9.5, fig.height=3}
CNAqc:::get_reference("hg19")

CNAqc:::get_reference("GRCh38")
```

The reference genomes can be specified as a parameter once you create a `CNAqc` object. Coordinates are used to translate relative to absolute genome coordinates of somatic mutations and CNA segments; this is done only 
when plotting the data. Note that mapping of mutations to CNA segments is independent of the reference genome, and it will work as far as both mutation and CNA segments are mapped to the same reference.

You can use a hidden function to plot a reference, and make it explicit with a title.

```{r, fig.width=9.5, fig.height=3}
print(CNAqc:::blank_genome(ref = 'hg19') + labs(title = "HG19 genome reference"))
```



