---
title: "8. Drivers annotation"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{8. Drivers annotation}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

options(crayon.enabled=FALSE)
```

```{r setup, warning=FALSE}
library(CNAqc)

# Extra packages
require(dplyr)

#if you use RStudio you will probably have to run this line
# options(connectionObserver = NULL)
```

CNAqc  has a number of simple annotation functions that can be useful for giving a preliminary idea of what somatic mutations might be _driver events_. This function does not replace more complex and in-depth analyzes of mutations, using specialised tools like [Intogen](https://www.intogen.org/search), [COSMIC](https://cancer.sanger.ac.uk/cosmic), etc..

Given a table of mutation we annotate:

1. Position of the mutation, any of:
    - `coding`, 
    - `intron`, 
    - `fiveUTR`, 
    - `threeUTR`,
    - `intron`,
    - `intergenic`, 
    - `spliceSite`, 
    - `promoter`.
    
2. Consequence on the protein for coding mutations (with eventual change in the AA):
    - `nonsynonymous`, 
    - `synonymous`, 
    -  `frameshift`,
    -  `stop`.
    
3. Synonymous mutations are then compared to known driver genes from the [intOGen](https://www.intogen.org/search) database, and flagged as drivers.

4. Optionally if the `dbSNP` id is present the `PolyPhen` score can also be annotated

The annotation phase is heavily based on the [VariantAnnotation](https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html) package.


# Install Required Packages

We will annotate the example dataset available in CNAqc.

```{r}
# Load template data
data('example_dataset_CNAqc', package = 'CNAqc')

mutations <- example_dataset_CNAqc$mutations

mutations %>%  head
```

This tibble has already have driver annotated which have been generated using ([ANNOVAR](https://annovar.openbioinformatics.org/en/latest/)). We get rid of them and re-annotate the data.

```{r, warning=FALSE, message=FALSE}
# What are drivers (labels)
drivers_true <- mutations %>% 
  filter(is_driver == T) %>% 
  pull(driver_label) %>%  
  unique

mutations <- mutations %>% 
  dplyr::select(-is_driver, -driver_label)
```

The function uses databases from [Bioconductor](https://www.bioconductor.org/) to annotate the variants; installation of these
databases might take a bit of time because ~1GB of data have to be downloaded. This will happen only the first time the annotation is run.

```{r, warning=FALSE, message=FALSE}
# Reference against which we mapped the reads
reference_genome <- example_dataset_CNAqc$reference

# All those packages are distributed in Bioconductor
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager", repos="http://cran.us.r-project.org")

# We have to install the corresponding txdb package for transcript annotations
paste0("TxDb.Hsapiens.UCSC.",reference_genome, ".knownGene") %>% BiocManager::install()

# We have to install also the BS database for the sequences (it may take some time)
paste0("BSgenome.Hsapiens.UCSC.",reference_genome) %>% BiocManager::install()

# Then these two packages provide useful utilities to deal with biological databases
"Organism.dplyr" %>% BiocManager::install()
"org.Hs.eg.db" %>% BiocManager::install()
```
# Annotation

Now we have everything to get our annotated file

```{r, warning=FALSE, message=FALSE}
# Run function
mutations_annotated <- annotate_variants(mutations, ref = reference_genome)

mutations_annotated
```
One thing to note is the presence of multiple locations and consequences for a single variant. 

This happens as we try to annotate the mutations in a transcript-agnostic manner, consequently we report all possible effects and locations for any transcript (separated by `:`). 

For this cohort, the drivers largely correspond to those previously noted (`TP53` and `CTCF` mutations) with some differences. Obviously using multiple annotations and validations guarantees more precise results. 

`CNAqc` uses a list of driver gene per cancer type from [intOGen](https://www.intogen.org/download) (_Release date 2020.02.01_), matching mutation by gene id (e.g., a missense mutation in `APC`). The list of genes can be pooled from multiple tumor types, or providing a tumour code. 

The list of tumor types and codes is available in the package. 
```{r}
data('intogen_ttypes', package = 'CNAqc')

# intOGen
intogen_ttypes
```


```{r,  warning=FALSE, message=FALSE}
# Example: OV is ovarian cancer
mutations_annotated_OV <- annotate_variants(mutations, 
                                       ref = reference_genome, 
                                       filter_tumor_type = "OV")

mutations_annotated_OV
```

Another option is to supply a list of drivers in a one-coloumn `data.frame`, via argument `driver_list`.

```{r,  warning=FALSE, message=FALSE}
mutations_annotated_custom <- annotate_variants(mutations,
                                           ref = reference_genome, 
                                           driver_list = c("TP53", "CHDS") %>% as.data.frame
                                           )

mutations_annotated_custom
```

# Plot

We can finally perform QC and plot drivers together with the data distribution.

```{r, warning=FALSE, message=FALSE, fig.width=10, fig.height=4}
x = init(
  mutations = mutations_annotated, 
  cna = example_dataset_CNAqc$cna,
  purity = example_dataset_CNAqc$purity,
  ref = 'hg19'
  )

ggpubr::ggarrange(
  plot_data_histogram(x, which = 'VAF'),
  plot_data_histogram(x, which = 'DP'),
  plot_data_histogram(x, which = 'NV'),
  ncol = 3,
  nrow = 1
  )

```


