---
title: "CNAqc"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{CNAqc}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

options(crayon.enabled=F)
```



CNAqc requires in input

* read counts from somatic mutations such as single-nucleotide variants (SNVs) or insertion-deletions (indels); 
* allele-specific copy number segments (CNAs), for both clonal and subclonal segments;
* tumor purity estimate.

CNAqc uses chromosome coordinates the supported references (GRCh38 and hg19) to map mutations to segments. Then, the tool can elaborate a number of analysis that help quality control the consistency among mutations, CNAs and tumour purity (see Articles).

### The model

The following concepts are used to develop `CNAqc` analyses.

The **expected Variant Allele Frequency (VAF)** for mutations present in $m$ copies of the tumour genome, when the tumour sample purity is $\pi$ and segment ploidy is $p$ (i.e., total number of copies of the minor and major allele at the mutation locus) are given by
\[
v = \dfrac{m \pi}{
2 (1 - \pi) + \pi p 
} \, .
\]

This formula gives as an expectation for the allelic frequency of somatic mutations that map to a certain karyotype (e.g., a triploid `2:1` segment, or AAB),

Given VAF, tumour purity and CNAs, **Cancer Cell Fractions (CCF)** values can be computed as

\[
\text{CCF} = 
\dfrac{
v[
(p - 2)\pi + 2
]
}
{
m \pi
}
\]
where $p$ is total segment ploidy (e.g., 3 for a `2:1` state).

