<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Analyze CNA calls by peak detection. — analyze_peaks • CNAqc</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Analyze CNA calls by peak detection. — analyze_peaks" />
<meta property="og:description" content="This function carries out a peak-detection
analysis based on KDE and the package peakPick
in order to determine if the mutations that map to a
certain karyotype have an allelic frequency that is
consistent with the Major and minor alleles of the
segment, and the tumour purity. A score is produced
as a linear combination of the distance of the actual
peak to the expected one, derived with standard equations
for CNA analysis." />
<meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Analyze CNA calls by peak detection.</h1>
    <small class="dont-index">Source: <a href='https://github.com/caravagnalab/CNAqcR/analyze_peaks.R'><code>R/analyze_peaks.R</code></a></small>
    <div class="hidden name"><code>analyze_peaks.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function carries out a peak-detection
analysis based on KDE and the package <code>peakPick</code>
in order to determine if the mutations that map to a
certain karyotype have an allelic frequency that is
consistent with the Major and minor alleles of the
segment, and the tumour purity. A score is produced
as a linear combination of the distance of the actual
peak to the expected one, derived with standard equations
for CNA analysis.</p>
    </div>

    <pre class="usage"><span class='fu'>analyze_peaks</span><span class='op'>(</span>
  <span class='va'>x</span>,
  karyotypes <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"1:0"</span>, <span class='st'>"1:1"</span>, <span class='st'>"2:1"</span>, <span class='st'>"2:0"</span>, <span class='st'>"2:2"</span><span class='op'>)</span>,
  min_karyotype_size <span class='op'>=</span> <span class='fl'>0</span>,
  min_absolute_karyotype_mutations <span class='op'>=</span> <span class='fl'>10</span>,
  p_binsize_peaks <span class='op'>=</span> <span class='fl'>0.005</span>,
  matching_epsilon <span class='op'>=</span> <span class='fl'>0.03</span>,
  n_bootstrap <span class='op'>=</span> <span class='fl'>1</span>,
  kernel_adjust <span class='op'>=</span> <span class='fl'>1</span>,
  matching_strategy <span class='op'>=</span> <span class='st'>"closest"</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>An object of class <code>cnaqc</code>, created by the <code>init</code> function.</p></td>
    </tr>
    <tr>
      <th>karyotypes</th>
      <td><p>The list of karyotypes to test. By default LOH regions (A, AA),
diploid regions (AB), and amplification regions (AAB, AABB). These correspond to
<code>'1:0', '1:1', '2:1', '2:0', '2:2'</code> in "Major:minor" notation.</p></td>
    </tr>
    <tr>
      <th>min_karyotype_size</th>
      <td><p>Karyotypes are subset by their size (normalized for
the number of input mutations). Karyotypes smaller than this value are removed
from analysis.</p></td>
    </tr>
    <tr>
      <th>min_absolute_karyotype_mutations</th>
      <td><p>As <code>min_karyotype_size</code> this imposes a
minimum number of mutations on a karyotype to be analysed (in absolute number of
mutations). Karyotypes smaller than this value are removed from analysis.</p></td>
    </tr>
    <tr>
      <th>p_binsize_peaks</th>
      <td><p>Peaks detected will be filtered if, in a peak, we map
less than <code>p_binsize_peaks * N</code> mutations. The value <code>N</code> is obtained
couting all mutations that map in all peaks.</p></td>
    </tr>
    <tr>
      <th>matching_epsilon</th>
      <td><p>Peaks at a location are matched with a range of <code>epsilon * 0.5</code>,
left and right. By default (<code>0.025</code>) a maxium 5% tolerance is adopted. The status of
overall PASS/ FAIL for this analysis is determined with the same tolerance, given the score
computed across all karyotypes.</p></td>
    </tr>
    <tr>
      <th>n_bootstrap</th>
      <td><p>Number of times peak detection is bootstrapped (this helps sometimes
finding peaks that might be visually observable but fail to be detected by the underlying
peak detection heuristic). The default of this parameter is 1, meaning that no bootstrap
is performed.</p></td>
    </tr>
    <tr>
      <th>kernel_adjust</th>
      <td><p>KDE adjust density parameter; see <code>density</code>. A Gaussian
kernel is used (<code>kernel = 'gaussian'</code>).</p></td>
    </tr>
    <tr>
      <th>matching_strategy</th>
      <td><p>If <code>"closest"</code>, the closest peak will be used to
match the expected peak. If <code>"rightmost"</code>, peaks are matched prioritizing
right to left peaks (the higher get matched first); this strategy is more correct
in principle but works only if there are no spurious peaks in the estimated
density. By defaule the first strategy is used.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object of class <code>cnaqc</code>, modified to hold the results from this analysis.
See the vignette to see how to extract and plot the results.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='st'>'example_dataset_CNAqc'</span>, package <span class='op'>=</span> <span class='st'>'CNAqc'</span><span class='op'>)</span>
<span class='va'>x</span> <span class='op'>=</span> <span class='fu'><a href='init.html'>init</a></span><span class='op'>(</span><span class='va'>example_dataset_CNAqc</span><span class='op'>$</span><span class='va'>snvs</span>, <span class='va'>example_dataset_CNAqc</span><span class='op'>$</span><span class='va'>cna</span>,<span class='va'>example_dataset_CNAqc</span><span class='op'>$</span><span class='va'>purity</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span style='color: #BBBBBB;background-color: #0000BB;'> [ CNAqc - CNA Quality Check ] 
#&gt; </span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Using reference genome coordinates for: </span><span style='color: #00BB00;'>GRCh38</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Drivers are annotated, but 'gene' column is missing, using mutation location.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Missing CCF column from CNA calls, adding CCF = 1 assuming clonal CNA calls.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Input n = 12963 mutations for 267 CNA segments (267 clonal, 0 subclonal)</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> Mapped n = 12963 mutations to clonal segments (100% of input)</span></div><div class='input'>
<span class='va'>x</span> <span class='op'>=</span> <span class='fu'>analyze_peaks</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Karyotypes </span><span style='color: #00BB00;'>1:0</span><span>, </span><span style='color: #00BB00;'>1:1</span><span>, </span><span style='color: #00BB00;'>2:1</span><span>, </span><span style='color: #00BB00;'>2:0</span><span>, and </span><span style='color: #00BB00;'>2:2</span><span>. Matching strategy </span><span style='color: #00BB00;'>closest</span><span>. KDE = </span><span style='color: #00BB00;'>TRUE</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Found n = 9122 mutations in 2:2, 2:1, 2:0 (skipping those with n &lt; 0 mutations).</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Some peaks have been removed</span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 2 x 5</span><span>
#&gt;       x     y counts_per_bin discarded from 
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span>  0.13   0.3              1 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'>2</span><span>  0.32   0.3              0 TRUE      KDE  </div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Some peaks have been removed</span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 10 x 5</span><span>
#&gt;        x     y counts_per_bin discarded from 
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span>  0.11  0.27             20 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span>  0.67  0.01              0 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span>  0.72  0.02              2 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span>  0.75  0.01              2 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span>  0.79  0.01              1 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span>  0.82  0.01              1 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span>  0.85  0.01              1 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span>  0.91  0.01              1 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span>  0.93  0.01              0 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'>10</span><span>  0.97  0.02              0 TRUE      KDE  </div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 6 x 17</span><span>
#&gt;   mutation_multiplic… karyotype  peak     x     y counts_per_bin discarded from 
#&gt;                 </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span>                   1 2:1       0.308 0.306 3.60              59 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>2</span><span>                   2 2:1       0.616 0.602 2.94              60 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>3</span><span>                   1 2:0       0.445 0.423 0.172              0 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>4</span><span>                   2 2:0       0.89  0.885 7.99               7 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>5</span><span>                   1 2:2       0.235 0.24  0.86              65 FALSE     KDE  
#&gt; </span><span style='color: #BCBCBC;'>6</span><span>                   2 2:2       0.471 0.46  8.14             625 FALSE     KDE  
#&gt; </span><span style='color: #949494;'># … with 9 more variables: offset_VAF &lt;dbl&gt;, offset &lt;dbl&gt;, weight &lt;dbl&gt;,</span><span>
#&gt; </span><span style='color: #949494;'>#   epsilon &lt;dbl&gt;, VAF_tolerance &lt;dbl&gt;, matched &lt;lgl&gt;, score &lt;dbl&gt;,</span><span>
#&gt; </span><span style='color: #949494;'>#   purity_error &lt;dbl&gt;, QC &lt;chr&gt;</span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> Peak detection </span><span style='color: #00BB00;'>PASS</span><span> with </span><span style='color: #00BB00;'>r = 0.00503623467528875</span><span> and tolerance e = </span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>── </span><span style='color: #000000;background-color: #BBBB00;'>[ CNAqc ] </span><span> n = 12963 mutations in </span><span style='color: #00BB00;'>267</span><span> segments (267 clonal + 0 subclonal). Ge</span></div><div class='output co'>#&gt; 
#&gt;    2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { </span><span style='color: #BBBB00;'>CTCF</span><span> }
#&gt;    4:2  [n = 1893, L =  331 Mb] ■■■■■■■
#&gt;    3:2  [n = 1625, L =  357 Mb] ■■■■■■
#&gt;    2:1  [n = 1563, L =  420 Mb] ■■■■■■  { </span><span style='color: #BBBB00;'>TTN</span><span> }
#&gt;    3:0  [n =  312, L =  137 Mb] ■
#&gt;    2:0  [n =   81, L =   39 Mb]   { </span><span style='color: #BBBB00;'>TP53</span><span> }
#&gt;   16:2  [n =    4, L =    0 Mb] 
#&gt;   25:2  [n =    2, L =    1 Mb] 
#&gt;    3:1  [n =    2, L =    1 Mb] 
#&gt;  106:1  [n =    1, L =    0 Mb] 
#&gt;   26:2  [n =    1, L =    0 Mb] 
#&gt;   99:1  [n =    1, L =    0 Mb] 
#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Sample Purity: 89% ~ Ploidy: 4.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> There are 3 annotated driver(s).</span></div><div class='output co'>#&gt;          chr      from        to ref alt  DP NV       VAF driver_label is_driver
#&gt;         chr2 179431633 179431634   C   T 117 77 0.6581197          TTN      TRUE
#&gt;        chr16  67646006  67646007   C   T 120 54 0.4500000         CTCF      TRUE
#&gt;        chr17   7577106   7577107   G   C  84 78 0.9285714         TP53      TRUE</div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='color: #000000;background-color: #00BB00;'> PASS </span><span> Peaks QC </span><span style='font-weight: bold;'>closest</span><span>:</span>
#&gt; <span class='message'></span><span style='color: #00BB00;'>100%</span><span>, </span><span style='color: #00BB00;'>λ = 0.005</span><span> </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> </span><span style='color: #0000BB;'>2:1</span><span> ~ n = 1563  (17%) → </span><span style='color: #000000;background-color: #00BB00;'> PASS </span><span> 0.01    </span><span style='color: #000000;background-color: #00BB00;'> PASS </span><span> 0.028  </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> </span><span style='color: #0000BB;'>2:0</span><span> ~ n = 81    ( 1%) → </span><span style='color: #000000;background-color: #00BB00;'> PASS </span><span> 0.043   </span><span style='color: #000000;background-color: #00BB00;'> PASS </span><span> 0.005  </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> </span><span style='color: #0000BB;'>2:2</span><span> ~ n = 7478  (82%) → </span><span style='color: #000000;background-color: #00BB00;'> PASS </span><span> -0.034  </span><span style='color: #000000;background-color: #00BB00;'> PASS </span><span> 0.037  </span></div><div class='input'>
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>$</span><span class='va'>peaks_analysis</span><span class='op'>)</span>
</div><div class='output co'>#&gt; $score
#&gt; [1] 0.005036235
#&gt; 
#&gt; $fits
#&gt; $fits$`1:0`
#&gt; $fits$`1:0`$matching
#&gt; NULL
#&gt; 
#&gt; 
#&gt; $fits$`1:1`
#&gt; $fits$`1:1`$matching
#&gt; NULL
#&gt; 
#&gt; 
#&gt; $fits$`2:1`
#&gt; $fits$`2:1`$matching
#&gt; </span><span style='color: #949494;'># A tibble: 2 x 14</span><span>
#&gt;   mutation_multiplic… karyotype  peak     x     y counts_per_bin discarded from 
#&gt;                 </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span>                   1 2:1       0.308 0.306  3.60             59 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>2</span><span>                   2 2:1       0.616 0.602  2.94             60 FALSE     BMix 
#&gt; </span><span style='color: #949494;'># … with 6 more variables: offset_VAF &lt;dbl&gt;, offset &lt;dbl&gt;, weight &lt;dbl&gt;,</span><span>
#&gt; </span><span style='color: #949494;'>#   epsilon &lt;dbl&gt;, VAF_tolerance &lt;dbl&gt;, matched &lt;lgl&gt;</span><span>
#&gt; 
#&gt; $fits$`2:1`$density
#&gt; 
#&gt; Call:
#&gt; 	density.default(x = y, adjust = kernel_adjust, kernel = "gaussian")
#&gt; 
#&gt; Data: y (1563 obs.);	Bandwidth 'bw' = 0.0341
#&gt; 
#&gt;        x                  y          
#&gt;  Min.   :-0.05578   Min.   :0.00016  
#&gt;  1st Qu.: 0.17384   1st Qu.:0.21293  
#&gt;  Median : 0.40346   Median :0.51690  
#&gt;  Mean   : 0.40346   Mean   :1.08768  
#&gt;  3rd Qu.: 0.63308   3rd Qu.:1.99965  
#&gt;  Max.   : 0.86270   Max.   :3.60365  
#&gt; 
#&gt; $fits$`2:1`$xy_peaks
#&gt; </span><span style='color: #949494;'># A tibble: 6 x 5</span><span>
#&gt;       x     y counts_per_bin discarded from 
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 0.08  0.27               5 FALSE     KDE  
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> 0.3   3.6               78 FALSE     KDE  
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> 0.6   2.94              60 FALSE     KDE  
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> 0.106 0.248              2 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> 0.306 3.60              59 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> 0.602 2.94              60 FALSE     BMix 
#&gt; 
#&gt; 
#&gt; $fits$`2:0`
#&gt; $fits$`2:0`$matching
#&gt; </span><span style='color: #949494;'># A tibble: 2 x 14</span><span>
#&gt;   mutation_multiplic… karyotype  peak     x     y counts_per_bin discarded from 
#&gt;                 </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span>                   1 2:0       0.445 0.423 0.172              0 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>2</span><span>                   2 2:0       0.89  0.885 7.99               7 FALSE     BMix 
#&gt; </span><span style='color: #949494;'># … with 6 more variables: offset_VAF &lt;dbl&gt;, offset &lt;dbl&gt;, weight &lt;dbl&gt;,</span><span>
#&gt; </span><span style='color: #949494;'>#   epsilon &lt;dbl&gt;, VAF_tolerance &lt;dbl&gt;, matched &lt;lgl&gt;</span><span>
#&gt; 
#&gt; $fits$`2:0`$density
#&gt; 
#&gt; Call:
#&gt; 	density.default(x = y, adjust = kernel_adjust, kernel = "gaussian")
#&gt; 
#&gt; Data: y (81 obs.);	Bandwidth 'bw' = 0.0165
#&gt; 
#&gt;        x                y           
#&gt;  Min.   :0.0305   Min.   :0.000000  
#&gt;  1st Qu.:0.2815   1st Qu.:0.000732  
#&gt;  Median :0.5325   Median :0.193553  
#&gt;  Mean   :0.5325   Mean   :0.994921  
#&gt;  3rd Qu.:0.7836   3rd Qu.:0.556356  
#&gt;  Max.   :1.0346   Max.   :9.001287  
#&gt; 
#&gt; $fits$`2:0`$xy_peaks
#&gt; </span><span style='color: #949494;'># A tibble: 8 x 5</span><span>
#&gt;       x     y counts_per_bin discarded from 
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 0.13  0.3                1 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> 0.32  0.3                0 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> 0.38  0.56               1 FALSE     KDE  
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> 0.47  0.79               1 FALSE     KDE  
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> 0.87  9                  5 FALSE     KDE  
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> 0.423 0.172              0 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>7</span><span> 0.101 0.199              0 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>8</span><span> 0.885 7.99               7 FALSE     BMix 
#&gt; 
#&gt; 
#&gt; $fits$`2:2`
#&gt; $fits$`2:2`$matching
#&gt; </span><span style='color: #949494;'># A tibble: 2 x 14</span><span>
#&gt;   mutation_multiplic… karyotype  peak     x     y counts_per_bin discarded from 
#&gt;                 </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span>                   1 2:2       0.235  0.24  0.86             65 FALSE     KDE  
#&gt; </span><span style='color: #BCBCBC;'>2</span><span>                   2 2:2       0.471  0.46  8.14            625 FALSE     KDE  
#&gt; </span><span style='color: #949494;'># … with 6 more variables: offset_VAF &lt;dbl&gt;, offset &lt;dbl&gt;, weight &lt;dbl&gt;,</span><span>
#&gt; </span><span style='color: #949494;'>#   epsilon &lt;dbl&gt;, VAF_tolerance &lt;dbl&gt;, matched &lt;lgl&gt;</span><span>
#&gt; 
#&gt; $fits$`2:2`$density
#&gt; 
#&gt; Call:
#&gt; 	density.default(x = y, adjust = kernel_adjust, kernel = "gaussian")
#&gt; 
#&gt; Data: y (7478 obs.);	Bandwidth 'bw' = 0.008033
#&gt; 
#&gt;        x                 y           
#&gt;  Min.   :0.02101   Min.   :0.000244  
#&gt;  1st Qu.:0.27179   1st Qu.:0.007045  
#&gt;  Median :0.52256   Median :0.203133  
#&gt;  Mean   :0.52256   Mean   :0.995946  
#&gt;  3rd Qu.:0.77333   3rd Qu.:0.787272  
#&gt;  Max.   :1.02410   Max.   :8.138867  
#&gt; 
#&gt; $fits$`2:2`$xy_peaks
#&gt; </span><span style='color: #949494;'># A tibble: 17 x 5</span><span>
#&gt;         x     y counts_per_bin discarded from 
#&gt;     </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> 0.11   0.27              20 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> 0.22   0.84              69 FALSE     KDE  
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> 0.24   0.86              65 FALSE     KDE  
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> 0.46   8.14             625 FALSE     KDE  
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> 0.67   0.01               0 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> 0.72   0.02               2 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> 0.75   0.01               2 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> 0.79   0.01               1 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> 0.82   0.01               1 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> 0.85   0.01               1 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'>11</span><span> 0.91   0.01               1 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'>12</span><span> 0.93   0.01               0 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'>13</span><span> 0.97   0.02               0 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'>14</span><span> 0.074</span><span style='text-decoration: underline;'>0</span><span> 0.558             57 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>15</span><span> 0.496  5.70             508 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>16</span><span> 0.223  0.825             69 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>17</span><span> 0.439  6.65             477 FALSE     BMix 
#&gt; 
#&gt; 
#&gt; 
#&gt; $matches
#&gt; </span><span style='color: #949494;'># A tibble: 6 x 17</span><span>
#&gt;   mutation_multiplic… karyotype  peak     x     y counts_per_bin discarded from 
#&gt;                 </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span>                   1 2:1       0.308 0.306 3.60              59 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>2</span><span>                   2 2:1       0.616 0.602 2.94              60 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>3</span><span>                   1 2:0       0.445 0.423 0.172              0 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>4</span><span>                   2 2:0       0.89  0.885 7.99               7 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>5</span><span>                   1 2:2       0.235 0.24  0.86              65 FALSE     KDE  
#&gt; </span><span style='color: #BCBCBC;'>6</span><span>                   2 2:2       0.471 0.46  8.14             625 FALSE     KDE  
#&gt; </span><span style='color: #949494;'># … with 9 more variables: offset_VAF &lt;dbl&gt;, offset &lt;dbl&gt;, weight &lt;dbl&gt;,</span><span>
#&gt; </span><span style='color: #949494;'>#   epsilon &lt;dbl&gt;, VAF_tolerance &lt;dbl&gt;, matched &lt;lgl&gt;, score &lt;dbl&gt;,</span><span>
#&gt; </span><span style='color: #949494;'>#   purity_error &lt;dbl&gt;, QC &lt;chr&gt;</span><span>
#&gt; 
#&gt; $matching_strategy
#&gt; [1] "closest"
#&gt; 
#&gt; $purity_error
#&gt; [1] 0.05
#&gt; 
#&gt; $min_karyotype_size
#&gt; [1] 0
#&gt; 
#&gt; $p_binsize_peaks
#&gt; [1] 0.005
#&gt; 
#&gt; $QC
#&gt; [1] "PASS"
#&gt; 
#&gt; $KDE
#&gt; [1] TRUE
#&gt; </div><div class='input'>

<span class='va'>x</span> <span class='op'>=</span> <span class='fu'>analyze_peaks</span><span class='op'>(</span><span class='va'>x</span>, matching_strategy <span class='op'>=</span> <span class='st'>"rightmost"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Karyotypes </span><span style='color: #00BB00;'>1:0</span><span>, </span><span style='color: #00BB00;'>1:1</span><span>, </span><span style='color: #00BB00;'>2:1</span><span>, </span><span style='color: #00BB00;'>2:0</span><span>, and </span><span style='color: #00BB00;'>2:2</span><span>. Matching strategy </span><span style='color: #00BB00;'>rightmost</span><span>. KDE = </span><span style='color: #00BB00;'>TRUE</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Found n = 9122 mutations in 2:2, 2:1, 2:0 (skipping those with n &lt; 0 mutations).</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Some KDE peaks have been removed</span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 2 x 5</span><span>
#&gt;       x     y counts_per_bin discarded from 
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span>  0.13   0.3              1 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'>2</span><span>  0.32   0.3              0 TRUE      KDE  </div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Some KDE peaks have been removed</span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 10 x 5</span><span>
#&gt;        x     y counts_per_bin discarded from 
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span>  0.11  0.27             20 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span>  0.67  0.01              0 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span>  0.72  0.02              2 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span>  0.75  0.01              2 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span>  0.79  0.01              1 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span>  0.82  0.01              1 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span>  0.85  0.01              1 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span>  0.91  0.01              1 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span>  0.93  0.01              0 TRUE      KDE  
#&gt; </span><span style='color: #BCBCBC;'>10</span><span>  0.97  0.02              0 TRUE      KDE  </div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 6 x 17</span><span>
#&gt;   mutation_multiplic… karyotype  peak     x     y counts_per_bin discarded from 
#&gt;                 </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span>                   2 2:1       0.616 0.602  2.94             60 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>2</span><span>                   1 2:1       0.308 0.6    2.94             60 FALSE     KDE  
#&gt; </span><span style='color: #BCBCBC;'>3</span><span>                   2 2:0       0.89  0.885  7.99              7 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>4</span><span>                   1 2:0       0.445 0.87   9                 5 FALSE     KDE  
#&gt; </span><span style='color: #BCBCBC;'>5</span><span>                   2 2:2       0.471 0.461  8.14            625 FALSE     BMix 
#&gt; </span><span style='color: #BCBCBC;'>6</span><span>                   1 2:2       0.235 0.46   8.14            625 FALSE     KDE  
#&gt; </span><span style='color: #949494;'># … with 9 more variables: offset_VAF &lt;dbl&gt;, offset &lt;dbl&gt;, weight &lt;dbl&gt;,</span><span>
#&gt; </span><span style='color: #949494;'>#   epsilon &lt;dbl&gt;, VAF_tolerance &lt;dbl&gt;, matched &lt;lgl&gt;, score &lt;dbl&gt;,</span><span>
#&gt; </span><span style='color: #949494;'>#   purity_error &lt;dbl&gt;, QC &lt;chr&gt;</span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> Peak detection </span><span style='color: #00BB00;'>PASS</span><span> with </span><span style='color: #00BB00;'>r = -29.0623856969481</span><span> and tolerance e = </span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>── </span><span style='color: #000000;background-color: #BBBB00;'>[ CNAqc ] </span><span> n = 12963 mutations in </span><span style='color: #00BB00;'>267</span><span> segments (267 clonal + 0 subclonal). Ge</span></div><div class='output co'>#&gt; 
#&gt;    2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { </span><span style='color: #BBBB00;'>CTCF</span><span> }
#&gt;    4:2  [n = 1893, L =  331 Mb] ■■■■■■■
#&gt;    3:2  [n = 1625, L =  357 Mb] ■■■■■■
#&gt;    2:1  [n = 1563, L =  420 Mb] ■■■■■■  { </span><span style='color: #BBBB00;'>TTN</span><span> }
#&gt;    3:0  [n =  312, L =  137 Mb] ■
#&gt;    2:0  [n =   81, L =   39 Mb]   { </span><span style='color: #BBBB00;'>TP53</span><span> }
#&gt;   16:2  [n =    4, L =    0 Mb] 
#&gt;   25:2  [n =    2, L =    1 Mb] 
#&gt;    3:1  [n =    2, L =    1 Mb] 
#&gt;  106:1  [n =    1, L =    0 Mb] 
#&gt;   26:2  [n =    1, L =    0 Mb] 
#&gt;   99:1  [n =    1, L =    0 Mb] 
#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Sample Purity: 89% ~ Ploidy: 4.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> There are 3 annotated driver(s).</span></div><div class='output co'>#&gt;          chr      from        to ref alt  DP NV       VAF driver_label is_driver
#&gt;         chr2 179431633 179431634   C   T 117 77 0.6581197          TTN      TRUE
#&gt;        chr16  67646006  67646007   C   T 120 54 0.4500000         CTCF      TRUE
#&gt;        chr17   7577106   7577107   G   C  84 78 0.9285714         TP53      TRUE</div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='color: #000000;background-color: #00BB00;'> PASS </span><span> Peaks QC </span><span style='font-weight: bold;'>rightmost</span><span>:</span>
#&gt; <span class='message'></span><span style='color: #00BB00;'>100%</span><span>, </span><span style='color: #00BB00;'>λ = -29.0624</span><span> </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> </span><span style='color: #0000BB;'>2:1</span><span> ~ n = 1563  (17%) → </span><span style='color: #000000;background-color: #00BB00;'> PASS </span><span> 0.028   </span><span style='color: #000000;background-color: #00BB00;'> PASS </span><span> -3.651 </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> </span><span style='color: #0000BB;'>2:0</span><span> ~ n = 81    ( 1%) → </span><span style='color: #000000;background-color: #00BB00;'> PASS </span><span> 0.005   </span><span style='color: #000000;background-color: #00BB00;'> PASS </span><span> -0.85  </span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> </span><span style='color: #0000BB;'>2:2</span><span> ~ n = 7478  (82%) → </span><span style='color: #000000;background-color: #00BB00;'> PASS </span><span> 0.035   </span><span style='color: #000000;background-color: #00BB00;'> PASS </span><span> -70.172</span></div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://www.caravagnalab.org'>Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


