<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Import VEP annotations. — augment_with_vep • CNAqc</title><!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Import VEP annotations. — augment_with_vep"><meta property="og:description" content='This function imports
Ensembl Variant Effect Predictor (VEP)
annotations into a CNAqc object, or to a mutations
dataframe in the format ready for CNAqc. Therefore the expected uses are

"[Option 1]"Mutations + CNA + Purity -&amp;gt; CNAqc -&amp;gt; VEP annotation
"[Option 2]"Mutations -&amp;gt; VEP annotation -&amp;gt; Mutations (with VEP) + CNA + Purity -&amp;gt; CNAqc


VEP annotations should have been created using the scripts discussed at
the
CNAqc webpage
utility. At this point, if one has created the input data for CNAqc from the
original VCF file, the VEP can be added to the CNAqc object (Option 1), or to
the available mutations before creating the CNAqc obejct (Option 2).
VEP annotations and input mutations are matched by exact genomic coordinates.'><meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
    <li>
      <a href="../articles/a9_automatic.html">9. Automatic CNA calling</a>
    </li>
    <li>
      <a href="../articles/b10_MultiCNAqc.html">10. multiCNAqc</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/caravagnalab/CNAqc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Import VEP annotations.</h1>
    <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqc/R/vep.R" class="external-link"><code>R/vep.R</code></a></small>
    <div class="hidden name"><code>augment_with_vep.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function imports
<a href="https://www.ensembl.org/info/docs/tools/vep/index.html" class="external-link">Ensembl Variant Effect Predictor (VEP)</a>
annotations into a CNAqc object, or to a mutations
dataframe in the format ready for CNAqc. Therefore the expected uses are</p>
<ul><li><p>"[Option 1]"Mutations + CNA + Purity -&gt; CNAqc -&gt; VEP annotation</p></li>
<li><p>"[Option 2]"Mutations -&gt; VEP annotation -&gt; Mutations (with VEP) + CNA + Purity -&gt; CNAqc</p></li>
</ul><p>VEP annotations should have been created using the scripts discussed at
the
<a href="https://github.com/caravagnalab/CNAqc" class="external-link">CNAqc webpage</a>
utility. At this point, if one has created the input data for CNAqc from the
original VCF file, the VEP can be added to the CNAqc object (Option 1), or to
the available mutations before creating the CNAqc obejct (Option 2).</p>
<p>VEP annotations and input mutations are matched by exact genomic coordinates.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">augment_with_vep</span><span class="op">(</span><span class="va">x</span>, <span class="va">vep</span>, columns <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>A CNAqc object, or a dataframe of mutations in the input format
for CNAqc.</p></dd>


<dt id="arg-maf">maf<a class="anchor" aria-label="anchor" href="#arg-maf"></a></dt>
<dd><p>The file MAF associated containing the annotations in MAF format.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>It depends on `x`</p><ul><li><p>"[Option 1]"
 A CNAqc object like `x` where the mutations are associated to the
 MAF annotations, if `x` is a CNAqc object. In this case the S3 print method for `x` will report the
presence of MAF annotations.</p></li>
<li><p>"[Option 2]"
 If `x` is a dataframe, the same dataframe augmented with MAF annotations.</p></li>
</ul></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>function <code><a href="as_maftools_cohort.html">as_maftools_cohort</a></code> to convert multiple CNAqc
objects with MAF annotations into a single MAF cohort;  package
[maftools](https://bioconductor.org/packages/release/bioc/html/maftools.html) to
summarize, analyze and visualize MAF Files; the utility [vcf2maf](https://github.com/mskcc/vcf2maf)
to create MAF files from VCFs, using the [Ensembl Variant Effect Predictor (VEP)](https://www.ensembl.org/info/docs/tools/vep/index.html)
utility.</p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Example with a CNAqc input object</span></span></span>
<span class="r-in"><span><span class="kw">if</span><span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">{</span></span></span>
<span class="r-in"><span>   <span class="co"># Create your CNAqc object (omissis here) from an original "file.vcf"</span></span></span>
<span class="r-in"><span>   <span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="init.html">init</a></span><span class="op">(</span>mutations <span class="op">=</span> <span class="va">...</span>, cna <span class="op">=</span> <span class="va">...</span>, purity <span class="op">=</span> <span class="va">...</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>   <span class="co"># Offline, create your MAF annotations as file "file_vcf.maf" from "file.vcf"</span></span></span>
<span class="r-in"><span>   <span class="co"># vcf2maf file.vcf .... file_vcf.maf</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>   <span class="co"># Import into R/CNAqc</span></span></span>
<span class="r-in"><span>   <span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="augment_with_maf.html">augment_with_maf</a></span><span class="op">(</span><span class="va">x</span>, maf <span class="op">=</span> <span class="st">"file_vcf.maf"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>   <span class="co"># check they are in (there should be many columns with "MAF." prefix)</span></span></span>
<span class="r-in"><span>   <span class="va">x</span> <span class="op"><a href="pipe.html">%&gt;%</a></span> <span class="va">Mutations</span> <span class="op"><a href="pipe.html">%&gt;%</a></span> <span class="va">colnames</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://www.caravagnalab.org" class="external-link">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer></div>






  </body></html>

