<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>CNAqc: CNAqc - Copy Number Analysis quality check — CNAqc-package • CNAqc</title><!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="CNAqc: CNAqc - Copy Number Analysis quality check — CNAqc-package"><meta property="og:description" content="CNAqc is a package to provide a set of metrics to quality control (QC) allele-specific Copy Number Alteration (CNA) calls, somatic mutations and tumour purity determined from bulk sequencing assays. The package provides statistical measures to quantify the concordance between mutation and CNAs, exploiting the combinatorial relation between allelic imbalance in CNA segments and variant allelic frequencies (VAFs) of somatic mutations. Distinct peak matching procedures for VAFs can be used to QC clonal and subclonal copy numbers. Quantitative metrics and plots for data exploration are also available, allowing for an intuitive assessment of the goodness of somatic calls. QC metrics computed by CNAqc can also be used to re-parametrize a copy number caller and increase the quality of the somatic calls; the Sequenza tool is supported by the CNAqc with an automatica calling-QC pipeline. The tools is general and can be applied downstream of any bioinformatics pipeline. Routines to determine Cancer Cell Franctions (CCFs) for each mutation are also available, along with a model of CCF uncertainty that can be used to identify high-quality CCF calls for downstream analyses."><meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
    <li>
      <a href="../articles/a9_automatic.html">9. Automatic CNA calling</a>
    </li>
    <li>
      <a href="../articles/b10_MultiCNAqc.html">10. multiCNAqc</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/caravagnalab/CNAqc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>CNAqc: CNAqc - Copy Number Analysis quality check</h1>
    <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqc/R/CNAqc-package.R" class="external-link"><code>R/CNAqc-package.R</code></a></small>
    <div class="hidden name"><code>CNAqc-package.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>CNAqc is a package to provide a set of metrics to quality control (QC) allele-specific Copy Number Alteration (CNA) calls, somatic mutations and tumour purity determined from bulk sequencing assays. The package provides statistical measures to quantify the concordance between mutation and CNAs, exploiting the combinatorial relation between allelic imbalance in CNA segments and variant allelic frequencies (VAFs) of somatic mutations. Distinct peak matching procedures for VAFs can be used to QC clonal and subclonal copy numbers. Quantitative metrics and plots for data exploration are also available, allowing for an intuitive assessment of the goodness of somatic calls. QC metrics computed by CNAqc can also be used to re-parametrize a copy number caller and increase the quality of the somatic calls; the Sequenza tool is supported by the CNAqc with an automatica calling-QC pipeline. The tools is general and can be applied downstream of any bioinformatics pipeline. Routines to determine Cancer Cell Franctions (CCFs) for each mutation are also available, along with a model of CCF uncertainty that can be used to identify high-quality CCF calls for downstream analyses.</p>
    </div>


    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>Useful links:</p><ul><li><p><a href="https://github.com/caravagnalab/CNAqc" class="external-link">https://github.com/caravagnalab/CNAqc</a></p></li>
<li><p>Report bugs at <a href="https://github.com/caravagnalab/CNAqc/issues" class="external-link">https://github.com/caravagnalab/CNAqc/issues</a></p></li>
</ul></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p><strong>Maintainer</strong>: Giulio Caravagna <a href="mailto:gcaravagn@gmail.com">gcaravagn@gmail.com</a> (<a href="https://orcid.org/0000-0003-4240-3265" class="external-link">ORCID</a>)</p>
<p>Other contributors:</p><ul><li><p>Riccardo Bergamin <a href="mailto:ric.b.91@gmail.com">ric.b.91@gmail.com</a> [contributor]</p></li>
<li><p>Nicola Calonaci <a href="mailto:nicola.calonaci@gmail.com">nicola.calonaci@gmail.com</a> [contributor]</p></li>
<li><p>Salvatore Milite <a href="mailto:militesalvatore@gmail.com">militesalvatore@gmail.com</a> [contributor]</p></li>
<li><p>Alice Antonello <a href="mailto:aliceantonello9@gmail.com">aliceantonello9@gmail.com</a> [contributor]</p></li>
</ul></div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://www.caravagnalab.org" class="external-link">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer></div>






  </body></html>

