<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ CNAqc</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
    <li>
      <a href="../articles/a9_automatic.html">9. Automatic CNA calling</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/caravagnalab/CNAqc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="available-data">Available data <a href="#available-data" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>References and datasets included in the package.</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="chr_coordinates_hg19.html">chr_coordinates_hg19</a></code> </p>
        </td>
        <td><p>Coordinates for hg19 chromosomes.</p></td>
      </tr><tr><td>
          <p><code><a href="chr_coordinates_GRCh38.html">chr_coordinates_GRCh38</a></code> </p>
        </td>
        <td><p>Coordinates for GRCh38 chromosomes.</p></td>
      </tr><tr><td>
          <p><code><a href="gene_coordinates_GRCh38.html">gene_coordinates_GRCh38</a></code> </p>
        </td>
        <td><p>Coordinates for GRCh38 genes</p></td>
      </tr><tr><td>
          <p><code><a href="DriverDBv3.html">DriverDBv3</a></code> </p>
        </td>
        <td><p>Driver mutations released with the DriverDBv3 database
<a href="http://driverdb.tms.cmu.edu.tw/" class="external-link">http://driverdb.tms.cmu.edu.tw/</a></p></td>
      </tr><tr><td>
          <p><code><a href="example_dataset_CNAqc.html">example_dataset_CNAqc</a></code> </p>
        </td>
        <td><p>Example CNAqc dataset.</p></td>
      </tr><tr><td>
          <p><code><a href="example_PCAWG.html">example_PCAWG</a></code> </p>
        </td>
        <td><p>Example PCAWG tumour</p></td>
      </tr><tr><td>
          <p><code><a href="fpr_test.html">fpr_test</a></code> </p>
        </td>
        <td><p>Data (simulation performance) from the trainig set to auto-tune epsilon.</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="data-processing">Data processing <a href="#data-processing" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions to manipulate and visualise the data.</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="annotate_variants.html">annotate_variants()</a></code> </p>
        </td>
        <td><p>Annotate variant with VariantAnnotation.</p></td>
      </tr><tr><td>
          <p><code><a href="init.html">init()</a></code> </p>
        </td>
        <td><p>Create a CNAqc object.</p></td>
      </tr><tr><td>
          <p><code><a href="Mutations.html">Mutations()</a></code> </p>
        </td>
        <td><p>Extract mutation information.</p></td>
      </tr><tr><td>
          <p><code><a href="CNA.html">CNA()</a></code> </p>
        </td>
        <td><p>Extract CNA information.</p></td>
      </tr><tr><td>
          <p><code><a href="inspect_segment.html">inspect_segment()</a></code> </p>
        </td>
        <td><p>Plot the VAF of a set of segments</p></td>
      </tr><tr><td>
          <p><code><a href="get_drivers.html">get_drivers()</a></code> </p>
        </td>
        <td><p>Extract driver data.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_segment_size_distribution.html">plot_segment_size_distribution()</a></code> </p>
        </td>
        <td><p>Plot the segments' size distribution.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_karyotypes.html">plot_karyotypes()</a></code> </p>
        </td>
        <td><p>Plot a barplot for the segments.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_data_histogram.html">plot_data_histogram()</a></code> </p>
        </td>
        <td><p>Plot the data histogram.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_gw_counts.html">plot_gw_counts()</a></code> </p>
        </td>
        <td><p>Plot a genome-wide histogram of mutation counts.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_gw_depth.html">plot_gw_depth()</a></code> </p>
        </td>
        <td><p>Plot a genome-wide scatter plot of coverage.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_gw_vaf.html">plot_gw_vaf()</a></code> </p>
        </td>
        <td><p>Plot a genome-wide scatter plot of VAF</p></td>
      </tr><tr><td>
          <p><code><a href="plot_segments.html">plot_segments()</a></code> </p>
        </td>
        <td><p>Plot CNA segments.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_multisample_CNA.html">plot_multisample_CNA()</a></code> </p>
        </td>
        <td><p>Plots CNAs from multiple samples.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_icon_CNA.html">plot_icon_CNA()</a></code> </p>
        </td>
        <td><p>Plot icon for these segments.</p></td>
      </tr><tr><td>
          <p><code><a href="split_by_chromosome.html">split_by_chromosome()</a></code> </p>
        </td>
        <td><p>Split a dataset by chromosome.</p></td>
      </tr><tr><td>
          <p><code><a href="subsample.html">subsample()</a></code> </p>
        </td>
        <td><p>Randomly subsample a dataset's mutations.</p></td>
      </tr><tr><td>
          <p><code><a href="subset_by_segment_karyotype.html">subset_by_segment_karyotype()</a></code> </p>
        </td>
        <td><p>Subset calls by segments' karyotype.</p></td>
      </tr><tr><td>
          <p><code><a href="subset_by_segment_totalcn.html">subset_by_segment_totalcn()</a></code> </p>
        </td>
        <td><p>Subset calls by segments total copy number</p></td>
      </tr><tr><td>
          <p><code><a href="subset_by_minimum_CCF.html">subset_by_minimum_CCF()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr><td>
          <p><code><a href="subset_snvs.html">subset_snvs()</a></code> </p>
        </td>
        <td><p>Retain only SNVs.</p></td>
      </tr><tr><td>
          <p><code><a href="print.cnaqc.html">print.cnaqc()</a></code> </p>
        </td>
        <td><p>Print for class <code>'cnaqc'</code>.</p></td>
      </tr><tr><td>
          <p><code><a href="plot.cnaqc.html">plot.cnaqc()</a></code> </p>
        </td>
        <td><p>Plot for class <code>'cnaqc'</code>.</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="purity-and-ploidy-qc">Purity and ploidy QC <a href="#purity-and-ploidy-qc" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Analysis of CNA calls by peak detection.</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="analyze_peaks.html">analyze_peaks()</a></code> </p>
        </td>
        <td><p>Analyze calls by peak detection.</p></td>
      </tr><tr><td>
          <p><code><a href="auto_tolerance.html">auto_tolerance()</a></code> </p>
        </td>
        <td><p>Determine the optimal error tolerance based on data</p></td>
      </tr><tr><td>
          <p><code><a href="plot_peaks_analysis.html">plot_peaks_analysis()</a></code> </p>
        </td>
        <td><p>Plot the results of peak analysis.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_qc.html">plot_qc()</a></code> </p>
        </td>
        <td><p>Plot a summary of QC results.</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="ccf-estimation">CCF estimation <a href="#ccf-estimation" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Computation of Cancer Cell Fractions data.</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="compute_CCF.html">compute_CCF()</a></code> </p>
        </td>
        <td><p>Compute CCF values for the available mutations.</p></td>
      </tr><tr><td>
          <p><code><a href="CCF.html">CCF()</a></code> </p>
        </td>
        <td><p>Return the available mutation CCF estimates in the data.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_CCF.html">plot_CCF()</a></code> </p>
        </td>
        <td><p>Plot the CCF estimates in the data.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_gw_ccf.html">plot_gw_ccf()</a></code> </p>
        </td>
        <td><p>Plot a genome-wide scatter plot of VAF</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="smoothing">Smoothing <a href="#smoothing" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Analysis and plotting functions to smooth segments.</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="smooth_segments.html">smooth_segments()</a></code> </p>
        </td>
        <td><p>Smooth copy number segments.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_smoothing.html">plot_smoothing()</a></code> </p>
        </td>
        <td><p>Plot smoothed segments data</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="overfragmentation-detection">Overfragmentation detection <a href="#overfragmentation-detection" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Analysis and plotting functions to smooth segments.</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="detect_arm_overfragmentation.html">detect_arm_overfragmentation()</a></code> </p>
        </td>
        <td><p>Determines fragmentation of copy number segments</p></td>
      </tr><tr><td>
          <p><code><a href="plot_arm_fragmentation.html">plot_arm_fragmentation()</a></code> </p>
        </td>
        <td><p>Plot patterns of arm level fragmentation.</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="pipelines">Pipelines <a href="#pipelines" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Pipelines implemented with CNAqc.</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="Sequenza_CNAqc.html">Sequenza_CNAqc()</a></code> </p>
        </td>
        <td><p>CNAqc-based purity-optimisation pipeline for Sequenza CNA calling.</p></td>
      </tr><tr><td>
          <p><code><a href="parse_Battenberg.html">parse_Battenberg()</a></code> </p>
        </td>
        <td><p>Parse Battenberg clonal/ subclonal calls</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://www.caravagnalab.org" class="external-link">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer></div>

  


  

  </body></html>

