<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>CNAqc-based purity-optimisation pipeline for Sequenza CNA calling. — Sequenza_CNAqc • CNAqc</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="CNAqc-based purity-optimisation pipeline for Sequenza CNA calling. — Sequenza_CNAqc" />
<meta property="og:description" content="This pipeline implements optimised allele-specific
copy number calling with Sequenza and CNAqc, optimising purity estimated
from data, and allele-specific segments.
Requirements:
1. (Prerequisites) Run Sequenza utils steps outside R
   a. Process a FASTA file to produce a GC Wiggle track file
   b. Process BAM and Wiggle files to produce a seqz file:
   c. Post-process by binning the original seqz file:
Instructions to customize these steps are available at
   https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html
2. Then this pipeline (Sequenza_CNAqc) will optimize CNA calling and purity
estimation using peak detection from CNAqc. Starting from a broad set of initial conditions
for purity (5
Results are then quality controlled by peak detection via CNAqc. The analysis with CNAqc 
can be carried out using either somatic mutations called by Sequenza, or an external set 
of input calls. At every run the best solution by Sequenza is stored in a cache, and up to two alternative 
solutions are generated: one optional by Sequenza, which might propose different exact values for 
cellularity and ploidy; one by CNAqc, adjusting the current cellularity of the best Sequenza 
solution. The alternative solutions are enqueued in a list L of cellularity and ploidy values 
to be tested, based on their presence in the cache.
3. Until the list L is empty, the point values of cellularity and ploidy are 
tested repeating step 2, using ranges around the proposed values specified by
input parameters. Further refinements steps will restrict the Sequenza purity
range based on the score computed by analyse_peaks, which determines a purity
gradient.
4. Each run is saved in a folder named run-1, run-2, etc.
When the list L is empty the pipeline stops and a symbolic link (final) 
pointing to the best Sequenza solution, based on CNAqc score is created." />
<meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>CNAqc-based purity-optimisation pipeline for Sequenza CNA calling.</h1>
    <small class="dont-index">Source: <a href='https://github.com/caravagnalab/CNAqcR/pipelines.R'><code>R/pipelines.R</code></a></small>
    <div class="hidden name"><code>Sequenza_CNAqc.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This pipeline implements optimised allele-specific
copy number calling with Sequenza and CNAqc, optimising purity estimated
from data, and allele-specific segments.</p>
<p>Requirements:</p>
<p>1. (Prerequisites) Run Sequenza utils steps outside R
   a. Process a FASTA file to produce a GC Wiggle track file
   b. Process BAM and Wiggle files to produce a seqz file:
   c. Post-process by binning the original seqz file:</p>
<p>Instructions to customize these steps are available at
   <a href='https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html'>https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html</a></p>
<p>2. Then this pipeline (<code>Sequenza_CNAqc</code>) will optimize CNA calling and purity
estimation using peak detection from CNAqc. Starting from a broad set of initial conditions
for purity (5<!-- % to 100%) the pipeline fits cellularity and ploidy values and dumps results. -->
Results are then quality controlled by peak detection via CNAqc. The analysis with CNAqc 
can be carried out using either somatic mutations called by Sequenza, or an external set 
of input calls. At every run the best solution by Sequenza is stored in a cache, and up to two alternative 
solutions are generated: one optional by Sequenza, which might propose different exact values for 
cellularity and ploidy; one by CNAqc, adjusting the current cellularity of the best Sequenza 
solution. The alternative solutions are enqueued in a list L of cellularity and ploidy values 
to be tested, based on their presence in the cache.</p>
<p>3. Until the list L is empty, the point values of cellularity and ploidy are 
tested repeating step 2, using ranges around the proposed values specified by
input parameters. Further refinements steps will restrict the Sequenza purity
range based on the score computed by <code>analyse_peaks</code>, which determines a purity
gradient.</p>
<p>4. Each run is saved in a folder named <code>run-1</code>, <code>run-2</code>, etc.
When the list L is empty the pipeline stops and a symbolic link (<code>final</code>) 
pointing to the best Sequenza solution, based on CNAqc score is created.</p>
    </div>

    <pre class="usage"><span class='fu'>Sequenza_CNAqc</span><span class='op'>(</span>
  <span class='va'>sample_id</span>,
  <span class='va'>seqz_file</span>,
  mutations <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>sex</span>,
  cellularity <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0.05</span>, <span class='fl'>1</span><span class='op'>)</span>,
  ploidy <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1.8</span>, <span class='fl'>5.4</span><span class='op'>)</span>,
  reference <span class='op'>=</span> <span class='st'>"GRCh38"</span>,
  normalization.method <span class='op'>=</span> <span class='st'>"median"</span>,
  window <span class='op'>=</span> <span class='fl'>1e+05</span>,
  gamma <span class='op'>=</span> <span class='fl'>280</span>,
  kmin <span class='op'>=</span> <span class='fl'>300</span>,
  min.reads.baf <span class='op'>=</span> <span class='fl'>50</span>,
  min.reads <span class='op'>=</span> <span class='fl'>50</span>,
  min.reads.normal <span class='op'>=</span> <span class='fl'>15</span>,
  max.mut.types <span class='op'>=</span> <span class='fl'>1</span>,
  delta_cellularity <span class='op'>=</span> <span class='fl'>0.05</span>,
  delta_ploidy <span class='op'>=</span> <span class='fl'>0.25</span>,
  verbose <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>sample_id</th>
      <td><p>The id of the sample.</p></td>
    </tr>
    <tr>
      <th>seqz_file</th>
      <td><p>A binned <code>.seqz</code> file.</p></td>
    </tr>
    <tr>
      <th>mutations</th>
      <td><p>If not NULL (default), an external set of mutation calls,
in tibble format. Must include colummns "chr", "from", "to", "NV", "NR", "VAF".</p></td>
    </tr>
    <tr>
      <th>sex</th>
      <td><p>Sex of the patient from whom the sample is drawn.</p></td>
    </tr>
    <tr>
      <th>cellularity</th>
      <td><p>Input to <code>sequenza.fit</code>, see https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html.</p></td>
    </tr>
    <tr>
      <th>ploidy</th>
      <td><p>Input to <code>sequenza.fit</code>, see https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html.</p></td>
    </tr>
    <tr>
      <th>reference</th>
      <td><p>Reference genome among supported by CNAqc (GRCh38 or hg19/GRCh37).</p></td>
    </tr>
    <tr>
      <th>normalization.method</th>
      <td><p>Input to <code>sequenza.extract</code>, see https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html.</p></td>
    </tr>
    <tr>
      <th>window</th>
      <td><p>Input to <code>sequenza.extract</code>, see https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html.</p></td>
    </tr>
    <tr>
      <th>gamma</th>
      <td><p>Input to <code>sequenza.extract</code>, see https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html.</p></td>
    </tr>
    <tr>
      <th>kmin</th>
      <td><p>Input to <code>sequenza.extract</code>, see https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html.</p></td>
    </tr>
    <tr>
      <th>min.reads.baf</th>
      <td><p>Input to <code>sequenza.extract</code>, see https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html.</p></td>
    </tr>
    <tr>
      <th>min.reads</th>
      <td><p>Input to <code>sequenza.extract</code>, see https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html.</p></td>
    </tr>
    <tr>
      <th>min.reads.normal</th>
      <td><p>Input to <code>sequenza.extract</code>, see https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html.</p></td>
    </tr>
    <tr>
      <th>max.mut.types</th>
      <td><p>Input to <code>sequenza.extract</code>, see https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html.</p></td>
    </tr>
    <tr>
      <th>delta_cellularity</th>
      <td><p>When a proposed solution of purity and ploidy is tested,
Sequenza is run on an interval of purity centered at the proposed value, of half 
width <code>delta_cellularity</code>.</p></td>
    </tr>
    <tr>
      <th>delta_ploidy</th>
      <td><p>When a proposed solution of purity and ploidy is tested,
Sequenza is run on an interval of ploidy centered at the proposed value, of half 
width <code>delta_ploidy</code>.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>If TRUE (default FALSE) at each Sequenza fitting step, print the updated
list of purity and ploidy pairs to test.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Optional parameters passed to the <code>analyse_peaks</code> function
by CNAqc. Tune these to change error tolerance or karyotypes to use for QC.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object of class <code>tibble</code>, containing for each run its identifier
<code>run</code>, the proposed values of <code>purity</code> and <code>ploidy</code>, the corresponding
CNAqc <code>score</code> and quality control status <code>QC</code>, a list containing the
input and output of Sequenza fitting procedure, and the object of class <code>CNAqc</code>
created by the <code>init</code> and <code>analyze_peaks</code> functions.</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>If at any time a purity proposal made by CNAqc leads
to inconsistent values (i.e., outside $[0,1]$) the pipeline stops prompting to the user
to adjust manually the range of <code>ploidy</code>.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://www.caravagnalab.org'>Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


