<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Smooth copy number segments. — smooth_segments • CNAqc</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Smooth copy number segments. — smooth_segments" />
<meta property="og:description" content="This functions join segments that have the same Major and minor alleles
(absolute copy number), and that are split by at most a certain number
of nucleotides Delta. The pre-smoothing copy number segments are
retained in the output computation." />
<meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Smooth copy number segments.</h1>
    <small class="dont-index">Source: <a href='https://github.com/caravagnalab/CNAqcR/smooth_segments.R'><code>R/smooth_segments.R</code></a></small>
    <div class="hidden name"><code>smooth_segments.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This functions join segments that have the same Major and minor alleles
(absolute copy number), and that are split by at most a certain number
of nucleotides <code>Delta</code>. The pre-smoothing copy number segments are
retained in the output computation.</p>
    </div>

    <pre class="usage"><span class='fu'>smooth_segments</span><span class='op'>(</span><span class='va'>x</span>, maximum_distance <span class='op'>=</span> <span class='fl'>1e+06</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>An object of class <code>cnaqc</code>, created by the <code>init</code> function.</p></td>
    </tr>
    <tr>
      <th>maximum_distance</th>
      <td><p>The <code>Delta</code> maximum of nucleotides to allow
to join two equivalent copy number segments.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object of class <code>cnaqc</code>, created by the <code>init</code> function.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='r-in'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='st'>'example_dataset_CNAqc'</span>, package <span class='op'>=</span> <span class='st'>'CNAqc'</span><span class='op'>)</span></span>
<span class='r-in'><span class='va'>x</span> <span class='op'>=</span> <span class='fu'><a href='init.html'>init</a></span><span class='op'>(</span><span class='va'>example_dataset_CNAqc</span><span class='op'>$</span><span class='va'>snvs</span>, <span class='va'>example_dataset_CNAqc</span><span class='op'>$</span><span class='va'>cna</span>,<span class='va'>example_dataset_CNAqc</span><span class='op'>$</span><span class='va'>purity</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BBBBBB; background-color: #0000BB;'> [ CNAqc - CNA Quality Check ] </span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> <span style='color: #00BBBB;'>ℹ</span> Using reference genome coordinates for: <span style='color: #00BB00;'>GRCh38</span>.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> <span style='color: #00BBBB;'>ℹ</span> Drivers are annotated, but 'gene' column is missing, using mutation location.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> <span style='color: #BBBB00;'>!</span> Missing CCF column from CNA calls, adding CCF = 1 assuming clonal CNA calls.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> <span style='color: #00BBBB;'>ℹ</span> Input n = 12963 mutations for 267 CNA segments (267 clonal, 0 subclonal)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> <span style='color: #00BB00;'>✔</span> Mapped n = 12963 mutations to clonal segments (100% of input)</span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>x</span> <span class='op'>=</span> <span class='fu'>smooth_segments</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr1</span> with <span style='color: #BB0000;'>37</span> segments: [1-24] [27-30] [31-33] [34-37] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr10</span> with <span style='color: #BB0000;'>8</span> segments: [1-2] [3-4] [5-8] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr11</span> with <span style='color: #BB0000;'>22</span> segments: [1-7] [8-9] [10-22] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr12</span> with <span style='color: #BB0000;'>13</span> segments: [2-3] [12-13] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr14</span> with <span style='color: #BB0000;'>2</span> segments: [1-2] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr15</span> with <span style='color: #BB0000;'>9</span> segments: [1-5] [7-9] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr16</span> with <span style='color: #BB0000;'>10</span> segments: [1-6] [8-10] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr17</span> with <span style='color: #BB0000;'>10</span> segments: [4-5] [6-7] [8-10] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr18</span> with <span style='color: #BB0000;'>8</span> segments: [1-2] [3-8] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr19</span> with <span style='color: #BB0000;'>5</span> segments: [2-5] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr2</span> with <span style='color: #BB0000;'>18</span> segments: [1-8] [10-12] [14-18] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr20</span> with <span style='color: #BB0000;'>9</span> segments: [1-3] [4-9] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr21</span> with <span style='color: #BB0000;'>2</span> segments: [1-2] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr22</span> with <span style='color: #BB0000;'>3</span> segments: </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr3</span> with <span style='color: #BB0000;'>19</span> segments: [1-7] [8-13] [15-19] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr4</span> with <span style='color: #BB0000;'>8</span> segments: [1-4] [5-8] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr5</span> with <span style='color: #BB0000;'>6</span> segments: [1-2] [3-5] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr6</span> with <span style='color: #BB0000;'>4</span> segments: [1-2] [3-4] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr7</span> with <span style='color: #BB0000;'>46</span> segments: [1-5] [9-27] [36-38] [39-41] [43-46] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr8</span> with <span style='color: #BB0000;'>18</span> segments: [1-4] [5-11] [12-18] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chr9</span> with <span style='color: #BB0000;'>3</span> segments: [1-2] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Smoothing <span style='color: #0000BB;'>chrX</span> with <span style='color: #BB0000;'>6</span> segments: [1-2] [3-6] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> <span style='color: #00BB00;'>✔</span> Smoothed from 267 to 87 segments with 1e+06 gap (bases).</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> <span style='color: #00BBBB;'>ℹ</span> Creating a new CNAqc object. The old object will be retained in the $before_smoothing field.</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BBBBBB; background-color: #0000BB;'> [ CNAqc - CNA Quality Check ] </span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> <span style='color: #00BBBB;'>ℹ</span> Using reference genome coordinates for: <span style='color: #00BB00;'>GRCh38</span>.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> <span style='color: #00BBBB;'>ℹ</span> Input n = 12963 mutations for 87 CNA segments (87 clonal, 0 subclonal)</span>
<span class='r-wrn co'><span class='r-pr'>#&gt;</span> <span class='warning'>Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> <span style='color: #00BB00;'>✔</span> Mapped n = 12963 mutations to clonal segments (100% of input)</span>
<span class='r-in'><span class='fu'><a href='plot_smoothing.html'>plot_smoothing</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> will replace the existing scale.</span>
<span class='r-wrn co'><span class='r-pr'>#&gt;</span> <span class='warning'>Warning: </span>Graphs cannot be horizontally aligned unless the axis parameter is set. Placing graphs unaligned.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> will replace the existing scale.</span>
<span class='r-wrn co'><span class='r-pr'>#&gt;</span> <span class='warning'>Warning: </span>Graphs cannot be horizontally aligned unless the axis parameter is set. Placing graphs unaligned.</span>
<span class='r-plt img'><img src='smooth_segments-1.png' alt='' width='700' height='433' /></span>
</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://www.caravagnalab.org'>Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


