<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>CNAqc - Copy Number Analysis quality check â€¢ CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="CNAqc - Copy Number Analysis quality check">
<meta property="og:description" content="CNAqc is a package to provide a set of metrics to assess the quality of 
  Copy Number Alteration (CNA) calls, somatic mutations and other tumour covariates from bulk sequencing. 
  The package provides statistical measures to quantify the concordance 
  between mutation and CNAs, exploiting allelic imbalance in allele-specific CNA segments and allelic 
  frequencies of somatic mutations. Quantitative metrics and plots for data exploration and quality control 
  are available, allowing for an intuitive assessment of the quality of calls. Quantitative measures can 
  also be used to suggest adjustments of the current purity estimates to increase the quality of CNA calls.
  The tools is general and can be applied downstream of any bioinformatics pipeline.
  Routines to determine Cancer Cell Franctions for each mutation are available as well. CNAqc is part of the
  evoverse, a package that gathers multiple R packages to implement Cancer Evolution analyses.">
<meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="cnaqc-" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#cnaqc-" class="anchor"></a>CNAqc <img src="reference/figures/logo.png" align="right" height="139">
</h1></div>
<!-- badges: start -->

<p><a href="https://www.tidyverse.org/lifecycle/#stable"><img src="https://img.shields.io/badge/lifecycle-stable-green.svg" alt="Lifecycle: maturing"></a> <!-- badges: end --></p>
<p><code>CNAqc</code> contains different methods to assess quality of allele-specific Copy Number Alterations (CNA), somatic mutations and purity estimates generated from tumour bulk sequencing.</p>
<p>The package provides methods to estimate Cancer Cell Fractions (CCFs) normalizing somatic allele frequencies for copy number states and tumour purity. The package contains also statistical tests to identify patterns of over-fragmentation of chromosome arms.</p>
<div id="citation" class="section level4">
<h4 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h4>
<p><a href="https://doi.org/10.1101/2021.02.13.429885"><img src="https://img.shields.io/badge/doi-10.1101/2021.02.13.429885-red.svg"></a></p>
<p>If you use <code>CNAqc</code>, please cite:</p>
<ul>
<li>
<em>A fully automated approach for quality control of cancer mutations in the era of high-resolution whole genome sequencing</em> Jacob Househam, William CH Cross and Giulio Caravagna. <a href="https://www.biorxiv.org/content/10.1101/2021.02.13.429885v1.full">biorXiv preprint</a>, 2021</li>
</ul>
</div>
<div id="help-and-support" class="section level4">
<h4 class="hasAnchor">
<a href="#help-and-support" class="anchor"></a>Help and support</h4>
</div>
<div id="section" class="section level2">
<h2 class="hasAnchor">
<a href="#section" class="anchor"></a><a href="https://caravagnalab.github.io/CNAqc"><img src="https://img.shields.io/badge/GitHub%20Pages-https://caravagnalab.github.io/CNAqc/-yellow.svg"></a>
</h2>
<div id="installation" class="section level3">
<h3 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h3>
<p>You can install the released version of <code>CNAqc</code> from <a href="https://github.com/">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("devtools")</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"caravagnalab/CNAqc"</span><span class="op">)</span></code></pre></div>
<hr>
<div id="copyright-and-contacts" class="section level4">
<h4 class="hasAnchor">
<a href="#copyright-and-contacts" class="anchor"></a>Copyright and contacts</h4>
<p>Giulio Caravagna. Cancer Data Science (CDS) Laboratory.</p>
<p><a href="mailto:gcaravagn@gmail.com"><img src="https://img.shields.io/badge/Email-gcaravagn@gmail.com-steelblue.svg"></a> <a href="https://github.com/caravagnalab"><img src="https://img.shields.io/badge/CDS%20Lab%20Github-caravagnalab-seagreen.svg"></a> <a href="https://www.caravagnalab.org/"><img src="https://img.shields.io/badge/CDS%20Lab%20webpage-https://www.caravagnalab.org/-red.svg"></a></p>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/caravagnalab/CNAqc/">https://â€‹github.com/â€‹caravagnalab/â€‹CNAqc/â€‹</a>
</li>
<li>Report a bug at <br><a href="https://github.com/caravagnalab/CNAqc/issues">https://â€‹github.com/â€‹caravagnalab/â€‹CNAqc/â€‹issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>GPL</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.caravagnalab.org">Giulio Caravagna</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-4240-3265" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/caravagnalab/CNAqc/actions"><img src="https://github.com/caravagnalab/CNAqc/workflows/R-CMD-check/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://github.com/caravagnalab/CNAqc/actions/workflows/pkgdown.yaml"><img src="https://github.com/caravagnalab/CNAqc/actions/workflows/pkgdown.yaml/badge.svg" alt="pkgdown"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://www.caravagnalab.org">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
