<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>2. Plotting data • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="2. Plotting data">
<meta property="og:description" content="CNAqc">
<meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
    <li>
      <a href="../articles/a9_automatic.html">9. Automatic CNA calling</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>2. Plotting data</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqcvignettes/a2_Plots.Rmd" class="external-link"><code>vignettes/a2_Plots.Rmd</code></a></small>
      <div class="hidden name"><code>a2_Plots.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/CNAqc" class="external-link">CNAqc</a></span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> Loading <span style="color: #00BB00;">CNAqc</span>, <span style="font-style: italic;">'Copy Number Alteration quality check'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagn.github.io/CNAqc/&gt;</span></span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span></code></pre></div>
<p>All CNAqc plotting functions use:</p>
<ul>
<li>
<a href="https://ggplot2.tidyverse.org/" class="external-link">ggplot</a> for panel
plots;</li>
<li>
<a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">ggpubr</a> and <a href="https://cran.r-project.org/web/packages/cowplot" class="external-link">cowplot</a> for
multi-panel figures assembly.</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Dataset available with the package </span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">'example_dataset_CNAqc'</span>, package <span class="op">=</span> <span class="st">'CNAqc'</span><span class="op">)</span>

<span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span>
  mutations <span class="op">=</span> <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">mutations</span>, 
  cna <span class="op">=</span> <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">cna</span>,
  purity <span class="op">=</span> <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">purity</span>,
  ref <span class="op">=</span> <span class="st">'hg19'</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">CNAqc - CNA Quality Check</span> <span style="color: #00BBBB;">───────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Drivers are annotated, but 'gene' column is missing, using mutation location.</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">12963</span> somatic mutations: <span style="color: #00BB00;">12963</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">0</span> indels.</span>
<span class="co">#&gt; <span style="color: #BBBB00;">!</span> CNAs have no CCF, assuming clonal CNAs (CCF = 1).</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">267</span> segments: <span style="color: #00BB00;">267</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">12963</span> mutations mapped to clonal CNAs.</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">12963</span> mutations in <span style="color: #00BB00;">267</span> segments (<span style="color: #00BB00;">267</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Clonal CNAs</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;    2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { <span style="color: #BBBB00;">CTCF</span> }</span>
<span class="co">#&gt;    4:2  [n = 1893, L =  331 Mb] ■■■■■■■</span>
<span class="co">#&gt;    3:2  [n = 1625, L =  357 Mb] ■■■■■■</span>
<span class="co">#&gt;    2:1  [n = 1563, L =  420 Mb] ■■■■■■  { <span style="color: #BBBB00;">TTN</span> }</span>
<span class="co">#&gt;    3:0  [n =  312, L =  137 Mb] ■</span>
<span class="co">#&gt;    2:0  [n =   81, L =   39 Mb]   { <span style="color: #BBBB00;">TP53</span> }</span>
<span class="co">#&gt;   16:2  [n =    4, L =    0 Mb] </span>
<span class="co">#&gt;   25:2  [n =    2, L =    1 Mb] </span>
<span class="co">#&gt;    3:1  [n =    2, L =    1 Mb] </span>
<span class="co">#&gt;  106:1  [n =    1, L =    0 Mb]</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample Purity: 89% ~ Ploidy: 4.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> There are 3 annotated driver(s) mapped to clonal CNAs.</span>
<span class="co">#&gt;          chr      from        to ref alt  DP NV       VAF driver_label is_driver</span>
<span class="co">#&gt;         chr2 179431633 179431634   C   T 117 77 0.6581197          TTN      TRUE</span>
<span class="co">#&gt;        chr16  67646006  67646007   C   T 120 54 0.4500000         CTCF      TRUE</span>
<span class="co">#&gt;        chr17   7577106   7577107   G   C  84 78 0.9285714         TP53      TRUE</span></code></pre></div>
<div class="section level2">
<h2 id="plotting-copy-number-segments">Plotting copy number segments<a class="anchor" aria-label="anchor" href="#plotting-copy-number-segments"></a>
</h2>
<div class="section level3">
<h3 id="segments-plots">Segments plots<a class="anchor" aria-label="anchor" href="#segments-plots"></a>
</h3>
<p>CNAqc can plot genome-wide segments showing major and minor allele
counts as red and blue bars. Bottom circles annotate breakpoints; by
default this plot has limited y-axis and breakpoints for segments
outside the plot (e.g. very high amplifications) are in black. Areas in
the genome that are mapped to the most prevalent karyotype are shadowed
by default.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Default plot</span>
<span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-3-1.png" width="912"></p>
<blockquote>
<p>Note: the colour scheme in CNAqc is fixed for certain segments. The
colour scheme is adopted also for other information, e.g., to report the
segment where a certain driver is mapped.</p>
</blockquote>
<p>An alternative circular layout plot is available.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Circular layout</span>
<span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span>, circular <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-4-1.png" width="384"></p>
<div class="section level4">
<h4 id="customising-segments-plot">Customising segments plot<a class="anchor" aria-label="anchor" href="#customising-segments-plot"></a>
</h4>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Shadow other segments</span>
<span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span>, highlight <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"2:1"</span>, <span class="st">"2:0"</span>, <span class="st">"2:2"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Annotate different karyotypes"</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-5-1.png" width="912"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># Subset the genome</span>
<span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span>, chromosomes <span class="op">=</span> <span class="st">'chr17'</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Chromosome 17 zoom"</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-5-2.png" width="912"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span>, chromosomes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'chr17'</span>, <span class="st">'chr13'</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Chromosome 17 and 13 zoom"</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-5-3.png" width="912"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># Total copy number</span>
<span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span>, cn <span class="op">=</span> <span class="st">'total'</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Total copy number"</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-5-4.png" width="912"></p>
</div>
<div class="section level4">
<h4 id="subclonal-cnas">Subclonal CNAs<a class="anchor" aria-label="anchor" href="#subclonal-cnas"></a>
</h4>
<p>We use the <a href="https://dcc.icgc.org/pcawg" class="external-link">PCAWG</a> object
released with the package</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">CNAqc</span><span class="fu">::</span><span class="va"><a href="../reference/example_PCAWG.html">example_PCAWG</a></span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">293736</span> mutations in <span style="color: #00BB00;">667</span> segments (<span style="color: #00BB00;">654</span> clonal, <span style="color: #00BB00;">13</span> subclonal). Genom</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Clonal CNAs</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   2:1  [n = 88422, L =   692 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;   3:2  [n = 58384, L =   417 Mb] ■■■■■■■■■■■■■■■■■■  { <span style="color: #BBBB00;">BRAF</span> }</span>
<span class="co">#&gt;   3:1  [n = 48704, L =   380 Mb] ■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;   3:0  [n = 26622, L =   360 Mb] ■■■■■■■■  { <span style="color: #BBBB00;">CDKN2A</span> }</span>
<span class="co">#&gt;   2:2  [n = 25290, L =   253 Mb] ■■■■■■■■</span>
<span class="co">#&gt;   3:3  [n = 16790, L =   115 Mb] ■■■■■</span>
<span class="co">#&gt;   2:0  [n =  5374, L =    67 Mb] ■■</span>
<span class="co">#&gt;   4:0  [n =  1752, L =    22 Mb] ■  { <span style="color: #BBBB00;">TP53</span> }</span>
<span class="co">#&gt;   4:2  [n =  1441, L =    11 Mb] </span>
<span class="co">#&gt;   1:1  [n =   855, L =     9 Mb]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Subclonal CNAs (showing up to 10 segments)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  chr11@55700000  [n =  10468, L =  78.75 Mb]   2:1 (0.21)   2:2 (0.79) ■■■■■■■■■■</span>
<span class="co">#&gt;  chr11@17365005  [n =   5389, L =  31.55 Mb]   2:1 (0.21)   2:2 (0.79) ■■■■■</span>
<span class="co">#&gt;   chr11@5372292  [n =   1014, L =  11.99 Mb]   2:1 (0.21)   2:2 (0.79) </span>
<span class="co">#&gt;    chr11@202253  [n =    610, L =   5.17 Mb]   2:1 (0.22)   2:2 (0.78) </span>
<span class="co">#&gt;  chr11@48918601  [n =    542, L =   2.68 Mb]   2:1 (0.25)   2:2 (0.75) </span>
<span class="co">#&gt;   chr6@82432583  [n =    301, L =   1.81 Mb]   2:1 (0.19)   2:2 (0.81) </span>
<span class="co">#&gt;  chr11@51600000  [n =    290, L =    4.1 Mb]   2:1 (0.26)   2:2 (0.74) </span>
<span class="co">#&gt;   chr6@81896364  [n =     69, L =   0.54 Mb]   2:1 (0.19)   2:2 (0.81) </span>
<span class="co">#&gt;   chr6@93956180  [n =     41, L =   0.11 Mb]    2:1 (0.2)    2:2 (0.8) </span>
<span class="co">#&gt;   chr8@42633277  [n =     13, L =   0.26 Mb]   2:1 (0.28)   2:2 (0.72)</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 3.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> There are 3 annotated driver(s) mapped to clonal CNAs.</span>
<span class="co">#&gt;          chr      from        to ref alt DP NV       VAF driver_label is_driver</span>
<span class="co">#&gt;        chr17   7577082   7577082   C   T 78 70 0.8974359         TP53      TRUE</span>
<span class="co">#&gt;         chr7 140453136 140453136   A   T 95 54 0.5684211         BRAF      TRUE</span>
<span class="co">#&gt;         chr9  21971120  21971120   G   A 23 14 0.6086957       CDKN2A      TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="color: #000000; background-color: #00BB00; font-weight: bold;"> PASS </span><span style="font-weight: bold;"> Peaks QC closest: </span><span style="color: #00BB00; font-weight: bold;">100%</span><span style="font-weight: bold;">, </span><span style="color: #00BB00; font-weight: bold;">λ = -0.0104</span><span style="font-weight: bold;">. Purity correction: -1%.</span> <span style="color: #00BBBB;">─────────</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">1:1</span> ~ n = 855   (0.7%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> -0.006</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">2:0</span> ~ n = 5374  (  4%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> -0.003  <span style="color: #000000; background-color: #00BB00;"> PASS </span> -0.002</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">2:1</span> ~ n = 88422 ( 74%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> -0.006  <span style="color: #000000; background-color: #00BB00;"> PASS </span> -0.006</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">2:2</span> ~ n = 25290 ( 21%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> -0.053  <span style="color: #000000; background-color: #00BB00;"> PASS </span> -0.004</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">General peak QC (</span><span style="color: #00BB00; font-weight: bold;">154052</span><span style="font-weight: bold;"> mutations): </span><span style="color: #000000; background-color: #00BB00; font-weight: bold;"> PASS </span><span style="font-weight: bold;"> 21 </span><span style="color: #BBBBBB; background-color: #BB0000; font-weight: bold;"> FAIL </span><span style="font-weight: bold;"> 3 - epsilon = 0.03.</span> <span style="color: #00BBBB;">────</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">3:0</span> ~ n = 26622 ( 17%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 0</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">3:1</span> ~ n = 48704 ( 32%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 0</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">3:2</span> ~ n = 58384 ( 38%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 0</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">3:3</span> ~ n = 16790 ( 11%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 0</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">4:0</span> ~ n = 1752  (  1%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 1</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">4:2</span> ~ n = 1441  (  1%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 1</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">4:3</span> ~ n = 359   (  0%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Subclonal peaks QC (</span><span style="color: #00BB00; font-weight: bold;">7</span><span style="font-weight: bold;"> segments, initial state </span><span style="color: #00BB00; font-weight: bold;">2:1</span><span style="font-weight: bold;">): </span><span style="font-weight: bold; text-decoration: underline;">linear</span><span style="font-weight: bold;"> </span><span style="color: #00BB00; font-weight: bold;">3</span><span style="font-weight: bold;"> </span><span style="font-weight: bold; text-decoration: underline;">branching</span><span style="font-weight: bold;"> </span><span style="color: #BB0000; font-weight: bold;">0</span><span style="font-weight: bold;"> </span><span style="font-weight: bold; text-decoration: underline;">eith</span></span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── <span style="color: #000000; background-color: #00BB00;"> PASS </span> Linear models</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">chr11@17365005</span> ~ (31.6Mb, n = 5389) <span style="color: #BBBB00;">2:1 (21) + 2:2 (79)</span> : A1A2B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">chr11@51600000</span> ~ (4.1Mb, n = 290) <span style="color: #BBBB00;">2:1 (26) + 2:2 (74)</span> : A1A2B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #BBBB00;">75</span>]</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;"> chr11@5372292</span> ~ (12Mb, n = 1014) <span style="color: #BBBB00;">2:1 (21) + 2:2 (79)</span> : A1A2B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── <span style="color: #BBBBBB; background-color: #BBBB00;"> UNKNOWN </span> Either branching or linear models</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">  chr11@202253</span> ~ (5.2Mb, n = 610) <span style="color: #BBBB00;">2:1 (22) + 2:2 (78)</span> : A1A2B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1A2B1 -&gt; A1A2B1 | A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1A2B1 -&gt; A1A2B1B2 -&gt; A2B1B2 [<span style="color: #00BB00;">100</span>]</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">chr11@48918601</span> ~ (2.7Mb, n = 542) <span style="color: #BBBB00;">2:1 (25) + 2:2 (75)</span> : A1A2B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1A2B1 -&gt; A1A2B1 | A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1A2B1 -&gt; A1A2B1B2 -&gt; A2B1B2 [<span style="color: #00BB00;">100</span>]</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">chr11@55700000</span> ~ (78.8Mb, n = 10468) <span style="color: #BBBB00;">2:1 (21) + 2:2 (79)</span> : A1A2B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1A2B1 -&gt; A1A2B1 | A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1A2B1 -&gt; A1A2B1B2 -&gt; A2B1B2 [<span style="color: #00BB00;">100</span>]</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;"> chr6@82432583</span> ~ (1.8Mb, n = 301) <span style="color: #BBBB00;">2:1 (19) + 2:2 (81)</span> : A1A2B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1A2B1 -&gt; A1A2B1 | A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1A2B1 -&gt; A1A2B1B2 -&gt; A2B1B2 [<span style="color: #00BB00;">100</span>]</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> Cancer Cell Fraction (CCF) data available for karyotypes:1:0, 1:1, 2:0, 2:1, and 2:2.</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #BBBBBB; background-color: #00BB00;"> PASS </span> CCF via <span style="font-weight: bold;">ENTROPY</span>.</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #BBBBBB; background-color: #00BB00;"> PASS </span> CCF via <span style="font-weight: bold;">ENTROPY</span>.</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #BBBBBB; background-color: #00BB00;"> PASS </span> CCF via <span style="font-weight: bold;">ENTROPY</span>.</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #BBBBBB; background-color: #00BB00;"> PASS </span> CCF via <span style="font-weight: bold;">ENTROPY</span>.</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #BBBBBB; background-color: #00BB00;"> PASS </span> CCF via <span style="font-weight: bold;">ENTROPY</span>.</span></code></pre></div>
<p>Subclonal segments of each subclone have different colours compared
to red/blue clonal segments.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="fu">CNAqc</span><span class="fu">::</span><span class="va"><a href="../reference/example_PCAWG.html">example_PCAWG</a></span><span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-7-1.png" width="912"></p>
<blockquote>
<p>Note: the CCF of each subclone is not reported in this plot.</p>
</blockquote>
</div>
</div>
<div class="section level3">
<h3 id="segments-summaries">Segments summaries<a class="anchor" aria-label="anchor" href="#segments-summaries"></a>
</h3>
<p>Based on parameter <code>type</code>, function
<code>plot_karyotypes</code> reports:</p>
<ul>
<li>
<code>type = 'percentage'</code>, the proportion of genome covered
by each segment (default);</li>
<li>
<code>type = 'number'</code>, the counts of segments per
karyotype.</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span>
  <span class="fu"><a href="../reference/plot_karyotypes.html">plot_karyotypes</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_karyotypes.html">plot_karyotypes</a></span><span class="op">(</span><span class="va">x</span>, type <span class="op">=</span> <span class="st">'number'</span><span class="op">)</span>,
  common.legend <span class="op">=</span> <span class="cn">TRUE</span>,
  legend <span class="op">=</span> <span class="st">'bottom'</span>
<span class="op">)</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-8-1.png" width="432"></p>
<p>We can also plot the segments length distribution, which is used to
detect fragmentation patterns.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_segment_size_distribution.html">plot_segment_size_distribution</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-9-1.png" width="432"></p>
</div>
</div>
<div class="section level2">
<h2 id="plotting-mutations">Plotting mutations<a class="anchor" aria-label="anchor" href="#plotting-mutations"></a>
</h2>
<div class="section level3">
<h3 id="data-histograms">Data histograms<a class="anchor" aria-label="anchor" href="#data-histograms"></a>
</h3>
<p>Read counts information can be plot using
<code>plot_data_histogram</code> as histograms.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span>
  <span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x</span>, which <span class="op">=</span> <span class="st">'VAF'</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x</span>, which <span class="op">=</span> <span class="st">'DP'</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x</span>, which <span class="op">=</span> <span class="st">'NV'</span><span class="op">)</span>,
  ncol <span class="op">=</span> <span class="fl">3</span>,
  nrow <span class="op">=</span> <span class="fl">1</span>
  <span class="op">)</span>
<span class="co">#&gt; Warning: Removed 8 rows containing missing values (geom_bar).</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-10-1.png" width="960"></p>
<blockquote>
<p>Note: if you compute CCFs you can use
<code>plot_data_histogram(x, which = 'CCF')</code> as well.</p>
</blockquote>
</div>
<div class="section level3">
<h3 id="segment-specific-vafs">Segment-specific VAFs<a class="anchor" aria-label="anchor" href="#segment-specific-vafs"></a>
</h3>
<p>VAFs per segment, split by chromosome, can help identify miscalled
segments (i.e., for segments with the same major/minor allele copies the
VAF distributions should be similar)</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/inspect_segment.html">inspect_segment</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span>
<span class="co">#&gt; Warning: Removed 30 rows containing missing values (geom_bar).</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-11-1.png" width="1152"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># With facet from a data histogram</span>
<span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x</span>, which <span class="op">=</span> <span class="st">'VAF'</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="va">karyotype</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span>
<span class="co">#&gt; Warning: Removed 56 rows containing missing values (geom_bar).</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-12-1.png" width="2880"></p>
</div>
<div class="section level3">
<h3 id="genome-wide-mutation-data">Genome-wide mutation data<a class="anchor" aria-label="anchor" href="#genome-wide-mutation-data"></a>
</h3>
<p>Genome-wide plots that follow the layout of
<code>plot_segments</code> are avaiable to view the genome-wide
distributions of the number of mutations, their VAF and depth. VAF and
depth-plotting functions (scatterplots) can subset the input data to
speed up rendering and reduce the size of output files - by default
<span class="math inline">\(5000\)</span> points are shown Counts plot,
instead, bin the genome locations by one megabase (<span class="math inline">\(10^6\)</span> bases).</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># All genome, all data - counts of mutations per megabase</span>
<span class="fu"><a href="../reference/plot_gw_counts.html">plot_gw_counts</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-13-1.png" width="768"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># An example effect of downsampling the data</span>
<span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span>
  <span class="fu"><a href="../reference/plot_gw_depth.html">plot_gw_depth</a></span><span class="op">(</span><span class="va">x</span>, N <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_gw_depth.html">plot_gw_depth</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_gw_depth.html">plot_gw_depth</a></span><span class="op">(</span><span class="va">x</span>, N <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span>,
  ncol <span class="op">=</span> <span class="fl">1</span>
<span class="op">)</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_segment).</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_rect).</span>
<span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_hline).</span>
<span class="co">#&gt; Removed 1 rows containing missing values (geom_hline).</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_segment).</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_rect).</span>
<span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_hline).</span>
<span class="co">#&gt; Removed 1 rows containing missing values (geom_hline).</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_segment).</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_rect).</span>
<span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_hline).</span>
<span class="co">#&gt; Removed 1 rows containing missing values (geom_hline).</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-14-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="combined-plots">Combined plots<a class="anchor" aria-label="anchor" href="#combined-plots"></a>
</h3>
<p>The default S3 <code>plot</code> function layouts a panel with
segments in bottom, and all other genome-wide plots on top. This
function aligns plots on the vertical axis, and stretch their relative
height to obtain a nice layout.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The actual plot function</span>
<span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span>
  <span class="fu"><a href="../reference/plot_gw_counts.html">plot_gw_counts</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_gw_vaf.html">plot_gw_vaf</a></span><span class="op">(</span><span class="va">x</span>, N <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_gw_depth.html">plot_gw_depth</a></span><span class="op">(</span><span class="va">x</span>, N <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,
  align <span class="op">=</span> <span class="st">'v'</span>,
  nrow <span class="op">=</span> <span class="fl">4</span>,
  rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">.15</span>, <span class="fl">.15</span>, <span class="fl">.15</span>, <span class="fl">.8</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_segment).</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_rect).</span>
<span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_hline).</span>
<span class="co">#&gt; Removed 1 rows containing missing values (geom_hline).</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-15-1.png" width="912"></p>
</div>
</div>
<div class="section level2">
<h2 id="cohort-plots">Cohort plots<a class="anchor" aria-label="anchor" href="#cohort-plots"></a>
</h2>
<p>One can have multiple CNAqc objects - i.e., work with a cohort of
objects - when working with</p>
<ul>
<li>a group of distinct patients;</li>
<li>multiple samples from the same patient;</li>
<li>multiple calls from the same sample;</li>
</ul>
<p>Assuming all input objects have the same reference, cohort
cohort-level plots are possible.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Object 1</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span>
  mutations <span class="op">=</span> <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">mutations</span>, 
  cna <span class="op">=</span> <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">cna</span>,
  purity <span class="op">=</span> <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">purity</span>,
  ref <span class="op">=</span> <span class="st">'hg19'</span>
  <span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">CNAqc - CNA Quality Check</span> <span style="color: #00BBBB;">───────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Drivers are annotated, but 'gene' column is missing, using mutation location.</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">12963</span> somatic mutations: <span style="color: #00BB00;">12963</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">0</span> indels.</span>
<span class="co">#&gt; <span style="color: #BBBB00;">!</span> CNAs have no CCF, assuming clonal CNAs (CCF = 1).</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">267</span> segments: <span style="color: #00BB00;">267</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">12963</span> mutations mapped to clonal CNAs.</span>

<span class="co"># Object 1 with some modifications</span>
<span class="va">y</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span>
  mutations <span class="op">=</span> <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">mutations</span>, 
  cna <span class="op">=</span> <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">cna</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>
      Major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr3"</span>, <span class="st">"chr4"</span>, <span class="st">"chr1"</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">Major</span><span class="op">)</span>,
      minor <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr3"</span>, <span class="st">"chr4"</span>, <span class="st">"chr1"</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">minor</span><span class="op">)</span>
      <span class="op">)</span>,
  purity <span class="op">=</span> <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">purity</span>,
  ref <span class="op">=</span> <span class="st">'hg19'</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">CNAqc - CNA Quality Check</span> <span style="color: #00BBBB;">───────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Drivers are annotated, but 'gene' column is missing, using mutation location.</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">12963</span> somatic mutations: <span style="color: #00BB00;">12963</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">0</span> indels.</span>
<span class="co">#&gt; <span style="color: #BBBB00;">!</span> CNAs have no CCF, assuming clonal CNAs (CCF = 1).</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">267</span> segments: <span style="color: #00BB00;">267</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">12963</span> mutations mapped to clonal CNAs.</span>

<span class="co"># PCAWG sample</span>
<span class="va">z</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="va"><a href="../reference/example_PCAWG.html">example_PCAWG</a></span>

<span class="co"># Inputs need to be wrapped in a named list</span>
<span class="va">inputs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>`Original` <span class="op">=</span> <span class="va">x</span>, `Copy` <span class="op">=</span> <span class="va">x</span>, `Faked_diploid` <span class="op">=</span> <span class="va">y</span>, `PCAWG` <span class="op">=</span> <span class="va">z</span><span class="op">)</span></code></pre></div>
<div class="section level3">
<h3 id="copy-number-segments">Copy number segments<a class="anchor" aria-label="anchor" href="#copy-number-segments"></a>
</h3>
<p>The frequency of deletions and amplifications relative to diploid are
shown by using a certain binning (step <span class="math inline">\(\delta\)</span>) of the tumour genome. Here by
deletion we define anything with 0 copies of the minor allele, and by
amplification anything with <span class="math inline">\(\geq 3\)</span>
total copies (sum of the minor and major alleles).</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Comparative CNA plot - default delta</span>
<span class="fu"><a href="../reference/plot_multisample_CNA.html">plot_multisample_CNA</a></span><span class="op">(</span><span class="va">inputs</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Breaking input segments at <span style="color: #00BB00;">100 Kb</span> resolution</span>
<span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span>
<span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'easypar'</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-17-1.png" width="768"></p>
<p>The binning <span class="math inline">\(\delta\)</span> of the tumour
genome can be changed to get larger/ shorter segments</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Comparative CNA plot - 10 Mb delta</span>
<span class="fu"><a href="../reference/plot_multisample_CNA.html">plot_multisample_CNA</a></span><span class="op">(</span><span class="va">inputs</span>, delta <span class="op">=</span> <span class="fl">1e7</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Breaking input segments at <span style="color: #00BB00;">10000 Kb</span> resolution</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-18-1.png" width="768"></p>
<p>A circular layout is also possible, with no binning or transformation
of the data.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Comparative CNA plot</span>
<span class="fu"><a href="../reference/plot_multisample_CNA.html">plot_multisample_CNA</a></span><span class="op">(</span><span class="va">inputs</span>, layout <span class="op">=</span> <span class="st">'circular'</span><span class="op">)</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-19-1.png" width="576"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.caravagnalab.org" class="external-link">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
