<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>2. Plotting data • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="2. Plotting data">
<meta property="og:description" content="CNAqc">
<meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a2_Plots_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>2. Plotting data</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqcvignettes/a2_Plots.Rmd"><code>vignettes/a2_Plots.Rmd</code></a></small>
      <div class="hidden name"><code>a2_Plots.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/CNAqc">CNAqc</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span>
<span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'BMix'</span>
<span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading</span>
<span class="co">#&gt; 'BMix'</span>
<span class="co">#&gt; ✓ Loading BMix, 'Binomial and Beta-Binomial univariate mixtures'. Support : &lt;https://caravagnalab.github.io/BMix/&gt;</span>
<span class="co">#&gt; Warning: package 'tidyr' was built under R version 4.0.5</span>
<span class="co">#&gt; Warning: package 'readr' was built under R version 4.0.5</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.5</span>
<span class="co">#&gt; Warning: package 'crayon' was built under R version 4.0.5</span>
<span class="co">#&gt; ✓ Loading CNAqc, 'Copy Number Alteration quality check'. Support : &lt;https://caravagn.github.io/CNAqc/&gt;</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></code></pre></div>
<div id="plotting-copy-number-segments" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-copy-number-segments" class="anchor"></a>Plotting copy number segments</h1>
<p>All plotting functions use <code>ggplot</code> and multi-panel figures assembled with either <code>ggpubr</code> and <code>cowplot</code>. We work with the template dataset.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Dataset available with the package </span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">'example_dataset_CNAqc'</span>, package <span class="op">=</span> <span class="st">'CNAqc'</span><span class="op">)</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span>
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">snvs</span>, 
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">cna</span>,
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">purity</span>,
  ref <span class="op">=</span> <span class="st">'hg19'</span><span class="op">)</span>
<span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span>
<span class="co">#&gt; ℹ Using reference genome coordinates for: hg19.</span>
<span class="co">#&gt; ℹ Drivers are annotated, but 'gene' column is missing, using mutation location.</span>
<span class="co">#&gt; ✓ Fortified calls for 12963 somatic mutations: 12963 SNVs (100%) and 0 indels.</span>
<span class="co">#&gt; ! CNAs have no CCF, assuming clonal CNAs (CCF = 1).</span>
<span class="co">#&gt; ✓ Fortified CNAs for 267 segments: 267 clonal and 0 subclonal.</span>
<span class="co">#&gt; ✓ 12963 mutations mapped to clonal CNAs.</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  12963 mutations in 267 segments (267 clonal, 0 subclonal). Genome</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Clonal CNAs</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;    2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { CTCF }</span>
<span class="co">#&gt;    4:2  [n = 1893, L =  331 Mb] ■■■■■■■</span>
<span class="co">#&gt;    3:2  [n = 1625, L =  357 Mb] ■■■■■■</span>
<span class="co">#&gt;    2:1  [n = 1563, L =  420 Mb] ■■■■■■  { TTN }</span>
<span class="co">#&gt;    3:0  [n =  312, L =  137 Mb] ■</span>
<span class="co">#&gt;    2:0  [n =   81, L =   39 Mb]   { TP53 }</span>
<span class="co">#&gt;   16:2  [n =    4, L =    0 Mb] </span>
<span class="co">#&gt;   25:2  [n =    2, L =    1 Mb] </span>
<span class="co">#&gt;    3:1  [n =    2, L =    1 Mb] </span>
<span class="co">#&gt;  106:1  [n =    1, L =    0 Mb]</span>
<span class="co">#&gt; ℹ Sample Purity: 89% ~ Ploidy: 4.</span>
<span class="co">#&gt; ℹ There are 3 annotated driver(s) mapped to clonal CNAs.</span>
<span class="co">#&gt;          chr      from        to ref alt  DP NV       VAF driver_label is_driver</span>
<span class="co">#&gt;         chr2 179431633 179431634   C   T 117 77 0.6581197          TTN      TRUE</span>
<span class="co">#&gt;        chr16  67646006  67646007   C   T 120 54 0.4500000         CTCF      TRUE</span>
<span class="co">#&gt;        chr17   7577106   7577107   G   C  84 78 0.9285714         TP53      TRUE</span></code></pre></div>
<p>Function <code>plot_segments</code> plots all the genome, annotating in the caption summmary statistics. By default, it plots the major and minor alleles for each segment as red and blue bars. In the bottom, each circle annotates a breakpoint in the segments; grays are breakpoints of samples that are shown in the plot. Since by default this plot has limited y-axis, breakpoints of segments outside the plot (e.g. very high amplifications) are in black.</p>
<p>Since <code>x</code> has drivers annotated, these are also plot with a colour that reflects their karyotype. Areas in the genome that are mapped to the most prevalent karyotype (here, <code>"2:2"</code>), are shadowed by default.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Default plot</span>
<span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-3-1.png" width="912"></p>
<p>An alternative circular layout plot is available (default parameters).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Circular layout</span>
<span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span>, circular <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-4-1.png" width="384"></p>
<div id="custom-segments-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#custom-segments-plot" class="anchor"></a>Custom segments plot</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Shadow other karyotypes</span>
<span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span>, highlight <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"2:1"</span>, <span class="st">"2:0"</span>, <span class="st">"2:2"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Annotate different karyotypes"</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-5-1.png" width="912"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># Subset the genome</span>
<span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span>, chromosomes <span class="op">=</span> <span class="st">'chr17'</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Chromosome 17 zoom"</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-5-2.png" width="912"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span>, chromosomes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'chr17'</span>, <span class="st">'chr13'</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Chromosome 17 and 13 zoom"</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-5-3.png" width="912"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># Total copy number</span>
<span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span>, cn <span class="op">=</span> <span class="st">'total'</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Total copy number"</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-5-4.png" width="912"></p>
</div>
<div id="subclonal-cnas" class="section level2">
<h2 class="hasAnchor">
<a href="#subclonal-cnas" class="anchor"></a>Subclonal CNAs</h2>
<p>To show subclonal copy numbers we need to use the PCAWG object released with the package.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu">CNAqc</span><span class="fu">::</span><span class="va">example_PCAWG</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  12400 mutations in 552 segments (486 clonal, 66 subclonal). Genome</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Clonal CNAs</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   3:0  [n = 1772, L =  414 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { CDK12 }</span>
<span class="co">#&gt;   2:2  [n = 1713, L =  277 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;   2:1  [n = 1479, L =  289 Mb] ■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;   3:1  [n = 1444, L =  221 Mb] ■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;   2:0  [n =  949, L =  320 Mb] ■■■■■■■■■■■■■■  { TP53 }</span>
<span class="co">#&gt;   3:2  [n =  491, L =   54 Mb] ■■■■■■■</span>
<span class="co">#&gt;   4:1  [n =  445, L =   49 Mb] ■■■■■■■</span>
<span class="co">#&gt;   1:1  [n =  320, L =  104 Mb] ■■■■■</span>
<span class="co">#&gt;   4:2  [n =  310, L =   34 Mb] ■■■■■</span>
<span class="co">#&gt;   5:0  [n =  110, L =   20 Mb] ■■</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Subclonal CNAs (showing up to 10 segments)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  chr13@62300908  [n =  231, L =  24.27 Mb]   2:0 (0.08)   2:1 (0.92) ■■■■■■■■■■</span>
<span class="co">#&gt;  chr1@176138962  [n =  216, L =  25.93 Mb]   2:1 (0.26)   2:2 (0.74) ■■■■■■■■■</span>
<span class="co">#&gt;  chr1@220232950  [n =  137, L =  27.02 Mb]   2:1 (0.35)   2:2 (0.65) ■■■■■</span>
<span class="co">#&gt;   chr1@91503321  [n =  133, L =  14.64 Mb]   2:1 (0.25)   2:2 (0.75) ■■■■■</span>
<span class="co">#&gt;  chr18@30575049  [n =  129, L =  29.95 Mb]   1:0 (0.32)   1:1 (0.68) ■■■■■</span>
<span class="co">#&gt;   chr9@16500688  [n =  116, L =  13.48 Mb]   2:1 (0.87)   2:2 (0.13) ■■■■■</span>
<span class="co">#&gt;   chr1@48302609  [n =  111, L =  21.73 Mb]   2:1 (0.26)   2:2 (0.74) ■■■■</span>
<span class="co">#&gt;     chr8@159286  [n =  110, L =  27.39 Mb]   1:0 (0.06)   1:1 (0.94) ■■■■</span>
<span class="co">#&gt;  chr13@96297358  [n =  110, L =  17.85 Mb]   2:0 (0.09)   2:1 (0.91) ■■■■</span>
<span class="co">#&gt;   chr10@7016917  [n =   95, L =  11.51 Mb]   2:1 (0.67)   2:2 (0.33) ■■■■</span>
<span class="co">#&gt; ℹ Sample Purity: 88.5% ~ Ploidy: 3.</span>
<span class="co">#&gt; ℹ There are 2 annotated driver(s) mapped to clonal CNAs.</span>
<span class="co">#&gt;          chr     from       to ref alt DP NV       VAF driver_label is_driver</span>
<span class="co">#&gt;        chr17  7577058  7577058   C   A 26 23 0.8846154         TP53      TRUE</span>
<span class="co">#&gt;        chr17 37680973 37680973   C   T 41 39 0.9512195        CDK12      TRUE</span></code></pre></div>
<p>Note that the segments of each subclone have different colours; the CCF is not reported in this plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="fu">CNAqc</span><span class="fu">::</span><span class="va">example_PCAWG</span><span class="op">)</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-7-1.png" width="912"></p>
</div>
<div id="comparing-multiple-calls" class="section level2">
<h2 class="hasAnchor">
<a href="#comparing-multiple-calls" class="anchor"></a>Comparing multiple calls</h2>
<p>If you want to compare multiple CNAqc objects you can use <code>plot_multisample_CNA</code>. Here we show it by combining twice <code>x</code> with some changes in its calls, and putting together another sample.</p>
<p>This type of plot shows only clonal CNAs.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">y</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span>
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">snvs</span>, 
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">cna</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>
      Major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"chr3"</span>, <span class="st">"chr4"</span>, <span class="st">"chr1"</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">Major</span><span class="op">)</span>,
      minor <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"chr3"</span>, <span class="st">"chr4"</span>, <span class="st">"chr1"</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">minor</span><span class="op">)</span>
      <span class="op">)</span>,
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">purity</span>,
  ref <span class="op">=</span> <span class="st">'hg19'</span><span class="op">)</span>
<span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span>
<span class="co">#&gt; ℹ Using reference genome coordinates for: hg19.</span>
<span class="co">#&gt; ℹ Drivers are annotated, but 'gene' column is missing, using mutation location.</span>
<span class="co">#&gt; ✓ Fortified calls for 12963 somatic mutations: 12963 SNVs (100%) and 0 indels.</span>
<span class="co">#&gt; ! CNAs have no CCF, assuming clonal CNAs (CCF = 1).</span>
<span class="co">#&gt; ✓ Fortified CNAs for 267 segments: 267 clonal and 0 subclonal.</span>
<span class="co">#&gt; ✓ 12963 mutations mapped to clonal CNAs.</span>

<span class="co"># Comparative CNA plot</span>
<span class="fu"><a href="../reference/plot_multisample_CNA.html">plot_multisample_CNA</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>`Original` <span class="op">=</span> <span class="va">x</span>, `Copy` <span class="op">=</span> <span class="va">x</span>, `Faked_diploid` <span class="op">=</span> <span class="va">y</span>, `PCAWG` <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="va">example_PCAWG</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-8-1.png" width="576"></p>
</div>
<div id="plotting-segments-summaries" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-segments-summaries" class="anchor"></a>Plotting segments summaries</h2>
<p><code>plot_karyotypes</code> gives two visualizations, based on its parameter <code>type</code>:</p>
<ul>
<li>
<code>type = 'percentage'</code>, reporting the proportion of genome covered by each karyotypes (default);</li>
<li>
<code>type = 'number'</code>, reporting the counts of segments per karyotype.</li>
</ul>
<p>The package uses a pre-set color schema to colour simple CNAs.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span><span class="op">(</span>
  <span class="fu"><a href="../reference/plot_karyotypes.html">plot_karyotypes</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_karyotypes.html">plot_karyotypes</a></span><span class="op">(</span><span class="va">x</span>, type <span class="op">=</span> <span class="st">'number'</span><span class="op">)</span>,
  common.legend <span class="op">=</span> <span class="cn">TRUE</span>,
  legend <span class="op">=</span> <span class="st">'bottom'</span>
<span class="op">)</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-9-1.png" width="432"></p>
<p>You can also plot the segments’ length distribution, which is used to detect fragmentation patterns in the segments.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_segment_size_distribution.html">plot_segment_size_distribution</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-10-1.png" width="432"></p>
</div>
</div>
<div id="plotting-mutations" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-mutations" class="anchor"></a>Plotting mutations</h1>
<p>You have a function to plot all your data histograms</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span><span class="op">(</span>
  <span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x</span>, which <span class="op">=</span> <span class="st">'VAF'</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x</span>, which <span class="op">=</span> <span class="st">'DP'</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x</span>, which <span class="op">=</span> <span class="st">'NV'</span><span class="op">)</span>,
  ncol <span class="op">=</span> <span class="fl">3</span>,
  nrow <span class="op">=</span> <span class="fl">1</span>
  <span class="op">)</span>
<span class="co">#&gt; Warning: Removed 8 rows containing missing values (geom_bar).</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-11-1.png" width="960"></p>
<p>If you <a href="https://caravagn.github.io/CNAqc/articles/ccf_computation.html">compute Cancer Cell Fractions (CCF) values</a>, you can use <code>plot_data_histogram(x, which = 'CCF')</code> as well.</p>
<div id="segment-specific-vaf-distribution" class="section level2">
<h2 class="hasAnchor">
<a href="#segment-specific-vaf-distribution" class="anchor"></a>Segment-specific VAF distribution</h2>
<p>You can inspect the raw VAF distribution per segment, split by chromosome. Each colour represents a distinct segment; we want can use this plot to check that, for every karyotype and chromosome, the VAF distributions are similar.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Default parameters to filter segments are annotatetd in the plot</span>
<span class="fu"><a href="../reference/inspect_segment.html">inspect_segment</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span>
<span class="co">#&gt; Warning: Removed 30 rows containing missing values (geom_bar).</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-12-1.png" width="1152"></p>
<p>You can facet your data histograms and obtain a similar layout</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x</span>, which <span class="op">=</span> <span class="st">'VAF'</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_grid</span><span class="op">(</span><span class="va">karyotype</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span>
<span class="co">#&gt; Warning: Removed 56 rows containing missing values (geom_bar).</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-13-1.png" width="2880"></p>
</div>
<div id="genome-wide-mutation-data" class="section level2">
<h2 class="hasAnchor">
<a href="#genome-wide-mutation-data" class="anchor"></a>Genome-wide mutation data</h2>
<p>Genome-wide plots that follow the layout of <code>plot_segments</code> are avaiable to view the genome-wide distributions of the number of mutations, their VAF and depth. VAF and depth-plotting functions (scatterplots) can subset the input data to speed up rendering and reduce the size of output files - by default <code>N = 5000</code> points are randomly sampled. Counts plot, instead, bin the genome locations by <em>one megabase</em> (<span class="math inline">\(10^6\)</span> bases).</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># All genome, all data - counts of mutations per megabase</span>
<span class="fu"><a href="../reference/plot_gw_counts.html">plot_gw_counts</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-14-1.png" width="768"></p>
<p>An example effect of downsampling the data is the following</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Different subsamples</span>
<span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span><span class="op">(</span>
  <span class="fu"><a href="../reference/plot_gw_depth.html">plot_gw_depth</a></span><span class="op">(</span><span class="va">x</span>, N <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_gw_depth.html">plot_gw_depth</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_gw_depth.html">plot_gw_depth</a></span><span class="op">(</span><span class="va">x</span>, N <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span>,
  ncol <span class="op">=</span> <span class="fl">1</span>
<span class="op">)</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; = "none")` instead.</span>
<span class="co">#&gt; `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; = "none")` instead.</span>
<span class="co">#&gt; `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; = "none")` instead.</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_rect).</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_segment).</span>
<span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_hline).</span>
<span class="co">#&gt; Removed 1 rows containing missing values (geom_hline).</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_rect).</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_segment).</span>
<span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_hline).</span>
<span class="co">#&gt; Removed 1 rows containing missing values (geom_hline).</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_rect).</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_segment).</span>
<span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_hline).</span>
<span class="co">#&gt; Removed 1 rows containing missing values (geom_hline).</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-15-1.png" width="768"></p>
</div>
</div>
<div id="plotting-all-data" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-all-data" class="anchor"></a>Plotting all data</h1>
<p>A one-summary plot can be easily assembled combining the above functions and functions from <code>cowplot</code> or other ggplot-manipulation packages.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Layout a panel with segments in bottom, and all other genome-wide plots on top.</span>
<span class="co"># Via cowplot we can align plots on the vertical axis, and stretch their relative</span>
<span class="co"># height to obtain a nicer layout.</span>
<span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span>
  <span class="fu"><a href="../reference/plot_gw_counts.html">plot_gw_counts</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_gw_vaf.html">plot_gw_vaf</a></span><span class="op">(</span><span class="va">x</span>, N <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_gw_depth.html">plot_gw_depth</a></span><span class="op">(</span><span class="va">x</span>, N <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,
  align <span class="op">=</span> <span class="st">'v'</span>, 
  nrow <span class="op">=</span> <span class="fl">4</span>, 
  rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.15</span>, <span class="fl">.15</span>, <span class="fl">.15</span>, <span class="fl">.8</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; = "none")` instead.</span>
<span class="co">#&gt; `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; = "none")` instead.</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_rect).</span>
<span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_segment).</span>
<span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_hline).</span>
<span class="co">#&gt; Removed 1 rows containing missing values (geom_hline).</span></code></pre></div>
<p><img src="a2_Plots_files/figure-html/unnamed-chunk-16-1.png" width="912"></p>
<p>This is actually done by the default S3 <code>plot</code> function, which includes both segments and data histograms</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.caravagnalab.org">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
