<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>3. QC analysis via peaks detection • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="3. QC analysis via peaks detection">
<meta property="og:description" content="CNAqc">
<meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a3_peaks_detection_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>3. QC analysis via peaks detection</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqcvignettes/a3_peaks_detection.Rmd"><code>vignettes/a3_peaks_detection.Rmd</code></a></small>
      <div class="hidden name"><code>a3_peaks_detection.Rmd</code></div>

    </div>

    
    
<div id="peak-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#peak-analysis" class="anchor"></a>Peak analysis</h1>
<p><code>CNAqc</code> can perform quality control (QC) of somatic mutations, allele-specific CNAs and tumour purity. 3 distinct procedures are available, all called by function <code>analyze_peaks</code>. The idea is as follows: a CNA segment with <span class="math inline">\(m\)</span>/ <span class="math inline">\(M\)</span> copies of the minor/ major alleles is consistent with mutation data if VAFs peak at certain positions we can compute. CNAqc computes expected peaks, and compares them to the one it detects from data. The theory works for both clonal and subclonal segments, and a purity error parameter (<code>purity_error</code>) determines the maximum error we can tolerate before we fail a sample.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/CNAqc">CNAqc</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span>
<span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'BMix'</span>
<span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading</span>
<span class="co">#&gt; 'BMix'</span>
<span class="co">#&gt; ✓ Loading BMix, 'Binomial and Beta-Binomial univariate mixtures'. Support : &lt;https://caravagnalab.github.io/BMix/&gt;</span>
<span class="co">#&gt; Warning: package 'tidyr' was built under R version 4.0.5</span>
<span class="co">#&gt; Warning: package 'readr' was built under R version 4.0.5</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.5</span>
<span class="co">#&gt; Warning: package 'crayon' was built under R version 4.0.5</span>
<span class="co">#&gt; ✓ Loading CNAqc, 'Copy Number Alteration quality check'. Support : &lt;https://caravagn.github.io/CNAqc/&gt;</span>

<span class="co"># We work with the template dataset. </span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">'example_dataset_CNAqc'</span>, package <span class="op">=</span> <span class="st">'CNAqc'</span><span class="op">)</span>

<span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span>
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">snvs</span>, 
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">cna</span>,
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">purity</span>,
  ref <span class="op">=</span> <span class="st">'hg19'</span><span class="op">)</span>
<span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span>
<span class="co">#&gt; ℹ Using reference genome coordinates for: hg19.</span>
<span class="co">#&gt; ℹ Drivers are annotated, but 'gene' column is missing, using mutation location.</span>
<span class="co">#&gt; ✓ Fortified calls for 12963 somatic mutations: 12963 SNVs (100%) and 0 indels.</span>
<span class="co">#&gt; ! CNAs have no CCF, assuming clonal CNAs (CCF = 1).</span>
<span class="co">#&gt; ✓ Fortified CNAs for 267 segments: 267 clonal and 0 subclonal.</span>
<span class="co">#&gt; ✓ 12963 mutations mapped to clonal CNAs.</span>

<span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="va">example_PCAWG</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  293736 mutations in 667 segments (654 clonal, 13 subclonal). Genom</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Clonal CNAs</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   2:1  [n = 88422, L =   692 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;   3:2  [n = 58384, L =   417 Mb] ■■■■■■■■■■■■■■■■■■  { BRAF }</span>
<span class="co">#&gt;   3:1  [n = 48704, L =   380 Mb] ■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;   3:0  [n = 26622, L =   360 Mb] ■■■■■■■■  { CDKN2A }</span>
<span class="co">#&gt;   2:2  [n = 25290, L =   253 Mb] ■■■■■■■■</span>
<span class="co">#&gt;   3:3  [n = 16790, L =   115 Mb] ■■■■■</span>
<span class="co">#&gt;   2:0  [n =  5374, L =    67 Mb] ■■</span>
<span class="co">#&gt;   4:0  [n =  1752, L =    22 Mb] ■  { TP53 }</span>
<span class="co">#&gt;   4:2  [n =  1441, L =    11 Mb] </span>
<span class="co">#&gt;   1:1  [n =   855, L =     9 Mb]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Subclonal CNAs (showing up to 10 segments)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  chr11@55700000  [n =  10468, L =  78.75 Mb]   2:1 (0.21)   2:2 (0.79) ■■■■■■■■■■</span>
<span class="co">#&gt;  chr11@17365005  [n =   5389, L =  31.55 Mb]   2:1 (0.21)   2:2 (0.79) ■■■■■</span>
<span class="co">#&gt;   chr11@5372292  [n =   1014, L =  11.99 Mb]   2:1 (0.21)   2:2 (0.79) </span>
<span class="co">#&gt;    chr11@202253  [n =    610, L =   5.17 Mb]   2:1 (0.22)   2:2 (0.78) </span>
<span class="co">#&gt;  chr11@48918601  [n =    542, L =   2.68 Mb]   2:1 (0.25)   2:2 (0.75) </span>
<span class="co">#&gt;   chr6@82432583  [n =    301, L =   1.81 Mb]   2:1 (0.19)   2:2 (0.81) </span>
<span class="co">#&gt;  chr11@51600000  [n =    290, L =    4.1 Mb]   2:1 (0.26)   2:2 (0.74) </span>
<span class="co">#&gt;   chr6@81896364  [n =     69, L =   0.54 Mb]   2:1 (0.19)   2:2 (0.81) </span>
<span class="co">#&gt;   chr6@93956180  [n =     41, L =   0.11 Mb]    2:1 (0.2)    2:2 (0.8) </span>
<span class="co">#&gt;   chr8@42633277  [n =     13, L =   0.26 Mb]   2:1 (0.28)   2:2 (0.72)</span>
<span class="co">#&gt; ℹ Sample Purity: 73.4% ~ Ploidy: 3.</span>
<span class="co">#&gt; ℹ There are 3 annotated driver(s) mapped to clonal CNAs.</span>
<span class="co">#&gt;          chr      from        to ref alt DP NV       VAF driver_label is_driver</span>
<span class="co">#&gt;        chr17   7577082   7577082   C   T 78 70 0.8974359         TP53      TRUE</span>
<span class="co">#&gt;         chr7 140453136 140453136   A   T 95 54 0.5684211         BRAF      TRUE</span>
<span class="co">#&gt;         chr9  21971120  21971120   G   A 23 14 0.6086957       CDKN2A      TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ──  PASS  Peaks QC closest: 100%, λ = -0.0104. Purity correction: -1%. ─────────</span>
<span class="co">#&gt; ℹ 1:1 ~ n = 855   (  1%) →  PASS  -0.006</span>
<span class="co">#&gt; ℹ 2:0 ~ n = 5374  (  4%) →  PASS  -0.003   PASS  -0.002</span>
<span class="co">#&gt; ℹ 2:1 ~ n = 88422 ( 74%) →  PASS  -0.006   PASS  -0.006</span>
<span class="co">#&gt; ℹ 2:2 ~ n = 25290 ( 21%) →  PASS  -0.053   PASS  -0.004</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── General peak QC (154052 mutations):  PASS  21  FAIL  3 - epsilon = 0.03. ────</span>
<span class="co">#&gt; ℹ 3:0 ~ n = 26622 ( 17%) →  PASS  3  FAIL  0</span>
<span class="co">#&gt; ℹ 3:1 ~ n = 48704 ( 32%) →  PASS  3  FAIL  0</span>
<span class="co">#&gt; ℹ 3:2 ~ n = 58384 ( 38%) →  PASS  3  FAIL  0</span>
<span class="co">#&gt; ℹ 3:3 ~ n = 16790 ( 11%) →  PASS  3  FAIL  0</span>
<span class="co">#&gt; ℹ 4:0 ~ n = 1752  (  1%) →  PASS  3  FAIL  1</span>
<span class="co">#&gt; ℹ 4:2 ~ n = 1441  (  1%) →  PASS  3  FAIL  1</span>
<span class="co">#&gt; ℹ 4:3 ~ n = 359   (  0%) →  PASS  3  FAIL  1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Subclonal peaks QC (7 segments, initial state 2:1): linear 3 branching 0 eith</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ──  PASS  Linear models</span>
<span class="co">#&gt; ℹ chr11@17365005 ~ (31.6Mb, n = 5389) 2:1 (21) + 2:2 (79) : A1A2B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [100]</span>
<span class="co">#&gt; ℹ chr11@51600000 ~ (4.1Mb, n = 290) 2:1 (26) + 2:2 (74) : A1A2B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [75]</span>
<span class="co">#&gt; ℹ  chr11@5372292 ~ (12Mb, n = 1014) 2:1 (21) + 2:2 (79) : A1A2B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [100]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ──  UNKNOWN  Either branching or linear models</span>
<span class="co">#&gt; ℹ   chr11@202253 ~ (5.2Mb, n = 610) 2:1 (22) + 2:2 (78) : A1A2B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [100]; A1A2B1 -&gt; A1A2B1 | A1A2B1B2 [100]; A1A2B1 -&gt; A1A2B1B2 -&gt; A2B1B2 [100]</span>
<span class="co">#&gt; ℹ chr11@48918601 ~ (2.7Mb, n = 542) 2:1 (25) + 2:2 (75) : A1A2B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [100]; A1A2B1 -&gt; A1A2B1 | A1A2B1B2 [100]; A1A2B1 -&gt; A1A2B1B2 -&gt; A2B1B2 [100]</span>
<span class="co">#&gt; ℹ chr11@55700000 ~ (78.8Mb, n = 10468) 2:1 (21) + 2:2 (79) : A1A2B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [100]; A1A2B1 -&gt; A1A2B1 | A1A2B1B2 [100]; A1A2B1 -&gt; A1A2B1B2 -&gt; A2B1B2 [100]</span>
<span class="co">#&gt; ℹ  chr6@82432583 ~ (1.8Mb, n = 301) 2:1 (19) + 2:2 (81) : A1A2B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [100]; A1A2B1 -&gt; A1A2B1 | A1A2B1B2 [100]; A1A2B1 -&gt; A1A2B1B2 -&gt; A2B1B2 [100]</span>
<span class="co">#&gt; ✓ Cancer Cell Fraction (CCF) data available for karyotypes:1:0, 1:1, 2:0, 2:1, and 2:2.</span>
<span class="co">#&gt; ✓  PASS  CCF via ENTROPY.</span>
<span class="co">#&gt; ✓  PASS  CCF via ENTROPY.</span>
<span class="co">#&gt; ✓  PASS  CCF via ENTROPY.</span>
<span class="co">#&gt; ✓  PASS  CCF via ENTROPY.</span>
<span class="co">#&gt; ✓  PASS  CCF via ENTROPY.</span></code></pre></div>
<p>We run it with default parameters, otherwise one can select only certain segments to fit or change purity error tolerance etc.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/analyze_peaks.html">analyze_peaks</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Peak analysis: simple CNAs ──────────────────────────────────────────────────</span>
<span class="co">#&gt; ℹ Analysing 120065 mutations mapping to karyotype(s) 2:1, 2:2, 2:0, 1:1, and 1:0.</span>
<span class="co">#&gt; ℹ Mixed type peak detection for karyotype 1:0 (124 mutations)</span>
<span class="co">#&gt; ℹ Mixed type peak detection for karyotype 1:1 (855 mutations)</span>
<span class="co">#&gt; ℹ Mixed type peak detection for karyotype 2:0 (5374 mutations)</span>
<span class="co">#&gt; ℹ Mixed type peak detection for karyotype 2:1 (88422 mutations)</span>
<span class="co">#&gt; ℹ Mixed type peak detection for karyotype 2:2 (25290 mutations)</span>
<span class="co">#&gt; # A tibble: 8 × 16</span>
<span class="co">#&gt; # Rowwise: </span>
<span class="co">#&gt;   mutation_multiplicity karyotype  peak delta_vaf     x     y counts_per_bin</span>
<span class="co">#&gt;                   &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;int&gt;</span>
<span class="co">#&gt; 1                     1 2:1       0.268   0.0134  0.264 4.14            3138</span>
<span class="co">#&gt; 2                     2 2:1       0.537   0.0268  0.54  3               2729</span>
<span class="co">#&gt; 3                     1 2:2       0.212   0.00831 0.21  0.93             198</span>
<span class="co">#&gt; 4                     2 2:2       0.423   0.0166  0.423 6.85            1732</span>
<span class="co">#&gt; 5                     1 2:0       0.367   0.025   0.359 0.864             50</span>
<span class="co">#&gt; 6                     2 2:0       0.734   0.05    0.735 5.43             265</span>
<span class="co">#&gt; 7                     1 1:1       0.367   0.025   0.366 2.42              15</span>
<span class="co">#&gt; 8                     1 1:0       0.580   0.0624  0.570 4.16               7</span>
<span class="co">#&gt; # … with 9 more variables: discarded &lt;lgl&gt;, from &lt;chr&gt;, offset_VAF &lt;dbl&gt;,</span>
<span class="co">#&gt; #   offset &lt;dbl&gt;, weight &lt;dbl&gt;, epsilon &lt;dbl&gt;, VAF_tolerance &lt;dbl&gt;,</span>
<span class="co">#&gt; #   matched &lt;lgl&gt;, QC &lt;chr&gt;</span>
<span class="co">#&gt; ✓ Peak detection PASS with r = 0.0111491116404993 - maximum purity error ε = 0.05.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Peak analysis: complex CNAs ─────────────────────────────────────────────────</span>
<span class="co">#&gt; ℹ Karyotypes 3:2, 3:1, 3:0, 3:3, 4:0, 4:2, 4:3, 5:2, 5:3, and 6:3 with &gt;100 mutation(s). Using epsilon = 0.05.</span>
<span class="co">#&gt; # A tibble: 10 × 5</span>
<span class="co">#&gt; # Groups:   karyotype, matched [10]</span>
<span class="co">#&gt;    karyotype n       mismatched matched  prop</span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;table&gt;      &lt;int&gt;   &lt;int&gt; &lt;dbl&gt;</span>
<span class="co">#&gt;  1 4:2        1441            1       3 0.75 </span>
<span class="co">#&gt;  2 3:2       58384            1       2 0.667</span>
<span class="co">#&gt;  3 5:2         132            2       3 0.6  </span>
<span class="co">#&gt;  4 4:0        1752            2       2 0.5  </span>
<span class="co">#&gt;  5 5:3         132            3       2 0.4  </span>
<span class="co">#&gt;  6 3:0       26622            2       1 0.333</span>
<span class="co">#&gt;  7 3:1       48704            2       1 0.333</span>
<span class="co">#&gt;  8 3:3       16790            2       1 0.333</span>
<span class="co">#&gt;  9 6:3         114            4       2 0.333</span>
<span class="co">#&gt; 10 4:3         359            3       1 0.25</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Peak analysis: subclonal CNAs ───────────────────────────────────────────────</span>
<span class="co">#&gt; → Computing evolution models for subclonal CNAs - starting from 1:1</span>
<span class="co">#&gt; # A tibble: 11 × 6</span>
<span class="co">#&gt;    segment_id     model_id                   model      prop size         clones</span>
<span class="co">#&gt;    &lt;chr&gt;          &lt;chr&gt;                      &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt;        &lt;chr&gt; </span>
<span class="co">#&gt;  1 chr6@82432583  A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear     1    (1.8Mb, n =… 2:1 0…</span>
<span class="co">#&gt;  2 chr6@82432583  A1B1 -&gt; A1A2B1 | A1A2B1B2  branching  1    (1.8Mb, n =… 2:1 0…</span>
<span class="co">#&gt;  3 chr6@82432583  A1B1 -&gt; A1A2B1B2 -&gt; A2B1B2 linear     1    (1.8Mb, n =… 2:1 0…</span>
<span class="co">#&gt;  4 chr11@202253   A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear     1    (5.2Mb, n =… 2:1 0…</span>
<span class="co">#&gt;  5 chr11@5372292  A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear     1    (12Mb, n = … 2:1 0…</span>
<span class="co">#&gt;  6 chr11@17365005 A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear     1    (31.6Mb, n … 2:1 0…</span>
<span class="co">#&gt;  7 chr11@48918601 A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear     1    (2.7Mb, n =… 2:1 0…</span>
<span class="co">#&gt;  8 chr11@48918601 A1B1 -&gt; A1A2B1 | A1A2B1B2  branching  1    (2.7Mb, n =… 2:1 0…</span>
<span class="co">#&gt;  9 chr11@48918601 A1B1 -&gt; A1A2B1B2 -&gt; A2B1B2 linear     1    (2.7Mb, n =… 2:1 0…</span>
<span class="co">#&gt; 10 chr11@51600000 A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear     0.75 (4.1Mb, n =… 2:1 0…</span>
<span class="co">#&gt; 11 chr11@55700000 A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear     1    (78.8Mb, n … 2:1 0…</span></code></pre></div>
<p>An S3 methods print to screen all QC results based on peak-detection.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  293736 mutations in 667 segments (654 clonal, 13 subclonal). Genom</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Clonal CNAs</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   2:1  [n = 88422, L =   692 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;   3:2  [n = 58384, L =   417 Mb] ■■■■■■■■■■■■■■■■■■  { BRAF }</span>
<span class="co">#&gt;   3:1  [n = 48704, L =   380 Mb] ■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;   3:0  [n = 26622, L =   360 Mb] ■■■■■■■■  { CDKN2A }</span>
<span class="co">#&gt;   2:2  [n = 25290, L =   253 Mb] ■■■■■■■■</span>
<span class="co">#&gt;   3:3  [n = 16790, L =   115 Mb] ■■■■■</span>
<span class="co">#&gt;   2:0  [n =  5374, L =    67 Mb] ■■</span>
<span class="co">#&gt;   4:0  [n =  1752, L =    22 Mb] ■  { TP53 }</span>
<span class="co">#&gt;   4:2  [n =  1441, L =    11 Mb] </span>
<span class="co">#&gt;   1:1  [n =   855, L =     9 Mb]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Subclonal CNAs (showing up to 10 segments)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  chr11@55700000  [n =  10468, L =  78.75 Mb]   2:1 (0.21)   2:2 (0.79) ■■■■■■■■■■</span>
<span class="co">#&gt;  chr11@17365005  [n =   5389, L =  31.55 Mb]   2:1 (0.21)   2:2 (0.79) ■■■■■</span>
<span class="co">#&gt;   chr11@5372292  [n =   1014, L =  11.99 Mb]   2:1 (0.21)   2:2 (0.79) </span>
<span class="co">#&gt;    chr11@202253  [n =    610, L =   5.17 Mb]   2:1 (0.22)   2:2 (0.78) </span>
<span class="co">#&gt;  chr11@48918601  [n =    542, L =   2.68 Mb]   2:1 (0.25)   2:2 (0.75) </span>
<span class="co">#&gt;   chr6@82432583  [n =    301, L =   1.81 Mb]   2:1 (0.19)   2:2 (0.81) </span>
<span class="co">#&gt;  chr11@51600000  [n =    290, L =    4.1 Mb]   2:1 (0.26)   2:2 (0.74) </span>
<span class="co">#&gt;   chr6@81896364  [n =     69, L =   0.54 Mb]   2:1 (0.19)   2:2 (0.81) </span>
<span class="co">#&gt;   chr6@93956180  [n =     41, L =   0.11 Mb]    2:1 (0.2)    2:2 (0.8) </span>
<span class="co">#&gt;   chr8@42633277  [n =     13, L =   0.26 Mb]   2:1 (0.28)   2:2 (0.72)</span>
<span class="co">#&gt; ℹ Sample Purity: 73.4% ~ Ploidy: 3.</span>
<span class="co">#&gt; ℹ There are 3 annotated driver(s) mapped to clonal CNAs.</span>
<span class="co">#&gt;          chr      from        to ref alt DP NV       VAF driver_label is_driver</span>
<span class="co">#&gt;        chr17   7577082   7577082   C   T 78 70 0.8974359         TP53      TRUE</span>
<span class="co">#&gt;         chr7 140453136 140453136   A   T 95 54 0.5684211         BRAF      TRUE</span>
<span class="co">#&gt;         chr9  21971120  21971120   G   A 23 14 0.6086957       CDKN2A      TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ──  PASS  Peaks QC closest: 199%, λ = 0.0111. Purity correction: 1%. ───────────</span>
<span class="co">#&gt; ℹ 2:1 ~ n = 88422 ( 74%) →  PASS  0.017    PASS  -0.006</span>
<span class="co">#&gt; ℹ 2:2 ~ n = 25290 ( 21%) →  PASS  0.01     PASS  0.002</span>
<span class="co">#&gt; ℹ 2:0 ~ n = 5374  (  4%) →  PASS  0.015    PASS  -0.001</span>
<span class="co">#&gt; ℹ 1:1 ~ n = 855   (  1%) →  PASS  0.002</span>
<span class="co">#&gt; ℹ 1:0 ~ n = 124   (  0%) →  PASS  0.008</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── General peak QC (154430 mutations):  PASS  18  FAIL  22 - epsilon = 0.05. ───</span>
<span class="co">#&gt; ℹ 4:2 ~ n = 1441  (  1%) →  PASS  3  FAIL  1</span>
<span class="co">#&gt; ℹ 3:2 ~ n = 58384 ( 38%) →  PASS  2  FAIL  1</span>
<span class="co">#&gt; ℹ 5:2 ~ n = 132   (  0%) →  PASS  3  FAIL  2</span>
<span class="co">#&gt; ℹ 4:0 ~ n = 1752  (  1%) →  PASS  2  FAIL  2</span>
<span class="co">#&gt; ℹ 5:3 ~ n = 132   (  0%) →  PASS  2  FAIL  3</span>
<span class="co">#&gt; ℹ 3:0 ~ n = 26622 ( 17%) →  PASS  1  FAIL  2</span>
<span class="co">#&gt; ℹ 3:1 ~ n = 48704 ( 32%) →  PASS  1  FAIL  2</span>
<span class="co">#&gt; ℹ 3:3 ~ n = 16790 ( 11%) →  PASS  1  FAIL  2</span>
<span class="co">#&gt; ℹ 6:3 ~ n = 114   (  0%) →  PASS  2  FAIL  4</span>
<span class="co">#&gt; ℹ 4:3 ~ n = 359   (  0%) →  PASS  1  FAIL  3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Subclonal peaks QC (7 segments, initial state 2:1): linear 5 branching 0 eith</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ──  PASS  Linear models</span>
<span class="co">#&gt; ℹ chr11@17365005 ~ (31.6Mb, n = 5389) 2:1 (21) + 2:2 (79) : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [100]</span>
<span class="co">#&gt; ℹ   chr11@202253 ~ (5.2Mb, n = 610) 2:1 (22) + 2:2 (78) : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [100]</span>
<span class="co">#&gt; ℹ chr11@51600000 ~ (4.1Mb, n = 290) 2:1 (26) + 2:2 (74) : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [75]</span>
<span class="co">#&gt; ℹ  chr11@5372292 ~ (12Mb, n = 1014) 2:1 (21) + 2:2 (79) : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [100]</span>
<span class="co">#&gt; ℹ chr11@55700000 ~ (78.8Mb, n = 10468) 2:1 (21) + 2:2 (79) : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [100]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ──  UNKNOWN  Either branching or linear models</span>
<span class="co">#&gt; ℹ chr11@48918601 ~ (2.7Mb, n = 542) 2:1 (25) + 2:2 (75) : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [100]; A1B1 -&gt; A1A2B1 | A1A2B1B2 [100]; A1B1 -&gt; A1A2B1B2 -&gt; A2B1B2 [100]</span>
<span class="co">#&gt; ℹ  chr6@82432583 ~ (1.8Mb, n = 301) 2:1 (19) + 2:2 (81) : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [100]; A1B1 -&gt; A1A2B1 | A1A2B1B2 [100]; A1B1 -&gt; A1A2B1B2 -&gt; A2B1B2 [100]</span>
<span class="co">#&gt; ✓ Cancer Cell Fraction (CCF) data available for karyotypes:1:0, 1:1, 2:0, 2:1, and 2:2.</span>
<span class="co">#&gt; ✓  PASS  CCF via ENTROPY.</span>
<span class="co">#&gt; ✓  PASS  CCF via ENTROPY.</span>
<span class="co">#&gt; ✓  PASS  CCF via ENTROPY.</span>
<span class="co">#&gt; ✓  PASS  CCF via ENTROPY.</span>
<span class="co">#&gt; ✓  PASS  CCF via ENTROPY.</span></code></pre></div>
<div id="simple-clonal-cnas" class="section level2">
<h2 class="hasAnchor">
<a href="#simple-clonal-cnas" class="anchor"></a>Simple clonal CNAs</h2>
<p>This QC is the most important as it can be used to detect adjustments required to better fit purity or ploidy of the sample (to re-run the copy number caller), or to select among alternative copy number solutions returned from a caller (tetraploid versus triploid).</p>
<p>For these segments the package computes a linear score from the distance between data peaks, and their theoretical expectation. The peaks are determined via:</p>
<ul>
<li><p>a kernel-density (unitary adjustment, Gaussian kernel) used to smooth the data distribution, and the <a href="https://cran.rstudio.com/web/packages/peakPick/index.html">peakPick</a> package to determine peaks in the smoothed density.</p></li>
<li><p>a Binomial mixture model using <a href="https://caravagnalab.github.io/BMix/">Bmix</a>.</p></li>
</ul>
<p>There are two possible algorithmic strategies to match peaks with clonal simple CNAs (configured by parameter <code>matching_strategy</code>): one matches the closest peaks as defined by euclidean distance, the other ranks peaks right to left, and prioritises right-most peaks (see the manual for more details). A call to <code>plot_peaks_analysis</code> with default parameters shows the results from peak detection.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_peaks_analysis.html">plot_peaks_analysis</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<p><img src="a3_peaks_detection_files/figure-html/unnamed-chunk-4-1.png" width="1056"></p>
<p>Plots are ordered by fixed positions (gray panels represent karyotypes that have not been used). Each vertical dashed line is the expected peak, the bandwidth around being the tolerance we use to match peaks (based on <code>purity_error</code>, adjusted for segment ploidy and tumour purity). Each dot is a peak detected from data, with a bandwidth of tolerance (fixed) around it. The colours green/red reflect if a peak is matched or not; the overall QC status for a karyotype is given by the colour of the facet and the box surrounding the whole figure assembly reports the QC status for the sample.</p>
<p>Plots can also be generated without assembly or empty plots (see the parameters of function <code>plot_peaks_analysis</code>).</p>
</div>
<div id="complex-clonal-cnas" class="section level2">
<h2 class="hasAnchor">
<a href="#complex-clonal-cnas" class="anchor"></a>Complex clonal CNAs</h2>
<p>The QC procedure for these CNAs uses only the kernel density estimation to make the computation faster.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_peaks_analysis.html">plot_peaks_analysis</a></span><span class="op">(</span><span class="va">x</span>, what <span class="op">=</span> <span class="st">'general'</span><span class="op">)</span></code></pre></div>
<p><img src="a3_peaks_detection_files/figure-html/unnamed-chunk-5-1.png" width="768"> The plot reports dashed lines for expected peaks (for mutation with multiplicity from 1 to the maximum of the Major/minor allele counts), and dots for each data peak.The colours green/red report the QC status for each peak; with these CNAs no karyotype-level or sample-level scores are produced.</p>
</div>
</div>
<div id="subclonal-cnas" class="section level1">
<h1 class="hasAnchor">
<a href="#subclonal-cnas" class="anchor"></a>Subclonal CNAs</h1>
<p>The QC procedure for these CNAs uses only the kernel density estimation (as for complex CNAs), and considers 2 subclones with distinct mixing proportions. Differently from clonal CNAs, however, here the analysis is carried out at the level of each segment (without pooling segments with the same karyotypes) so to support callers that report segment-specific CCF values (e.g., Battenberg).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_peaks_analysis.html">plot_peaks_analysis</a></span><span class="op">(</span><span class="va">x</span>, what <span class="op">=</span> <span class="st">'subclonal'</span><span class="op">)</span></code></pre></div>
<p><img src="a3_peaks_detection_files/figure-html/unnamed-chunk-6-1.png" width="768"> The visual layout of this plot is the same of complex clonal CNAs; not that the facet reports the distinct evolutionary models that have been generated to QC subclonal CNAs.</p>
<div id="per-chromosome-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#per-chromosome-analysis" class="anchor"></a>Per-chromosome analysis</h2>
</div>
</div>
<div id="summary-results" class="section level1">
<h1 class="hasAnchor">
<a href="#summary-results" class="anchor"></a>Summary results</h1>
<p>For every type of copy number analyzed several information are reported, tables with summary peaks are available inside <code>x$peaks_analysis</code>. More precisely</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Simple clonal CNAs</span>
<span class="va">x</span><span class="op">$</span><span class="va">peaks_analysis</span><span class="op">$</span><span class="va">matches</span>
<span class="co">#&gt; # A tibble: 8 × 16</span>
<span class="co">#&gt; # Rowwise: </span>
<span class="co">#&gt;   mutation_multiplicity karyotype  peak delta_vaf     x     y counts_per_bin</span>
<span class="co">#&gt;                   &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;int&gt;</span>
<span class="co">#&gt; 1                     1 2:1       0.268   0.0134  0.264 4.14            3138</span>
<span class="co">#&gt; 2                     2 2:1       0.537   0.0268  0.54  3               2729</span>
<span class="co">#&gt; 3                     1 2:2       0.212   0.00831 0.21  0.93             198</span>
<span class="co">#&gt; 4                     2 2:2       0.423   0.0166  0.423 6.85            1732</span>
<span class="co">#&gt; 5                     1 2:0       0.367   0.025   0.359 0.864             50</span>
<span class="co">#&gt; 6                     2 2:0       0.734   0.05    0.735 5.43             265</span>
<span class="co">#&gt; 7                     1 1:1       0.367   0.025   0.366 2.42              15</span>
<span class="co">#&gt; 8                     1 1:0       0.580   0.0624  0.570 4.16               7</span>
<span class="co">#&gt; # … with 9 more variables: discarded &lt;lgl&gt;, from &lt;chr&gt;, offset_VAF &lt;dbl&gt;,</span>
<span class="co">#&gt; #   offset &lt;dbl&gt;, weight &lt;dbl&gt;, epsilon &lt;dbl&gt;, VAF_tolerance &lt;dbl&gt;,</span>
<span class="co">#&gt; #   matched &lt;lgl&gt;, QC &lt;chr&gt;</span>

<span class="co"># Complex clonal CNAs</span>
<span class="va">x</span><span class="op">$</span><span class="va">peaks_analysis</span><span class="op">$</span><span class="va">general</span><span class="op">$</span><span class="va">expected_peaks</span>
<span class="co">#&gt; # A tibble: 40 × 9</span>
<span class="co">#&gt;    minor Major ploidy multiplicity purity  peak karyotype matched n      </span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;        &lt;int&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;     &lt;lgl&gt;   &lt;table&gt;</span>
<span class="co">#&gt;  1     2     4      6            1  0.734 0.149 4:2       TRUE    1441   </span>
<span class="co">#&gt;  2     2     4      6            2  0.734 0.297 4:2       TRUE    1441   </span>
<span class="co">#&gt;  3     2     4      6            3  0.734 0.446 4:2       FALSE   1441   </span>
<span class="co">#&gt;  4     2     4      6            4  0.734 0.595 4:2       TRUE    1441   </span>
<span class="co">#&gt;  5     2     5      7            1  0.734 0.129 5:2       FALSE    132   </span>
<span class="co">#&gt;  6     2     5      7            2  0.734 0.259 5:2       TRUE     132   </span>
<span class="co">#&gt;  7     2     5      7            3  0.734 0.388 5:2       TRUE     132   </span>
<span class="co">#&gt;  8     2     5      7            4  0.734 0.518 5:2       FALSE    132   </span>
<span class="co">#&gt;  9     2     5      7            5  0.734 0.647 5:2       TRUE     132   </span>
<span class="co">#&gt; 10     0     4      4            1  0.734 0.212 4:0       TRUE    1752   </span>
<span class="co">#&gt; # … with 30 more rows</span>

<span class="co"># Subclonal CNAs</span>
<span class="va">x</span><span class="op">$</span><span class="va">peaks_analysis</span><span class="op">$</span><span class="va">subclonal</span><span class="op">$</span><span class="va">expected_peaks</span>
<span class="co">#&gt; # A tibble: 91 × 16</span>
<span class="co">#&gt;    mutation karyotype_1 genotype_1 karyotype_2 genotype_2 n.clone_1 n.clone_2</span>
<span class="co">#&gt;    &lt;chr&gt;    &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;       &lt;chr&gt;          &lt;int&gt;     &lt;int&gt;</span>
<span class="co">#&gt;  1 RLZTVJRP 2:1         A1A2B1     &lt;NA&gt;        &lt;NA&gt;               1         0</span>
<span class="co">#&gt;  2 ASLLFJXT &lt;NA&gt;        &lt;NA&gt;       2:2         A1A2B1B2           0         1</span>
<span class="co">#&gt;  3 MZRDFTAI 2:1         A1A2B1     2:2         A1A2B1B2           1         2</span>
<span class="co">#&gt;  4 ABFAVRWO 2:1         A1A2B1     2:2         A1A2B1B2           2         2</span>
<span class="co">#&gt;  5 HPZQTVZZ &lt;NA&gt;        &lt;NA&gt;       2:2         A1A2B1B2           0         1</span>
<span class="co">#&gt;  6 WUZJLYNX 2:1         A1A2B1     2:2         A1A2B1B2           1         1</span>
<span class="co">#&gt;  7 JWBFLKDT 2:1         A1A2B1     2:2         A1A2B1B2           1         2</span>
<span class="co">#&gt;  8 LBIFOUAO 2:1         A1A2B1     2:2         A1A2B1B2           2         2</span>
<span class="co">#&gt;  9 MCIYFTJM &lt;NA&gt;        &lt;NA&gt;       2:1         A2B1B2             0         1</span>
<span class="co">#&gt; 10 ABFXUOTF 2:2         A1A2B1B2   &lt;NA&gt;        &lt;NA&gt;               1         0</span>
<span class="co">#&gt; # … with 81 more rows, and 9 more variables: peak &lt;dbl&gt;,</span>
<span class="co">#&gt; #   genotype_initial &lt;chr&gt;, model &lt;chr&gt;, model_id &lt;chr&gt;, role &lt;chr&gt;,</span>
<span class="co">#&gt; #   segment_id &lt;chr&gt;, matched &lt;lgl&gt;, size &lt;chr&gt;, clones &lt;chr&gt;</span></code></pre></div>
<p>The most helpful table is usually the one for simple clonal CNAs <code>x$peaks_analysis$matches</code>, which reports several information:</p>
<ul>
<li>
<code>mutation_multiplicity</code> and <code>karyotype</code>, which are the number of copies of the mutation, and the segment’s karyotype. Each segment with <code>discarded = FALSE</code> has been analysed;</li>
<li>
<code>peak</code>, <code>x</code>, <code>y</code> report where the expected peak should be found, and where it has been found (<code>x</code> and <code>y</code>);</li>
<li>
<code>offset</code>, <code>weight</code> and <code>score</code> that report the factors of the final <code>score</code>;</li>
<li>
<code>QC</code> that reports if the peak is considered matched.</li>
</ul>
<p>The overall sample-level QC result - <code>"PASS"/"FAIL"</code> - is available in</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span><span class="op">$</span><span class="va">peaks_analysis</span><span class="op">$</span><span class="va">QC</span>
<span class="co">#&gt; [1] "PASS"</span></code></pre></div>
<p>You can visually summarise QC results for all the analyses available in CNAqc (peaks-based and <a href="https://caravagn.github.io/CNAqc/articles/ccf_computation.html">CCF-based</a>).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_qc.html">plot_qc</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<p><img src="a3_peaks_detection_files/figure-html/unnamed-chunk-9-1.png" width="576"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.caravagnalab.org">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
