<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>3. QC analysis via peaks detection • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="3. QC analysis via peaks detection">
<meta property="og:description" content="CNAqc">
<meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a3_peaks_detection_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>3. QC analysis via peaks detection</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqcvignettes/a3_peaks_detection.Rmd"><code>vignettes/a3_peaks_detection.Rmd</code></a></small>
      <div class="hidden name"><code>a3_peaks_detection.Rmd</code></div>

    </div>

    
    
<div id="peak-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#peak-analysis" class="anchor"></a>Peak analysis</h1>
<p><code>CNAqc</code> can perform quality control (QC) of somatic mutations, allele-specific CNAs and tumour purity. 3 distinct procedures are available, all called by function <code>analyze_peaks</code>. The idea is as follows: a CNA segment with <span class="math inline">\(m\)</span>/ <span class="math inline">\(M\)</span> copies of the minor/ major alleles is consistent with mutation data if VAFs peak at certain positions we can compute. CNAqc computes expected peaks, and compares them to the one it detects from data. The theory works for both clonal and subclonal segemtns with some modifications.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/CNAqc">CNAqc</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span>
<span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'BMix'</span>
<span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading</span>
<span class="co">#&gt; 'BMix'</span>
<span class="co">#&gt; ✓ Loading BMix, 'Binomial and Beta-Binomial univariate mixtures'. Support : &lt;https://caravagnalab.github.io/BMix/&gt;</span>
<span class="co">#&gt; Warning: package 'tidyr' was built under R version 4.0.5</span>
<span class="co">#&gt; Warning: package 'readr' was built under R version 4.0.5</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.5</span>
<span class="co">#&gt; Warning: package 'crayon' was built under R version 4.0.5</span>
<span class="co">#&gt; ✓ Loading CNAqc, 'Copy Number Alteration quality check'. Support : &lt;https://caravagn.github.io/CNAqc/&gt;</span>

<span class="co"># We work with the template dataset. </span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">'example_dataset_CNAqc'</span>, package <span class="op">=</span> <span class="st">'CNAqc'</span><span class="op">)</span>

<span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span>
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">snvs</span>, 
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">cna</span>,
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">purity</span>,
  ref <span class="op">=</span> <span class="st">'hg19'</span><span class="op">)</span>
<span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span>
<span class="co">#&gt; ℹ Using reference genome coordinates for: hg19.</span>
<span class="co">#&gt; ℹ Drivers are annotated, but 'gene' column is missing, using mutation location.</span>
<span class="co">#&gt; ✓ Fortified calls for 12963 somatic mutations: 12963 SNVs (100%) and 0 indels.</span>
<span class="co">#&gt; ! CNAs have no CCF, assuming clonal CNAs (CCF = 1).</span>
<span class="co">#&gt; ✓ Fortified CNAs for 267 segments: 267 clonal and 0 subclonal.</span>
<span class="co">#&gt; ✓ 12963 mutations mapped to clonal CNAs.</span>

<span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="va">example_PCAWG</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  12400 mutations in 552 segments (486 clonal, 66 subclonal). Genome</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Clonal CNAs</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   3:0  [n = 1772, L =  414 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { CDK12 }</span>
<span class="co">#&gt;   2:2  [n = 1713, L =  277 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;   2:1  [n = 1479, L =  289 Mb] ■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;   3:1  [n = 1444, L =  221 Mb] ■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;   2:0  [n =  949, L =  320 Mb] ■■■■■■■■■■■■■■  { TP53 }</span>
<span class="co">#&gt;   3:2  [n =  491, L =   54 Mb] ■■■■■■■</span>
<span class="co">#&gt;   4:1  [n =  445, L =   49 Mb] ■■■■■■■</span>
<span class="co">#&gt;   1:1  [n =  320, L =  104 Mb] ■■■■■</span>
<span class="co">#&gt;   4:2  [n =  310, L =   34 Mb] ■■■■■</span>
<span class="co">#&gt;   5:0  [n =  110, L =   20 Mb] ■■</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Subclonal CNAs (showing up to 10 segments)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  chr13@62300908  [n =  231, L =  24.27 Mb]   2:0 (0.08)   2:1 (0.92) ■■■■■■■■■■</span>
<span class="co">#&gt;  chr1@176138962  [n =  216, L =  25.93 Mb]   2:1 (0.26)   2:2 (0.74) ■■■■■■■■■</span>
<span class="co">#&gt;  chr1@220232950  [n =  137, L =  27.02 Mb]   2:1 (0.35)   2:2 (0.65) ■■■■■</span>
<span class="co">#&gt;   chr1@91503321  [n =  133, L =  14.64 Mb]   2:1 (0.25)   2:2 (0.75) ■■■■■</span>
<span class="co">#&gt;  chr18@30575049  [n =  129, L =  29.95 Mb]   1:0 (0.32)   1:1 (0.68) ■■■■■</span>
<span class="co">#&gt;   chr9@16500688  [n =  116, L =  13.48 Mb]   2:1 (0.87)   2:2 (0.13) ■■■■■</span>
<span class="co">#&gt;   chr1@48302609  [n =  111, L =  21.73 Mb]   2:1 (0.26)   2:2 (0.74) ■■■■</span>
<span class="co">#&gt;     chr8@159286  [n =  110, L =  27.39 Mb]   1:0 (0.06)   1:1 (0.94) ■■■■</span>
<span class="co">#&gt;  chr13@96297358  [n =  110, L =  17.85 Mb]   2:0 (0.09)   2:1 (0.91) ■■■■</span>
<span class="co">#&gt;   chr10@7016917  [n =   95, L =  11.51 Mb]   2:1 (0.67)   2:2 (0.33) ■■■■</span>
<span class="co">#&gt; ℹ Sample Purity: 88.5% ~ Ploidy: 3.</span>
<span class="co">#&gt; ℹ There are 2 annotated driver(s) mapped to clonal CNAs.</span>
<span class="co">#&gt;          chr     from       to ref alt DP NV       VAF driver_label is_driver</span>
<span class="co">#&gt;        chr17  7577058  7577058   C   A 26 23 0.8846154         TP53      TRUE</span>
<span class="co">#&gt;        chr17 37680973 37680973   C   T 41 39 0.9512195        CDK12      TRUE</span></code></pre></div>
<p>Run default analysis</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/analyze_peaks.html">analyze_peaks</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Peak analysis: simple CNAs ──────────────────────────────────────────────────</span>
<span class="co">#&gt; ℹ Analysing 4461 mutations mapping to karyotype(s) 2:2, 2:1, 2:0, and 1:1.</span>
<span class="co">#&gt; ℹ Mixed type peak detection for karyotype 1:1 (320 mutations)</span>
<span class="co">#&gt; ℹ Mixed type peak detection for karyotype 2:0 (949 mutations)</span>
<span class="co">#&gt; ℹ Mixed type peak detection for karyotype 2:1 (1479 mutations)</span>
<span class="co">#&gt; ℹ Mixed type peak detection for karyotype 2:2 (1713 mutations)</span>
<span class="co">#&gt; # A tibble: 7 × 16</span>
<span class="co">#&gt; # Rowwise: </span>
<span class="co">#&gt;   mutation_multiplicity karyotype  peak delta_vaf     x     y counts_per_bin</span>
<span class="co">#&gt;                   &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;int&gt;</span>
<span class="co">#&gt; 1                     1 2:2       0.235   0.00704 0.221 1.64               4</span>
<span class="co">#&gt; 2                     2 2:2       0.469   0.0141  0.448 2.28               7</span>
<span class="co">#&gt; 3                     1 2:1       0.307   0.0120  0.35  1.93              18</span>
<span class="co">#&gt; 4                     2 2:1       0.614   0.0240  0.460 1.77              19</span>
<span class="co">#&gt; 5                     1 2:0       0.442   0.025   0.47  0.55              11</span>
<span class="co">#&gt; 6                     2 2:0       0.885   0.05    0.472 0.550             11</span>
<span class="co">#&gt; 7                     1 1:1       0.442   0.025   0.316 2.76              50</span>
<span class="co">#&gt; # … with 9 more variables: discarded &lt;lgl&gt;, from &lt;chr&gt;, offset_VAF &lt;dbl&gt;,</span>
<span class="co">#&gt; #   offset &lt;dbl&gt;, weight &lt;dbl&gt;, epsilon &lt;dbl&gt;, VAF_tolerance &lt;dbl&gt;,</span>
<span class="co">#&gt; #   matched &lt;lgl&gt;, QC &lt;chr&gt;</span>
<span class="co">#&gt; ✓ Peak detection PASS with r = 0.173135550354146 - maximum purity error ε = 0.05.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Peak analysis: complex CNAs ─────────────────────────────────────────────────</span>
<span class="co">#&gt; ℹ Karyotypes 3:0, 3:1, 3:2, 4:1, 4:2, 5:0, and 7:4 with &gt;100 mutation(s). Using epsilon = 0.05.</span>
<span class="co">#&gt; # A tibble: 7 × 5</span>
<span class="co">#&gt; # Groups:   karyotype, matched [7]</span>
<span class="co">#&gt;   karyotype n       mismatched matched  prop</span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;table&gt;      &lt;dbl&gt;   &lt;int&gt; &lt;dbl&gt;</span>
<span class="co">#&gt; 1 3:2        491             0       3 1    </span>
<span class="co">#&gt; 2 4:1        445             1       3 0.75 </span>
<span class="co">#&gt; 3 3:0       1772             1       2 0.667</span>
<span class="co">#&gt; 4 3:1       1444             1       2 0.667</span>
<span class="co">#&gt; 5 5:0        110             2       3 0.6  </span>
<span class="co">#&gt; 6 4:2        310             2       2 0.5  </span>
<span class="co">#&gt; 7 7:4        101             4       3 0.429</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Peak analysis: subclonal CNAs ───────────────────────────────────────────────</span>
<span class="co">#&gt; → Computing evolution models for subclonal CNAs - starting from 1:1</span>
<span class="co">#&gt; # A tibble: 15 × 6</span>
<span class="co">#&gt;    segment_id     model_id                   model      prop size         clones</span>
<span class="co">#&gt;    &lt;chr&gt;          &lt;chr&gt;                      &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt;        &lt;chr&gt; </span>
<span class="co">#&gt;  1 chr1@48302609  A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear    0.5   (21.7Mb, n … 2:1 0…</span>
<span class="co">#&gt;  2 chr1@48302609  A1B1 -&gt; A1A2B1 | A1A2B1B2  branching 0.5   (21.7Mb, n … 2:1 0…</span>
<span class="co">#&gt;  3 chr1@91503321  A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear    0.25  (14.6Mb, n … 2:1 0…</span>
<span class="co">#&gt;  4 chr1@91503321  A1B1 -&gt; A1A2B1 | A1A2B1B2  branching 0.25  (14.6Mb, n … 2:1 0…</span>
<span class="co">#&gt;  5 chr1@176138962 A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear    0.5   (25.9Mb, n … 2:1 0…</span>
<span class="co">#&gt;  6 chr1@176138962 A1B1 -&gt; A1A2B1 | A1A2B1B2  branching 0.5   (25.9Mb, n … 2:1 0…</span>
<span class="co">#&gt;  7 chr1@220232950 A1B1 -&gt; A1A2B1 | A1A2B1B2  branching 0.75  (27Mb, n = … 2:1 0…</span>
<span class="co">#&gt;  8 chr8@159286    A1B1 -&gt; A1B1 -&gt; B1         linear    0.333 (27.4Mb, n … 1:0 0…</span>
<span class="co">#&gt;  9 chr8@159286    A1B1 -&gt; B1 | A1B1          branching 0.333 (27.4Mb, n … 1:0 0…</span>
<span class="co">#&gt; 10 chr9@16500688  A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear    0.25  (13.5Mb, n … 2:1 0…</span>
<span class="co">#&gt; 11 chr9@16500688  A1B1 -&gt; A1A2B1 | A1A2B1B2  branching 0.25  (13.5Mb, n … 2:1 0…</span>
<span class="co">#&gt; 12 chr13@62300908 A1B1 -&gt; A1A2B1 -&gt; A1A2     linear    0.75  (24.3Mb, n … 2:0 0…</span>
<span class="co">#&gt; 13 chr13@96297358 A1B1 -&gt; B1B2 | A1B1B2      branching 0.333 (17.8Mb, n … 2:0 0…</span>
<span class="co">#&gt; 14 chr18@30575049 A1B1 -&gt; A1B1 -&gt; B1         linear    0.333 (30Mb, n = … 1:0 0…</span>
<span class="co">#&gt; 15 chr18@30575049 A1B1 -&gt; B1 | A1B1          branching 0.333 (30Mb, n = … 1:0 0…</span></code></pre></div>
<p>An S3 methods print to screen all QC results based on peak-detection.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  12400 mutations in 552 segments (486 clonal, 66 subclonal). Genome</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Clonal CNAs</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   3:0  [n = 1772, L =  414 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { CDK12 }</span>
<span class="co">#&gt;   2:2  [n = 1713, L =  277 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;   2:1  [n = 1479, L =  289 Mb] ■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;   3:1  [n = 1444, L =  221 Mb] ■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;   2:0  [n =  949, L =  320 Mb] ■■■■■■■■■■■■■■  { TP53 }</span>
<span class="co">#&gt;   3:2  [n =  491, L =   54 Mb] ■■■■■■■</span>
<span class="co">#&gt;   4:1  [n =  445, L =   49 Mb] ■■■■■■■</span>
<span class="co">#&gt;   1:1  [n =  320, L =  104 Mb] ■■■■■</span>
<span class="co">#&gt;   4:2  [n =  310, L =   34 Mb] ■■■■■</span>
<span class="co">#&gt;   5:0  [n =  110, L =   20 Mb] ■■</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Subclonal CNAs (showing up to 10 segments)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  chr13@62300908  [n =  231, L =  24.27 Mb]   2:0 (0.08)   2:1 (0.92) ■■■■■■■■■■</span>
<span class="co">#&gt;  chr1@176138962  [n =  216, L =  25.93 Mb]   2:1 (0.26)   2:2 (0.74) ■■■■■■■■■</span>
<span class="co">#&gt;  chr1@220232950  [n =  137, L =  27.02 Mb]   2:1 (0.35)   2:2 (0.65) ■■■■■</span>
<span class="co">#&gt;   chr1@91503321  [n =  133, L =  14.64 Mb]   2:1 (0.25)   2:2 (0.75) ■■■■■</span>
<span class="co">#&gt;  chr18@30575049  [n =  129, L =  29.95 Mb]   1:0 (0.32)   1:1 (0.68) ■■■■■</span>
<span class="co">#&gt;   chr9@16500688  [n =  116, L =  13.48 Mb]   2:1 (0.87)   2:2 (0.13) ■■■■■</span>
<span class="co">#&gt;   chr1@48302609  [n =  111, L =  21.73 Mb]   2:1 (0.26)   2:2 (0.74) ■■■■</span>
<span class="co">#&gt;     chr8@159286  [n =  110, L =  27.39 Mb]   1:0 (0.06)   1:1 (0.94) ■■■■</span>
<span class="co">#&gt;  chr13@96297358  [n =  110, L =  17.85 Mb]   2:0 (0.09)   2:1 (0.91) ■■■■</span>
<span class="co">#&gt;   chr10@7016917  [n =   95, L =  11.51 Mb]   2:1 (0.67)   2:2 (0.33) ■■■■</span>
<span class="co">#&gt; ℹ Sample Purity: 88.5% ~ Ploidy: 3.</span>
<span class="co">#&gt; ℹ There are 2 annotated driver(s) mapped to clonal CNAs.</span>
<span class="co">#&gt;          chr     from       to ref alt DP NV       VAF driver_label is_driver</span>
<span class="co">#&gt;        chr17  7577058  7577058   C   A 26 23 0.8846154         TP53      TRUE</span>
<span class="co">#&gt;        chr17 37680973 37680973   C   T 41 39 0.9512195        CDK12      TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ──  PASS  Peaks QC closest: 119%, λ = 0.1731. Purity correction: 17%. ──────────</span>
<span class="co">#&gt; ℹ 2:2 ~ n = 1713  ( 38%) →  PASS  0.088    PASS  0.071</span>
<span class="co">#&gt; ℹ 2:1 ~ n = 1479  ( 33%) →  FAIL  -0.205   FAIL  0.259</span>
<span class="co">#&gt; ℹ 2:0 ~ n = 949   ( 21%) →  PASS  -0.055   PASS  0.413</span>
<span class="co">#&gt; ℹ 1:1 ~ n = 320   (  7%) →  FAIL  0.253</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── General peak QC (4673 mutations):  PASS  18  FAIL  11 - epsilon = 0.05. ─────</span>
<span class="co">#&gt; ℹ 3:2 ~ n = 491   ( 11%) →  PASS  3  FAIL  0</span>
<span class="co">#&gt; ℹ 4:1 ~ n = 445   ( 10%) →  PASS  3  FAIL  1</span>
<span class="co">#&gt; ℹ 3:0 ~ n = 1772  ( 38%) →  PASS  2  FAIL  1</span>
<span class="co">#&gt; ℹ 3:1 ~ n = 1444  ( 31%) →  PASS  2  FAIL  1</span>
<span class="co">#&gt; ℹ 5:0 ~ n = 110   (  2%) →  PASS  3  FAIL  2</span>
<span class="co">#&gt; ℹ 4:2 ~ n = 310   (  7%) →  PASS  2  FAIL  2</span>
<span class="co">#&gt; ℹ 7:4 ~ n = 101   (  2%) →  PASS  3  FAIL  4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Subclonal peaks QC (9 segments, initial state 3:0): linear 1 branching 2 eith</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ──  PASS  Linear models</span>
<span class="co">#&gt; ℹ chr13@62300908 ~ (24.3Mb, n = 231) 2:0 (8) + 2:1 (92) : A1B1 -&gt; A1A2B1 -&gt; A1A2 [75]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ──  PASS  Branching models</span>
<span class="co">#&gt; ℹ chr1@220232950 ~ (27Mb, n = 137) 2:1 (35) + 2:2 (65) : A1B1 -&gt; A1A2B1 | A1A2B1B2 [75]</span>
<span class="co">#&gt; ℹ chr13@96297358 ~ (17.8Mb, n = 110) 2:0 (9) + 2:1 (91) : A1B1 -&gt; B1B2 | A1B1B2 [33.3333333333333]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ──  UNKNOWN  Either branching or linear models</span>
<span class="co">#&gt; ℹ chr1@176138962 ~ (25.9Mb, n = 216) 2:1 (26) + 2:2 (74) : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [50]; A1B1 -&gt; A1A2B1 | A1A2B1B2 [50]</span>
<span class="co">#&gt; ℹ  chr1@48302609 ~ (21.7Mb, n = 111) 2:1 (26) + 2:2 (74) : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [50]; A1B1 -&gt; A1A2B1 | A1A2B1B2 [50]</span>
<span class="co">#&gt; ℹ  chr1@91503321 ~ (14.6Mb, n = 133) 2:1 (25) + 2:2 (75) : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [25]; A1B1 -&gt; A1A2B1 | A1A2B1B2 [25]</span>
<span class="co">#&gt; ℹ chr18@30575049 ~ (30Mb, n = 129) 1:0 (32) + 1:1 (68) : A1B1 -&gt; A1B1 -&gt; B1 [33.3333333333333]; A1B1 -&gt; B1 | A1B1 [33.3333333333333]</span>
<span class="co">#&gt; ℹ    chr8@159286 ~ (27.4Mb, n = 110) 1:0 (6) + 1:1 (94) : A1B1 -&gt; A1B1 -&gt; B1 [33.3333333333333]; A1B1 -&gt; B1 | A1B1 [33.3333333333333]</span>
<span class="co">#&gt; ℹ  chr9@16500688 ~ (13.5Mb, n = 116) 2:1 (87) + 2:2 (13) : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [25]; A1B1 -&gt; A1A2B1 | A1A2B1B2 [25]</span></code></pre></div>
<div id="simple-clonal-cnas" class="section level2">
<h2 class="hasAnchor">
<a href="#simple-clonal-cnas" class="anchor"></a>Simple clonal CNAs</h2>
<p>This QC is the most important as it can be used to detect adjustments required to better fit purity or ploidy of the sample (to re-run the copy number caller), or to select among alternative copy number solutions returned from a caller (tetraploid versus triploid).</p>
<p>For these segments the package computes a linear score from the distance between data peaks, and their theoretical expectation. The peaks are determined via:</p>
<ul>
<li><p>a kernel-density (unitary adjustment, Gaussian kernel) used to smooth the data distribution, and the <a href="https://cran.rstudio.com/web/packages/peakPick/index.html">peakPick</a> package to determine peaks in the smoothed density.</p></li>
<li>
<p>a Binomial mixture model using <a href="https://caravagnalab.github.io/BMix/">Bmix</a>.</p>
<p>Peaks below a mnimum density (noise and miscalled segments) are filtered, and the others are matched to a predefined location with some tolerance (e.g., plus/minus <code>0.5 * epsilon</code>, where <code>epsilon = 0.015</code> is the default).</p>
</li>
</ul>
<p>Theis analysis produces a set of scores that account for karyotype size and peaks offset, and that can be combined into a linear model to compute an overall QC metrics, called <code>rho</code>.</p>
<p>A summary table is stored inside the named list <code>peaks_analysis</code>.</p>
</div>
</div>
<div id="plotting-results" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-results" class="anchor"></a>Plotting results</h1>
<p>For every karyotype, a plot with the estimated KDE and the matched peaks is available (assembled <code>ggpubr</code> figure with); gray panels represent karyotypes that have not been used.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_peaks_analysis.html">plot_peaks_analysis</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<p><img src="a3_peaks_detection_files/figure-html/unnamed-chunk-4-1.png" width="1056"></p>
<p>Individual karyotypes plots are available inside the named lists <code>peaks_analysis$plots</code>, named afte each required karyotype.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Tetraploid segments</span>
<span class="va">x</span><span class="op">$</span><span class="va">peaks_analysis</span><span class="op">$</span><span class="va">plots</span><span class="op">$</span><span class="va">`2:2`</span>
<span class="co">#&gt; NULL</span></code></pre></div>
<p>You can integrate plots from QC analysis with data.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span>
  <span class="fu"><a href="../reference/plot_peaks_analysis.html">plot_peaks_analysis</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,
  rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.8</span>, <span class="fl">1</span><span class="op">)</span>,
  nrow <span class="op">=</span> <span class="fl">2</span>
<span class="op">)</span></code></pre></div>
<p><img src="a3_peaks_detection_files/figure-html/unnamed-chunk-6-1.png" width="1056"></p>
</div>
<div id="summary-results-and-quality-check-qc" class="section level1">
<h1 class="hasAnchor">
<a href="#summary-results-and-quality-check-qc" class="anchor"></a>Summary results and Quality Check (QC)</h1>
<p>The summary technical table is available as <code>peaks_analysis$matches</code> inside <code>x</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span><span class="op">$</span><span class="va">peaks_analysis</span><span class="op">$</span><span class="va">matches</span>
<span class="co">#&gt; # A tibble: 7 × 16</span>
<span class="co">#&gt; # Rowwise: </span>
<span class="co">#&gt;   mutation_multiplicity karyotype  peak delta_vaf     x     y counts_per_bin</span>
<span class="co">#&gt;                   &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;int&gt;</span>
<span class="co">#&gt; 1                     1 2:2       0.235   0.00704 0.221 1.64               4</span>
<span class="co">#&gt; 2                     2 2:2       0.469   0.0141  0.448 2.28               7</span>
<span class="co">#&gt; 3                     1 2:1       0.307   0.0120  0.35  1.93              18</span>
<span class="co">#&gt; 4                     2 2:1       0.614   0.0240  0.460 1.77              19</span>
<span class="co">#&gt; 5                     1 2:0       0.442   0.025   0.47  0.55              11</span>
<span class="co">#&gt; 6                     2 2:0       0.885   0.05    0.472 0.550             11</span>
<span class="co">#&gt; 7                     1 1:1       0.442   0.025   0.316 2.76              50</span>
<span class="co">#&gt; # … with 9 more variables: discarded &lt;lgl&gt;, from &lt;chr&gt;, offset_VAF &lt;dbl&gt;,</span>
<span class="co">#&gt; #   offset &lt;dbl&gt;, weight &lt;dbl&gt;, epsilon &lt;dbl&gt;, VAF_tolerance &lt;dbl&gt;,</span>
<span class="co">#&gt; #   matched &lt;lgl&gt;, QC &lt;chr&gt;</span></code></pre></div>
<p>The table reports several information. The following might be of some use after the analysis:</p>
<ul>
<li>
<code>mutation_multiplicity</code> and <code>karyotype</code>, which are the number of copies of the mutation, and the segment’s karyotype. Each segment with <code>discarded = FALSE</code> has been analysed;</li>
<li>
<code>peak</code>, <code>x</code>, <code>y</code> report where the expected peak should be found, and where it has been found (<code>x</code> and <code>y</code>);</li>
<li>
<code>offset</code>, <code>weight</code> and <code>score</code> that report the factors of the final <code>score</code>;</li>
<li>
<code>QC</code> that reports if the peak is considered matched.</li>
</ul>
<p>The overall <code>QC</code> result - <code>"PASS"/"FAIL"</code> - is available in <code>x</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span><span class="op">$</span><span class="va">peaks_analysis</span><span class="op">$</span><span class="va">QC</span>
<span class="co">#&gt; [1] "PASS"</span></code></pre></div>
<p>You can visually summarise the result of the QC analysis, for both peaks and <a href="https://caravagn.github.io/CNAqc/articles/ccf_computation.html">CCF computation</a>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_qc.html">plot_qc</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<p><img src="a3_peaks_detection_files/figure-html/unnamed-chunk-9-1.png" width="288"></p>
<p>Here since CCF have not been computed, the only available QC tests are for peak-detection.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.caravagnalab.org">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
