<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>CNAqc • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="CNAqc">
<meta property="og:description" content="CNAqc">
<meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
    <li>
      <a href="../articles/a9_automatic.html">9. Automatic CNA calling</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>CNAqc</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqcvignettes/CNAqc.Rmd" class="external-link"><code>vignettes/CNAqc.Rmd</code></a></small>
      <div class="hidden name"><code>CNAqc.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p><img src="logo_summary.png" alt="CNAqc Input/Output" width="650"><br><em>CNAqc Input/Output</em></p>
<div class="section level3">
<h3 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h3>
<p>CNAqc requires in input</p>
<ul>
<li>read counts from somatic mutations such as single-nucleotide
variants (SNVs) or insertion-deletions (indels);</li>
<li>allele-specific copy number segments (CNAs) for clonal segments and,
optionally, for subclonal segments;</li>
<li>a tumor purity estimate.</li>
</ul>
<p>CNAqc uses chromosome coordinates the to map mutations to segments.
The conversion of relative to absolute genome coordinates requires to
fix a reference genome build; supported references are GRCh38/hg17 and
hg19/GRCh37, but custom references can also be built.</p>
<p>The tool can elaborate a number of analysis to assess the consistency
among mutations, CNAs and tumour purity (see Articles).</p>
<p>CNAqc can be used to:</p>
<ol style="list-style-type: decimal">
<li>QC concordance across the input mutations, CNAs and purity;</li>
<li>select among alternative tumour segmentations and purity/ ploidy
estimates;</li>
<li>optimize CNA calling with automatic QC procedures leveraging the
Sequenza copy number caller;</li>
<li>estimate CCF values of the input variants, and estimate their
uncertainty;</li>
<li>identify patterns of over-fragmentation of chromosome arms;</li>
<li>smooth and subset segments with various filters;</li>
<li>annotate putative driver mutations among the input variants, using
<a href="https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html" class="external-link">VariantAnnotation</a>.</li>
</ol>
</div>
<div class="section level3">
<h3 id="the-model">The model<a class="anchor" aria-label="anchor" href="#the-model"></a>
</h3>
<p>The following concepts are used to develop CNAqc.</p>
<div class="section level4">
<h4 id="vaf-peaks">VAF peaks<a class="anchor" aria-label="anchor" href="#vaf-peaks"></a>
</h4>
<p><img src="peaks.png" alt="CNAqc Input/Output" width="650"><br><em>Expected VAF peaks for mutations mapped to diploid heterozygous and
triploid clonal CNAs (at purity <span class="math inline">\(\pi=1\)</span>).</em></p>
</div>
<div class="section level4">
<h4 id="clonal-cnas">Clonal CNAs<a class="anchor" aria-label="anchor" href="#clonal-cnas"></a>
</h4>
<p>Consider:</p>
<ul>
<li>mutations present in a percentage <span class="math inline">\(0&lt;c&lt;1\)</span> of tumour cells, sitting on a
segment <span class="math inline">\(nA:nB\)</span>;</li>
<li>tumour purity <span class="math inline">\(\pi\)</span>;</li>
<li>a healthy diploid normal;</li>
</ul>
<p>Since the proportion of all reads from the tumour is <span class="math inline">\(\pi(n_A+n_B)\)</span>, and from the normal is
<span class="math inline">\(2(1-\pi)\)</span>. Then, muations present in
<span class="math inline">\(m\)</span> copies of the tumour genome
should peak at VAF value <span class="math display">\[
v_m(c) = \dfrac{m \pi c}{
2 (1 - \pi) + \pi (n_A+n_B)
} \, .
\]</span></p>
</div>
<div class="section level4">
<h4 id="subclonal-cnas">Subclonal CNAs<a class="anchor" aria-label="anchor" href="#subclonal-cnas"></a>
</h4>
<p>Consider a mixture of 2 subclones with segments <span class="math inline">\(n_{A,1}:n_{B,1}\)</span> and <span class="math inline">\(n_{A,2}:n_{B,2}\)</span>, proportions <span class="math inline">\(\rho_1\)</span> and <span class="math inline">\(\rho_2\)</span> (<span class="math inline">\(\rho_1+\rho_2=1\)</span>).</p>
<p>The expected peak for a shared mutation with multiplicity <span class="math inline">\(m_1\)</span>/ <span class="math inline">\(m_2\)</span> in the first/ second subclone is
<span class="math display">\[
v_{m_1,m_2} = \dfrac{
(m_1\rho_1 + m_2\rho_2) \pi
}
{
2 (1 - \pi) + \pi [(n_{A,1}+n_{B,1})\rho_1 + (n_{A,2}+n_{B,2})\rho_2]
} \, .
\]</span></p>
<p>The expected peak for a private mutation with multiplicity <span class="math inline">\(m\)</span> in subclone <span class="math inline">\(i\)</span> is <span class="math display">\[
v_{m_i} = \dfrac{
m\rho_i \pi
}
{
2 (1 - \pi) + \pi [(n_{A,1}+n_{B,1})\rho_1 + (n_{A,2}+n_{B,2})\rho_2]
} \, .
\]</span></p>
</div>
<div class="section level4">
<h4 id="cancer-cell-fractions-ccfs">Cancer Cell Fractions (CCFs)<a class="anchor" aria-label="anchor" href="#cancer-cell-fractions-ccfs"></a>
</h4>
<p>Given VAF, tumour purity and CNAs, CCF values can be computed as</p>
<p><span class="math display">\[
c =
\dfrac{
v[
(n_a+n_B - 2)\pi + 2
]
}
{
m \pi
}
\]</span></p>
<p>Therefore the problem of computing CCFs is essentially linked to
computing <code>m</code>, mutation multiplicity. This type of
computation can be done by “phasing” the VAFs against the
multiplicity.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.caravagnalab.org" class="external-link">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
