<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Split a dataset by chromosome. — split_by_chromosome • CNAqc</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Split a dataset by chromosome. — split_by_chromosome"><meta property="og:description" content="Split a CNAqc object by chromosome, returning a named list
with data splits by chromosomes."><meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
    <li>
      <a href="../articles/a9_automatic.html">9. Automatic CNA calling</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/caravagnalab/CNAqc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Split a dataset by chromosome.</h1>
    <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqcR/subsample.R" class="external-link"><code>R/subsample.R</code></a></small>
    <div class="hidden name"><code>split_by_chromosome.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Split a CNAqc object by chromosome, returning a named list
with data splits by chromosomes.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">split_by_chromosome</span><span class="op">(</span><span class="va">x</span>, chromosomes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">22</span>, <span class="st">"X"</span>, <span class="st">"Y"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>x</dt>
<dd><p>An object created by CNAqc.</p></dd>
<dt>chromosomes</dt>
<dd><p>A list of chromosomes to retain.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A named list of CNAqc objects.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"example_PCAWG"</span>, package <span class="op">=</span> <span class="st">'CNAqc'</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="fu">split_by_chromosome</span><span class="op">(</span><span class="va">example_PCAWG</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">11027</span> somatic mutations: <span style="color: #00BB00;">10983</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">44</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">40</span> segments: <span style="color: #00BB00;">40</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">11027</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">29957</span> somatic mutations: <span style="color: #00BB00;">29857</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">100</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">13</span> segments: <span style="color: #00BB00;">13</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">29957</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">26192</span> somatic mutations: <span style="color: #00BB00;">26103</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">89</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">17</span> segments: <span style="color: #00BB00;">17</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">26192</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr4 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">21187</span> somatic mutations: <span style="color: #00BB00;">21104</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">83</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">377</span> segments: <span style="color: #00BB00;">377</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">21179</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">24728</span> somatic mutations: <span style="color: #00BB00;">24624</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">104</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">14</span> segments: <span style="color: #00BB00;">14</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">24728</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">19446</span> somatic mutations: <span style="color: #00BB00;">19383</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">63</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">14</span> segments: <span style="color: #00BB00;">14</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">19035</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">21030</span> somatic mutations: <span style="color: #00BB00;">20955</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">75</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">17</span> segments: <span style="color: #00BB00;">17</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">21030</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">19332</span> somatic mutations: <span style="color: #00BB00;">19270</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">62</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">62</span> segments: <span style="color: #00BB00;">62</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">19308</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">7122</span> somatic mutations: <span style="color: #00BB00;">7101</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">21</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">7</span> segments: <span style="color: #00BB00;">7</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">7122</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">5065</span> somatic mutations: <span style="color: #00BB00;">5052</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">13</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">19</span> segments: <span style="color: #00BB00;">19</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">5065</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr12 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">14351</span> somatic mutations: <span style="color: #00BB00;">14287</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">64</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">4</span> segments: <span style="color: #00BB00;">4</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">14351</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr13 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">8024</span> somatic mutations: <span style="color: #00BB00;">7986</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">38</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">7</span> segments: <span style="color: #00BB00;">7</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">8024</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr14 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">11085</span> somatic mutations: <span style="color: #00BB00;">11041</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">44</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">1</span> segments: <span style="color: #00BB00;">1</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">11085</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr15 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">8566</span> somatic mutations: <span style="color: #00BB00;">8536</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">30</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">18</span> segments: <span style="color: #00BB00;">18</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">8566</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr16 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">10010</span> somatic mutations: <span style="color: #00BB00;">9978</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">32</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">10</span> segments: <span style="color: #00BB00;">10</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">10010</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr17 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">6329</span> somatic mutations: <span style="color: #00BB00;">6316</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">13</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">8</span> segments: <span style="color: #00BB00;">8</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">6329</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr18 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">13062</span> somatic mutations: <span style="color: #00BB00;">13009</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">53</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">9</span> segments: <span style="color: #00BB00;">9</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">13062</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr19 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">5228</span> somatic mutations: <span style="color: #00BB00;">5208</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">20</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">9</span> segments: <span style="color: #00BB00;">9</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">5228</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr20 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">5970</span> somatic mutations: <span style="color: #00BB00;">5954</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">16</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">2</span> segments: <span style="color: #00BB00;">2</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">5970</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr21 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">5117</span> somatic mutations: <span style="color: #00BB00;">5095</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">22</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">2</span> segments: <span style="color: #00BB00;">2</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">5117</span> mutations mapped to clonal CNAs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── chr22 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBBBB; background-color: #0000BB;"> [ CNAqc - CNA Quality Check ] </span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Detected indels mutation; do not forget to rely more on SNVs for data QC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">2595</span> somatic mutations: <span style="color: #00BB00;">2589</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">6</span> indels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">4</span> segments: <span style="color: #00BB00;">4</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>[CNAqc] a karyotype column is present in CNA calls, and will be overwritten</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">2595</span> mutations mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">11027</span> mutations in <span style="color: #00BB00;">40</span> segments (<span style="color: #00BB00;">40</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome re</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:2  [n = 7968, L =   68 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:1  [n = 2017, L =   19 Mb] ■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:2  [n =  462, L =    4 Mb] ■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:1  [n =  428, L =    3 Mb] ■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:2  [n =   93, L =    2 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:3  [n =   24, L =    0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:2  [n =   19, L =    0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:2  [n =    8, L =    0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:0  [n =    3, L =    0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:2  [n =    3, L =    0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 4.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">29957</span> mutations in <span style="color: #00BB00;">13</span> segments (<span style="color: #00BB00;">13</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome re</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:1  [n = 29957, L =   237 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 4.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">26192</span> mutations in <span style="color: #00BB00;">17</span> segments (<span style="color: #00BB00;">17</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome re</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:2  [n = 13692, L =   104 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:1  [n = 12378, L =    83 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:0  [n =   100, L =     4 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:2  [n =    22, L =     0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 5.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">21179</span> mutations in <span style="color: #00BB00;">377</span> segments (<span style="color: #00BB00;">377</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   3:1  [n = 10448, L =    67 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   3:0  [n =  8069, L =    97 Mb] ■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   3:2  [n =   803, L =     6 Mb] ■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   3:3  [n =   801, L =     4 Mb] ■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4:3  [n =   359, L =     3 Mb] ■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   5:3  [n =   132, L =     1 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   6:3  [n =   114, L =     1 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4:2  [n =    59, L =     0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  10:3  [n =    56, L =     0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   9:3  [n =    52, L =     0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 3.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">24728</span> mutations in <span style="color: #00BB00;">14</span> segments (<span style="color: #00BB00;">14</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome re</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:1  [n = 24592, L =   175 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:1  [n =   136, L =     1 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 3.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">19035</span> mutations in <span style="color: #00BB00;">14</span> segments (<span style="color: #00BB00;">14</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome re</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:1  [n = 16330, L =   122 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:2  [n =  2705, L =    36 Mb] ■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 3.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">21030</span> mutations in <span style="color: #00BB00;">17</span> segments (<span style="color: #00BB00;">17</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome re</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:2  [n = 21030, L =   155 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { <span style="color: #BBBB00;">BRAF</span> }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 5.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 1 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         chr      from        to ref alt DP NV       VAF driver_label is_driver</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        chr7 140453136 140453136   A   T 95 54 0.5684211         BRAF      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">19308</span> mutations in <span style="color: #00BB00;">62</span> segments (<span style="color: #00BB00;">62</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome re</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:3  [n = 9995, L =   64 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:2  [n = 4836, L =   27 Mb] ■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:0  [n = 3063, L =   38 Mb] ■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:2  [n = 1289, L =    8 Mb] ■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:2  [n =  105, L =    1 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:2  [n =   20, L =    0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 6.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">7122</span> mutations in <span style="color: #00BB00;">7</span> segments (<span style="color: #00BB00;">7</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome refer</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:0  [n = 7098, L =  114 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { <span style="color: #BBBB00;">CDKN2A</span> }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:0  [n =   24, L =    5 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 3.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 1 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         chr     from       to ref alt DP NV       VAF driver_label is_driver</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        chr9 21971120 21971120   G   A 23 14 0.6086957       CDKN2A      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">5065</span> mutations in <span style="color: #00BB00;">19</span> segments (<span style="color: #00BB00;">19</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome ref</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:0  [n = 2715, L =   38 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:0  [n = 2287, L =   29 Mb] ■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:1  [n =   26, L =    0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:0  [n =   21, L =    0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:0  [n =   15, L =    0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:1  [n =    1, L =    0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 3.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr12</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">14351</span> mutations in <span style="color: #00BB00;">4</span> segments (<span style="color: #00BB00;">4</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome refe</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:1  [n = 14351, L =   129 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 3.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr13</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">8024</span> mutations in <span style="color: #00BB00;">7</span> segments (<span style="color: #00BB00;">7</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome refer</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:0  [n = 8024, L =   99 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 3.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr14</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">11085</span> mutations in <span style="color: #00BB00;">1</span> segments (<span style="color: #00BB00;">1</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome refe</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:1  [n = 11085, L =    87 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 3.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr15</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">8566</span> mutations in <span style="color: #00BB00;">18</span> segments (<span style="color: #00BB00;">18</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome ref</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:1  [n = 7617, L =   77 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:1  [n =  718, L =    9 Mb] ■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:1  [n =  146, L =    1 Mb] ■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:1  [n =   85, L =    0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 3.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr16</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">10010</span> mutations in <span style="color: #00BB00;">10</span> segments (<span style="color: #00BB00;">10</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome re</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:2  [n = 10002, L =    90 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:0  [n =     8, L =     0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 4.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr17</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">6329</span> mutations in <span style="color: #00BB00;">8</span> segments (<span style="color: #00BB00;">8</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome refer</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:2  [n = 4593, L =   59 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:0  [n = 1732, L =   22 Mb] ■■■■■■■■■■  { <span style="color: #BBBB00;">TP53</span> }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:0  [n =    4, L =    0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 4.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 1 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          chr    from      to ref alt DP NV       VAF driver_label is_driver</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        chr17 7577082 7577082   C   T 78 70 0.8974359         TP53      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">13062</span> mutations in <span style="color: #00BB00;">9</span> segments (<span style="color: #00BB00;">9</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome refe</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:2  [n = 13062, L =    78 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 5.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr19</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">5228</span> mutations in <span style="color: #00BB00;">9</span> segments (<span style="color: #00BB00;">9</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome refer</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:2  [n = 4499, L =   42 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:0  [n =  716, L =   12 Mb] ■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:1  [n =   13, L =    0 Mb] </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 5.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr20</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">5970</span> mutations in <span style="color: #00BB00;">2</span> segments (<span style="color: #00BB00;">2</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome refer</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:3  [n = 5970, L =   46 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 6.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">5117</span> mutations in <span style="color: #00BB00;">2</span> segments (<span style="color: #00BB00;">2</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome refer</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:1  [n = 5117, L =   34 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 4.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $chr22</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">2595</span> mutations in <span style="color: #00BB00;">4</span> segments (<span style="color: #00BB00;">4</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome refer</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ── Clonal CNAs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:1  [n = 2595, L =   39 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 4.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> There are 0 annotated driver(s) mapped to clonal CNAs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [1] chr          from         to           ref          alt         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [6] DP           NV           VAF          driver_label is_driver   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://www.caravagnalab.org" class="external-link">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

