<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Compute WT and mutant alleles per gene — wt_mutant_alleles • CNAqc</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Compute WT and mutant alleles per gene — wt_mutant_alleles"><meta property="og:description" content="This function works on a CNAqc object with annotated driver mutations and
a column (in mutation data) reporting gene symbols in human-readable format
(e.g., TP53, KRAS, etc).
This function retains only driver genes, maps them to clonal copy number
segments, and then computes mutation phasing by VAFs. This phasing is the
same operation carried out to compute CCFs in CNAqc; despite being an
approximation to the canonical SNP-based phasing, this operation can easily
determine from VAFs how many genome copies carry out a somatic mutation.
With the complete information altogether (major/minor allele copies and mutation
multiplicity), it is straightforward to identify, for instance, driver genes
that are mutated with a matched LOH status (e.g., complete inactivation of
suppressor genes, or amplification of oncogenic mutations)."><meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
    <li>
      <a href="../articles/a9_automatic.html">9. Automatic CNA calling</a>
    </li>
    <li>
      <a href="../articles/b10_MultiCNAqc.html">10. multiCNAqc</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/caravagnalab/CNAqc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Compute WT and mutant alleles per gene</h1>
    <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqc/R/maf.R" class="external-link"><code>R/maf.R</code></a></small>
    <div class="hidden name"><code>wt_mutant_alleles.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function works on a CNAqc object with annotated driver mutations and
a column (in mutation data) reporting gene symbols in human-readable format
(e.g., TP53, KRAS, etc).</p>
<p>This function retains only driver genes, maps them to clonal copy number
segments, and then computes mutation phasing by VAFs. This phasing is the
same operation carried out to compute CCFs in CNAqc; despite being an
approximation to the canonical SNP-based phasing, this operation can easily
determine from VAFs how many genome copies carry out a somatic mutation.</p>
<p>With the complete information altogether (major/minor allele copies and mutation
multiplicity), it is straightforward to identify, for instance, driver genes
that are mutated with a matched LOH status (e.g., complete inactivation of
suppressor genes, or amplification of oncogenic mutations).</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">wt_mutant_alleles</span><span class="op">(</span><span class="va">x</span>, gene_column <span class="op">=</span> <span class="st">"VEP.SYMBOL"</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>A CNAqc object.</p></dd>


<dt id="arg-gene-column">gene_column<a class="anchor" aria-label="anchor" href="#arg-gene-column"></a></dt>
<dd><p>The gene column where the human-redable gene name should
be. By default it is `"VEP.SYMBOL"` assuming that this function gets run on
a dataset where VEP annotations have been augmented.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>an unpdated CNAqc object.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>function <code><a href="augment_with_vep.html">augment_with_vep</a></code> to add VEP annotations to
a CNAqc object.</p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Example with a CNAqc input object, and MAF annotations</span></span></span>
<span class="r-in"><span><span class="kw">if</span><span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">{</span></span></span>
<span class="r-in"><span>   <span class="co"># Create your CNAqc object (omissis here) from an original "file.vcf"</span></span></span>
<span class="r-in"><span>   <span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="init.html">init</a></span><span class="op">(</span>mutations <span class="op">=</span> <span class="va">...</span>, cna <span class="op">=</span> <span class="va">...</span>, purity <span class="op">=</span> <span class="va">...</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>   <span class="co"># Offline, create your MAF annotations as file "file_vcf.maf" from "file.vcf"</span></span></span>
<span class="r-in"><span>   <span class="co"># vcf2maf file.vcf .... file_vcf.maf</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>   <span class="co"># Import into R/CNAqc</span></span></span>
<span class="r-in"><span>   <span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="augment_with_maf.html">augment_with_maf</a></span><span class="op">(</span><span class="va">x</span>, maf <span class="op">=</span> <span class="st">"file_vcf.maf"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>   <span class="co"># check they are in (there should be many columns with "MAF." prefix)</span></span></span>
<span class="r-in"><span>   <span class="va">x</span> <span class="op"><a href="pipe.html">%&gt;%</a></span> <span class="va">Mutations</span> <span class="op"><a href="pipe.html">%&gt;%</a></span> <span class="va">colnames</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>   <span class="co"># With MAF-imported that is the target column</span></span></span>
<span class="r-in"><span>   <span class="fu">wt_mutant_alleles</span><span class="op">(</span><span class="va">x</span>, gene_column <span class="op">=</span> <span class="st">'MAF.Hugo_Symbol'</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://www.caravagnalab.org" class="external-link">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

