<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>1. Introduction • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="1. Introduction">
<meta property="og:description" content="CNAqc">
<meta property="og:image" content="caravagn.github.io/CNAqc/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Annotatation of SNVs</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a1_Introduction_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>1. Introduction</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqcvignettes/a1_Introduction.Rmd"><code>vignettes/a1_Introduction.Rmd</code></a></small>
      <div class="hidden name"><code>a1_Introduction.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/CNAqc">CNAqc</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span>
<span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'BMix'</span>
<span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading</span>
<span class="co">#&gt; 'BMix'</span>
<span class="co">#&gt; ✔ Loading BMix, 'Binomial and Beta-Binomial univariate mixtures'. Support : &lt;https://caravagnalab.github.io/BMix/&gt;</span>
<span class="co">#&gt; ✔ Loading CNAqc, 'Copy Number Alteration quality check'. Support : &lt;https://caravagn.github.io/CNAqc/&gt;</span></code></pre></div>
<p>We discuss the input format and references used in <code>CNAqc</code>.</p>
<div id="input-format" class="section level1">
<h1 class="hasAnchor">
<a href="#input-format" class="anchor"></a>Input format</h1>
<p>You need to provide data for <em>somatic mutations</em>, <em>copy number segments</em> and <em>tumour purity</em>. The package comes with a template dataset available.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load template data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">'example_dataset_CNAqc'</span>, package <span class="op">=</span> <span class="st">'CNAqc'</span><span class="op">)</span></code></pre></div>
<p>These fields are required for <strong>mutation</strong> calls:</p>
<ul>
<li>the mutation location, as <code>chr</code>, <code>from</code>, <code>to</code>.</li>
<li>the mutation reference and alternative alleles <code>ref</code> and <code>alt</code>;</li>
<li>the total number of reads covering the mutated base(s), <code>DP</code> (depth);</li>
<li>the total number of reads covering only the mutant allele(s), <code>NV</code> (number of reads with variant);</li>
<li>the Variant Allele Frequency, <code>VAF</code>, defined as <code>NV/DP</code>.</li>
</ul>
<p>Chromosome names and alleles should be in character format; chromosomes must be in the format <code>chr1</code>, <code>chr2</code>, etc..</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Example input SNVs</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span>
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">snvs</span> <span class="op">%&gt;%</span> 
        <span class="fu">select</span><span class="op">(</span>
          <span class="va">chr</span>, <span class="va">from</span>, <span class="va">to</span>, <span class="co"># Genomic coordinates</span>
          <span class="va">ref</span>, <span class="va">alt</span>,      <span class="co"># Alleles (reference and alternative)</span>
          <span class="va">DP</span>, <span class="va">NV</span>, <span class="va">VAF</span>    <span class="co"># Read counts (depth, number of variant reads, tumour VAF)</span>
          <span class="op">)</span>
  <span class="op">)</span>
<span class="co">#&gt; # A tibble: 12,963 x 8</span>
<span class="co">#&gt;    chr      from      to ref   alt      DP    NV    VAF</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;</span>
<span class="co">#&gt;  1 chr1  1027104 1027105 T     G        60     6 0.1   </span>
<span class="co">#&gt;  2 chr1  2248588 2248589 A     C       127     9 0.0709</span>
<span class="co">#&gt;  3 chr1  2461999 2462000 G     A       156    65 0.417 </span>
<span class="co">#&gt;  4 chr1  2727935 2727936 T     C       180    90 0.5   </span>
<span class="co">#&gt;  5 chr1  2763397 2763398 C     T       183    61 0.333 </span>
<span class="co">#&gt;  6 chr1  2768208 2768209 C     T       203   130 0.640 </span>
<span class="co">#&gt;  7 chr1  2935590 2935591 C     T       228   132 0.579 </span>
<span class="co">#&gt;  8 chr1  2980032 2980033 C     T       196    85 0.434 </span>
<span class="co">#&gt;  9 chr1  3387161 3387162 T     G       124     6 0.0484</span>
<span class="co">#&gt; 10 chr1  3502517 3502518 G     A        88    10 0.114 </span>
<span class="co">#&gt; # … with 12,953 more rows</span></code></pre></div>
<p>Optionally, you can annotate also <em>driver mutations</em>:</p>
<ul>
<li>whether the mutation is a driver, using a boolean flag <code>is_driver</code>;</li>
<li>a <code>driver_label</code> reportin the label to show in the plots with drivers.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span>
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">snvs</span> <span class="op">%&gt;%</span> 
        <span class="fu">select</span><span class="op">(</span>
          <span class="va">chr</span>, <span class="va">from</span>, <span class="va">to</span>, <span class="va">ref</span>, <span class="va">alt</span>, <span class="va">is_driver</span>, <span class="va">driver_label</span>, <span class="fu">everything</span><span class="op">(</span><span class="op">)</span>
        <span class="op">)</span>
  <span class="op">)</span>
<span class="co">#&gt; # A tibble: 12,963 x 13</span>
<span class="co">#&gt;    chr      from     to ref   alt   is_driver driver_label    FILTER    DP    NV</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;lgl&gt;     &lt;chr&gt;           &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;</span>
<span class="co">#&gt;  1 chr1  1027104 1.03e6 T     G     FALSE     C1orf159(ENST0… PASS      60     6</span>
<span class="co">#&gt;  2 chr1  2248588 2.25e6 A     C     FALSE     SKI(dist=7030)… PASS     127     9</span>
<span class="co">#&gt;  3 chr1  2461999 2.46e6 G     A     FALSE     HES5            PASS     156    65</span>
<span class="co">#&gt;  4 chr1  2727935 2.73e6 T     C     FALSE     RP11-740P5.2    PASS     180    90</span>
<span class="co">#&gt;  5 chr1  2763397 2.76e6 C     T     FALSE     RP11-740P5.3(d… PASS     183    61</span>
<span class="co">#&gt;  6 chr1  2768208 2.77e6 C     T     FALSE     RP11-740P5.3(d… PASS     203   130</span>
<span class="co">#&gt;  7 chr1  2935590 2.94e6 C     T     FALSE     RP11-740P5.3(d… PASS     228   132</span>
<span class="co">#&gt;  8 chr1  2980032 2.98e6 C     T     FALSE     LINC00982       PASS     196    85</span>
<span class="co">#&gt;  9 chr1  3387161 3.39e6 T     G     FALSE     ARHGEF16        PASS     124     6</span>
<span class="co">#&gt; 10 chr1  3502517 3.50e6 G     A     FALSE     MEGF6           PASS      88    10</span>
<span class="co">#&gt; # … with 12,953 more rows, and 3 more variables: VAF &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ANNOVAR_FUNCTION &lt;chr&gt;, GENE &lt;chr&gt;</span></code></pre></div>
<p>These fields are required for <strong>copy number segments</strong>:</p>
<ul>
<li>the segment location, as <code>chr</code>, <code>from</code> and <code>to</code>;</li>
<li>the segment absolute number of copies of the major and minor alleles, as <code>Major</code> and <code>minor</code>;</li>
</ul>
<p>In <code>CNAqc</code> format, a diploid balanced segment has <code>Major = minor = 1</code> and will be shortly denoted as <em>karyotype</em> <code>1:1</code>. Similarly, a tetraploid segment has <code>Major = minor = 2</code> and <em>karyotype</em> <code>2:2</code>. A fixed set of colours is used to identify the main karyotypes supported by <code>CNAqc</code>: <code>"1:0"</code>, <code>"2:0"</code>, <code>"1:1"</code>, <code>"2:1"</code> and <code>"2:2"</code>.</p>
<p>Optionally, you can annotate also annotate the Cancer Cell Fraction (CCF) <code>CCF</code> for each segment:</p>
<ul>
<li>segments with <code>CCF = 1</code> are <em>clonal</em>, otherwise <em>subclonal</em>;</li>
<li>by default, if missing, all segments are assumed to be clonal and assigned <code>CCF = 1</code>. Some computations subset only clonal copy number segments (see the manual of each function).</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Example input CNA</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span>
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">cna</span> <span class="op">%&gt;%</span> 
        <span class="fu">select</span><span class="op">(</span>
          <span class="va">chr</span>, <span class="va">from</span>, <span class="va">to</span>, <span class="co"># Genomic coordinates</span>
          <span class="va">Major</span>, <span class="va">minor</span>  <span class="co"># Number of copies of major/ and minor allele (B-allele)</span>
        <span class="op">)</span>
  <span class="op">)</span>
<span class="co">#&gt; # A tibble: 267 x 5</span>
<span class="co">#&gt;    chr       from       to Major minor</span>
<span class="co">#&gt;    &lt;chr&gt;    &lt;int&gt;    &lt;int&gt; &lt;dbl&gt; &lt;dbl&gt;</span>
<span class="co">#&gt;  1 chr1    840009  1689987     3     2</span>
<span class="co">#&gt;  2 chr1   1689988  1815015     3     2</span>
<span class="co">#&gt;  3 chr1   1815016  9799969     3     2</span>
<span class="co">#&gt;  4 chr1  10479910 12079917     3     2</span>
<span class="co">#&gt;  5 chr1  12079917 12154980     3     2</span>
<span class="co">#&gt;  6 chr1  12154981 12839977     3     2</span>
<span class="co">#&gt;  7 chr1  13780016 17790026     3     2</span>
<span class="co">#&gt;  8 chr1  17849962 21080067     3     2</span>
<span class="co">#&gt;  9 chr1  21080068 21559998     3     2</span>
<span class="co">#&gt; 10 chr1  21559998 24830001     3     2</span>
<span class="co">#&gt; # … with 257 more rows</span></code></pre></div>
<p>Tumour <strong>purity</strong>, defined as the percentage of reads coming from tumour cells), which must be a value in <span class="math inline">\([0, 1]\)</span>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Example purity</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">purity</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.89</span></code></pre></div>
<p><strong>Note</strong> In the current implementation no use of subclonal copy number segments is done. Most function compute statics only with clonal copy number segments.</p>
<div id="initialisation-of-a-new-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#initialisation-of-a-new-dataset" class="anchor"></a>Initialisation of a new dataset</h2>
<p>To use <code>CNAqc</code>, you need to initialize a <code>cnaqc</code> S3 object with the initialisation function <code>init</code>.</p>
<p>This function will check input formats, and will map mutations to CNA segments. This call does not subset the data, unless there are errors with respect to the required formats. It is required to explicit the reference genome for the assembly (see below).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Use SNVs, CNAs and tumour purity (hg19 reference, see below)</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span>
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">snvs</span>, 
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">cna</span>,
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">purity</span>,
  ref <span class="op">=</span> <span class="st">'hg19'</span>
  <span class="op">)</span>
<span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span>
<span class="co">#&gt; ℹ Using reference genome coordinates for: hg19.</span>
<span class="co">#&gt; ℹ Drivers are annotated, but 'gene' column is missing, using mutation location.</span>
<span class="co">#&gt; ! Missing CCF column from CNA calls, adding CCF = 1 assuming clonal CNA calls.</span>
<span class="co">#&gt; ℹ Input n = 12963 mutations for 267 CNA segments (267 clonal, 0 subclonal)</span>
<span class="co">#&gt; ✔ Mapped n = 12963 mutations to clonal segments (100% of input)</span></code></pre></div>
<p>The summary of <code>x</code> can be <code>print</code> to see the number of mutations per karyotype, sorted by size.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  n = 12963 mutations in 267 segments (267 clonal + 0 subclonal). Ge</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;    2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { CTCF }</span>
<span class="co">#&gt;    4:2  [n = 1893, L =  331 Mb] ■■■■■■■</span>
<span class="co">#&gt;    3:2  [n = 1625, L =  357 Mb] ■■■■■■</span>
<span class="co">#&gt;    2:1  [n = 1563, L =  420 Mb] ■■■■■■  { TTN }</span>
<span class="co">#&gt;    3:0  [n =  312, L =  137 Mb] ■</span>
<span class="co">#&gt;    2:0  [n =   81, L =   39 Mb]   { TP53 }</span>
<span class="co">#&gt;   16:2  [n =    4, L =    0 Mb] </span>
<span class="co">#&gt;   25:2  [n =    2, L =    1 Mb] </span>
<span class="co">#&gt;    3:1  [n =    2, L =    1 Mb] </span>
<span class="co">#&gt;  106:1  [n =    1, L =    0 Mb] </span>
<span class="co">#&gt;   26:2  [n =    1, L =    0 Mb] </span>
<span class="co">#&gt;   99:1  [n =    1, L =    0 Mb]</span>
<span class="co">#&gt; ℹ Sample Purity: 89% ~ Ploidy: 4.</span></code></pre></div>
</div>
<div id="subsetting-data" class="section level2">
<h2 class="hasAnchor">
<a href="#subsetting-data" class="anchor"></a>Subsetting data</h2>
<p>You can <em>subset randomly</em> the data; if drivers are annotated, they can be forced to say in.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">y_5000</span> <span class="op">=</span> <span class="fu"><a href="../reference/subsample.html">subsample</a></span><span class="op">(</span><span class="va">x</span>, N <span class="op">=</span> <span class="fl">5000</span>, keep_drivers <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span>
<span class="co">#&gt; ℹ Using reference genome coordinates for: GRCh38.</span>
<span class="co">#&gt; ! Missing CCF column from CNA calls, adding CCF = 1 assuming clonal CNA calls.</span>
<span class="co">#&gt; ℹ Input n = 5003 mutations for 267 CNA segments (267 clonal, 0 subclonal)</span>
<span class="co">#&gt; Warning in map_mutations_to_segments(snvs, cna %&gt;% filter(CCF == 1)): [CNAqc] a</span>
<span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span>
<span class="co">#&gt; ✔ Mapped n = 5003 mutations to clonal segments (100% of input)</span>

<span class="co"># 5000 + the ranomd entries that we sampled before</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">y_5000</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  n = 5003 mutations in 267 segments (267 clonal + 0 subclonal). Gen</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   2:2  [n = 2881, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { CTCF }</span>
<span class="co">#&gt;   4:2  [n =  716, L =  331 Mb] ■■■■■■■</span>
<span class="co">#&gt;   3:2  [n =  625, L =  357 Mb] ■■■■■■</span>
<span class="co">#&gt;   2:1  [n =  621, L =  420 Mb] ■■■■■■  { TTN }</span>
<span class="co">#&gt;   3:0  [n =  123, L =  137 Mb] ■</span>
<span class="co">#&gt;   2:0  [n =   34, L =   39 Mb]   { TP53 }</span>
<span class="co">#&gt;  16:2  [n =    2, L =    0 Mb] </span>
<span class="co">#&gt;   3:1  [n =    1, L =    1 Mb]</span>
<span class="co">#&gt; ℹ Sample Purity: 89% ~ Ploidy: 4.</span></code></pre></div>
<p>You can also subset data by <em>karyotype of the segments</em>, and by <em>total copy number</em> of the segment. Both subset do not keep drivers that map off the selected segments.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Triploid and copy-neutral LOH segments </span>
<span class="va">y_tripl_cnloh</span> <span class="op">=</span> <span class="fu"><a href="../reference/subset_by_segment_karyotype.html">subset_by_segment_karyotype</a></span><span class="op">(</span><span class="va">x</span>, karyotypes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'2:1'</span>, <span class="st">'2:0'</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span>
<span class="co">#&gt; ℹ Using reference genome coordinates for: hg19.</span>
<span class="co">#&gt; ! Missing CCF column from CNA calls, adding CCF = 1 assuming clonal CNA calls.</span>
<span class="co">#&gt; ℹ Input n = 12963 mutations for 58 CNA segments (58 clonal, 0 subclonal)</span>
<span class="co">#&gt; Warning in map_mutations_to_segments(snvs, cna %&gt;% filter(CCF == 1)): [CNAqc] a</span>
<span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span>
<span class="co">#&gt; ✖ n = 11319 mutations  cannot be mapped to segments and will be removed.</span>
<span class="co">#&gt; ✔ Mapped n = 1644 mutations to clonal segments (100% of input)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">y_tripl_cnloh</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  n = 1644 mutations in 58 segments (58 clonal + 0 subclonal). Genom</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  2:1  [n = 1563, L =  420 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { TTN }</span>
<span class="co">#&gt;  2:0  [n =   81, L =   39 Mb] ■  { TP53 }</span>
<span class="co">#&gt; ℹ Sample Purity: 89% ~ Ploidy: 3.</span>

<span class="co"># Two and four copies</span>
<span class="va">y_2_4</span> <span class="op">=</span> <span class="fu"><a href="../reference/subset_by_segment_totalcn.html">subset_by_segment_totalcn</a></span><span class="op">(</span><span class="va">x</span>, totalcn <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span>
<span class="co">#&gt; ℹ Using reference genome coordinates for: hg19.</span>
<span class="co">#&gt; ℹ Input n = 12963 mutations for 90 CNA segments (90 clonal, 0 subclonal)</span>
<span class="co">#&gt; Warning in map_mutations_to_segments(snvs, cna %&gt;% filter(CCF == 1)): [CNAqc] a</span>
<span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span>
<span class="co">#&gt; ✖ n = 5402 mutations  cannot be mapped to segments and will be removed.</span>
<span class="co">#&gt; ✔ Mapped n = 7561 mutations to clonal segments (100% of input)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">y_2_4</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  n = 7561 mutations in 90 segments (90 clonal + 0 subclonal). Genom</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { CTCF }</span>
<span class="co">#&gt;  2:0  [n =   81, L =   39 Mb]   { TP53 }</span>
<span class="co">#&gt;  3:1  [n =    2, L =    1 Mb]</span>
<span class="co">#&gt; ℹ Sample Purity: 89% ~ Ploidy: 4.</span></code></pre></div>
</div>
<div id="reference-genome-coordinates" class="section level2">
<h2 class="hasAnchor">
<a href="#reference-genome-coordinates" class="anchor"></a>Reference genome coordinates</h2>
<p><code>CNAqc</code> contains two coordinates for two reference genomes, <code>hg19</code> and <code>GRCh38</code> (default). For each reference genome the package stores the location of the chromosomes, and their centromers; you can access these data.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">CNAqc</span><span class="fu">:::</span><span class="fu">get_reference</span><span class="op">(</span><span class="st">"hg19"</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 24 x 6</span>
<span class="co">#&gt;    chr      length       from         to centromerStart centromerEnd</span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;int&gt;      &lt;dbl&gt;      &lt;dbl&gt;          &lt;dbl&gt;        &lt;dbl&gt;</span>
<span class="co">#&gt;  1 chr1  249250621          0  249250621      121535434    124535434</span>
<span class="co">#&gt;  2 chr2  243199373  249250621  492449994      341576792    344576792</span>
<span class="co">#&gt;  3 chr3  198022430  492449994  690472424      582954848    585954848</span>
<span class="co">#&gt;  4 chr4  191154276  690472424  881626700      740132541    743132541</span>
<span class="co">#&gt;  5 chr5  180915260  881626700 1062541960      928032341    931032341</span>
<span class="co">#&gt;  6 chr6  171115067 1062541960 1233657027     1121372126   1124372126</span>
<span class="co">#&gt;  7 chr7  159138663 1233657027 1392795690     1291711358   1294711358</span>
<span class="co">#&gt;  8 chr8  146364022 1392795690 1539159712     1436634577   1439634577</span>
<span class="co">#&gt;  9 chr9  141213431 1539159712 1680373143     1586527391   1589527391</span>
<span class="co">#&gt; 10 chr10 135534747 1680373143 1815907890     1719628078   1722628078</span>
<span class="co">#&gt; # … with 14 more rows</span>

<span class="fu">CNAqc</span><span class="fu">:::</span><span class="fu">get_reference</span><span class="op">(</span><span class="st">"GRCh38"</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 24 x 6</span>
<span class="co">#&gt;    chr      length       from         to centromerStart centromerEnd</span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;dbl&gt;      &lt;dbl&gt;      &lt;dbl&gt;          &lt;dbl&gt;        &lt;dbl&gt;</span>
<span class="co">#&gt;  1 chr1  248956422          0  248956422      122026459    124849229</span>
<span class="co">#&gt;  2 chr2  242193529  248956422  491149951      341144567    341144567</span>
<span class="co">#&gt;  3 chr3  198295559  491149951  689445510      581922409    582703370</span>
<span class="co">#&gt;  4 chr4  190214555  689445510  879660065      739157571    739157571</span>
<span class="co">#&gt;  5 chr5  181538259  879660065 1061198324      926145965    929381368</span>
<span class="co">#&gt;  6 chr6  170805979 1061198324 1232004303     1119752212   1119752212</span>
<span class="co">#&gt;  7 chr7  159345973 1232004303 1391350276     1290173956   1293382091</span>
<span class="co">#&gt;  8 chr8  145138636 1391350276 1536488912     1435384020   1435384020</span>
<span class="co">#&gt;  9 chr9  138394717 1536488912 1674883629     1579878547   1579878547</span>
<span class="co">#&gt; 10 chr10 133797422 1674883629 1808681051     1714570311   1716429449</span>
<span class="co">#&gt; # … with 14 more rows</span></code></pre></div>
<p>The reference genomes can be specified as a parameter once you create a <code>CNAqc</code> object. Coordinates are used to translate relative to absolute genome coordinates of somatic mutations and CNA segments; this is done only when plotting the data. Note that mapping of mutations to CNA segments is independent of the reference genome, and it will work as far as both mutation and CNA segments are mapped to the same reference.</p>
<p>You can use a hidden function to plot a reference, and make it explicit with a title.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu">CNAqc</span><span class="fu">:::</span><span class="fu">blank_genome</span><span class="op">(</span>ref <span class="op">=</span> <span class="st">'hg19'</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"HG19 genome reference"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="a1_Introduction_files/figure-html/unnamed-chunk-12-1.png" width="912"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.caravagnalab.org">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
