<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>3. QC analysis via peaks detection • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="3. QC analysis via peaks detection">
<meta property="og:description" content="CNAqc">
<meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
    <li>
      <a href="../articles/a9_automatic.html">9. Automatic CNA calling</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a3_peaks_detection_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>3. QC analysis via peaks detection</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqcvignettes/a3_peaks_detection.Rmd" class="external-link"><code>vignettes/a3_peaks_detection.Rmd</code></a></small>
      <div class="hidden name"><code>a3_peaks_detection.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/CNAqc" class="external-link">CNAqc</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Loading <span style="color: #00BB00;">CNAqc</span>, <span style="font-style: italic;">'Copy Number Alteration quality check'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagn.github.io/CNAqc/&gt;</span></span></span>
<span></span>
<span><span class="co"># We work with the PCAWG object</span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="va"><a href="../reference/example_PCAWG.html">example_PCAWG</a></span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">293736</span> mutations in <span style="color: #00BB00;">667</span> segments (<span style="color: #00BB00;">654</span> clonal, <span style="color: #00BB00;">13</span> subclonal). Genom</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Clonal CNAs</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   2:1  [n = 88422, L =   692 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span></span>
<span><span class="co">#&gt;   3:2  [n = 58384, L =   417 Mb] ■■■■■■■■■■■■■■■■■■  { <span style="color: #BBBB00;">BRAF</span> }</span></span>
<span><span class="co">#&gt;   3:1  [n = 48704, L =   380 Mb] ■■■■■■■■■■■■■■■</span></span>
<span><span class="co">#&gt;   3:0  [n = 26622, L =   360 Mb] ■■■■■■■■  { <span style="color: #BBBB00;">CDKN2A</span> }</span></span>
<span><span class="co">#&gt;   2:2  [n = 25290, L =   253 Mb] ■■■■■■■■</span></span>
<span><span class="co">#&gt;   3:3  [n = 16790, L =   115 Mb] ■■■■■</span></span>
<span><span class="co">#&gt;   2:0  [n =  5374, L =    67 Mb] ■■</span></span>
<span><span class="co">#&gt;   4:0  [n =  1752, L =    22 Mb] ■  { <span style="color: #BBBB00;">TP53</span> }</span></span>
<span><span class="co">#&gt;   4:2  [n =  1441, L =    11 Mb] </span></span>
<span><span class="co">#&gt;   1:1  [n =   855, L =     9 Mb]</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Subclonal CNAs (showing up to 10 segments)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  chr11@55700000  [n =  10468, L =  78.75 Mb]   2:1 (0.21)   2:2 (0.79) ■■■■■■■■■■</span></span>
<span><span class="co">#&gt;  chr11@17365005  [n =   5389, L =  31.55 Mb]   2:1 (0.21)   2:2 (0.79) ■■■■■</span></span>
<span><span class="co">#&gt;   chr11@5372292  [n =   1014, L =  11.99 Mb]   2:1 (0.21)   2:2 (0.79) </span></span>
<span><span class="co">#&gt;    chr11@202253  [n =    610, L =   5.17 Mb]   2:1 (0.22)   2:2 (0.78) </span></span>
<span><span class="co">#&gt;  chr11@48918601  [n =    542, L =   2.68 Mb]   2:1 (0.25)   2:2 (0.75) </span></span>
<span><span class="co">#&gt;   chr6@82432583  [n =    301, L =   1.81 Mb]   2:1 (0.19)   2:2 (0.81) </span></span>
<span><span class="co">#&gt;  chr11@51600000  [n =    290, L =    4.1 Mb]   2:1 (0.26)   2:2 (0.74) </span></span>
<span><span class="co">#&gt;   chr6@81896364  [n =     69, L =   0.54 Mb]   2:1 (0.19)   2:2 (0.81) </span></span>
<span><span class="co">#&gt;   chr6@93956180  [n =     41, L =   0.11 Mb]    2:1 (0.2)    2:2 (0.8) </span></span>
<span><span class="co">#&gt;   chr8@42633277  [n =     13, L =   0.26 Mb]   2:1 (0.28)   2:2 (0.72)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 3.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> There are 3 annotated driver(s) mapped to clonal CNAs.</span></span>
<span><span class="co">#&gt;          chr      from        to ref alt DP NV       VAF driver_label is_driver</span></span>
<span><span class="co">#&gt;        chr17   7577082   7577082   C   T 78 70 0.8974359         TP53      TRUE</span></span>
<span><span class="co">#&gt;         chr7 140453136 140453136   A   T 95 54 0.5684211         BRAF      TRUE</span></span>
<span><span class="co">#&gt;         chr9  21971120  21971120   G   A 23 14 0.6086957       CDKN2A      TRUE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="color: #000000; background-color: #00BB00; font-weight: bold;"> PASS </span><span style="font-weight: bold;"> Peaks QC closest: </span><span style="color: #00BB00; font-weight: bold;">199%</span><span style="font-weight: bold;">, </span><span style="color: #00BB00; font-weight: bold;">λ = 0.0059</span><span style="font-weight: bold;">. Purity correction: 1%.</span> <span style="color: #00BBBB;">───────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">2:1</span> ~ n = 88422 ( 74%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 0.01    <span style="color: #000000; background-color: #00BB00;"> PASS </span> -0.006</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">2:2</span> ~ n = 25290 ( 21%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 0.01    <span style="color: #000000; background-color: #00BB00;"> PASS </span> 0.002</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">2:0</span> ~ n = 5374  (  4%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 0.015   <span style="color: #000000; background-color: #00BB00;"> PASS </span> -0.001</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">1:1</span> ~ n = 855   (0.7%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> -0.006</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">1:0</span> ~ n = 124   (0.1%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 0.008</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">General peak QC (</span><span style="color: #00BB00; font-weight: bold;">154430</span><span style="font-weight: bold;"> mutations): </span><span style="color: #000000; background-color: #00BB00; font-weight: bold;"> PASS </span><span style="font-weight: bold;"> 27 </span><span style="color: #BBBBBB; background-color: #BB0000; font-weight: bold;"> FAIL </span><span style="font-weight: bold;"> 13 - epsilon = 0.05.</span> <span style="color: #00BBBB;">───</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">3:0</span> ~ n = 26622 ( 17%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 0</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">3:1</span> ~ n = 48704 ( 32%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 0</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">3:2</span> ~ n = 58384 ( 38%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 0</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">3:3</span> ~ n = 16790 ( 11%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 0</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">4:2</span> ~ n = 1441  (  1%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 1</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">4:3</span> ~ n = 359   (  0%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 1</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">5:3</span> ~ n = 132   (  0%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">4:0</span> ~ n = 1752  (  1%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 2 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">5:2</span> ~ n = 132   (  0%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 2 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 3</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">6:3</span> ~ n = 114   (  0%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 2 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 4</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Subclonal peaks QC (</span><span style="color: #00BB00; font-weight: bold;">7</span><span style="font-weight: bold;"> segments, initial state </span><span style="color: #00BB00; font-weight: bold;">2:1</span><span style="font-weight: bold;">): </span><span style="font-weight: bold; text-decoration: underline;">linear</span><span style="font-weight: bold;"> </span><span style="color: #00BB00; font-weight: bold;">5</span><span style="font-weight: bold;"> </span><span style="font-weight: bold; text-decoration: underline;">branching</span><span style="font-weight: bold;"> </span><span style="color: #BB0000; font-weight: bold;">0</span><span style="font-weight: bold;"> </span><span style="font-weight: bold; text-decoration: underline;">eith</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #00BB00;"> PASS </span> Linear models</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">chr11@17365005</span> ~ (31.6Mb, n = 5389) <span style="color: #BBBB00;">2:1 (21) + 2:2 (79)</span> : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">  chr11@202253</span> ~ (5.2Mb, n = 610) <span style="color: #BBBB00;">2:1 (22) + 2:2 (78)</span> : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">chr11@51600000</span> ~ (4.1Mb, n = 290) <span style="color: #BBBB00;">2:1 (26) + 2:2 (74)</span> : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #BBBB00;">75</span>]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;"> chr11@5372292</span> ~ (12Mb, n = 1014) <span style="color: #BBBB00;">2:1 (21) + 2:2 (79)</span> : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">chr11@55700000</span> ~ (78.8Mb, n = 10468) <span style="color: #BBBB00;">2:1 (21) + 2:2 (79)</span> : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="color: #BBBBBB; background-color: #BBBB00;"> UNKNOWN </span> Either branching or linear models</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">chr11@48918601</span> ~ (2.7Mb, n = 542) <span style="color: #BBBB00;">2:1 (25) + 2:2 (75)</span> : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1B1 -&gt; A1A2B1 | A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1B1 -&gt; A1A2B1B2 -&gt; A2B1B2 [<span style="color: #00BB00;">100</span>]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;"> chr6@82432583</span> ~ (1.8Mb, n = 301) <span style="color: #BBBB00;">2:1 (19) + 2:2 (81)</span> : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1B1 -&gt; A1A2B1 | A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1B1 -&gt; A1A2B1B2 -&gt; A2B1B2 [<span style="color: #00BB00;">100</span>]</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Cancer Cell Fraction (CCF) data available for karyotypes:1:0, 1:1, 2:0, 2:1, and 2:2.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #BBBBBB; background-color: #00BB00;"> PASS </span> CCF via <span style="font-weight: bold;">ENTROPY</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #BBBBBB; background-color: #00BB00;"> PASS </span> CCF via <span style="font-weight: bold;">ENTROPY</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #BBBBBB; background-color: #00BB00;"> PASS </span> CCF via <span style="font-weight: bold;">ENTROPY</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #BBBBBB; background-color: #00BB00;"> PASS </span> CCF via <span style="font-weight: bold;">ENTROPY</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #BBBBBB; background-color: #00BB00;"> PASS </span> CCF via <span style="font-weight: bold;">ENTROPY</span>.</span></span></code></pre></div>
<div class="section level2">
<h2 id="peak-analysis">Peak analysis<a class="anchor" aria-label="anchor" href="#peak-analysis"></a>
</h2>
<p>CNAqc uses peak-detection algorithms to QC data; all leverage the idea that VAFs peaks are known for mutations mapped to a segment with given minor/ major allele copies. CNAqc therefore computes expected peaks, and compares them to peaks detected from data. The theory works with minor modifications for both clonal and subclonal segments.</p>
<p>Three distinct algorithms are available, each one working with a different type of copy number segment; all analyses are called by function <code>analyze_peaks</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/analyze_peaks.html">analyze_peaks</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Peak analysis: simple CNAs</span> <span style="color: #00BBBB;">──────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Analysing <span style="color: #00BB00;">120065</span> mutations mapping to karyotype(s) <span style="color: #00BB00;">2:1</span>, <span style="color: #00BB00;">2:2</span>, <span style="color: #00BB00;">2:0</span>, <span style="color: #00BB00;">1:1</span>, and <span style="color: #00BB00;">1:0</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Mixed type peak detection for karyotype <span style="color: #00BB00;">1:0</span> (<span style="color: #00BB00;">124</span> mutations)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Loading <span style="color: #00BB00;">BMix</span>, <span style="font-style: italic;">'Binomial and Beta-Binomial univariate mixtures'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagnalab.github.io/BMix/&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Mixed type peak detection for karyotype <span style="color: #00BB00;">1:1</span> (<span style="color: #00BB00;">855</span> mutations)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Mixed type peak detection for karyotype <span style="color: #00BB00;">2:0</span> (<span style="color: #00BB00;">5374</span> mutations)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Mixed type peak detection for karyotype <span style="color: #00BB00;">2:1</span> (<span style="color: #00BB00;">88422</span> mutations)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Mixed type peak detection for karyotype <span style="color: #00BB00;">2:2</span> (<span style="color: #00BB00;">25290</span> mutations)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 8 × 16</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Rowwise: </span></span></span>
<span><span class="co">#&gt;   mutation_multiplicity karyotype  peak delta_vaf     x     y counts_per_bin</span></span>
<span><span class="co">#&gt;                   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>                     1 2:1       0.268   0.013<span style="text-decoration: underline;">4</span>  0.264 4.14            <span style="text-decoration: underline;">3</span>138</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span>                     2 2:1       0.537   0.026<span style="text-decoration: underline;">8</span>  0.54  3               <span style="text-decoration: underline;">2</span>729</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span>                     1 2:2       0.212   0.008<span style="text-decoration: underline;">31</span> 0.21  0.93             198</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span>                     2 2:2       0.423   0.016<span style="text-decoration: underline;">6</span>  0.423 6.85            <span style="text-decoration: underline;">1</span>732</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span>                     1 2:0       0.367   0.025   0.357 0.868             50</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span>                     2 2:0       0.734   0.05    0.735 5.43             265</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span>                     1 1:1       0.367   0.025   0.37  2.42              15</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">8</span>                     1 1:0       0.580   0.062<span style="text-decoration: underline;">4</span>  0.570 4.16               7</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 9 more variables: discarded &lt;lgl&gt;, from &lt;chr&gt;, offset_VAF &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   offset &lt;dbl&gt;, weight &lt;dbl&gt;, epsilon &lt;dbl&gt;, VAF_tolerance &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   matched &lt;lgl&gt;, QC &lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Peak detection <span style="color: #00BB00;">PASS</span> with <span style="color: #00BB00;">r = 0.0112655534075274</span> - maximum purity error ε = <span style="color: #00BB00;">0.05</span>.</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(Major, minor, QC_PASS)`</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(karyotype, QC_PASS)`</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Peak analysis: complex CNAs</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">─────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Karyotypes <span style="color: #00BB00;">3:2</span>, <span style="color: #00BB00;">3:1</span>, <span style="color: #00BB00;">3:0</span>, <span style="color: #00BB00;">3:3</span>, <span style="color: #00BB00;">4:0</span>, <span style="color: #00BB00;">4:2</span>, <span style="color: #00BB00;">4:3</span>, <span style="color: #00BB00;">5:2</span>, <span style="color: #00BB00;">5:3</span>, and <span style="color: #00BB00;">6:3</span> with &gt;<span style="color: #00BB00;">100</span> mutation(s). Using epsilon = <span style="color: #00BB00;">0.05</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 5</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Groups:   karyotype, matched [10]</span></span></span>
<span><span class="co">#&gt;    karyotype n           matched mismatched  prop</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;table[1d]&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> 3:0       26622             3          0 1    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 3:1       48704             3          0 1    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> 3:2       58384             3          0 1    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> 3:3       16790             3          0 1    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> 4:2        1441             3          1 0.75 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> 4:3         359             3          1 0.75 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> 5:3         132             3          2 0.6  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> 4:0        1752             2          2 0.5  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> 5:2         132             2          3 0.4  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 6:3         114             2          4 0.333</span></span>
<span><span class="co">#&gt; Adding missing grouping variables: `matched`</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(Major, minor, QC_PASS, matched)`</span></span>
<span><span class="co">#&gt; Adding missing grouping variables: `matched`</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(karyotype, QC_PASS, matched)`</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Peak analysis: subclonal CNAs</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">───────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; → Computing evolution models for subclonal CNAs - starting from <span style="color: #00BB00;">1:1</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 11 × 6</span></span></span>
<span><span class="co">#&gt;    segment_id     model_id                   model      prop size         clones</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr6@82432583  A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear     1    (1.8Mb, n =… 2:1 0…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr6@82432583  A1B1 -&gt; A1A2B1 | A1A2B1B2  branching  1    (1.8Mb, n =… 2:1 0…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr6@82432583  A1B1 -&gt; A1A2B1B2 -&gt; A2B1B2 linear     1    (1.8Mb, n =… 2:1 0…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr11@202253   A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear     1    (5.2Mb, n =… 2:1 0…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr11@5372292  A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear     1    (12Mb, n = … 2:1 0…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr11@17365005 A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear     1    (31.6Mb, n … 2:1 0…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr11@48918601 A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear     1    (2.7Mb, n =… 2:1 0…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr11@48918601 A1B1 -&gt; A1A2B1 | A1A2B1B2  branching  1    (2.7Mb, n =… 2:1 0…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr11@48918601 A1B1 -&gt; A1A2B1B2 -&gt; A2B1B2 linear     1    (2.7Mb, n =… 2:1 0…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr11@51600000 A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear     0.75 (4.1Mb, n =… 2:1 0…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> chr11@55700000 A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 linear     1    (78.8Mb, n … 2:1 0…</span></span>
<span></span>
<span><span class="co"># Shows results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">293736</span> mutations in <span style="color: #00BB00;">667</span> segments (<span style="color: #00BB00;">654</span> clonal, <span style="color: #00BB00;">13</span> subclonal). Genom</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Clonal CNAs</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   2:1  [n = 88422, L =   692 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span></span>
<span><span class="co">#&gt;   3:2  [n = 58384, L =   417 Mb] ■■■■■■■■■■■■■■■■■■  { <span style="color: #BBBB00;">BRAF</span> }</span></span>
<span><span class="co">#&gt;   3:1  [n = 48704, L =   380 Mb] ■■■■■■■■■■■■■■■</span></span>
<span><span class="co">#&gt;   3:0  [n = 26622, L =   360 Mb] ■■■■■■■■  { <span style="color: #BBBB00;">CDKN2A</span> }</span></span>
<span><span class="co">#&gt;   2:2  [n = 25290, L =   253 Mb] ■■■■■■■■</span></span>
<span><span class="co">#&gt;   3:3  [n = 16790, L =   115 Mb] ■■■■■</span></span>
<span><span class="co">#&gt;   2:0  [n =  5374, L =    67 Mb] ■■</span></span>
<span><span class="co">#&gt;   4:0  [n =  1752, L =    22 Mb] ■  { <span style="color: #BBBB00;">TP53</span> }</span></span>
<span><span class="co">#&gt;   4:2  [n =  1441, L =    11 Mb] </span></span>
<span><span class="co">#&gt;   1:1  [n =   855, L =     9 Mb]</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Subclonal CNAs (showing up to 10 segments)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  chr11@55700000  [n =  10468, L =  78.75 Mb]   2:1 (0.21)   2:2 (0.79) ■■■■■■■■■■</span></span>
<span><span class="co">#&gt;  chr11@17365005  [n =   5389, L =  31.55 Mb]   2:1 (0.21)   2:2 (0.79) ■■■■■</span></span>
<span><span class="co">#&gt;   chr11@5372292  [n =   1014, L =  11.99 Mb]   2:1 (0.21)   2:2 (0.79) </span></span>
<span><span class="co">#&gt;    chr11@202253  [n =    610, L =   5.17 Mb]   2:1 (0.22)   2:2 (0.78) </span></span>
<span><span class="co">#&gt;  chr11@48918601  [n =    542, L =   2.68 Mb]   2:1 (0.25)   2:2 (0.75) </span></span>
<span><span class="co">#&gt;   chr6@82432583  [n =    301, L =   1.81 Mb]   2:1 (0.19)   2:2 (0.81) </span></span>
<span><span class="co">#&gt;  chr11@51600000  [n =    290, L =    4.1 Mb]   2:1 (0.26)   2:2 (0.74) </span></span>
<span><span class="co">#&gt;   chr6@81896364  [n =     69, L =   0.54 Mb]   2:1 (0.19)   2:2 (0.81) </span></span>
<span><span class="co">#&gt;   chr6@93956180  [n =     41, L =   0.11 Mb]    2:1 (0.2)    2:2 (0.8) </span></span>
<span><span class="co">#&gt;   chr8@42633277  [n =     13, L =   0.26 Mb]   2:1 (0.28)   2:2 (0.72)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample Purity: 73.4% ~ Ploidy: 3.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> There are 3 annotated driver(s) mapped to clonal CNAs.</span></span>
<span><span class="co">#&gt;          chr      from        to ref alt DP NV       VAF driver_label is_driver</span></span>
<span><span class="co">#&gt;        chr17   7577082   7577082   C   T 78 70 0.8974359         TP53      TRUE</span></span>
<span><span class="co">#&gt;         chr7 140453136 140453136   A   T 95 54 0.5684211         BRAF      TRUE</span></span>
<span><span class="co">#&gt;         chr9  21971120  21971120   G   A 23 14 0.6086957       CDKN2A      TRUE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="color: #000000; background-color: #00BB00; font-weight: bold;"> PASS </span><span style="font-weight: bold;"> Peaks QC closest: </span><span style="color: #00BB00; font-weight: bold;">199%</span><span style="font-weight: bold;">, </span><span style="color: #00BB00; font-weight: bold;">λ = 0.0113</span><span style="font-weight: bold;">. Purity correction: 1%.</span> <span style="color: #00BBBB;">───────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">2:1</span> ~ n = 88422 ( 74%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 0.017   <span style="color: #000000; background-color: #00BB00;"> PASS </span> -0.006 </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">2:2</span> ~ n = 25290 ( 21%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 0.01    <span style="color: #000000; background-color: #00BB00;"> PASS </span> 0.002  </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">2:0</span> ~ n = 5374  (  4%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 0.019   <span style="color: #000000; background-color: #00BB00;"> PASS </span> -0.001 </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">1:1</span> ~ n = 855   (0.7%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> -0.006 </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">1:0</span> ~ n = 124   (0.1%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 0.008  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">General peak QC (</span><span style="color: #00BB00; font-weight: bold;">154430</span><span style="font-weight: bold;"> mutations): </span><span style="color: #000000; background-color: #00BB00; font-weight: bold;"> PASS </span><span style="font-weight: bold;"> 27 </span><span style="color: #BBBBBB; background-color: #BB0000; font-weight: bold;"> FAIL </span><span style="font-weight: bold;"> 13 - epsilon = 0.05.</span> <span style="color: #00BBBB;">───</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">3:0</span> ~ n = 26622 ( 17%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 0</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">3:1</span> ~ n = 48704 ( 32%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 0</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">3:2</span> ~ n = 58384 ( 38%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 0</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">3:3</span> ~ n = 16790 ( 11%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 0</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">4:2</span> ~ n = 1441  (  1%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 1</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">4:3</span> ~ n = 359   (  0%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 1</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">5:3</span> ~ n = 132   (  0%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 3 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">4:0</span> ~ n = 1752  (  1%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 2 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">5:2</span> ~ n = 132   (  0%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 2 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 3</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">6:3</span> ~ n = 114   (  0%) → <span style="color: #000000; background-color: #00BB00;"> PASS </span> 2 <span style="color: #BBBBBB; background-color: #BB0000;"> FAIL </span> 4</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Subclonal peaks QC (</span><span style="color: #00BB00; font-weight: bold;">7</span><span style="font-weight: bold;"> segments, initial state </span><span style="color: #00BB00; font-weight: bold;">2:1</span><span style="font-weight: bold;">): </span><span style="font-weight: bold; text-decoration: underline;">linear</span><span style="font-weight: bold;"> </span><span style="color: #00BB00; font-weight: bold;">5</span><span style="font-weight: bold;"> </span><span style="font-weight: bold; text-decoration: underline;">branching</span><span style="font-weight: bold;"> </span><span style="color: #BB0000; font-weight: bold;">0</span><span style="font-weight: bold;"> </span><span style="font-weight: bold; text-decoration: underline;">eith</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #00BB00;"> PASS </span> Linear models </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">chr11@17365005</span> ~ (31.6Mb, n = 5389) <span style="color: #BBBB00;">2:1 (21) + 2:2 (79)</span> : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">  chr11@202253</span> ~ (5.2Mb, n = 610) <span style="color: #BBBB00;">2:1 (22) + 2:2 (78)</span> : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">chr11@51600000</span> ~ (4.1Mb, n = 290) <span style="color: #BBBB00;">2:1 (26) + 2:2 (74)</span> : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #BBBB00;">75</span>]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;"> chr11@5372292</span> ~ (12Mb, n = 1014) <span style="color: #BBBB00;">2:1 (21) + 2:2 (79)</span> : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">chr11@55700000</span> ~ (78.8Mb, n = 10468) <span style="color: #BBBB00;">2:1 (21) + 2:2 (79)</span> : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="color: #BBBBBB; background-color: #BBBB00;"> UNKNOWN </span> Either branching or linear models </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">chr11@48918601</span> ~ (2.7Mb, n = 542) <span style="color: #BBBB00;">2:1 (25) + 2:2 (75)</span> : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1B1 -&gt; A1A2B1 | A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1B1 -&gt; A1A2B1B2 -&gt; A2B1B2 [<span style="color: #00BB00;">100</span>]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;"> chr6@82432583</span> ~ (1.8Mb, n = 301) <span style="color: #BBBB00;">2:1 (19) + 2:2 (81)</span> : A1B1 -&gt; A1A2B1 -&gt; A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1B1 -&gt; A1A2B1 | A1A2B1B2 [<span style="color: #00BB00;">100</span>]; A1B1 -&gt; A1A2B1B2 -&gt; A2B1B2 [<span style="color: #00BB00;">100</span>]</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Cancer Cell Fraction (CCF) data available for karyotypes:1:0, 1:1, 2:0, 2:1, and 2:2.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #BBBBBB; background-color: #00BB00;"> PASS </span> CCF via <span style="font-weight: bold;">ENTROPY</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #BBBBBB; background-color: #00BB00;"> PASS </span> CCF via <span style="font-weight: bold;">ENTROPY</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #BBBBBB; background-color: #00BB00;"> PASS </span> CCF via <span style="font-weight: bold;">ENTROPY</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #BBBBBB; background-color: #00BB00;"> PASS </span> CCF via <span style="font-weight: bold;">ENTROPY</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #BBBBBB; background-color: #00BB00;"> PASS </span> CCF via <span style="font-weight: bold;">ENTROPY</span>.</span></span></code></pre></div>
<div class="section level3">
<h3 id="simple-clonal-segments-10-20-11-21-22">Simple clonal segments (<code>1:0</code>, <code>2:0</code>, <code>1:1</code>, <code>2:1</code>, <code>2:2</code>)<a class="anchor" aria-label="anchor" href="#simple-clonal-segments-10-20-11-21-22"></a>
</h3>
<p>This QC measures an error for the precision of the current purity estimate, failing a whole sample or a subset of segments the value is over a desired maximum value. The error is determined as a linear combination from the distance between VAF peaks and their theoretical expectation. For this analysis, all mutations mapping across any segment with the same major/minor alleles are pooled.</p>
<blockquote>
<p>Note: the score can be used to select among alternative copy number solutions, i.e., favouring a solution with lower score.</p>
</blockquote>
<p>The peaks are determined via:</p>
<ul>
<li><p>peak-detection algorithms from the <a href="https://cran.rstudio.com/web/packages/peakPick/index.html" class="external-link">peakPick</a> package, applied to a Gaussian kernel density estimate (gKDE) smooth of the VAF distribution;</p></li>
<li><p>the <a href="https://caravagnalab.github.io/BMix/" class="external-link">Bmix</a> Binomial mixture model.</p></li>
</ul>
<p>Peak-matching (i.e., determining what data peak is closest to the expected peak) has two possible implementations:</p>
<ul>
<li>one matcheing the closest peaks by euclidean distance;</li>
<li>the other ranking peaks from higher to lowr VAFs, and prioritising the former.</li>
</ul>
<p>Results from peak-based QC are available via <code>plot_peaks_analysis</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_peaks_analysis.html">plot_peaks_analysis</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></code></pre></div>
<p><img src="a3_peaks_detection_files/figure-html/unnamed-chunk-3-1.png" width="1056"></p>
<p>Gray panels are placeholders for segments among <code>1:0</code>, <code>2:0</code>, <code>1:1</code>, <code>2:1</code>, <code>2:2</code> that are not available for the sample. Each vertical dashed line is an expected peak, the bandwidth around being the tolerance we use to match peaks (based on <code>purity_error</code>, adjusted for segment ploidy and tumour purity). Each dot is a peak detected from data, with a bandwidth of tolerance (fixed) around it.</p>
<p>Note that:</p>
<ul>
<li>A green peak is matched, a red one is mismatched;</li>
<li>The overall segment QC is given by the colour of the facet;</li>
<li>The overall sample QC is given by the box surrounding the whole figure assembly.</li>
</ul>
<p>Options of function <code>plot_peaks_analysis</code> allow to separate the plots.</p>
<blockquote>
<p>Note: a chromosome-level analysis is possible by using function <code>split_by_chromosome</code> to separate a CNAqc object into chromosomes, and then running a standard analysis on each chromosome.</p>
</blockquote>
</div>
<div class="section level3">
<h3 id="complex-clonal-segments">Complex clonal segments<a class="anchor" aria-label="anchor" href="#complex-clonal-segments"></a>
</h3>
<p>The QC procedure for these “general” segments uses only the gKDE and, as for simple segments, pools all mutations mapping across any segment with the same major/minor alleles.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_peaks_analysis.html">plot_peaks_analysis</a></span><span class="op">(</span><span class="va">x</span>, what <span class="op">=</span> <span class="st">'general'</span><span class="op">)</span></span></code></pre></div>
<p><img src="a3_peaks_detection_files/figure-html/unnamed-chunk-4-1.png" width="768"></p>
<p>The plot is similar to the one for simple segments, but no segment-level or sample-level scores are produced. A complex segment with many matched peaks is likely to be correct.</p>
</div>
<div class="section level3">
<h3 id="subclonal-simple-segments">Subclonal simple segments<a class="anchor" aria-label="anchor" href="#subclonal-simple-segments"></a>
</h3>
<p>The QC procedure for these segments uses the gKDE and considers 2 subclones with distinct mixing proportions. Differently from clonal CNAs, however, here the analysis is carried out at the level of each segment, i.e., without pooling segments with the same karyotypes. This makes it possible to use subclonal calls fromcallers that report segment-specific CCF values, e.g., <a href="https://github.com/Wedge-lab/battenberg" class="external-link">Battenberg</a>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_peaks_analysis.html">plot_peaks_analysis</a></span><span class="op">(</span><span class="va">x</span>, what <span class="op">=</span> <span class="st">'subclonal'</span><span class="op">)</span></span></code></pre></div>
<p><img src="a3_peaks_detection_files/figure-html/unnamed-chunk-5-1.png" width="768"> The visual layout of this plot is the same of complex clonal CNAs; not that the facet reports the distinct evolutionary models that have been generated to QC subclonal CNAs. The model in CNAqc ranks the proposed evolutionary alternatives (linear versus branching) based on the number of matched peaks. A subclonal segment with many matched peaks is likely to be correct.</p>
</div>
</div>
<div class="section level2">
<h2 id="summary-results">Summary results<a class="anchor" aria-label="anchor" href="#summary-results"></a>
</h2>
<p>For every type of segment analyzed tables with summary peaks are available in <code>x$peaks_analysis</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Simple clonal CNAs - each segment with `discarded = FALSE` has been analysed</span></span>
<span><span class="va">x</span><span class="op">$</span><span class="va">peaks_analysis</span><span class="op">$</span><span class="va">matches</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 8 × 16</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Rowwise: </span></span></span>
<span><span class="co">#&gt;   mutation_multiplicity karyotype  peak delta_vaf     x     y counts_per_bin</span></span>
<span><span class="co">#&gt;                   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>                     1 2:1       0.268   0.013<span style="text-decoration: underline;">4</span>  0.264 4.14            <span style="text-decoration: underline;">3</span>138</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span>                     2 2:1       0.537   0.026<span style="text-decoration: underline;">8</span>  0.54  3               <span style="text-decoration: underline;">2</span>729</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span>                     1 2:2       0.212   0.008<span style="text-decoration: underline;">31</span> 0.21  0.93             198</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span>                     2 2:2       0.423   0.016<span style="text-decoration: underline;">6</span>  0.423 6.85            <span style="text-decoration: underline;">1</span>732</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span>                     1 2:0       0.367   0.025   0.357 0.868             50</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span>                     2 2:0       0.734   0.05    0.735 5.43             265</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span>                     1 1:1       0.367   0.025   0.37  2.42              15</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">8</span>                     1 1:0       0.580   0.062<span style="text-decoration: underline;">4</span>  0.570 4.16               7</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 9 more variables: discarded &lt;lgl&gt;, from &lt;chr&gt;, offset_VAF &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   offset &lt;dbl&gt;, weight &lt;dbl&gt;, epsilon &lt;dbl&gt;, VAF_tolerance &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   matched &lt;lgl&gt;, QC &lt;chr&gt;</span></span></span>
<span></span>
<span><span class="co"># Complex clonal CNAs</span></span>
<span><span class="va">x</span><span class="op">$</span><span class="va">peaks_analysis</span><span class="op">$</span><span class="va">general</span><span class="op">$</span><span class="va">expected_peaks</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 40 × 9</span></span></span>
<span><span class="co">#&gt;    minor Major ploidy multiplicity purity  peak karyotype matched n          </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;table[1d]&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     2     4      6            1  0.734 0.149 4:2       TRUE    1441       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     2     4      6            2  0.734 0.297 4:2       TRUE    1441       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>     2     4      6            3  0.734 0.446 4:2       FALSE   1441       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>     2     4      6            4  0.734 0.595 4:2       TRUE    1441       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>     2     5      7            1  0.734 0.129 5:2       FALSE    132       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     2     5      7            2  0.734 0.259 5:2       TRUE     132       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>     2     5      7            3  0.734 0.388 5:2       FALSE    132       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>     2     5      7            4  0.734 0.518 5:2       FALSE    132       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>     2     5      7            5  0.734 0.647 5:2       TRUE     132       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>     0     4      4            1  0.734 0.212 4:0       FALSE   1752       </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 30 more rows</span></span></span>
<span></span>
<span><span class="co"># Subclonal CNAs</span></span>
<span><span class="va">x</span><span class="op">$</span><span class="va">peaks_analysis</span><span class="op">$</span><span class="va">subclonal</span><span class="op">$</span><span class="va">expected_peaks</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 91 × 16</span></span></span>
<span><span class="co">#&gt;    mutation karyotype_1 genotype_1 karyotype_2 genotype_2 n.clone_1 n.clone_2</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> DGULFIAL 2:1         A1A2B1     <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>                 1         0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> ALLBCSCT <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>         2:2         A1A2B1B2           0         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> OEIIUDKO 2:1         A1A2B1     2:2         A1A2B1B2           1         2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> LJXWOIBG 2:1         A1A2B1     2:2         A1A2B1B2           2         2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> ARGWCXPK <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>         2:2         A1A2B1B2           0         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> BJADBTCM 2:1         A1A2B1     2:2         A1A2B1B2           1         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> HNUSCBPO 2:1         A1A2B1     2:2         A1A2B1B2           1         2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> GEITMUAX 2:1         A1A2B1     2:2         A1A2B1B2           2         2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> GQSEIUHU <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>         2:1         A2B1B2             0         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> OOEVHPBC 2:2         A1A2B1B2   <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>                 1         0</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 81 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 9 more variables: peak &lt;dbl&gt;, genotype_initial &lt;chr&gt;, model &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   model_id &lt;chr&gt;, role &lt;chr&gt;, segment_id &lt;chr&gt;, matched &lt;lgl&gt;, size &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   clones &lt;chr&gt;</span></span></span></code></pre></div>
<p>The most helpful table is usually the one for simple clonal CNAs <code>x$peaks_analysis$matches</code>, which reports several information:</p>
<ul>
<li>
<code>mutation_multiplicity</code>, the number of copies of the mutation (i.e., a phasing information);</li>
<li>
<code>peak</code>, <code>x</code>, <code>y</code> the expected peak, and the matched peak (<code>x</code> and <code>y</code>);</li>
<li>
<code>offset</code>, <code>weight</code> and <code>score</code>, the factors of the final <code>score</code>;</li>
<li>
<code>QC</code>, a <code>"PASS"/"FAIL"</code> status for the peak.</li>
</ul>
<p>The overall sample-level QC result - <code>"PASS"/"FAIL"</code> - is available in</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span><span class="op">$</span><span class="va">peaks_analysis</span><span class="op">$</span><span class="va">QC</span></span>
<span><span class="co">#&gt; [1] "PASS"</span></span></code></pre></div>
<p>You can summarise QC results in a plot.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_qc.html">plot_qc</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></code></pre></div>
<p><img src="a3_peaks_detection_files/figure-html/unnamed-chunk-8-1.png" width="576"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.caravagnalab.org" class="external-link">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
