<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>6. Genome fragmentation analysis • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="6. Genome fragmentation analysis">
<meta property="og:description" content="CNAqc">
<meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
    <li>
      <a href="../articles/a9_automatic.html">9. Automatic CNA calling</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a6_fragmentation_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>6. Genome fragmentation analysis</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqcvignettes/a6_fragmentation.Rmd" class="external-link"><code>vignettes/a6_fragmentation.Rmd</code></a></small>
      <div class="hidden name"><code>a6_fragmentation.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/CNAqc" class="external-link">CNAqc</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Loading <span style="color: #00BB00;">CNAqc</span>, <span style="font-style: italic;">'Copy Number Alteration quality check'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagn.github.io/CNAqc/&gt;</span></span></span></code></pre></div>
<div class="section level3">
<h3 id="fragmentation-of-individual-arms">Fragmentation of individual arms<a class="anchor" aria-label="anchor" href="#fragmentation-of-individual-arms"></a>
</h3>
<p>The fragmentation of a chromosome arm is assessed with a statistical test based on counting the size of the copy number segments mapping to the arm. This analysis works only at the level of clonal segments</p>
<p>We work with the template dataset.</p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { <span style="color: #BBBB00;">CTCF</span> }</span></span>
<span><span class="co">#&gt;    4:2  [n = 1893, L =  331 Mb] ■■■■■■■</span></span>
<span><span class="co">#&gt;    3:2  [n = 1625, L =  357 Mb] ■■■■■■</span></span>
<span><span class="co">#&gt;    2:1  [n = 1563, L =  420 Mb] ■■■■■■  { <span style="color: #BBBB00;">TTN</span> }</span></span>
<span><span class="co">#&gt;    3:0  [n =  312, L =  137 Mb] ■</span></span>
<span><span class="co">#&gt;    2:0  [n =   81, L =   39 Mb]   { <span style="color: #BBBB00;">TP53</span> }</span></span>
<span><span class="co">#&gt;   16:2  [n =    4, L =    0 Mb] </span></span>
<span><span class="co">#&gt;   25:2  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;    3:1  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;  106:1  [n =    1, L =    0 Mb] </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;          chr      from        to ref alt  DP NV       VAF driver_label is_driver</span></span>
<span><span class="co">#&gt;         chr2 179431633 179431634   C   T 117 77 0.6581197          TTN      TRUE</span></span>
<span><span class="co">#&gt;        chr16  67646006  67646007   C   T 120 54 0.4500000         CTCF      TRUE</span></span>
<span><span class="co">#&gt;        chr17   7577106   7577107   G   C  84 78 0.9285714         TP53      TRUE</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># A histogram of segments' lenght</span></span>
<span><span class="fu"><a href="../reference/plot_segment_size_distribution.html">plot_segment_size_distribution</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></code></pre></div>
<p><img src="a6_fragmentation_files/figure-html/unnamed-chunk-3-1.png" width="288"></p>
<p><code>CNAqc</code> counts, for every arm with lenght <span class="math inline">\(L\)</span> nucleotides:</p>
<ul>
<li>
<span class="math inline">\(n_s\)</span>, the number of mapped CNA segments shorter than <span class="math inline">\(\delta\%\)</span> of <span class="math inline">\(L\)</span>;</li>
<li>
<span class="math inline">\(n_l\)</span>, the number of mapped CNA segments longer than <span class="math inline">\(\delta\%\)</span> sof <span class="math inline">\(L\)</span>.</li>
</ul>
<p>A one-sided Binomial test is used to compute a p-value for the null hypothesis of seeing <span class="math inline">\(n_s\)</span> observations in <span class="math inline">\(n_l\)</span> trials, assuming a Binomial success probability <span class="math inline">\(p = \delta &gt; 0\)</span>. <span class="math inline">\(p\)</span> represents a model where each segment length is equally likely (uniform distribution).</p>
<p>In this way the test accounts for the difference in lenghts of the chromsome arms; a p-value per arm is reported and adjusted for multiple hyoptheses (Bonferroni).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Test with default parameters (small segments: &lt; 20% of chromosome arm)</span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/detect_arm_overfragmentation.html">detect_arm_overfragmentation</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> One-tailed Binomial test: 8 tests, alpha 0.01. Short segments: 0.2% of the reference arm.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">chr7p</span>,  p = 1.71798691840001e-24 ~ 34 segments, 34 short.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">chr1p</span>,  p = 1.62738995200002e-15 ~ 24 segments, 23 short.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">chr1q</span>,  p = 4.34176000000001e-08 ~ 13 segments, 12 short.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">chr11q</span>,  p = 1.0657792e-06 ~ 13 segments, 11 short.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">chr12q</span>,  p = 2.00704e-07 ~ 12 segments, 11 short.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">chr3q</span>,  p = 4.52608e-06 ~ 12 segments, 10 short.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">chr7q</span>,  p = 4.52608e-06 ~ 12 segments, 10 short.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">chr8p</span>,  p = 9.21599999999998e-07 ~ 11 segments, 10 short.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> 8 significantly overfragmented chromosome arms (alpha level 0.01).</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">12963</span> mutations in <span style="color: #00BB00;">267</span> segments (<span style="color: #00BB00;">267</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Clonal CNAs</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { <span style="color: #BBBB00;">CTCF</span> }</span></span>
<span><span class="co">#&gt;    4:2  [n = 1893, L =  331 Mb] ■■■■■■■</span></span>
<span><span class="co">#&gt;    3:2  [n = 1625, L =  357 Mb] ■■■■■■</span></span>
<span><span class="co">#&gt;    2:1  [n = 1563, L =  420 Mb] ■■■■■■  { <span style="color: #BBBB00;">TTN</span> }</span></span>
<span><span class="co">#&gt;    3:0  [n =  312, L =  137 Mb] ■</span></span>
<span><span class="co">#&gt;    2:0  [n =   81, L =   39 Mb]   { <span style="color: #BBBB00;">TP53</span> }</span></span>
<span><span class="co">#&gt;   16:2  [n =    4, L =    0 Mb] </span></span>
<span><span class="co">#&gt;   25:2  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;    3:1  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;  106:1  [n =    1, L =    0 Mb]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample Purity: 89% ~ Ploidy: 4.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> There are 3 annotated driver(s) mapped to clonal CNAs.</span></span>
<span><span class="co">#&gt;          chr      from        to ref alt  DP NV       VAF driver_label is_driver</span></span>
<span><span class="co">#&gt;         chr2 179431633 179431634   C   T 117 77 0.6581197          TTN      TRUE</span></span>
<span><span class="co">#&gt;        chr16  67646006  67646007   C   T 120 54 0.4500000         CTCF      TRUE</span></span>
<span><span class="co">#&gt;        chr17   7577106   7577107   G   C  84 78 0.9285714         TP53      TRUE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Arm-level fragmentation analysis: 8 segments overfragmented.</span></span></code></pre></div>
<p>You can produce a arm-level report for the fragmentation test, with:</p>
<ul>
<li>a scatter of the counts per arm, with scaled the p-values;</li>
<li>a jump statistics per arm, <span class="math inline">\(J\)</span>.</li>
</ul>
<p><span class="math inline">\(J\)</span> is the sum of the variation in total copy number profiles, evaluated among each pair of contiguous segments.</p>
<p>Significantly overfragmented arms with high <span class="math inline">\(J\)</span> have a “scattered” copy number profile. Those with low <span class="math inline">\(J\)</span> are more uniform, as they show little no copy number change, and can be possibly smoothed (see below).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_arm_fragmentation.html">plot_arm_fragmentation</a></span><span class="op">(</span><span class="va">x</span>, zoom <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p><img src="a6_fragmentation_files/figure-html/unnamed-chunk-5-1.png" width="576"></p>
<p>Once available, these results appear in any call to <code>plot_segments</code> as annotated purple squares sorrounding the arms.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Default plot has now segments</span></span>
<span><span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span></span>
<span><span class="co">#&gt; will replace the existing scale.</span></span></code></pre></div>
<p><img src="a6_fragmentation_files/figure-html/unnamed-chunk-6-1.png" width="912"></p>
<p>Smoothing is a good way to start cleaning up the fragmented sets of arms.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Smooth with default parameters</span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/smooth_segments.html">smooth_segments</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr1</span> <span style="color: #BB0000;">37</span> -<span style="color: #00BB00;">6</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr10</span> <span style="color: #BB0000;">8</span> -<span style="color: #00BB00;">3</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr11</span> <span style="color: #BB0000;">22</span> -<span style="color: #00BB00;">3</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr12</span> <span style="color: #BB0000;">13</span> -<span style="color: #00BB00;">11</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr14</span> <span style="color: #BB0000;">2</span> -<span style="color: #00BB00;">1</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr15</span> <span style="color: #BB0000;">9</span> -<span style="color: #00BB00;">3</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr16</span> <span style="color: #BB0000;">10</span> -<span style="color: #00BB00;">3</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr17</span> <span style="color: #BB0000;">10</span> -<span style="color: #00BB00;">6</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr18</span> <span style="color: #BB0000;">8</span> -<span style="color: #00BB00;">2</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr19</span> <span style="color: #BB0000;">5</span> -<span style="color: #00BB00;">2</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr2</span> <span style="color: #BB0000;">18</span> -<span style="color: #00BB00;">5</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr20</span> <span style="color: #BB0000;">9</span> -<span style="color: #00BB00;">2</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr21</span> <span style="color: #BB0000;">2</span> -<span style="color: #00BB00;">1</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr22</span> <span style="color: #BB0000;">3</span> -<span style="color: #00BB00;">3</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr3</span> <span style="color: #BB0000;">19</span> -<span style="color: #00BB00;">4</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr4</span> <span style="color: #BB0000;">8</span> -<span style="color: #00BB00;">2</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr5</span> <span style="color: #BB0000;">6</span> -<span style="color: #00BB00;">3</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr6</span> <span style="color: #BB0000;">4</span> -<span style="color: #00BB00;">2</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr7</span> <span style="color: #BB0000;">46</span> -<span style="color: #00BB00;">17</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr8</span> <span style="color: #BB0000;">18</span> -<span style="color: #00BB00;">3</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chr9</span> <span style="color: #BB0000;">3</span> -<span style="color: #00BB00;">2</span> @</span></span>
<span><span class="co">#&gt; → <span style="color: #0000BB;">chrX</span> <span style="color: #BB0000;">6</span> -<span style="color: #00BB00;">2</span> @</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Smoothed from 267 to 87 segments with 1e+06 gap (bases).</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Creating a new CNAqc object. The old object will be retained in the $before_smoothing field.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">CNAqc - CNA Quality Check</span> <span style="color: #00BBBB;">───────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">12963</span> somatic mutations: <span style="color: #00BB00;">12963</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">0</span> indels.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">87</span> segments: <span style="color: #00BB00;">87</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span></span>
<span><span class="co">#&gt; Warning in map_mutations_to_clonal_segments(mutations, cna_clonal): [CNAqc] a</span></span>
<span><span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">12963</span> mutations mapped to clonal CNAs.</span></span>
<span></span>
<span><span class="co"># Re-compute the fragmentation</span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/detect_arm_overfragmentation.html">detect_arm_overfragmentation</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> One-tailed Binomial test: 2 tests, alpha 0.01. Short segments: 0.2% of the reference arm.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">chr7p</span>,  p = 4.52608e-06 ~ 12 segments, 10 short.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">chr12q</span>,  p = 4.19839999999999e-06 ~ 10 segments, 9 short.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> 2 significantly overfragmented chromosome arms (alpha level 0.01).</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] </span> <span style="color: #00BB00;">12963</span> mutations in <span style="color: #00BB00;">87</span> segments (<span style="color: #00BB00;">87</span> clonal, <span style="color: #00BB00;">0</span> subclonal). Genome re</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Clonal CNAs</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    2:2  [n = 7478, L = 1493 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { <span style="color: #BBBB00;">CTCF</span> }</span></span>
<span><span class="co">#&gt;    4:2  [n = 1893, L =  333 Mb] ■■■■■■■</span></span>
<span><span class="co">#&gt;    3:2  [n = 1625, L =  362 Mb] ■■■■■■</span></span>
<span><span class="co">#&gt;    2:1  [n = 1563, L =  424 Mb] ■■■■■■  { <span style="color: #BBBB00;">TTN</span> }</span></span>
<span><span class="co">#&gt;    3:0  [n =  312, L =  139 Mb] ■</span></span>
<span><span class="co">#&gt;    2:0  [n =   81, L =   39 Mb]   { <span style="color: #BBBB00;">TP53</span> }</span></span>
<span><span class="co">#&gt;   16:2  [n =    4, L =    0 Mb] </span></span>
<span><span class="co">#&gt;   25:2  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;    3:1  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;  106:1  [n =    1, L =    0 Mb]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample Purity: 89% ~ Ploidy: 4.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> There are 3 annotated driver(s) mapped to clonal CNAs.</span></span>
<span><span class="co">#&gt;          chr      from        to ref alt  DP NV       VAF driver_label is_driver</span></span>
<span><span class="co">#&gt;         chr2 179431633 179431634   C   T 117 77 0.6581197          TTN      TRUE</span></span>
<span><span class="co">#&gt;        chr16  67646006  67646007   C   T 120 54 0.4500000         CTCF      TRUE</span></span>
<span><span class="co">#&gt;        chr17   7577106   7577107   G   C  84 78 0.9285714         TP53      TRUE</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> These segments are smoothed; before smoothing there were 267 segments.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Arm-level fragmentation analysis: 2 segments overfragmented.</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_arm_fragmentation.html">plot_arm_fragmentation</a></span><span class="op">(</span><span class="va">x</span>, zoom <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p><img src="a6_fragmentation_files/figure-html/unnamed-chunk-8-1.png" width="576"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.caravagnalab.org" class="external-link">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
