<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>6. Genome fragmentation analysis • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="6. Genome fragmentation analysis">
<meta property="og:description" content="CNAqc">
<meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a6_fragmentation_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>6. Genome fragmentation analysis</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqcvignettes/a6_fragmentation.Rmd"><code>vignettes/a6_fragmentation.Rmd</code></a></small>
      <div class="hidden name"><code>a6_fragmentation.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/CNAqc">CNAqc</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span>
<span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'BMix'</span>
<span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading</span>
<span class="co">#&gt; 'BMix'</span>
<span class="co">#&gt; ✔ Loading BMix, 'Binomial and Beta-Binomial univariate mixtures'. Support : &lt;https://caravagnalab.github.io/BMix/&gt;</span>
<span class="co">#&gt; ✔ Loading CNAqc, 'Copy Number Alteration quality check'. Support : &lt;https://caravagn.github.io/CNAqc/&gt;</span></code></pre></div>
<p>We work with the template dataset.</p>
<pre><code>#&gt;  [ CNAqc - CNA Quality Check ]
#&gt; 
#&gt;    2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { CTCF }
#&gt;    4:2  [n = 1893, L =  331 Mb] ■■■■■■■
#&gt;    3:2  [n = 1625, L =  357 Mb] ■■■■■■
#&gt;    2:1  [n = 1563, L =  420 Mb] ■■■■■■  { TTN }
#&gt;    3:0  [n =  312, L =  137 Mb] ■
#&gt;    2:0  [n =   81, L =   39 Mb]   { TP53 }
#&gt;   16:2  [n =    4, L =    0 Mb] 
#&gt;   25:2  [n =    2, L =    1 Mb] 
#&gt;    3:1  [n =    2, L =    1 Mb] 
#&gt;  106:1  [n =    1, L =    0 Mb] 
#&gt;   26:2  [n =    1, L =    0 Mb] 
#&gt;   99:1  [n =    1, L =    0 Mb] 
#&gt; 
#&gt; 
#&gt;          chr      from        to ref alt  DP NV       VAF driver_label is_driver
#&gt;         chr2 179431633 179431634   C   T 117 77 0.6581197          TTN      TRUE
#&gt;        chr16  67646006  67646007   C   T 120 54 0.4500000         CTCF      TRUE
#&gt;        chr17   7577106   7577107   G   C  84 78 0.9285714         TP53      TRUE</code></pre>
<div id="fragmentation-of-individual-arms" class="section level3">
<h3 class="hasAnchor">
<a href="#fragmentation-of-individual-arms" class="anchor"></a>Fragmentation of individual arms</h3>
<p>The fragmentation of a chromosome arm is assessed with a statistical test based on counting the size of the copy number segments mapping to the arm.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># A histogram of segments' lenght</span>
<span class="fu"><a href="../reference/plot_segment_size_distribution.html">plot_segment_size_distribution</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<p><img src="a6_fragmentation_files/figure-html/unnamed-chunk-3-1.png" width="288"></p>
<p><code>CNAqc</code> counts, for every arm with lenght <span class="math inline">\(L\)</span> nucleotides:</p>
<ul>
<li>
<span class="math inline">\(n_s\)</span>, the number of mapped CNA segments shorter than <span class="math inline">\(\delta\%\)</span> of <span class="math inline">\(L\)</span>;</li>
<li>
<span class="math inline">\(n_l\)</span>, the number of mapped CNA segments longer than <span class="math inline">\(\delta\%\)</span> sof <span class="math inline">\(L\)</span>.</li>
</ul>
<p>A one-sided Binomial test is used to compute a p-value for the null hypothesis of seeing <span class="math inline">\(n_s\)</span> observations in <span class="math inline">\(n_l\)</span> trials, assuming a Binomial success probability <span class="math inline">\(p = \delta &gt; 0\)</span>. <span class="math inline">\(p\)</span> represents a model where each segment length is equally likely (uniform distribution).</p>
<p>In this way the test accounts for the difference in lenghts of the chromsome arms; a p-value per arm is reported and adjusted for multiple hyoptheses (Bonferroni).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Test with default parameters (small segments: &lt; 20% of chromosome arm)</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/detect_arm_overfragmentation.html">detect_arm_overfragmentation</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ℹ One-tailed Binomial test: 8 tests, alpha 0.01. Short segments: 0.2% of the reference arm.</span>
<span class="co">#&gt; ✔ chr7p,  p = 1.71798691840001e-24 ~ 34 segments, 34 short.</span>
<span class="co">#&gt; ✔ chr1p,  p = 1.62738995200002e-15 ~ 24 segments, 23 short.</span>
<span class="co">#&gt; ✔ chr1q,  p = 4.34176000000001e-08 ~ 13 segments, 12 short.</span>
<span class="co">#&gt; ✔ chr11q,  p = 1.0657792e-06 ~ 13 segments, 11 short.</span>
<span class="co">#&gt; ✔ chr12q,  p = 2.00704e-07 ~ 12 segments, 11 short.</span>
<span class="co">#&gt; ✔ chr3q,  p = 4.52608e-06 ~ 12 segments, 10 short.</span>
<span class="co">#&gt; ✔ chr7q,  p = 4.52608e-06 ~ 12 segments, 10 short.</span>
<span class="co">#&gt; ✔ chr8p,  p = 9.21599999999998e-07 ~ 11 segments, 10 short.</span>
<span class="co">#&gt; ℹ 8 significantly overfragmented chromosome arms (alpha level 0.01).</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  n = 12963 mutations in 267 segments (267 clonal + 0 subclonal). Ge</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;    2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { CTCF }</span>
<span class="co">#&gt;    4:2  [n = 1893, L =  331 Mb] ■■■■■■■</span>
<span class="co">#&gt;    3:2  [n = 1625, L =  357 Mb] ■■■■■■</span>
<span class="co">#&gt;    2:1  [n = 1563, L =  420 Mb] ■■■■■■  { TTN }</span>
<span class="co">#&gt;    3:0  [n =  312, L =  137 Mb] ■</span>
<span class="co">#&gt;    2:0  [n =   81, L =   39 Mb]   { TP53 }</span>
<span class="co">#&gt;   16:2  [n =    4, L =    0 Mb] </span>
<span class="co">#&gt;   25:2  [n =    2, L =    1 Mb] </span>
<span class="co">#&gt;    3:1  [n =    2, L =    1 Mb] </span>
<span class="co">#&gt;  106:1  [n =    1, L =    0 Mb] </span>
<span class="co">#&gt;   26:2  [n =    1, L =    0 Mb] </span>
<span class="co">#&gt;   99:1  [n =    1, L =    0 Mb]</span>
<span class="co">#&gt; ℹ Sample Purity: 89% ~ Ploidy: 4.</span>
<span class="co">#&gt; ℹ There are 3 annotated driver(s).</span>
<span class="co">#&gt;          chr      from        to ref alt  DP NV       VAF driver_label is_driver</span>
<span class="co">#&gt;         chr2 179431633 179431634   C   T 117 77 0.6581197          TTN      TRUE</span>
<span class="co">#&gt;        chr16  67646006  67646007   C   T 120 54 0.4500000         CTCF      TRUE</span>
<span class="co">#&gt;        chr17   7577106   7577107   G   C  84 78 0.9285714         TP53      TRUE</span>
<span class="co">#&gt; ✔ Arm-level fragmentation analysis: 8 segments overfragmented.</span></code></pre></div>
<p>You can produce a arm-level report for the fragmentation test, with:</p>
<ul>
<li>a scatter of the counts per arm, with scaled the p-values;</li>
<li>a jump statistics per arm, <span class="math inline">\(J\)</span>.</li>
</ul>
<p><span class="math inline">\(J\)</span> is the sum of the variation in total copy number profiles, evaluated among each pair of contiguous segments.</p>
<p>Significantly overfragmented arms with high <span class="math inline">\(J\)</span> have a “scattered” copy number profile. Those with low <span class="math inline">\(J\)</span> are more uniform, as they show little no copy number change, and can be possibly smoothed (see below).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_arm_fragmentation.html">plot_arm_fragmentation</a></span><span class="op">(</span><span class="va">x</span>, zoom <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span></code></pre></div>
<p><img src="a6_fragmentation_files/figure-html/unnamed-chunk-5-1.png" width="576"></p>
<p>Once available, these results appear in any call to <code>plot_segments</code> as annotated purple squares sorrounding the arms.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Default plot has now segments</span>
<span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="a6_fragmentation_files/figure-html/unnamed-chunk-6-1.png" width="912"></p>
<p><a href="https://caravagn.github.io/CNAqc/articles/smoothing.html">Smoothing</a> is a good way to start cleaning up the fragmented sets of arms, as it merges segments that have no karyotype change, and are closer than a certain distance.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Smooth with default parameters</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/smooth_segments.html">smooth_segments</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; Smoothing chr1 with 37 segments: [1-24] [27-30] [31-33] [34-37] </span>
<span class="co">#&gt; Smoothing chr10 with 8 segments: [1-2] [3-4] [5-8] </span>
<span class="co">#&gt; Smoothing chr11 with 22 segments: [1-7] [8-9] [10-22] </span>
<span class="co">#&gt; Smoothing chr12 with 13 segments: [2-3] [12-13] </span>
<span class="co">#&gt; Smoothing chr14 with 2 segments: [1-2] </span>
<span class="co">#&gt; Smoothing chr15 with 9 segments: [1-5] [7-9] </span>
<span class="co">#&gt; Smoothing chr16 with 10 segments: [1-6] [8-10] </span>
<span class="co">#&gt; Smoothing chr17 with 10 segments: [4-5] [6-7] [8-10] </span>
<span class="co">#&gt; Smoothing chr18 with 8 segments: [1-2] [3-8] </span>
<span class="co">#&gt; Smoothing chr19 with 5 segments: [2-5] </span>
<span class="co">#&gt; Smoothing chr2 with 18 segments: [1-8] [10-12] [14-18] </span>
<span class="co">#&gt; Smoothing chr20 with 9 segments: [1-3] [4-9] </span>
<span class="co">#&gt; Smoothing chr21 with 2 segments: [1-2] </span>
<span class="co">#&gt; Smoothing chr22 with 3 segments: </span>
<span class="co">#&gt; Smoothing chr3 with 19 segments: [1-7] [8-13] [15-19] </span>
<span class="co">#&gt; Smoothing chr4 with 8 segments: [1-4] [5-8] </span>
<span class="co">#&gt; Smoothing chr5 with 6 segments: [1-2] [3-5] </span>
<span class="co">#&gt; Smoothing chr6 with 4 segments: [1-2] [3-4] </span>
<span class="co">#&gt; Smoothing chr7 with 46 segments: [1-5] [9-27] [36-38] [39-41] [43-46] </span>
<span class="co">#&gt; Smoothing chr8 with 18 segments: [1-4] [5-11] [12-18] </span>
<span class="co">#&gt; Smoothing chr9 with 3 segments: [1-2] </span>
<span class="co">#&gt; Smoothing chrX with 6 segments: [1-2] [3-6]</span>
<span class="co">#&gt; ✔ Smoothed from 267 to 87 segments with 1e+06 gap (bases).</span>
<span class="co">#&gt; ℹ Creating a new CNAqc object. The old object will be retained in the $before_smoothing field.</span>
<span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span>
<span class="co">#&gt; ℹ Using reference genome coordinates for: hg19.</span>
<span class="co">#&gt; ℹ Input n = 12963 mutations for 87 CNA segments (87 clonal, 0 subclonal)</span>
<span class="co">#&gt; Warning in map_mutations_to_segments(snvs, cna %&gt;% filter(CCF == 1)): [CNAqc] a</span>
<span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span>
<span class="co">#&gt; ✔ Mapped n = 12963 mutations to clonal segments (100% of input)</span>

<span class="co"># Re-compute the fragmentation</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/detect_arm_overfragmentation.html">detect_arm_overfragmentation</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ℹ One-tailed Binomial test: 2 tests, alpha 0.01. Short segments: 0.2% of the reference arm.</span>
<span class="co">#&gt; ✔ chr7p,  p = 4.52608e-06 ~ 12 segments, 10 short.</span>
<span class="co">#&gt; ✔ chr12q,  p = 4.19839999999999e-06 ~ 10 segments, 9 short.</span>
<span class="co">#&gt; ℹ 2 significantly overfragmented chromosome arms (alpha level 0.01).</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  n = 12963 mutations in 87 segments (87 clonal + 0 subclonal). Geno</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;    2:2  [n = 7478, L = 1493 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { CTCF }</span>
<span class="co">#&gt;    4:2  [n = 1893, L =  333 Mb] ■■■■■■■</span>
<span class="co">#&gt;    3:2  [n = 1625, L =  362 Mb] ■■■■■■</span>
<span class="co">#&gt;    2:1  [n = 1563, L =  424 Mb] ■■■■■■  { TTN }</span>
<span class="co">#&gt;    3:0  [n =  312, L =  139 Mb] ■</span>
<span class="co">#&gt;    2:0  [n =   81, L =   39 Mb]   { TP53 }</span>
<span class="co">#&gt;   16:2  [n =    4, L =    0 Mb] </span>
<span class="co">#&gt;   25:2  [n =    2, L =    1 Mb] </span>
<span class="co">#&gt;    3:1  [n =    2, L =    1 Mb] </span>
<span class="co">#&gt;  106:1  [n =    1, L =    0 Mb] </span>
<span class="co">#&gt;   26:2  [n =    1, L =    0 Mb] </span>
<span class="co">#&gt;   99:1  [n =    1, L =    0 Mb]</span>
<span class="co">#&gt; ℹ Sample Purity: 89% ~ Ploidy: 4.</span>
<span class="co">#&gt; ℹ There are 3 annotated driver(s).</span>
<span class="co">#&gt;          chr      from        to ref alt  DP NV       VAF driver_label is_driver</span>
<span class="co">#&gt;         chr2 179431633 179431634   C   T 117 77 0.6581197          TTN      TRUE</span>
<span class="co">#&gt;        chr16  67646006  67646007   C   T 120 54 0.4500000         CTCF      TRUE</span>
<span class="co">#&gt;        chr17   7577106   7577107   G   C  84 78 0.9285714         TP53      TRUE</span>
<span class="co">#&gt; ✔ These segments are smoothed; before smoothing there were 267 segments.</span>
<span class="co">#&gt; ✔ Arm-level fragmentation analysis: 2 segments overfragmented.</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_arm_fragmentation.html">plot_arm_fragmentation</a></span><span class="op">(</span><span class="va">x</span>, zoom <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span></code></pre></div>
<p><img src="a6_fragmentation_files/figure-html/unnamed-chunk-8-1.png" width="576"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.caravagnalab.org">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
