<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>7. Example data analysis • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="7. Example data analysis">
<meta property="og:description" content="CNAqc">
<meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a7_example_MSeq_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>7. Example data analysis</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqcvignettes/a7_example_MSeq.Rmd"><code>vignettes/a7_example_MSeq.Rmd</code></a></small>
      <div class="hidden name"><code>a7_example_MSeq.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/CNAqc">CNAqc</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span>
<span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'BMix'</span>
<span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading</span>
<span class="co">#&gt; 'BMix'</span>
<span class="co">#&gt; ✔ Loading BMix, 'Binomial and Beta-Binomial univariate mixtures'. Support : &lt;https://caravagnalab.github.io/BMix/&gt;</span>
<span class="co">#&gt; ✔ Loading CNAqc, 'Copy Number Alteration quality check'. Support : &lt;https://caravagn.github.io/CNAqc/&gt;</span>

<span class="co"># Extra packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://github.com/knausb/vcfR">vcfR</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: vcfR</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;    *****       ***   vcfR   ***       *****</span>
<span class="co">#&gt;    This is vcfR 1.12.0 </span>
<span class="co">#&gt;      browseVignettes('vcfR') # Documentation</span>
<span class="co">#&gt;      citation('vcfR') # Citation</span>
<span class="co">#&gt;    *****       *****      *****       *****</span></code></pre></div>
<p>We work with MSeq data discussed in the main preprint for <code>CNAqc</code>, replicating one analysis for patient <code>Set06</code>.</p>
<p>The data we used is hosted at the GitHub repository <a href="https://github.com/caravagnalab/CNAqc_datasets">caravagnalab/CNAqc_datasets</a>.</p>
<div id="mutation-data" class="section level1">
<h1 class="hasAnchor">
<a href="#mutation-data" class="anchor"></a>Mutation data</h1>
<p>We download from Github a VCF file for MSeq sample <code>Set06</code>, and load it using the <code>vcfR</code> package.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>VCF_url =<span class="st"> "https://raw.githubusercontent.com/caravagnalab/CNAqc_datasets/main/MSeq_Set06/Mutations/Set.06.WGS.merged_filtered.vcf"</span></span>
<span id="cb2-2"><a href="#cb2-2"></a></span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="co"># Download, load and cancel data</span></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="kw">download.file</span>(VCF_url, <span class="st">"Set.06.WGS.merged_filtered.vcf"</span>,)</span>
<span id="cb2-5"><a href="#cb2-5"></a></span>
<span id="cb2-6"><a href="#cb2-6"></a>set6 =<span class="st"> </span>vcfR<span class="op">::</span><span class="kw">read.vcfR</span>(<span class="st">"Set.06.WGS.merged_filtered.vcf"</span>)</span>
<span id="cb2-7"><a href="#cb2-7"></a><span class="co">#&gt; Scanning file to determine attributes.</span></span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="co">#&gt; File attributes:</span></span>
<span id="cb2-9"><a href="#cb2-9"></a><span class="co">#&gt;   meta lines: 47</span></span>
<span id="cb2-10"><a href="#cb2-10"></a><span class="co">#&gt;   header_line: 48</span></span>
<span id="cb2-11"><a href="#cb2-11"></a><span class="co">#&gt;   variant count: 43053</span></span>
<span id="cb2-12"><a href="#cb2-12"></a><span class="co">#&gt;   column count: 16</span></span>
<span id="cb2-13"><a href="#cb2-13"></a><span class="co">#&gt; </span></span>
<span id="cb2-14"><a href="#cb2-14"></a>Meta line <span class="dv">47</span> read in.</span>
<span id="cb2-15"><a href="#cb2-15"></a><span class="co">#&gt; All meta lines processed.</span></span>
<span id="cb2-16"><a href="#cb2-16"></a><span class="co">#&gt; gt matrix initialized.</span></span>
<span id="cb2-17"><a href="#cb2-17"></a><span class="co">#&gt; Character matrix gt created.</span></span>
<span id="cb2-18"><a href="#cb2-18"></a><span class="co">#&gt;   Character matrix gt rows: 43053</span></span>
<span id="cb2-19"><a href="#cb2-19"></a><span class="co">#&gt;   Character matrix gt cols: 16</span></span>
<span id="cb2-20"><a href="#cb2-20"></a><span class="co">#&gt;   skip: 0</span></span>
<span id="cb2-21"><a href="#cb2-21"></a><span class="co">#&gt;   nrows: 43053</span></span>
<span id="cb2-22"><a href="#cb2-22"></a><span class="co">#&gt;   row_num: 0</span></span>
<span id="cb2-23"><a href="#cb2-23"></a><span class="co">#&gt; </span></span>
<span id="cb2-24"><a href="#cb2-24"></a>Processed variant <span class="dv">1000</span></span>
<span id="cb2-25"><a href="#cb2-25"></a>Processed variant <span class="dv">2000</span></span>
<span id="cb2-26"><a href="#cb2-26"></a>Processed variant <span class="dv">3000</span></span>
<span id="cb2-27"><a href="#cb2-27"></a>Processed variant <span class="dv">4000</span></span>
<span id="cb2-28"><a href="#cb2-28"></a>Processed variant <span class="dv">5000</span></span>
<span id="cb2-29"><a href="#cb2-29"></a>Processed variant <span class="dv">6000</span></span>
<span id="cb2-30"><a href="#cb2-30"></a>Processed variant <span class="dv">7000</span></span>
<span id="cb2-31"><a href="#cb2-31"></a>Processed variant <span class="dv">8000</span></span>
<span id="cb2-32"><a href="#cb2-32"></a>Processed variant <span class="dv">9000</span></span>
<span id="cb2-33"><a href="#cb2-33"></a>Processed variant <span class="dv">10000</span></span>
<span id="cb2-34"><a href="#cb2-34"></a>Processed variant <span class="dv">11000</span></span>
<span id="cb2-35"><a href="#cb2-35"></a>Processed variant <span class="dv">12000</span></span>
<span id="cb2-36"><a href="#cb2-36"></a>Processed variant <span class="dv">13000</span></span>
<span id="cb2-37"><a href="#cb2-37"></a>Processed variant <span class="dv">14000</span></span>
<span id="cb2-38"><a href="#cb2-38"></a>Processed variant <span class="dv">15000</span></span>
<span id="cb2-39"><a href="#cb2-39"></a>Processed variant <span class="dv">16000</span></span>
<span id="cb2-40"><a href="#cb2-40"></a>Processed variant <span class="dv">17000</span></span>
<span id="cb2-41"><a href="#cb2-41"></a>Processed variant <span class="dv">18000</span></span>
<span id="cb2-42"><a href="#cb2-42"></a>Processed variant <span class="dv">19000</span></span>
<span id="cb2-43"><a href="#cb2-43"></a>Processed variant <span class="dv">20000</span></span>
<span id="cb2-44"><a href="#cb2-44"></a>Processed variant <span class="dv">21000</span></span>
<span id="cb2-45"><a href="#cb2-45"></a>Processed variant <span class="dv">22000</span></span>
<span id="cb2-46"><a href="#cb2-46"></a>Processed variant <span class="dv">23000</span></span>
<span id="cb2-47"><a href="#cb2-47"></a>Processed variant <span class="dv">24000</span></span>
<span id="cb2-48"><a href="#cb2-48"></a>Processed variant <span class="dv">25000</span></span>
<span id="cb2-49"><a href="#cb2-49"></a>Processed variant <span class="dv">26000</span></span>
<span id="cb2-50"><a href="#cb2-50"></a>Processed variant <span class="dv">27000</span></span>
<span id="cb2-51"><a href="#cb2-51"></a>Processed variant <span class="dv">28000</span></span>
<span id="cb2-52"><a href="#cb2-52"></a>Processed variant <span class="dv">29000</span></span>
<span id="cb2-53"><a href="#cb2-53"></a>Processed variant <span class="dv">30000</span></span>
<span id="cb2-54"><a href="#cb2-54"></a>Processed variant <span class="dv">31000</span></span>
<span id="cb2-55"><a href="#cb2-55"></a>Processed variant <span class="dv">32000</span></span>
<span id="cb2-56"><a href="#cb2-56"></a>Processed variant <span class="dv">33000</span></span>
<span id="cb2-57"><a href="#cb2-57"></a>Processed variant <span class="dv">34000</span></span>
<span id="cb2-58"><a href="#cb2-58"></a>Processed variant <span class="dv">35000</span></span>
<span id="cb2-59"><a href="#cb2-59"></a>Processed variant <span class="dv">36000</span></span>
<span id="cb2-60"><a href="#cb2-60"></a>Processed variant <span class="dv">37000</span></span>
<span id="cb2-61"><a href="#cb2-61"></a>Processed variant <span class="dv">38000</span></span>
<span id="cb2-62"><a href="#cb2-62"></a>Processed variant <span class="dv">39000</span></span>
<span id="cb2-63"><a href="#cb2-63"></a>Processed variant <span class="dv">40000</span></span>
<span id="cb2-64"><a href="#cb2-64"></a>Processed variant <span class="dv">41000</span></span>
<span id="cb2-65"><a href="#cb2-65"></a>Processed variant <span class="dv">42000</span></span>
<span id="cb2-66"><a href="#cb2-66"></a>Processed variant <span class="dv">43000</span></span>
<span id="cb2-67"><a href="#cb2-67"></a>Processed variant<span class="op">:</span><span class="st"> </span><span class="dv">43053</span></span>
<span id="cb2-68"><a href="#cb2-68"></a><span class="co">#&gt; All variants processed</span></span>
<span id="cb2-69"><a href="#cb2-69"></a></span>
<span id="cb2-70"><a href="#cb2-70"></a><span class="kw">file.remove</span>(<span class="st">"Set.06.WGS.merged_filtered.vcf"</span>)</span>
<span id="cb2-71"><a href="#cb2-71"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb2-72"><a href="#cb2-72"></a></span>
<span id="cb2-73"><a href="#cb2-73"></a><span class="co"># VCF</span></span>
<span id="cb2-74"><a href="#cb2-74"></a><span class="kw">print</span>(set6)</span>
<span id="cb2-75"><a href="#cb2-75"></a><span class="co">#&gt; ***** Object of Class vcfR *****</span></span>
<span id="cb2-76"><a href="#cb2-76"></a><span class="co">#&gt; 7 samples</span></span>
<span id="cb2-77"><a href="#cb2-77"></a><span class="co">#&gt; 24 CHROMs</span></span>
<span id="cb2-78"><a href="#cb2-78"></a><span class="co">#&gt; 43,053 variants</span></span>
<span id="cb2-79"><a href="#cb2-79"></a><span class="co">#&gt; Object size: 43.5 Mb</span></span>
<span id="cb2-80"><a href="#cb2-80"></a><span class="co">#&gt; 0 percent missing data</span></span>
<span id="cb2-81"><a href="#cb2-81"></a><span class="co">#&gt; *****        *****         *****</span></span></code></pre></div>
<p>We extract all the information we need, using the tidy data representation format.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># INFO fields </span>
<span class="va">info_tidy</span> <span class="op">=</span> <span class="fu">vcfR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/vcfR_to_tidy_conversion.html">extract_info_tidy</a></span><span class="op">(</span><span class="va">set6</span><span class="op">)</span>

<span class="co"># Fixed fields (genomic coordinates)</span>
<span class="va">fix_tidy</span> <span class="op">=</span> <span class="va">set6</span><span class="op">@</span><span class="va">fix</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="va">as_tibble</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>
    chr <span class="op">=</span> <span class="va">CHROM</span>,
    from <span class="op">=</span> <span class="va">POS</span>,
    ref <span class="op">=</span> <span class="va">REF</span>,
    alt <span class="op">=</span> <span class="va">ALT</span>
  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">from</span><span class="op">)</span>, to <span class="op">=</span> <span class="va">from</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="va">alt</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Genotypes</span>
<span class="va">geno_tidy</span> <span class="op">=</span> <span class="fu">vcfR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/vcfR_to_tidy_conversion.html">extract_gt_tidy</a></span><span class="op">(</span><span class="va">set6</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_split.html">group_split</a></span><span class="op">(</span><span class="va">Indiv</span><span class="op">)</span>
<span class="co">#&gt; Extracting gt element GT</span>
<span class="co">#&gt; Extracting gt element GQ</span>
<span class="co">#&gt; Extracting gt element GOF</span>
<span class="co">#&gt; Extracting gt element NR</span>
<span class="co">#&gt; Extracting gt element GL</span>
<span class="co">#&gt; Extracting gt element NV</span>

<span class="co"># Sample mutations in the CNAqc format</span>
<span class="va">sample_mutations</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">geno_tidy</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_cols</a></span><span class="op">(</span><span class="va">info_tidy</span>, <span class="va">fix_tidy</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">full_join</a></span><span class="op">(</span><span class="va">x</span>, by <span class="op">=</span> <span class="st">"Key"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>DP <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">gt_NR</span><span class="op">)</span>, NV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">gt_NV</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>VAF <span class="op">=</span> <span class="va">NV</span> <span class="op">/</span> <span class="va">DP</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">chr</span>, <span class="va">from</span>, <span class="va">to</span>, <span class="va">ref</span>, <span class="va">alt</span>, <span class="va">NV</span>, <span class="va">DP</span>, <span class="va">VAF</span>, <span class="fu"><a href="https://tidyselect.r-lib.org/reference/everything.html">everything</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">VAF</span><span class="op">)</span>, <span class="va">VAF</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="co"># A list for all samples available</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">sample_mutations</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">sample_mutations</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">$</span><span class="va">Indiv</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="va">sample_mutations</span> <span class="op">=</span> <span class="va">sample_mutations</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">sample_mutations</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<p>We have all the somatic mutations called for <code>Set06</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">sample_mutations</span><span class="op">)</span>
<span class="co">#&gt; $Set6_42</span>
<span class="co">#&gt; # A tibble: 33,245 × 46</span>
<span class="co">#&gt;    chr     from     to ref     alt       NV    DP    VAF   Key FR     MMLQ   TCR</span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;int&gt; &lt;chr&gt; &lt;dbl&gt; &lt;int&gt;</span>
<span class="co">#&gt;  1 chr1   30923  30924 G       T          2    41 0.0488     1 0.42…    37    98</span>
<span class="co">#&gt;  2 chr1   83906  83907 A       G          1    56 0.0179     2 0.28…    37   132</span>
<span class="co">#&gt;  3 chr1   83911  83930 AAGAGA… TAGAA…     2    59 0.0339     3 0.07…    37   133</span>
<span class="co">#&gt;  4 chr1   83933  83934 A       G          1    59 0.0169     4 0.00…   100   141</span>
<span class="co">#&gt;  5 chr1  137123 137124 T       G          5    44 0.114      5 0.14…    37    87</span>
<span class="co">#&gt;  6 chr1  797474 797475 G       T          6    75 0.08       6 0.12…    37   222</span>
<span class="co">#&gt;  7 chr1  809123 809124 G       A          2    92 0.0217     7 0.14…    37   235</span>
<span class="co">#&gt;  8 chr1  809135 809136 C       T          2    91 0.0220     8 0.14…    37   236</span>
<span class="co">#&gt;  9 chr1  812625 812642 GCTGTG… TCTGT…     1    90 0.0111     9 0.14…    37   223</span>
<span class="co">#&gt; 10 chr1  815277 815278 T       C         10    70 0.143     10 0.35…    37   196</span>
<span class="co">#&gt; # … with 33,235 more rows, and 34 more variables: HP &lt;int&gt;, WE &lt;int&gt;,</span>
<span class="co">#&gt; #   Source &lt;chr&gt;, FS &lt;chr&gt;, WS &lt;int&gt;, PP &lt;chr&gt;, TR &lt;chr&gt;, NF &lt;chr&gt;, TCF &lt;int&gt;,</span>
<span class="co">#&gt; #   NR &lt;chr&gt;, TC &lt;int&gt;, END &lt;chr&gt;, MGOF &lt;chr&gt;, SbPval &lt;chr&gt;, START &lt;chr&gt;,</span>
<span class="co">#&gt; #   ReadPosRankSum &lt;chr&gt;, MQ &lt;chr&gt;, QD &lt;dbl&gt;, SC &lt;chr&gt;, BRF &lt;dbl&gt;,</span>
<span class="co">#&gt; #   HapScore &lt;chr&gt;, Size &lt;chr&gt;, ID &lt;chr&gt;, QUAL &lt;chr&gt;, FILTER &lt;chr&gt;, INFO &lt;chr&gt;,</span>
<span class="co">#&gt; #   Indiv &lt;chr&gt;, gt_GT &lt;chr&gt;, gt_GQ &lt;chr&gt;, gt_GOF &lt;chr&gt;, gt_NR &lt;chr&gt;,</span>
<span class="co">#&gt; #   gt_GL &lt;chr&gt;, gt_NV &lt;chr&gt;, gt_GT_alleles &lt;chr&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Set6_44</span>
<span class="co">#&gt; # A tibble: 32,960 × 46</span>
<span class="co">#&gt;    chr     from     to ref   alt      NV    DP    VAF   Key FR      MMLQ   TCR</span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;int&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;int&gt;</span>
<span class="co">#&gt;  1 chr1   30923  30924 G     T         2    20 0.1        1 0.4275    37    98</span>
<span class="co">#&gt;  2 chr1   83906  83907 A     G         3    39 0.0769     2 0.2857    37   132</span>
<span class="co">#&gt;  3 chr1   83933  83934 A     G         3    46 0.0652     4 0.0000   100   141</span>
<span class="co">#&gt;  4 chr1  137123 137124 T     G         1    29 0.0345     5 0.1438    37    87</span>
<span class="co">#&gt;  5 chr1  797474 797475 G     T         5    78 0.0641     6 0.1291    37   222</span>
<span class="co">#&gt;  6 chr1  809123 809124 G     A         3    67 0.0448     7 0.1429    37   235</span>
<span class="co">#&gt;  7 chr1  809135 809136 C     T         2    66 0.0303     8 0.1429    37   236</span>
<span class="co">#&gt;  8 chr1  815277 815278 T     C         4    70 0.0571    10 0.3569    37   196</span>
<span class="co">#&gt;  9 chr1  815316 815319 GTT   ATC       3    80 0.0375    11 0.1403    37   191</span>
<span class="co">#&gt; 10 chr1  819221 819225 GTTA  ATTC      2    55 0.0364    12 0.0715    23   198</span>
<span class="co">#&gt; # … with 32,950 more rows, and 34 more variables: HP &lt;int&gt;, WE &lt;int&gt;,</span>
<span class="co">#&gt; #   Source &lt;chr&gt;, FS &lt;chr&gt;, WS &lt;int&gt;, PP &lt;chr&gt;, TR &lt;chr&gt;, NF &lt;chr&gt;, TCF &lt;int&gt;,</span>
<span class="co">#&gt; #   NR &lt;chr&gt;, TC &lt;int&gt;, END &lt;chr&gt;, MGOF &lt;chr&gt;, SbPval &lt;chr&gt;, START &lt;chr&gt;,</span>
<span class="co">#&gt; #   ReadPosRankSum &lt;chr&gt;, MQ &lt;chr&gt;, QD &lt;dbl&gt;, SC &lt;chr&gt;, BRF &lt;dbl&gt;,</span>
<span class="co">#&gt; #   HapScore &lt;chr&gt;, Size &lt;chr&gt;, ID &lt;chr&gt;, QUAL &lt;chr&gt;, FILTER &lt;chr&gt;, INFO &lt;chr&gt;,</span>
<span class="co">#&gt; #   Indiv &lt;chr&gt;, gt_GT &lt;chr&gt;, gt_GQ &lt;chr&gt;, gt_GOF &lt;chr&gt;, gt_NR &lt;chr&gt;,</span>
<span class="co">#&gt; #   gt_GL &lt;chr&gt;, gt_NV &lt;chr&gt;, gt_GT_alleles &lt;chr&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Set6_45</span>
<span class="co">#&gt; # A tibble: 32,782 × 46</span>
<span class="co">#&gt;    chr     from     to ref     alt       NV    DP    VAF   Key FR     MMLQ   TCR</span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;int&gt; &lt;chr&gt; &lt;dbl&gt; &lt;int&gt;</span>
<span class="co">#&gt;  1 chr1   30923  30924 G       T          2    42 0.0476     1 0.42…    37    98</span>
<span class="co">#&gt;  2 chr1   83906  83907 A       G          3    57 0.0526     2 0.28…    37   132</span>
<span class="co">#&gt;  3 chr1   83911  83930 AAGAGA… TAGAA…     5    57 0.0877     3 0.07…    37   133</span>
<span class="co">#&gt;  4 chr1   83933  83934 A       G          3    59 0.0508     4 0.00…   100   141</span>
<span class="co">#&gt;  5 chr1  137123 137124 T       G          3    34 0.0882     5 0.14…    37    87</span>
<span class="co">#&gt;  6 chr1  797474 797475 G       T          3    64 0.0469     6 0.12…    37   222</span>
<span class="co">#&gt;  7 chr1  809123 809124 G       A          3    71 0.0423     7 0.14…    37   235</span>
<span class="co">#&gt;  8 chr1  809135 809136 C       T          3    75 0.04       8 0.14…    37   236</span>
<span class="co">#&gt;  9 chr1  815277 815278 T       C          9    81 0.111     10 0.35…    37   196</span>
<span class="co">#&gt; 10 chr1  815316 815319 GTT     ATC        3    89 0.0337    11 0.14…    37   191</span>
<span class="co">#&gt; # … with 32,772 more rows, and 34 more variables: HP &lt;int&gt;, WE &lt;int&gt;,</span>
<span class="co">#&gt; #   Source &lt;chr&gt;, FS &lt;chr&gt;, WS &lt;int&gt;, PP &lt;chr&gt;, TR &lt;chr&gt;, NF &lt;chr&gt;, TCF &lt;int&gt;,</span>
<span class="co">#&gt; #   NR &lt;chr&gt;, TC &lt;int&gt;, END &lt;chr&gt;, MGOF &lt;chr&gt;, SbPval &lt;chr&gt;, START &lt;chr&gt;,</span>
<span class="co">#&gt; #   ReadPosRankSum &lt;chr&gt;, MQ &lt;chr&gt;, QD &lt;dbl&gt;, SC &lt;chr&gt;, BRF &lt;dbl&gt;,</span>
<span class="co">#&gt; #   HapScore &lt;chr&gt;, Size &lt;chr&gt;, ID &lt;chr&gt;, QUAL &lt;chr&gt;, FILTER &lt;chr&gt;, INFO &lt;chr&gt;,</span>
<span class="co">#&gt; #   Indiv &lt;chr&gt;, gt_GT &lt;chr&gt;, gt_GQ &lt;chr&gt;, gt_GOF &lt;chr&gt;, gt_NR &lt;chr&gt;,</span>
<span class="co">#&gt; #   gt_GL &lt;chr&gt;, gt_NV &lt;chr&gt;, gt_GT_alleles &lt;chr&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Set6_46</span>
<span class="co">#&gt; # A tibble: 33,907 × 46</span>
<span class="co">#&gt;    chr     from     to ref     alt       NV    DP    VAF   Key FR     MMLQ   TCR</span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;int&gt; &lt;chr&gt; &lt;dbl&gt; &lt;int&gt;</span>
<span class="co">#&gt;  1 chr1   30923  30924 G       T          2    22 0.0909     1 0.42…    37    98</span>
<span class="co">#&gt;  2 chr1   83906  83907 A       G          3    35 0.0857     2 0.28…    37   132</span>
<span class="co">#&gt;  3 chr1   83933  83934 A       G          6    38 0.158      4 0.00…   100   141</span>
<span class="co">#&gt;  4 chr1  137123 137124 T       G          1    30 0.0333     5 0.14…    37    87</span>
<span class="co">#&gt;  5 chr1  797474 797475 G       T          4    65 0.0615     6 0.12…    37   222</span>
<span class="co">#&gt;  6 chr1  809123 809124 G       A          6    96 0.0625     7 0.14…    37   235</span>
<span class="co">#&gt;  7 chr1  809135 809136 C       T          5    94 0.0532     8 0.14…    37   236</span>
<span class="co">#&gt;  8 chr1  812625 812642 GCTGTG… TCTGT…     3    87 0.0345     9 0.14…    37   223</span>
<span class="co">#&gt;  9 chr1  815277 815278 T       C         10    69 0.145     10 0.35…    37   196</span>
<span class="co">#&gt; 10 chr1  815316 815319 GTT     ATC        3    85 0.0353    11 0.14…    37   191</span>
<span class="co">#&gt; # … with 33,897 more rows, and 34 more variables: HP &lt;int&gt;, WE &lt;int&gt;,</span>
<span class="co">#&gt; #   Source &lt;chr&gt;, FS &lt;chr&gt;, WS &lt;int&gt;, PP &lt;chr&gt;, TR &lt;chr&gt;, NF &lt;chr&gt;, TCF &lt;int&gt;,</span>
<span class="co">#&gt; #   NR &lt;chr&gt;, TC &lt;int&gt;, END &lt;chr&gt;, MGOF &lt;chr&gt;, SbPval &lt;chr&gt;, START &lt;chr&gt;,</span>
<span class="co">#&gt; #   ReadPosRankSum &lt;chr&gt;, MQ &lt;chr&gt;, QD &lt;dbl&gt;, SC &lt;chr&gt;, BRF &lt;dbl&gt;,</span>
<span class="co">#&gt; #   HapScore &lt;chr&gt;, Size &lt;chr&gt;, ID &lt;chr&gt;, QUAL &lt;chr&gt;, FILTER &lt;chr&gt;, INFO &lt;chr&gt;,</span>
<span class="co">#&gt; #   Indiv &lt;chr&gt;, gt_GT &lt;chr&gt;, gt_GQ &lt;chr&gt;, gt_GOF &lt;chr&gt;, gt_NR &lt;chr&gt;,</span>
<span class="co">#&gt; #   gt_GL &lt;chr&gt;, gt_NV &lt;chr&gt;, gt_GT_alleles &lt;chr&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Set6_47</span>
<span class="co">#&gt; # A tibble: 32,814 × 46</span>
<span class="co">#&gt;    chr     from     to ref     alt       NV    DP    VAF   Key FR     MMLQ   TCR</span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;int&gt; &lt;chr&gt; &lt;dbl&gt; &lt;int&gt;</span>
<span class="co">#&gt;  1 chr1   30923  30924 G       T          3    12 0.25       1 0.42…    37    98</span>
<span class="co">#&gt;  2 chr1   83906  83907 A       G          6    41 0.146      2 0.28…    37   132</span>
<span class="co">#&gt;  3 chr1   83933  83934 A       G          7    39 0.179      4 0.00…   100   141</span>
<span class="co">#&gt;  4 chr1  137123 137124 T       G          7    38 0.184      5 0.14…    37    87</span>
<span class="co">#&gt;  5 chr1  797474 797475 G       T          4    68 0.0588     6 0.12…    37   222</span>
<span class="co">#&gt;  6 chr1  809123 809124 G       A          4    84 0.0476     7 0.14…    37   235</span>
<span class="co">#&gt;  7 chr1  809135 809136 C       T          2    84 0.0238     8 0.14…    37   236</span>
<span class="co">#&gt;  8 chr1  812625 812642 GCTGTG… TCTGT…     4    71 0.0563     9 0.14…    37   223</span>
<span class="co">#&gt;  9 chr1  815277 815278 T       C          6    67 0.0896    10 0.35…    37   196</span>
<span class="co">#&gt; 10 chr1  815316 815319 GTT     ATC        2    68 0.0294    11 0.14…    37   191</span>
<span class="co">#&gt; # … with 32,804 more rows, and 34 more variables: HP &lt;int&gt;, WE &lt;int&gt;,</span>
<span class="co">#&gt; #   Source &lt;chr&gt;, FS &lt;chr&gt;, WS &lt;int&gt;, PP &lt;chr&gt;, TR &lt;chr&gt;, NF &lt;chr&gt;, TCF &lt;int&gt;,</span>
<span class="co">#&gt; #   NR &lt;chr&gt;, TC &lt;int&gt;, END &lt;chr&gt;, MGOF &lt;chr&gt;, SbPval &lt;chr&gt;, START &lt;chr&gt;,</span>
<span class="co">#&gt; #   ReadPosRankSum &lt;chr&gt;, MQ &lt;chr&gt;, QD &lt;dbl&gt;, SC &lt;chr&gt;, BRF &lt;dbl&gt;,</span>
<span class="co">#&gt; #   HapScore &lt;chr&gt;, Size &lt;chr&gt;, ID &lt;chr&gt;, QUAL &lt;chr&gt;, FILTER &lt;chr&gt;, INFO &lt;chr&gt;,</span>
<span class="co">#&gt; #   Indiv &lt;chr&gt;, gt_GT &lt;chr&gt;, gt_GQ &lt;chr&gt;, gt_GOF &lt;chr&gt;, gt_NR &lt;chr&gt;,</span>
<span class="co">#&gt; #   gt_GL &lt;chr&gt;, gt_NV &lt;chr&gt;, gt_GT_alleles &lt;chr&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Set6_48</span>
<span class="co">#&gt; # A tibble: 32,859 × 46</span>
<span class="co">#&gt;    chr     from     to ref     alt       NV    DP    VAF   Key FR     MMLQ   TCR</span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;int&gt; &lt;chr&gt; &lt;dbl&gt; &lt;int&gt;</span>
<span class="co">#&gt;  1 chr1   30923  30924 G       T          3    14 0.214      1 0.42…    37    98</span>
<span class="co">#&gt;  2 chr1   83906  83907 A       G          2    28 0.0714     2 0.28…    37   132</span>
<span class="co">#&gt;  3 chr1   83933  83934 A       G          3    36 0.0833     4 0.00…   100   141</span>
<span class="co">#&gt;  4 chr1  137123 137124 T       G          2    36 0.0556     5 0.14…    37    87</span>
<span class="co">#&gt;  5 chr1  797474 797475 G       T          6    69 0.0870     6 0.12…    37   222</span>
<span class="co">#&gt;  6 chr1  809123 809124 G       A          5    76 0.0658     7 0.14…    37   235</span>
<span class="co">#&gt;  7 chr1  809135 809136 C       T          5    78 0.0641     8 0.14…    37   236</span>
<span class="co">#&gt;  8 chr1  812625 812642 GCTGTG… TCTGT…     1    62 0.0161     9 0.14…    37   223</span>
<span class="co">#&gt;  9 chr1  815277 815278 T       C          6    58 0.103     10 0.35…    37   196</span>
<span class="co">#&gt; 10 chr1  815316 815319 GTT     ATC        4    71 0.0563    11 0.14…    37   191</span>
<span class="co">#&gt; # … with 32,849 more rows, and 34 more variables: HP &lt;int&gt;, WE &lt;int&gt;,</span>
<span class="co">#&gt; #   Source &lt;chr&gt;, FS &lt;chr&gt;, WS &lt;int&gt;, PP &lt;chr&gt;, TR &lt;chr&gt;, NF &lt;chr&gt;, TCF &lt;int&gt;,</span>
<span class="co">#&gt; #   NR &lt;chr&gt;, TC &lt;int&gt;, END &lt;chr&gt;, MGOF &lt;chr&gt;, SbPval &lt;chr&gt;, START &lt;chr&gt;,</span>
<span class="co">#&gt; #   ReadPosRankSum &lt;chr&gt;, MQ &lt;chr&gt;, QD &lt;dbl&gt;, SC &lt;chr&gt;, BRF &lt;dbl&gt;,</span>
<span class="co">#&gt; #   HapScore &lt;chr&gt;, Size &lt;chr&gt;, ID &lt;chr&gt;, QUAL &lt;chr&gt;, FILTER &lt;chr&gt;, INFO &lt;chr&gt;,</span>
<span class="co">#&gt; #   Indiv &lt;chr&gt;, gt_GT &lt;chr&gt;, gt_GQ &lt;chr&gt;, gt_GOF &lt;chr&gt;, gt_NR &lt;chr&gt;,</span>
<span class="co">#&gt; #   gt_GL &lt;chr&gt;, gt_NV &lt;chr&gt;, gt_GT_alleles &lt;chr&gt;</span></code></pre></div>
</div>
<div id="copy-number-data" class="section level1">
<h1 class="hasAnchor">
<a href="#copy-number-data" class="anchor"></a>Copy Number data</h1>
<p>Sequenza calls are available in the same repository.</p>
<p>We use an extra function to load a solution (so we can easily compare multiple runs etc etc.).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load Sequenza output</span>
<span class="va">load_SQ_output</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">URL</span>, <span class="va">sample</span>, <span class="va">run</span><span class="op">)</span>
<span class="op">{</span>
  <span class="co"># We can directly read them from remote URLs</span>
  <span class="va">segments_file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">URL</span>, <span class="va">run</span>, <span class="st">'/'</span>, <span class="va">sample</span>, <span class="st">'.smoothedSegs.txt'</span><span class="op">)</span>
  <span class="va">purity_file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">URL</span>, <span class="va">run</span>, <span class="st">'/'</span>, <span class="va">sample</span>, <span class="st">'_confints_CP.txt'</span><span class="op">)</span>

  <span class="co"># Get segments</span>
  <span class="va">segments</span> <span class="op">=</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_tsv</a></span><span class="op">(</span><span class="va">segments_file</span>, col_types <span class="op">=</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/cols.html">cols</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>
      chr <span class="op">=</span> <span class="va">chromosome</span>,
      from <span class="op">=</span> <span class="va">start.pos</span>,
      to <span class="op">=</span> <span class="va">end.pos</span>,
      Major <span class="op">=</span> <span class="va">A</span>,
      minor <span class="op">=</span> <span class="va">B</span>
    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">chr</span>, <span class="va">from</span>, <span class="va">to</span>, <span class="va">Major</span>, <span class="va">minor</span>, <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/everything.html">everything</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>

  <span class="co"># Get purity and ploidy</span>
  <span class="va">solutions</span> <span class="op">=</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_tsv</a></span><span class="op">(</span><span class="va">purity_file</span>, col_types <span class="op">=</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/cols.html">cols</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>

  <span class="va">purity</span> <span class="op">=</span> <span class="va">solutions</span><span class="op">$</span><span class="va">cellularity</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>
  <span class="va">ploidy</span> <span class="op">=</span> <span class="va">solutions</span><span class="op">$</span><span class="va">ploidy.estimate</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>
  
  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
      segments <span class="op">=</span> <span class="va">segments</span>,
      purity <span class="op">=</span> <span class="va">purity</span>,
      ploidy <span class="op">=</span> <span class="va">ploidy</span>
    <span class="op">)</span>
  <span class="op">)</span>
<span class="op">}</span></code></pre></div>
<p>We can load the calls for 2 solutions of sample <code>Set6_42</code>; we begin with our final, good solution.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Sequenza_URL</span> <span class="op">=</span> <span class="st">"https://raw.githubusercontent.com/caravagnalab/CNAqc_datasets/main/MSeq_Set06/Copy%20Number/"</span>

<span class="co"># Final sequenza run (good calls)</span>
<span class="va">Sequenza_good_calls</span> <span class="op">=</span> <span class="fu">load_SQ_output</span><span class="op">(</span><span class="va">Sequenza_URL</span>, sample <span class="op">=</span> <span class="st">'Set6_42'</span>, run <span class="op">=</span> <span class="st">'final'</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">Sequenza_good_calls</span><span class="op">)</span>
<span class="co">#&gt; $segments</span>
<span class="co">#&gt; # A tibble: 54 × 7</span>
<span class="co">#&gt;    chr        from        to Major minor   CNt      size</span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;</span>
<span class="co">#&gt;  1 chr1    2847119  27674851     1     0     1  24827732</span>
<span class="co">#&gt;  2 chr1   27681534 120332727     1     1     2  92651193</span>
<span class="co">#&gt;  3 chr1  120333201 121485435     2     0     2   1152234</span>
<span class="co">#&gt;  4 chr1  142535470 246750621     1     1     2 104215151</span>
<span class="co">#&gt;  5 chr2    2500000  90500000     1     1     2  88000000</span>
<span class="co">#&gt;  6 chr2   96800000 240699373     1     1     2 143899373</span>
<span class="co">#&gt;  7 chr3    2500000  87900000     1     1     2  85400000</span>
<span class="co">#&gt;  8 chr3   93900000 195522430     1     1     2 101622430</span>
<span class="co">#&gt;  9 chr4    2500000  48200000     1     1     2  45700000</span>
<span class="co">#&gt; 10 chr4   52700000 188654276     1     1     2 135954276</span>
<span class="co">#&gt; # … with 44 more rows</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $purity</span>
<span class="co">#&gt; [1] 0.69</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $ploidy</span>
<span class="co">#&gt; [1] 2</span></code></pre></div>
</div>
<div id="calls-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#calls-analysis" class="anchor"></a>Calls analysis</h1>
<p>For this example we work with mutations found in sample <code>Set6_42</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Single-nucleotide variants with VAF &gt;5%</span>
<span class="va">snvs</span> <span class="op">=</span> <span class="va">sample_mutations</span><span class="op">[[</span><span class="st">'Set6_42'</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">ref</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'A'</span>, <span class="st">'C'</span>, <span class="st">"T"</span>, <span class="st">'G'</span><span class="op">)</span>, <span class="va">alt</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'A'</span>, <span class="st">'C'</span>, <span class="st">"T"</span>, <span class="st">'G'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">VAF</span> <span class="op">&gt;</span> <span class="fl">0.05</span><span class="op">)</span>

<span class="co"># CNA segments and purity</span>
<span class="va">cna</span> <span class="op">=</span> <span class="va">Sequenza_good_calls</span><span class="op">$</span><span class="va">segments</span>
<span class="va">purity</span> <span class="op">=</span> <span class="va">Sequenza_good_calls</span><span class="op">$</span><span class="va">purity</span></code></pre></div>
<p>Full CNAqc analysis. First we create the object.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># CNAqc data object</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span><span class="va">snvs</span>,
                <span class="va">cna</span>,
                <span class="va">purity</span>,
                ref <span class="op">=</span> <span class="st">"GRCh38"</span><span class="op">)</span>
<span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span>
<span class="co">#&gt; ℹ Using reference genome coordinates for: GRCh38.</span>
<span class="co">#&gt; ! Missing CCF column from CNA calls, adding CCF = 1 assuming clonal CNA calls.</span>
<span class="co">#&gt; ! Missing segments length from CNA calls, adding it to CNA calls.</span>
<span class="co">#&gt; ℹ Input n = 18858 mutations for 54 CNA segments (54 clonal, 0 subclonal)</span>
<span class="co">#&gt; ✖ n = 1575 mutations  cannot be mapped to segments and will be removed.</span>
<span class="co">#&gt; ✔ Mapped n = 17283 mutations to clonal segments (100% of input)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  n = 17283 mutations in 54 segments (54 clonal + 0 subclonal). Geno</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  1:1  [n = 14488, L =  2398 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;  2:0  [n =  1552, L =   200 Mb] ■■■</span>
<span class="co">#&gt;  1:0  [n =   898, L =   141 Mb] ■■</span>
<span class="co">#&gt;  2:1  [n =   345, L =    43 Mb] ■</span>
<span class="co">#&gt; ℹ Sample Purity: 69% ~ Ploidy: 2.</span></code></pre></div>
<div id="data" class="section level2">
<h2 class="hasAnchor">
<a href="#data" class="anchor"></a>Data</h2>
<p>Show the CNA data for this sample.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span>
  <span class="fu"><a href="../reference/plot_gw_counts.html">plot_gw_counts</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_gw_vaf.html">plot_gw_vaf</a></span><span class="op">(</span><span class="va">x</span>, N <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_gw_depth.html">plot_gw_depth</a></span><span class="op">(</span><span class="va">x</span>, N <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span>, highlight <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"1:0"</span>, <span class="st">"1:1"</span>, <span class="st">"2:0"</span>, <span class="st">"2:1"</span>, <span class="st">'2:2'</span><span class="op">)</span><span class="op">)</span>,
  align <span class="op">=</span> <span class="st">'v'</span>, 
  nrow <span class="op">=</span> <span class="fl">4</span>, 
  rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.15</span>, <span class="fl">.15</span>, <span class="fl">.15</span>, <span class="fl">.8</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-9-1.png" width="912"></p>
<p>Show the mutation data for this sample.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span><span class="op">(</span>
  <span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x</span>, which <span class="op">=</span> <span class="st">'VAF'</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x</span>, which <span class="op">=</span> <span class="st">'DP'</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x</span>, which <span class="op">=</span> <span class="st">'NV'</span><span class="op">)</span>,
  ncol <span class="op">=</span> <span class="fl">3</span>,
  nrow <span class="op">=</span> <span class="fl">1</span>
  <span class="op">)</span>
<span class="co">#&gt; Warning: Removed 8 rows containing missing values (geom_bar).</span></code></pre></div>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-10-1.png" width="1056"></p>
</div>
<div id="peak-detection" class="section level2">
<h2 class="hasAnchor">
<a href="#peak-detection" class="anchor"></a>Peak detection</h2>
<p>Perform peak detection and show its results.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Peaks</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/analyze_peaks.html">analyze_peaks</a></span><span class="op">(</span><span class="va">x</span>, matching_strategy <span class="op">=</span> <span class="st">'closest'</span><span class="op">)</span>
<span class="co">#&gt; ℹ Karyotypes 1:0, 1:1, 2:0, 2:1, and 2:2. Matching strategy closest. KDE = TRUE.</span>
<span class="co">#&gt; ℹ Found n = 17283 mutations in 1:1, 2:0, 1:0, 2:1 (skipping those with n &lt; 0 mutations).</span>
<span class="co">#&gt; ! Some peaks have been removed</span>
<span class="co">#&gt; # A tibble: 2 × 5</span>
<span class="co">#&gt;       x     y counts_per_bin discarded from </span>
<span class="co">#&gt;   &lt;dbl&gt; &lt;dbl&gt;          &lt;int&gt; &lt;lgl&gt;     &lt;chr&gt;</span>
<span class="co">#&gt; 1  0.68  0.03              6 TRUE      KDE  </span>
<span class="co">#&gt; 2  0.74  0.03              5 TRUE      KDE</span>
<span class="co">#&gt; ! Some peaks have been removed</span>
<span class="co">#&gt; # A tibble: 1 × 5</span>
<span class="co">#&gt;       x     y counts_per_bin discarded from </span>
<span class="co">#&gt;   &lt;dbl&gt; &lt;dbl&gt;          &lt;int&gt; &lt;lgl&gt;     &lt;chr&gt;</span>
<span class="co">#&gt; 1  0.81  0.11              0 TRUE      KDE  </span>
<span class="co">#&gt; # A tibble: 6 × 17</span>
<span class="co">#&gt; # Rowwise: </span>
<span class="co">#&gt;   mutation_multiplic… karyotype  peak     x     y counts_per_bin discarded from </span>
<span class="co">#&gt;                 &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;int&gt; &lt;lgl&gt;     &lt;chr&gt;</span>
<span class="co">#&gt; 1                   1 1:0       0.527 0.540 1.26              13 FALSE     BMix </span>
<span class="co">#&gt; 2                   1 1:1       0.345 0.349 3.65             604 FALSE     BMix </span>
<span class="co">#&gt; 3                   1 2:0       0.345 0.34  1.43              19 FALSE     KDE  </span>
<span class="co">#&gt; 4                   2 2:0       0.69  0.673 0.537             12 FALSE     BMix </span>
<span class="co">#&gt; 5                   1 2:1       0.257 0.284 1.51               5 FALSE     BMix </span>
<span class="co">#&gt; 6                   2 2:1       0.513 0.523 0.617              1 FALSE     BMix </span>
<span class="co">#&gt; # … with 9 more variables: offset_VAF &lt;dbl&gt;, offset &lt;dbl&gt;, weight &lt;dbl&gt;,</span>
<span class="co">#&gt; #   epsilon &lt;dbl&gt;, VAF_tolerance &lt;dbl&gt;, matched &lt;lgl&gt;, score &lt;dbl&gt;,</span>
<span class="co">#&gt; #   purity_error &lt;dbl&gt;, QC &lt;chr&gt;</span>
<span class="co">#&gt; ✔ Peak detection PASS with r = -0.0068011386514684 and tolerance e =</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  n = 17283 mutations in 54 segments (54 clonal + 0 subclonal). Geno</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  1:1  [n = 14488, L =  2398 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;  2:0  [n =  1552, L =   200 Mb] ■■■</span>
<span class="co">#&gt;  1:0  [n =   898, L =   141 Mb] ■■</span>
<span class="co">#&gt;  2:1  [n =   345, L =    43 Mb] ■</span>
<span class="co">#&gt; ℹ Sample Purity: 69% ~ Ploidy: 2.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ──  PASS  Peaks QC closest: 100%, λ = -0.0068. Purity correction: -1%.</span>
<span class="co">#&gt; ℹ 1:0 ~ n = 898   ( 5%) →  PASS  -0.011</span>
<span class="co">#&gt; ℹ 1:1 ~ n = 14488 (84%) →  PASS  -0.007</span>
<span class="co">#&gt; ℹ 2:0 ~ n = 1552  ( 9%) →  PASS  0.01     PASS  0.017</span>
<span class="co">#&gt; ℹ 2:1 ~ n = 345   ( 2%) →  PASS  -0.105   PASS  -0.019</span></code></pre></div>
<p>For this sample these calls are passed by CNAqc.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Do not assemble plots, and remove karyotypes with no data associated</span>
<span class="fu"><a href="../reference/plot_peaks_analysis.html">plot_peaks_analysis</a></span><span class="op">(</span><span class="va">x</span>, empty_plot <span class="op">=</span> <span class="cn">FALSE</span>, assembly_plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; [[1]]</span></code></pre></div>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-12-1.png" width="384"></p>
<pre><code>#&gt; 
#&gt; [[2]]</code></pre>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-12-2.png" width="384"></p>
<pre><code>#&gt; 
#&gt; [[3]]</code></pre>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-12-3.png" width="384"></p>
<pre><code>#&gt; 
#&gt; [[4]]</code></pre>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-12-4.png" width="384"></p>
</div>
<div id="ccf" class="section level2">
<h2 class="hasAnchor">
<a href="#ccf" class="anchor"></a>CCF</h2>
<p>Perform CCF computation detection with the <code>ENTROPY</code> method.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># CCF</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/compute_CCF.html">compute_CCF</a></span><span class="op">(</span><span class="va">x</span>, method <span class="op">=</span> <span class="st">'ENTROPY'</span><span class="op">)</span>
<span class="co">#&gt; ── Computing mutation multiplicity for single-copy karyotype 1:0 ───────────────</span>
<span class="co">#&gt; ── Computing mutation multiplicity for single-copy karyotype 1:1 ───────────────</span>
<span class="co">#&gt; ── Computing mutation multiplicity for karyotype 2:0 using the entropy method. ─</span>
<span class="co">#&gt; ℹ Expected Binomial peak(s) for these calls (1 and 2 copies): 0.345 and 0.69</span>
<span class="co">#&gt; ℹ Mixing pre/ post aneuploidy: 0.74 and 0.26</span>
<span class="co">#&gt; ℹ Not assignamble area: [0.452830188679245; 0.622641509433962]</span>
<span class="co">#&gt; ── Computing mutation multiplicity for karyotype 2:1 using the entropy method. ─</span>
<span class="co">#&gt; ℹ Expected Binomial peak(s) for these calls (1 and 2 copies): 0.256505576208178 and 0.513011152416357</span>
<span class="co">#&gt; ℹ Mixing pre/ post aneuploidy: 0.7 and 0.3</span>
<span class="co">#&gt; ℹ Not assignamble area: [0.336842105263158; 0.442105263157895]</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  n = 17283 mutations in 54 segments (54 clonal + 0 subclonal). Geno</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  1:1  [n = 14488, L =  2398 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;  2:0  [n =  1552, L =   200 Mb] ■■■</span>
<span class="co">#&gt;  1:0  [n =   898, L =   141 Mb] ■■</span>
<span class="co">#&gt;  2:1  [n =   345, L =    43 Mb] ■</span>
<span class="co">#&gt; ℹ Sample Purity: 69% ~ Ploidy: 2.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ──  PASS  Peaks QC closest: 100%, λ = -0.0068. Purity correction: -1%.</span>
<span class="co">#&gt; ℹ 1:0 ~ n = 898   ( 5%) →  PASS  -0.011</span>
<span class="co">#&gt; ℹ 1:1 ~ n = 14488 (84%) →  PASS  -0.007</span>
<span class="co">#&gt; ℹ 2:0 ~ n = 1552  ( 9%) →  PASS  0.01     PASS  0.017</span>
<span class="co">#&gt; ℹ 2:1 ~ n = 345   ( 2%) →  PASS  -0.105   PASS  -0.019</span>
<span class="co">#&gt; ✔ Cancer Cell Fraction (CCF) data available for karyotypes:1:0, 1:1, 2:0, and 2:1.</span>
<span class="co">#&gt; ✔  PASS  CCF via ENTROPY.</span>
<span class="co">#&gt; ✔  PASS  CCF via ENTROPY.</span>
<span class="co">#&gt; ✔  PASS  CCF via ENTROPY.</span>
<span class="co">#&gt; ✔  PASS  CCF via ENTROPY.</span></code></pre></div>
<p>CCF can be estimated well for this sample.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Do not assemble plots, and remove karyotypes with no data associated</span>
<span class="fu"><a href="../reference/plot_CCF.html">plot_CCF</a></span><span class="op">(</span><span class="va">x</span>, assembly_plot <span class="op">=</span> <span class="cn">FALSE</span>, empty_plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; [[1]]</span></code></pre></div>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-14-1.png" width="528"></p>
<pre><code>#&gt; 
#&gt; [[2]]</code></pre>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-14-2.png" width="528"></p>
<pre><code>#&gt; 
#&gt; [[3]]</code></pre>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-14-3.png" width="528"></p>
<pre><code>#&gt; 
#&gt; [[4]]</code></pre>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-14-4.png" width="528"></p>
</div>
<div id="other-analyses" class="section level2">
<h2 class="hasAnchor">
<a href="#other-analyses" class="anchor"></a>Other analyses</h2>
<p>Smooth segments with gaps up to 10 megabases (does not affect segments in this sample).</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/smooth_segments.html">smooth_segments</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; Smoothing chr1 with 4 segments: </span>
<span class="co">#&gt; Smoothing chr2 with 2 segments: </span>
<span class="co">#&gt; Smoothing chr3 with 2 segments: </span>
<span class="co">#&gt; Smoothing chr4 with 2 segments: </span>
<span class="co">#&gt; Smoothing chr5 with 3 segments: </span>
<span class="co">#&gt; Smoothing chr6 with 2 segments: </span>
<span class="co">#&gt; Smoothing chr7 with 2 segments: </span>
<span class="co">#&gt; Smoothing chr8 with 4 segments: </span>
<span class="co">#&gt; Smoothing chr9 with 3 segments: </span>
<span class="co">#&gt; Smoothing chr10 with 4 segments: </span>
<span class="co">#&gt; Smoothing chr11 with 2 segments: </span>
<span class="co">#&gt; Smoothing chr12 with 2 segments: </span>
<span class="co">#&gt; Smoothing chr15 with 2 segments: </span>
<span class="co">#&gt; Smoothing chr16 with 3 segments: </span>
<span class="co">#&gt; Smoothing chr17 with 2 segments: </span>
<span class="co">#&gt; Smoothing chr18 with 2 segments: </span>
<span class="co">#&gt; Smoothing chr19 with 2 segments: </span>
<span class="co">#&gt; Smoothing chr20 with 2 segments: </span>
<span class="co">#&gt; Smoothing chr21 with 2 segments: </span>
<span class="co">#&gt; Smoothing chrX with 2 segments: </span>
<span class="co">#&gt; Smoothing chrY with 2 segments:</span>
<span class="co">#&gt; ✔ Smoothed from 54 to 54 segments with 1e+06 gap (bases).</span>
<span class="co">#&gt; ℹ Creating a new CNAqc object. The old object will be retained in the $before_smoothing field.</span>
<span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span>
<span class="co">#&gt; ℹ Using reference genome coordinates for: GRCh38.</span>
<span class="co">#&gt; ℹ Input n = 17283 mutations for 54 CNA segments (54 clonal, 0 subclonal)</span>
<span class="co">#&gt; Warning in map_mutations_to_segments(snvs, cna %&gt;% filter(CCF == 1)): [CNAqc] a</span>
<span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span>
<span class="co">#&gt; ✔ Mapped n = 17283 mutations to clonal segments (100% of input)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  n = 17283 mutations in 54 segments (54 clonal + 0 subclonal). Geno</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  1:1  [n = 14488, L =  2398 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;  2:0  [n =  1552, L =   200 Mb] ■■■</span>
<span class="co">#&gt;  1:0  [n =   898, L =   141 Mb] ■■</span>
<span class="co">#&gt;  2:1  [n =   345, L =    43 Mb] ■</span>
<span class="co">#&gt; ℹ Sample Purity: 69% ~ Ploidy: 2.</span>
<span class="co">#&gt; ✔ These segments are smoothed; before smoothing there were 54 segments.</span></code></pre></div>
<p>Perform fragmentation analysis (no excess of short segments in this sample).</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/detect_arm_overfragmentation.html">detect_arm_overfragmentation</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ℹ One-tailed Binomial test: 0 tests, alpha 0.01. Short segments: 0.2% of the reference arm.</span>
<span class="co">#&gt; ℹ 0 significantly overfragmented chromosome arms (alpha level 0.01).</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  n = 17283 mutations in 54 segments (54 clonal + 0 subclonal). Geno</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  1:1  [n = 14488, L =  2398 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;  2:0  [n =  1552, L =   200 Mb] ■■■</span>
<span class="co">#&gt;  1:0  [n =   898, L =   141 Mb] ■■</span>
<span class="co">#&gt;  2:1  [n =   345, L =    43 Mb] ■</span>
<span class="co">#&gt; ℹ Sample Purity: 69% ~ Ploidy: 2.</span>
<span class="co">#&gt; ✔ These segments are smoothed; before smoothing there were 54 segments.</span>
<span class="co">#&gt; ✔ Arm-level fragmentation analysis: 0 segments overfragmented.</span></code></pre></div>
</div>
</div>
<div id="alternative-solutions" class="section level1">
<h1 class="hasAnchor">
<a href="#alternative-solutions" class="anchor"></a>Alternative solutions</h1>
<p>We show how to discover that a tetraploid solution is not correct.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Tetraploid solution sequenza  (bad calls)</span>
<span class="va">Sequenza_bad_calls</span> <span class="op">=</span> <span class="fu">load_SQ_output</span><span class="op">(</span><span class="va">Sequenza_URL</span>, sample <span class="op">=</span> <span class="st">'Set6_42'</span>, run <span class="op">=</span> <span class="st">'tetra'</span><span class="op">)</span>

<span class="co"># CNA segments and purity</span>
<span class="va">cna</span> <span class="op">=</span> <span class="va">Sequenza_bad_calls</span><span class="op">$</span><span class="va">segments</span>
<span class="va">purity</span> <span class="op">=</span> <span class="va">Sequenza_bad_calls</span><span class="op">$</span><span class="va">purity</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">Sequenza_bad_calls</span><span class="op">$</span><span class="va">ploidy</span><span class="op">)</span> <span class="co"># Tetraploid</span>
<span class="co">#&gt; [1] 3.9</span></code></pre></div>
<p>Let’s see why this is wrong, using peak detection.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># CNAqc data object</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span><span class="va">snvs</span>, <span class="va">cna</span>, <span class="va">purity</span>, ref <span class="op">=</span> <span class="st">"GRCh38"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/analyze_peaks.html">analyze_peaks</a></span><span class="op">(</span>matching_strategy <span class="op">=</span> <span class="st">'closest'</span><span class="op">)</span>
<span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span>
<span class="co">#&gt; ℹ Using reference genome coordinates for: GRCh38.</span>
<span class="co">#&gt; ! Missing CCF column from CNA calls, adding CCF = 1 assuming clonal CNA calls.</span>
<span class="co">#&gt; ! Missing segments length from CNA calls, adding it to CNA calls.</span>
<span class="co">#&gt; ℹ Input n = 18858 mutations for 56 CNA segments (56 clonal, 0 subclonal)</span>
<span class="co">#&gt; ✖ n = 1575 mutations  cannot be mapped to segments and will be removed.</span>
<span class="co">#&gt; ✔ Mapped n = 17283 mutations to clonal segments (100% of input)</span>
<span class="co">#&gt; ℹ Karyotypes 1:0, 1:1, 2:0, 2:1, and 2:2. Matching strategy closest. KDE = TRUE.</span>
<span class="co">#&gt; ℹ Found n = 15166 mutations in 2:2, 2:0 (skipping those with n &lt; 0 mutations).</span>
<span class="co">#&gt; ! Some peaks have been removed</span>
<span class="co">#&gt; # A tibble: 2 × 5</span>
<span class="co">#&gt;       x     y counts_per_bin discarded from </span>
<span class="co">#&gt;   &lt;dbl&gt; &lt;dbl&gt;          &lt;int&gt; &lt;lgl&gt;     &lt;chr&gt;</span>
<span class="co">#&gt; 1  0.69  0.03              4 TRUE      KDE  </span>
<span class="co">#&gt; 2  0.74  0.03              5 TRUE      KDE  </span>
<span class="co">#&gt; # A tibble: 4 × 17</span>
<span class="co">#&gt;   mutation_multipli… karyotype  peak      x     y counts_per_bin discarded from </span>
<span class="co">#&gt;                &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;          &lt;int&gt; &lt;lgl&gt;     &lt;chr&gt;</span>
<span class="co">#&gt; 1                  1 2:0       0.275 0.269  0.411              4 FALSE     BMix </span>
<span class="co">#&gt; 2                  2 2:0       0.55  0.542  1.25              13 FALSE     BMix </span>
<span class="co">#&gt; 3                  1 2:2       0.177 0.0976 3.72             594 FALSE     BMix </span>
<span class="co">#&gt; 4                  2 2:2       0.355 0.343  3.73             499 FALSE     BMix </span>
<span class="co">#&gt; # … with 9 more variables: offset_VAF &lt;dbl&gt;, offset &lt;dbl&gt;, weight &lt;dbl&gt;,</span>
<span class="co">#&gt; #   epsilon &lt;dbl&gt;, VAF_tolerance &lt;dbl&gt;, matched &lt;lgl&gt;, score &lt;dbl&gt;,</span>
<span class="co">#&gt; #   purity_error &lt;dbl&gt;, QC &lt;chr&gt;</span>
<span class="co">#&gt; ✖ Peak detection FAIL with r = 0.129304886375948 and tolerance e =</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  n = 17283 mutations in 56 segments (56 clonal + 0 subclonal). Geno</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  2:2  [n = 14268, L =  2321 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;  3:1  [n =  1331, L =   188 Mb] ■■</span>
<span class="co">#&gt;  2:0  [n =   898, L =   141 Mb] ■■</span>
<span class="co">#&gt;  4:0  [n =   617, L =   101 Mb] ■</span>
<span class="co">#&gt;  4:2  [n =   169, L =    31 Mb]</span>
<span class="co">#&gt; ℹ Sample Purity: 55% ~ Ploidy: 4.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ──  FAIL  Peaks QC closest: 94%, λ = 0.1293. Purity correction: 13%.</span>
<span class="co">#&gt; ℹ 2:0 ~ n = 898   ( 6%) →  PASS  0.012    PASS  0.008</span>
<span class="co">#&gt; ℹ 2:2 ~ n = 14268 (94%) →  FAIL  0.246    FAIL  0.027</span></code></pre></div>
<p>For this sample these calls are flagged by CNAqc; note that most of the mutations are mapped to tetraploid segments, but one of the two peaks is completely off. Note the overall coloring is green because most of the mutations are underneath a peak that is passed by CNAqc; however the overall karyotype can be failed because it does not show the expected 2-peaks VAF spectrum for tetraploid SNVs.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Do not assemble plots, and remove karyotypes with no data associated</span>
<span class="fu"><a href="../reference/plot_peaks_analysis.html">plot_peaks_analysis</a></span><span class="op">(</span><span class="va">x</span>, empty_plot <span class="op">=</span> <span class="cn">FALSE</span>, assembly_plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; [[1]]</span></code></pre></div>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-19-1.png" width="384"></p>
<pre><code>#&gt; 
#&gt; [[2]]</code></pre>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-19-2.png" width="384"></p>
<p>We do the same for a low-cellularity one.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Low cellularity solution sequenza  (bad calls)</span>
<span class="va">Sequenza_bad_calls</span> <span class="op">=</span> <span class="fu">load_SQ_output</span><span class="op">(</span><span class="va">Sequenza_URL</span>, sample <span class="op">=</span> <span class="st">'Set6_42'</span>, run <span class="op">=</span> <span class="st">'lowcell'</span><span class="op">)</span>

<span class="co"># CNA segments and purity</span>
<span class="va">cna</span> <span class="op">=</span> <span class="va">Sequenza_bad_calls</span><span class="op">$</span><span class="va">segments</span>
<span class="va">purity</span> <span class="op">=</span> <span class="va">Sequenza_bad_calls</span><span class="op">$</span><span class="va">purity</span>

<span class="co"># CNAqc data object</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span><span class="va">snvs</span>, <span class="va">cna</span>, <span class="va">purity</span>, ref <span class="op">=</span> <span class="st">"GRCh38"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/analyze_peaks.html">analyze_peaks</a></span><span class="op">(</span>matching_strategy <span class="op">=</span> <span class="st">'closest'</span><span class="op">)</span>
<span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span>
<span class="co">#&gt; ℹ Using reference genome coordinates for: GRCh38.</span>
<span class="co">#&gt; ! Missing CCF column from CNA calls, adding CCF = 1 assuming clonal CNA calls.</span>
<span class="co">#&gt; ! Missing segments length from CNA calls, adding it to CNA calls.</span>
<span class="co">#&gt; ℹ Input n = 18858 mutations for 56 CNA segments (56 clonal, 0 subclonal)</span>
<span class="co">#&gt; ✖ n = 1575 mutations  cannot be mapped to segments and will be removed.</span>
<span class="co">#&gt; ✔ Mapped n = 17283 mutations to clonal segments (100% of input)</span>
<span class="co">#&gt; ℹ Karyotypes 1:0, 1:1, 2:0, 2:1, and 2:2. Matching strategy closest. KDE = TRUE.</span>
<span class="co">#&gt; ℹ Found n = 17283 mutations in 1:1, 2:0, 1:0, 2:1 (skipping those with n &lt; 0 mutations).</span>
<span class="co">#&gt; ! Some peaks have been removed</span>
<span class="co">#&gt; # A tibble: 2 × 5</span>
<span class="co">#&gt;       x     y counts_per_bin discarded from </span>
<span class="co">#&gt;   &lt;dbl&gt; &lt;dbl&gt;          &lt;int&gt; &lt;lgl&gt;     &lt;chr&gt;</span>
<span class="co">#&gt; 1  0.69  0.03              4 TRUE      KDE  </span>
<span class="co">#&gt; 2  0.74  0.03              5 TRUE      KDE</span>
<span class="co">#&gt; ! Some peaks have been removed</span>
<span class="co">#&gt; # A tibble: 1 × 5</span>
<span class="co">#&gt;       x     y counts_per_bin discarded from </span>
<span class="co">#&gt;   &lt;dbl&gt; &lt;dbl&gt;          &lt;int&gt; &lt;lgl&gt;     &lt;chr&gt;</span>
<span class="co">#&gt; 1  0.81  0.11              0 TRUE      KDE  </span>
<span class="co">#&gt; # A tibble: 6 × 17</span>
<span class="co">#&gt; # Rowwise: </span>
<span class="co">#&gt;   mutation_multiplic… karyotype  peak     x     y counts_per_bin discarded from </span>
<span class="co">#&gt;                 &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;int&gt; &lt;lgl&gt;     &lt;chr&gt;</span>
<span class="co">#&gt; 1                   1 1:0       0.429 0.538 1.26              13 FALSE     BMix </span>
<span class="co">#&gt; 2                   1 1:1       0.3   0.34  3.73             499 FALSE     KDE  </span>
<span class="co">#&gt; 3                   1 2:0       0.3   0.339 1.31              20 FALSE     BMix </span>
<span class="co">#&gt; 4                   2 2:0       0.6   0.641 0.492             13 FALSE     BMix </span>
<span class="co">#&gt; 5                   1 2:1       0.231 0.275 1.58               5 FALSE     BMix </span>
<span class="co">#&gt; 6                   2 2:1       0.462 0.521 0.615              1 FALSE     BMix </span>
<span class="co">#&gt; # … with 9 more variables: offset_VAF &lt;dbl&gt;, offset &lt;dbl&gt;, weight &lt;dbl&gt;,</span>
<span class="co">#&gt; #   epsilon &lt;dbl&gt;, VAF_tolerance &lt;dbl&gt;, matched &lt;lgl&gt;, score &lt;dbl&gt;,</span>
<span class="co">#&gt; #   purity_error &lt;dbl&gt;, QC &lt;chr&gt;</span>
<span class="co">#&gt; ✔ Peak detection PASS with r = -0.0797439724313142 and tolerance e =</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ── [ CNAqc ]  n = 17283 mutations in 56 segments (56 clonal + 0 subclonal). Geno</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  1:1  [n = 14268, L =  2321 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span>
<span class="co">#&gt;  2:0  [n =  1772, L =   277 Mb] ■■■</span>
<span class="co">#&gt;  1:0  [n =   898, L =   141 Mb] ■■</span>
<span class="co">#&gt;  2:1  [n =   345, L =    43 Mb] ■</span>
<span class="co">#&gt; ℹ Sample Purity: 60% ~ Ploidy: 2.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ──  PASS  Peaks QC closest: 93%, λ = -0.0797. Purity correction: -8%.</span>
<span class="co">#&gt; ℹ 1:0 ~ n = 898   ( 5%) →  FAIL  -0.092</span>
<span class="co">#&gt; ℹ 1:1 ~ n = 14268 (83%) →  PASS  -0.08</span>
<span class="co">#&gt; ℹ 2:0 ~ n = 1772  (10%) →  PASS  -0.078   PASS  -0.041</span>
<span class="co">#&gt; ℹ 2:1 ~ n = 345   ( 2%) →  FAIL  -0.17    FAIL  -0.109</span></code></pre></div>
<p>For this sample these calls are flagged by CNAqc because the purity is off by a <code>~10%</code> factor.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_peaks_analysis.html">plot_peaks_analysis</a></span><span class="op">(</span><span class="va">x</span>, empty_plot <span class="op">=</span> <span class="cn">FALSE</span>, assembly_plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; [[1]]</span></code></pre></div>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-21-1.png" width="384"></p>
<pre><code>#&gt; 
#&gt; [[2]]</code></pre>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-21-2.png" width="384"></p>
<pre><code>#&gt; 
#&gt; [[3]]</code></pre>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-21-3.png" width="384"></p>
<pre><code>#&gt; 
#&gt; [[4]]</code></pre>
<p><img src="a7_example_MSeq_files/figure-html/unnamed-chunk-21-4.png" width="384"></p>
<p>CNAqc can help increasing the caller purity in this case; the expected adjustment (score from VAF analysis) of 5% would correspond indeed to a purity increase of <code>10%</code>.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">Sequenza_bad_calls</span><span class="op">$</span><span class="va">purity</span> <span class="op">-</span> <span class="va">Sequenza_good_calls</span><span class="op">$</span><span class="va">purity</span><span class="op">)</span> <span class="co"># error in these calls</span>
<span class="co">#&gt; [1] -0.09</span>

<span class="co">#CNAqc suggested adjustment</span>
<span class="va">x</span><span class="op">$</span><span class="va">peaks_analysis</span><span class="op">$</span><span class="va">score</span>
<span class="co">#&gt; [1] -0.07974397</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.caravagnalab.org">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
