<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>8. Drivers annotation • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="8. Drivers annotation">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
    <li>
      <a href="../articles/a9_automatic.html">9. Automatic CNA calling</a>
    </li>
    <li>
      <a href="../articles/b10_MultiCNAqc.html">10. multiCNAqc</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>8. Drivers annotation</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqc/vignettes/a8_annotate_variants.Rmd" class="external-link"><code>vignettes/a8_annotate_variants.Rmd</code></a></small>
      <div class="hidden name"><code>a8_annotate_variants.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/CNAqc" class="external-link">CNAqc</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Loading <span style="color: #00BB00;">CNAqc</span>, <span style="font-style: italic;">'Copy Number Alteration quality check'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagn.github.io/CNAqc/&gt;</span></span></span>
<span></span>
<span><span class="co"># Extra packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: dplyr</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span></span>
<span><span class="co">#if you use RStudio you will probably have to run this line</span></span>
<span><span class="co"># options(connectionObserver = NULL)</span></span></code></pre></div>
<blockquote>
<p>Note: variant annotation should be carried out with dedicated tools.
CNAqc functions should only be used to get a preliminary idea of the
most important mutations annotated in a sample.</p>
</blockquote>
<div class="section level2">
<h2 id="annotating-driver-mutations-with-variantannotation-and-intogen">Annotating driver mutations with VariantAnnotation and Intogen<a class="anchor" aria-label="anchor" href="#annotating-driver-mutations-with-variantannotation-and-intogen"></a>
</h2>
<p>CNAqc can annotate input mutations and flag potential <em>driver
mutations</em>. Using the <a href="https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html" class="external-link">VariantAnnotation</a>
package and the <a href="https://www.intogen.org/search" class="external-link">intOGen</a>
database, CNAqc performs the following steps:</p>
<ol style="list-style-type: decimal">
<li>Annotates the position of each mutation, with <a href="https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html" class="external-link">VariantAnnotation</a>:
<ul>
<li>
<code>coding</code>,</li>
<li>
<code>intron</code>,</li>
<li>
<code>fiveUTR</code>,</li>
<li>
<code>threeUTR</code>,</li>
<li>
<code>intron</code>,</li>
<li>
<code>intergenic</code>,</li>
<li>
<code>spliceSite</code>,</li>
<li>
<code>promoter</code>.</li>
</ul>
</li>
<li>Annotates the consequence on the protein for coding mutations, with
change in the amino acid if known, with <a href="https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html" class="external-link">VariantAnnotation</a>:
<ul>
<li>
<code>nonsynonymous</code>,</li>
<li>
<code>synonymous</code>,</li>
<li>
<code>frameshift</code>,</li>
<li>
<code>stop</code>.</li>
</ul>
</li>
<li>Compares non-synonymous mutations to known driver genes from the <a href="https://www.intogen.org/search" class="external-link">intOGen</a> database or a custom
list, and flags drivers.</li>
</ol>
<p>This functionality works with a CNAqc object.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Dataset available with the package </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">'example_dataset_CNAqc'</span>, package <span class="op">=</span> <span class="st">'CNAqc'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span></span>
<span>  mutations <span class="op">=</span> <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">mutations</span>, </span>
<span>  cna <span class="op">=</span> <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">cna</span>,</span>
<span>  purity <span class="op">=</span> <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">purity</span>,</span>
<span>  ref <span class="op">=</span> <span class="st">'hg19'</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">CNAqc - CNA Quality Check</span> <span style="color: #00BBBB;">───────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Found annotated driver mutations: <span style="color: #00BB00;">TTN</span>, <span style="color: #00BB00;">CTCF</span>, and <span style="color: #00BB00;">TP53</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">12963</span> somatic mutations: <span style="color: #00BB00;">12963</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">0</span> indels.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> CNAs have no CCF, assuming clonal CNAs (CCF = 1).</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">267</span> segments: <span style="color: #00BB00;">267</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">12963</span> mutations mapped to clonal CNAs.</span></span>
<span></span>
<span><span class="co"># What we annotate</span></span>
<span><span class="va">x</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="va">Mutations</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 12,963 × 17</span></span></span>
<span><span class="co">#&gt;    chr      from      to ref   alt      NV    DP    VAF FILTER ANNOVAR_FUNCTION</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1  1<span style="text-decoration: underline;">027</span>104 1<span style="text-decoration: underline;">027</span>105 T     G         6    60 0.1    PASS   UTR5            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1  2<span style="text-decoration: underline;">248</span>588 2<span style="text-decoration: underline;">248</span>589 A     C         9   127 0.070<span style="text-decoration: underline;">9</span> PASS   intergenic      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1  2<span style="text-decoration: underline;">461</span>999 2<span style="text-decoration: underline;">462</span>000 G     A        65   156 0.417  PASS   upstream        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1  2<span style="text-decoration: underline;">727</span>935 2<span style="text-decoration: underline;">727</span>936 T     C        90   180 0.5    PASS   downstream      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1  2<span style="text-decoration: underline;">763</span>397 2<span style="text-decoration: underline;">763</span>398 C     T        61   183 0.333  PASS   intergenic      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1  2<span style="text-decoration: underline;">768</span>208 2<span style="text-decoration: underline;">768</span>209 C     T       130   203 0.640  PASS   intergenic      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr1  2<span style="text-decoration: underline;">935</span>590 2<span style="text-decoration: underline;">935</span>591 C     T       132   228 0.579  PASS   intergenic      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr1  2<span style="text-decoration: underline;">980</span>032 2<span style="text-decoration: underline;">980</span>033 C     T        85   196 0.434  PASS   ncRNA_exonic    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr1  3<span style="text-decoration: underline;">387</span>161 3<span style="text-decoration: underline;">387</span>162 T     G         6   124 0.048<span style="text-decoration: underline;">4</span> PASS   intronic        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr1  3<span style="text-decoration: underline;">502</span>517 3<span style="text-decoration: underline;">502</span>518 G     A        10    88 0.114  PASS   intronic        </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 12,953 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 7 more variables: GENE &lt;chr&gt;, is_driver &lt;lgl&gt;, driver_label &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   type &lt;chr&gt;, karyotype &lt;chr&gt;, segment_id &lt;chr&gt;, cna &lt;chr&gt;</span></span></span></code></pre></div>
<div class="section level3">
<h3 id="required-packages">Required packages<a class="anchor" aria-label="anchor" href="#required-packages"></a>
</h3>
<p>CNAqc uses databases from <a href="https://www.bioconductor.org/" class="external-link">Bioconductor</a> to annotate the
variants; installation of these databases might take a bit of time
because ~1GB of data have to be downloaded. This will happen only the
first time the annotation is run.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Reference against which we mapped the reads</span></span>
<span><span class="va">reference_genome</span> <span class="op">&lt;-</span> <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">reference</span></span>
<span></span>
<span><span class="co"># All those packages are distributed in Bioconductor</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, repos<span class="op">=</span><span class="st">"http://cran.us.r-project.org"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; The downloaded binary packages are in</span></span>
<span><span class="co">#&gt;  /var/folders/2s/h6hvv9ps03xgz_krkkstvq_r0000gn/T//RtmpdmM79Q/downloaded_packages</span></span>
<span></span>
<span><span class="co"># We have to install the corresponding txdb package for transcript annotations</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"TxDb.Hsapiens.UCSC."</span>,<span class="va">reference_genome</span>, <span class="st">".knownGene"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># We have to install also the BS database for the sequences (it may take some time)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"BSgenome.Hsapiens.UCSC."</span>,<span class="va">reference_genome</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Then these two packages provide useful utilities to deal with biological databases</span></span>
<span><span class="st">"Organism.dplyr"</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="st">"org.Hs.eg.db"</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="drivers-known-to-cnaqc">Drivers known to CNAqc<a class="anchor" aria-label="anchor" href="#drivers-known-to-cnaqc"></a>
</h3>
<p>CNAqc has pre-loaded a list of 568 driver genes for 66 cancer types,
compiled from <a href="https://www.intogen.org/download" class="external-link">intOGen</a>
release date 2020.02.01.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The available list:</span></span>
<span><span class="co"># - gene id</span></span>
<span><span class="co"># - tumour code where the gene has been flagged as driver</span></span>
<span><span class="co"># - tumour code longname (Esophageal cancer)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"intogen_drivers"</span>, package <span class="op">=</span> <span class="st">'CNAqc'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Number of genes (568)</span></span>
<span><span class="va">intogen_drivers</span><span class="op">$</span><span class="va">gene</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="va">unique</span></span>
<span><span class="co">#&gt;   [1] "ABCB1"      "ABI1"       "ABL1"       "ABL2"       "ACKR3"     </span></span>
<span><span class="co">#&gt;   [6] "ACSL3"      "ACVR1"      "ACVR2A"     "ADGRB1"     "AFF1"      </span></span>
<span><span class="co">#&gt;  [11] "AFF3"       "AJUBA"      "AKT1"       "AKT3"       "ALB"       </span></span>
<span><span class="co">#&gt;  [16] "ALK"        "AMER1"      "APC"        "AR"         "ARAF"      </span></span>
<span><span class="co">#&gt;  [21] "ARHGAP35"   "ARHGAP5"    "ARHGEF10"   "ARHGEF10L"  "ARHGEF12"  </span></span>
<span><span class="co">#&gt;  [26] "ARID1A"     "ARID1B"     "ARID2"      "ASXL1"      "ASXL2"     </span></span>
<span><span class="co">#&gt;  [31] "ATF7IP"     "ATG7"       "ATM"        "ATR"        "ATRX"      </span></span>
<span><span class="co">#&gt;  [36] "AXIN1"      "AXIN2"      "B2M"        "BAP1"       "BCL10"     </span></span>
<span><span class="co">#&gt;  [41] "BCL11A"     "BCL11B"     "BCL2"       "BCL2L12"    "BCL6"      </span></span>
<span><span class="co">#&gt;  [46] "BCL7A"      "BCL9"       "BCL9L"      "BCLAF1"     "BCOR"      </span></span>
<span><span class="co">#&gt;  [51] "BCORL1"     "BCR"        "BIRC3"      "BIRC6"      "BMPR1A"    </span></span>
<span><span class="co">#&gt;  [56] "BMPR2"      "BRAF"       "BRCA1"      "BRCA2"      "BRD4"      </span></span>
<span><span class="co">#&gt;  [61] "BRD7"       "BTG1"       "BTK"        "CACNA1D"    "CAMTA1"    </span></span>
<span><span class="co">#&gt;  [66] "CARD11"     "CARS"       "CASP8"      "CASZ1"      "CBFA2T3"   </span></span>
<span><span class="co">#&gt;  [71] "CBFB"       "CBL"        "CCDC6"      "CCND1"      "CCND2"     </span></span>
<span><span class="co">#&gt;  [76] "CCND3"      "CCR7"       "CD58"       "CD79B"      "CDC73"     </span></span>
<span><span class="co">#&gt;  [81] "CDH1"       "CDH10"      "CDH11"      "CDK12"      "CDK4"      </span></span>
<span><span class="co">#&gt;  [86] "CDKN1A"     "CDKN1B"     "CDKN2A"     "CDKN2C"     "CDX2"      </span></span>
<span><span class="co">#&gt;  [91] "CEBPA"      "CHD2"       "CHD4"       "CHEK2"      "CIC"       </span></span>
<span><span class="co">#&gt;  [96] "CIITA"      "CLIP1"      "CLTC"       "CLTCL1"     "CMTR2"     </span></span>
<span><span class="co">#&gt; [101] "CNOT3"      "CNOT9"      "COL1A1"     "CPEB3"      "CR1"       </span></span>
<span><span class="co">#&gt; [106] "CREBBP"     "CRNKL1"     "CRTC1"      "CSF3R"      "CTCF"      </span></span>
<span><span class="co">#&gt; [111] "CTNNB1"     "CUL3"       "CUX1"       "CXCR4"      "CYLD"      </span></span>
<span><span class="co">#&gt; [116] "CYP2C8"     "CYSLTR2"    "DAXX"       "DAZAP1"     "DCAF12L2"  </span></span>
<span><span class="co">#&gt; [121] "DCC"        "DCSTAMP"    "DDB2"       "DDX3X"      "DDX6"      </span></span>
<span><span class="co">#&gt; [126] "DGCR8"      "DHX9"       "DICER1"     "DIS3"       "DNAJB1"    </span></span>
<span><span class="co">#&gt; [131] "DNMT3A"     "DOT1L"      "DROSHA"     "DTX1"       "DUSP16"    </span></span>
<span><span class="co">#&gt; [136] "EBF1"       "EFTUD2"     "EGFR"       "EGR2"       "EHD2"      </span></span>
<span><span class="co">#&gt; [141] "EIF1AX"     "EIF3E"      "ELF3"       "ELF4"       "ELL"       </span></span>
<span><span class="co">#&gt; [146] "ELN"        "EML4"       "ENPEP"      "EP300"      "EPAS1"     </span></span>
<span><span class="co">#&gt; [151] "EPHA2"      "EPHA3"      "EPHA7"      "EPS15"      "ERBB2"     </span></span>
<span><span class="co">#&gt; [156] "ERBB3"      "ERBB4"      "ERCC2"      "ERCC3"      "ERG"       </span></span>
<span><span class="co">#&gt; [161] "ESR1"       "ESRRA"      "ETV4"       "ETV5"       "ETV6"      </span></span>
<span><span class="co">#&gt; [166] "EWSR1"      "EXT2"       "EZH2"       "FAM135B"    "FAM174B"   </span></span>
<span><span class="co">#&gt; [171] "FAM186A"    "FAM46C"     "FANCA"      "FANCC"      "FANCD2"    </span></span>
<span><span class="co">#&gt; [176] "FANCF"      "FAS"        "FAT1"       "FAT2"       "FAT3"      </span></span>
<span><span class="co">#&gt; [181] "FAT4"       "FBLN1"      "FBN2"       "FBXO11"     "FBXW7"     </span></span>
<span><span class="co">#&gt; [186] "FGD5"       "FGFR1"      "FGFR2"      "FGFR3"      "FGFR4"     </span></span>
<span><span class="co">#&gt; [191] "FH"         "FHIT"       "FLCN"       "FLT3"       "FLT4"      </span></span>
<span><span class="co">#&gt; [196] "FN1"        "FOXA1"      "FOXA2"      "FOXD4L1"    "FOXL2"     </span></span>
<span><span class="co">#&gt; [201] "FOXO1"      "FOXO3"      "FOXP1"      "FUBP1"      "GATA1"     </span></span>
<span><span class="co">#&gt; [206] "GATA2"      "GATA3"      "GLI1"       "GMPS"       "GNA11"     </span></span>
<span><span class="co">#&gt; [211] "GNA13"      "GNAI2"      "GNAQ"       "GNAS"       "GRIN2A"    </span></span>
<span><span class="co">#&gt; [216] "GTF2I"      "H3F3A"      "HERC2"      "HGF"        "HIP1"      </span></span>
<span><span class="co">#&gt; [221] "HIST1H3B"   "HIST1H4I"   "HLA-A"      "HLA-B"      "HNF1A"     </span></span>
<span><span class="co">#&gt; [226] "HNRNPA2B1"  "HOXA11"     "HOXC13"     "HOXD13"     "HRAS"      </span></span>
<span><span class="co">#&gt; [231] "HSP90AA1"   "HSP90AB1"   "HSPG2"      "HTRA2"      "ID3"       </span></span>
<span><span class="co">#&gt; [236] "IDH1"       "IDH2"       "IFNAR1"     "IFNGR1"     "IKBKB"     </span></span>
<span><span class="co">#&gt; [241] "IKZF1"      "IKZF3"      "IL6ST"      "IL7R"       "ING1"      </span></span>
<span><span class="co">#&gt; [246] "INO80"      "IRAK1"      "IRF1"       "IRF4"       "IRS4"      </span></span>
<span><span class="co">#&gt; [251] "JAK1"       "JAK2"       "JAK3"       "KAT6A"      "KAT6B"     </span></span>
<span><span class="co">#&gt; [256] "KDM3B"      "KDM5A"      "KDM5C"      "KDM6A"      "KDR"       </span></span>
<span><span class="co">#&gt; [261] "KEAP1"      "KEL"        "KIF5B"      "KIFC1"      "KIT"       </span></span>
<span><span class="co">#&gt; [266] "KLF4"       "KLF5"       "KLHL36"     "KLHL6"      "KMT2A"     </span></span>
<span><span class="co">#&gt; [271] "KMT2B"      "KMT2C"      "KMT2D"      "KRAS"       "LATS1"     </span></span>
<span><span class="co">#&gt; [276] "LATS2"      "LDB1"       "LIFR"       "LOX"        "LPAR4"     </span></span>
<span><span class="co">#&gt; [281] "LPP"        "LRIG3"      "LRP1B"      "LTB"        "LY75-CD302"</span></span>
<span><span class="co">#&gt; [286] "LZTR1"      "MAF"        "MALT1"      "MAML2"      "MAP2"      </span></span>
<span><span class="co">#&gt; [291] "MAP2K1"     "MAP2K4"     "MAP2K7"     "MAP3K1"     "MAPK1"     </span></span>
<span><span class="co">#&gt; [296] "MARK2"      "MAX"        "MB21D2"     "MCM3AP"     "MDM2"      </span></span>
<span><span class="co">#&gt; [301] "MDM4"       "MECOM"      "MED12"      "MEF2B"      "MEN1"      </span></span>
<span><span class="co">#&gt; [306] "MET"        "MGA"        "MLLT1"      "MLLT3"      "MSI2"      </span></span>
<span><span class="co">#&gt; [311] "MSN"        "MTCP1"      "MTOR"       "MYC"        "MYCN"      </span></span>
<span><span class="co">#&gt; [316] "MYD88"      "MYH11"      "MYH9"       "MYO5A"      "NBEA"      </span></span>
<span><span class="co">#&gt; [321] "NCOA1"      "NCOA2"      "NCOR1"      "NCOR2"      "NEFH"      </span></span>
<span><span class="co">#&gt; [326] "NF1"        "NF2"        "NFATC2"     "NFE2L2"     "NFKB2"     </span></span>
<span><span class="co">#&gt; [331] "NFKBIA"     "NFKBIE"     "NIN"        "NIPBL"      "NKTR"      </span></span>
<span><span class="co">#&gt; [336] "NKX2-1"     "NONO"       "NOTCH1"     "NOTCH2"     "NPEPPS"    </span></span>
<span><span class="co">#&gt; [341] "NPM1"       "NPRL2"      "NRAS"       "NRG1"       "NRK"       </span></span>
<span><span class="co">#&gt; [346] "NRP1"       "NSD1"       "NSD2"       "NT5C3A"     "NTRK1"     </span></span>
<span><span class="co">#&gt; [351] "NTRK3"      "NUMA1"      "NUP214"     "NXF1"       "P2RY8"     </span></span>
<span><span class="co">#&gt; [356] "PABPC1"     "PAK2"       "PARP4"      "PBRM1"      "PCBP1"     </span></span>
<span><span class="co">#&gt; [361] "PCDH17"     "PCMTD1"     "PDCD1LG2"   "PDE4DIP"    "PDGFRA"    </span></span>
<span><span class="co">#&gt; [366] "PDGFRB"     "PDPR"       "PEG3"       "PHF6"       "PIK3CA"    </span></span>
<span><span class="co">#&gt; [371] "PIK3CB"     "PIK3R1"     "PIM1"       "PLAG1"      "PLCB4"     </span></span>
<span><span class="co">#&gt; [376] "PLCG1"      "PML"        "PMS2"       "POLD1"      "POLE"      </span></span>
<span><span class="co">#&gt; [381] "POLQ"       "POT1"       "POU2F2"     "PPM1D"      "PPP2R1A"   </span></span>
<span><span class="co">#&gt; [386] "PPP3CA"     "PPP6C"      "PPT2"       "PRDM1"      "PRDM2"     </span></span>
<span><span class="co">#&gt; [391] "PREX2"      "PRF1"       "PRKAR1A"    "PRKCB"      "PRKCD"     </span></span>
<span><span class="co">#&gt; [396] "PRKD2"      "PRR14"      "PRRX1"      "PSIP1"      "PTCH1"     </span></span>
<span><span class="co">#&gt; [401] "PTEN"       "PTMA"       "PTPN11"     "PTPN13"     "PTPN14"    </span></span>
<span><span class="co">#&gt; [406] "PTPN6"      "PTPRB"      "PTPRC"      "PTPRD"      "PTPRK"     </span></span>
<span><span class="co">#&gt; [411] "PTPRT"      "QKI"        "RABEP1"     "RAC1"       "RAD21"     </span></span>
<span><span class="co">#&gt; [416] "RAF1"       "RALGDS"     "RANBP2"     "RAP1GDS1"   "RARA"      </span></span>
<span><span class="co">#&gt; [421] "RASA1"      "RASA2"      "RB1"        "RBFOX1"     "RBFOX2"    </span></span>
<span><span class="co">#&gt; [426] "RBM10"      "RBM15"      "RBM38"      "RBM39"      "RECQL4"    </span></span>
<span><span class="co">#&gt; [431] "RELA"       "RET"        "RGL3"       "RGPD3"      "RGS7"      </span></span>
<span><span class="co">#&gt; [436] "RHOA"       "RHPN2"      "RIPK1"      "RNF213"     "RNF43"     </span></span>
<span><span class="co">#&gt; [441] "RNF6"       "RPL10"      "RPL22"      "RPS3A"      "RPS6KA3"   </span></span>
<span><span class="co">#&gt; [446] "RRAGC"      "RRAS2"      "RSPH10B2"   "RUNX1"      "RUNX1T1"   </span></span>
<span><span class="co">#&gt; [451] "RXRA"       "SALL4"      "SATB1"      "SDC4"       "SEPT9"     </span></span>
<span><span class="co">#&gt; [456] "SET"        "SETBP1"     "SETD1B"     "SETD2"      "SETDB1"    </span></span>
<span><span class="co">#&gt; [461] "SF3B1"      "SFMBT2"     "SGK1"       "SH2B3"      "SIN3A"     </span></span>
<span><span class="co">#&gt; [466] "SIRPA"      "SIX1"       "SLC45A3"    "SMAD2"      "SMAD3"     </span></span>
<span><span class="co">#&gt; [471] "SMAD4"      "SMARCA1"    "SMARCA4"    "SMARCB1"    "SMARCD1"   </span></span>
<span><span class="co">#&gt; [476] "SMC1A"      "SMO"        "SOCS1"      "SOHLH2"     "SOS1"      </span></span>
<span><span class="co">#&gt; [481] "SOX17"      "SOX21"      "SOX9"       "SP140"      "SPEN"      </span></span>
<span><span class="co">#&gt; [486] "SPOP"       "SRGAP3"     "SRSF2"      "STAG2"      "STAT3"     </span></span>
<span><span class="co">#&gt; [491] "STAT5B"     "STAT6"      "STIL"       "STK11"      "STRN"      </span></span>
<span><span class="co">#&gt; [496] "SUSD2"      "SUZ12"      "TAF15"      "TBL1XR1"    "TBX3"      </span></span>
<span><span class="co">#&gt; [501] "TCF4"       "TCF7L2"     "TCIRG1"     "TEC"        "TET1"      </span></span>
<span><span class="co">#&gt; [506] "TET2"       "TFAP4"      "TFG"        "TGFBR2"     "TGIF1"     </span></span>
<span><span class="co">#&gt; [511] "TLL1"       "TNC"        "TNFAIP3"    "TNFRSF14"   "TOP1"      </span></span>
<span><span class="co">#&gt; [516] "TOP2A"      "TP53"       "TP63"       "TRAF3"      "TRIM24"    </span></span>
<span><span class="co">#&gt; [521] "TRIM33"     "TRIM49C"    "TRIP11"     "TRRAP"      "TSC1"      </span></span>
<span><span class="co">#&gt; [526] "TSC2"       "U2AF1"      "U2AF2"      "UBE2A"      "UBE2D2"    </span></span>
<span><span class="co">#&gt; [531] "UBR5"       "UGT2B17"    "USP44"      "USP6"       "USP8"      </span></span>
<span><span class="co">#&gt; [536] "USP9X"      "VAV1"       "VHL"        "WAS"        "WDR45"     </span></span>
<span><span class="co">#&gt; [541] "WNK2"       "WNK4"       "WRN"        "WT1"        "XPC"       </span></span>
<span><span class="co">#&gt; [546] "XPO1"       "ZBTB16"     "ZBTB20"     "ZBTB7B"     "ZCRB1"     </span></span>
<span><span class="co">#&gt; [551] "ZEB1"       "ZFHX3"      "ZFP36L1"    "ZFX"        "ZNF148"    </span></span>
<span><span class="co">#&gt; [556] "ZNF165"     "ZNF208"     "ZNF429"     "ZNF521"     "ZNF626"    </span></span>
<span><span class="co">#&gt; [561] "ZNF680"     "ZNF721"     "ZNF780A"    "ZNF814"     "ZNF93"     </span></span>
<span><span class="co">#&gt; [566] "ZNRF3"      "ZRSR2"      "ZXDB"</span></span>
<span></span>
<span><span class="co"># Tumour types (66)</span></span>
<span><span class="va">intogen_drivers</span><span class="op">$</span><span class="va">synopsis</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="va">unique</span></span>
<span><span class="co">#&gt;  [1] "Esophageal cancer"                    </span></span>
<span><span class="co">#&gt;  [2] "Hepatic cancer"                       </span></span>
<span><span class="co">#&gt;  [3] "Endometrial cancer"                   </span></span>
<span><span class="co">#&gt;  [4] "Breast adenocarcinoma"                </span></span>
<span><span class="co">#&gt;  [5] "Thyroid adenocarcinoma"               </span></span>
<span><span class="co">#&gt;  [6] "Ovary cancer"                         </span></span>
<span><span class="co">#&gt;  [7] "Acute myeloid leukemia"               </span></span>
<span><span class="co">#&gt;  [8] "High-grade Glioma"                    </span></span>
<span><span class="co">#&gt;  [9] "Colorectal adenocarcinoma"            </span></span>
<span><span class="co">#&gt; [10] "Pancreas adenocarcinoma"              </span></span>
<span><span class="co">#&gt; [11] "Skin squamous cell carcinoma"         </span></span>
<span><span class="co">#&gt; [12] "Stomach adenocarcinoma"               </span></span>
<span><span class="co">#&gt; [13] "Anus cancer"                          </span></span>
<span><span class="co">#&gt; [14] "Cutaneous melanoma of the skin"       </span></span>
<span><span class="co">#&gt; [15] "Head and neck squamous cell carcinoma"</span></span>
<span><span class="co">#&gt; [16] "Adenoid cystic carcinoma"             </span></span>
<span><span class="co">#&gt; [17] "Acute lymphoblastic leukemia"         </span></span>
<span><span class="co">#&gt; [18] "Cervix squamous cancer"               </span></span>
<span><span class="co">#&gt; [19] "Osteosarcoma"                         </span></span>
<span><span class="co">#&gt; [20] "Prostate adenocarcinoma"              </span></span>
<span><span class="co">#&gt; [21] "Neuroblastoma"                        </span></span>
<span><span class="co">#&gt; [22] "Non small cell lung cancer"           </span></span>
<span><span class="co">#&gt; [23] "Small-cell lung cancer"               </span></span>
<span><span class="co">#&gt; [24] "Wilms tumor"                          </span></span>
<span><span class="co">#&gt; [25] "Bowel cancer"                         </span></span>
<span><span class="co">#&gt; [26] "Cholangiocarcinoma"                   </span></span>
<span><span class="co">#&gt; [27] "Skin basal cell carcinoma"            </span></span>
<span><span class="co">#&gt; [28] "Uveal melanoma"                       </span></span>
<span><span class="co">#&gt; [29] "Vulva Cancer"                         </span></span>
<span><span class="co">#&gt; [30] "Diffuse large B-cell lymphoma"        </span></span>
<span><span class="co">#&gt; [31] "Multiple myeloma"                     </span></span>
<span><span class="co">#&gt; [32] "Nasopharyngeal cancer"                </span></span>
<span><span class="co">#&gt; [33] "Chronic lymphoblastic leukemia"       </span></span>
<span><span class="co">#&gt; [34] "Lung squamous cell carcinoma"         </span></span>
<span><span class="co">#&gt; [35] "Bladder cancer"                       </span></span>
<span><span class="co">#&gt; [36] "Angiosarcoma"                         </span></span>
<span><span class="co">#&gt; [37] "Medulloblastoma"                      </span></span>
<span><span class="co">#&gt; [38] "Burkitt lymphoma"                     </span></span>
<span><span class="co">#&gt; [39] "Glioblastoma"                         </span></span>
<span><span class="co">#&gt; [40] "Lower grade glioma"                   </span></span>
<span><span class="co">#&gt; [41] "Lung adenocarcinoma"                  </span></span>
<span><span class="co">#&gt; [42] "Non-hodking lymphoma"                 </span></span>
<span><span class="co">#&gt; [43] "Pilocityc astrocytoma"                </span></span>
<span><span class="co">#&gt; [44] "Renal clear cell carcinoma"           </span></span>
<span><span class="co">#&gt; [45] "Renal papillary cell carcinoma"       </span></span>
<span><span class="co">#&gt; [46] "Uterine carcinosarcoma"               </span></span>
<span><span class="co">#&gt; [47] "Pancreatic neuroendocrine cancer"     </span></span>
<span><span class="co">#&gt; [48] "Adrenocortical carcinoma"             </span></span>
<span><span class="co">#&gt; [49] "Leiomyosarcoma"                       </span></span>
<span><span class="co">#&gt; [50] "Sarcoma"                              </span></span>
<span><span class="co">#&gt; [51] "Lymphoma"                             </span></span>
<span><span class="co">#&gt; [52] "Mesothelioma"                         </span></span>
<span><span class="co">#&gt; [53] "Ewing's sarcoma"                      </span></span>
<span><span class="co">#&gt; [54] "Thymic carcinoma"                     </span></span>
<span><span class="co">#&gt; [55] "Male germ cell tumor"                 </span></span>
<span><span class="co">#&gt; [56] "Pheochromocytoma and paraganglioma"   </span></span>
<span><span class="co">#&gt; [57] "Chromophobe renal cell carcinoma"     </span></span>
<span><span class="co">#&gt; [58] "Small intestine cancer neuroendocrine"</span></span>
<span><span class="co">#&gt; [59] "Salivary glands cancer"               </span></span>
<span><span class="co">#&gt; [60] "Hepatic blastoma"                     </span></span>
<span><span class="co">#&gt; [61] "Rhabdomyosarcoma"                     </span></span>
<span><span class="co">#&gt; [62] "Myelodysplastic syndrome neoplasm"    </span></span>
<span><span class="co">#&gt; [63] "Lung neuroendocrine cancer"           </span></span>
<span><span class="co">#&gt; [64] "Ependymoma"                           </span></span>
<span><span class="co">#&gt; [65] "Retinoblastoma"                       </span></span>
<span><span class="co">#&gt; [66] "Atypical teratoid/rhabdoid tumor"</span></span>
<span></span>
<span><span class="co"># Organised table</span></span>
<span><span class="va">intogen_drivers</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2,080 × 3</span></span></span>
<span><span class="co">#&gt;    gene  tumour synopsis              </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> ABCB1 ESCA   Esophageal cancer     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> ABI1  ESCA   Esophageal cancer     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> ABL1  HC     Hepatic cancer        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> ABL1  UCEC   Endometrial cancer    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> ABL2  BRCA   Breast adenocarcinoma </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> ABL2  THCA   Thyroid adenocarcinoma</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> ABL2  UCEC   Endometrial cancer    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> ACKR3 OV     Ovary cancer          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> ACSL3 AML    Acute myeloid leukemia</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> ACVR1 HGG    High-grade Glioma     </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2,070 more rows</span></span></span></code></pre></div>
<p>We plot the list of genes that appear in at least 20 tumour
types.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">n</span> <span class="op">&lt;-</span> <span class="fl">60</span></span>
<span><span class="va">qual_col_pals</span> <span class="op">=</span> <span class="va">brewer.pal.info</span><span class="op">[</span><span class="va">brewer.pal.info</span><span class="op">$</span><span class="va">category</span> <span class="op">==</span> <span class="st">'qual'</span>,<span class="op">]</span></span>
<span><span class="va">col_vector</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mapply.html" class="external-link">mapply</a></span><span class="op">(</span><span class="va">brewer.pal</span>, <span class="va">qual_col_pals</span><span class="op">$</span><span class="va">maxcolors</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">qual_col_pals</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">intogen_drivers</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">20</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">gene</span>, fill <span class="op">=</span> <span class="va">synopsis</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">col_vector</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">CNAqc</span><span class="fu">:::</span><span class="fu">my_ggplot_theme</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="a8_annotate_variants_files/figure-html/unnamed-chunk-5-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="default-annotation">Default annotation<a class="anchor" aria-label="anchor" href="#default-annotation"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run default annotation function</span></span>
<span><span class="va">x_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotate_variants.html">annotate_variants</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 17</span></span></span>
<span><span class="co">#&gt;   chr        from       to ref   alt      NV    DP   VAF FILTER ANNOVAR_FUNCTION</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chr2  179<span style="text-decoration: underline;">431</span>633   1.79<span style="color: #949494;">e</span>8 C     T        77   117 0.658 PASS   exonic          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chr16  67<span style="text-decoration: underline;">646</span>006   6.76<span style="color: #949494;">e</span>7 C     T        54   120 0.45  PASS   exonic          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chr17   7<span style="text-decoration: underline;">577</span>106   7.58<span style="color: #949494;">e</span>6 G     C        78    84 0.929 PASS   exonic          </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 7 more variables: GENE &lt;chr&gt;, is_driver &lt;lgl&gt;, driver_label &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   type &lt;chr&gt;, karyotype &lt;chr&gt;, segment_id &lt;chr&gt;, cna &lt;chr&gt;</span></span></span>
<span><span class="co">#&gt;       chr      from        to consequence refAA varAA</span></span>
<span><span class="co">#&gt; 1    chr1  19181096  19181097  frameshift            </span></span>
<span><span class="co">#&gt; 2    chr1  22174512  22174513  frameshift            </span></span>
<span><span class="co">#&gt; 4    chr1  53555518  53555519  frameshift            </span></span>
<span><span class="co">#&gt; 8    chr1  53905657  53905658  frameshift            </span></span>
<span><span class="co">#&gt; 9    chr1  60331554  60331555  frameshift            </span></span>
<span><span class="co">#&gt; 12   chr1  82447611  82447612  frameshift            </span></span>
<span><span class="co">#&gt; 18   chr1  92446883  92446884  frameshift            </span></span>
<span><span class="co">#&gt; 25   chr1 104160689 104160690  frameshift            </span></span>
<span><span class="co">#&gt; 27   chr1 216538391 216538392  frameshift            </span></span>
<span><span class="co">#&gt; 29   chr1 237729903 237729904  frameshift            </span></span>
<span><span class="co">#&gt; 30   chr1 248263060 248263061  frameshift            </span></span>
<span><span class="co">#&gt; 32   chr2 164466573 164466574  frameshift            </span></span>
<span><span class="co">#&gt; 33   chr2 166850776 166850777  frameshift            </span></span>
<span><span class="co">#&gt; 36   chr2 179431633 179431634  frameshift            </span></span>
<span><span class="co">#&gt; 42   chr2 200213658 200213659  frameshift            </span></span>
<span><span class="co">#&gt; 47   chr3   8787442   8787443  frameshift            </span></span>
<span><span class="co">#&gt; 49   chr3  36779518  36779519  frameshift            </span></span>
<span><span class="co">#&gt; 50   chr3  38357967  38357968  frameshift            </span></span>
<span><span class="co">#&gt; 52   chr3  57283510  57283511  frameshift            </span></span>
<span><span class="co">#&gt; 55   chr3  58154222  58154223  frameshift            </span></span>
<span><span class="co">#&gt; 61   chr3  58792166  58792167  frameshift            </span></span>
<span><span class="co">#&gt; 63   chr3 119316821 119316822  frameshift            </span></span>
<span><span class="co">#&gt; 65   chr3 124398309 124398310  frameshift            </span></span>
<span><span class="co">#&gt; 67   chr3 168838908 168838909  frameshift            </span></span>
<span><span class="co">#&gt; 77   chr4  40104548  40104549  frameshift            </span></span>
<span><span class="co">#&gt; 80   chr4  71275513  71275514  frameshift            </span></span>
<span><span class="co">#&gt; 81   chr4  74453626  74453627  frameshift            </span></span>
<span><span class="co">#&gt; 85   chr4 115544477 115544478  frameshift            </span></span>
<span><span class="co">#&gt; 87   chr4 188924096 188924097  frameshift            </span></span>
<span><span class="co">#&gt; 90   chr4 189061712 189061713  frameshift            </span></span>
<span><span class="co">#&gt; 91   chr5    169493    169494  frameshift            </span></span>
<span><span class="co">#&gt; 92   chr5  23523442  23523443  frameshift            </span></span>
<span><span class="co">#&gt; 93   chr5  52183723  52183724  frameshift            </span></span>
<span><span class="co">#&gt; 94   chr5  74441953  74441954  frameshift            </span></span>
<span><span class="co">#&gt; 95   chr5 106722968 106722969  frameshift            </span></span>
<span><span class="co">#&gt; 96   chr5 135692398 135692399  frameshift            </span></span>
<span><span class="co">#&gt; 100  chr5 140263104 140263105  frameshift            </span></span>
<span><span class="co">#&gt; 103  chr5 158745750 158745751  frameshift            </span></span>
<span><span class="co">#&gt; 104  chr6  26157225  26157226  frameshift            </span></span>
<span><span class="co">#&gt; 105  chr6  27861573  27861574  frameshift            </span></span>
<span><span class="co">#&gt; 106  chr6  36946316  36946317  frameshift            </span></span>
<span><span class="co">#&gt; 109  chr6 110763563 110763564  frameshift            </span></span>
<span><span class="co">#&gt; 111  chr6 111621324 111621325  frameshift            </span></span>
<span><span class="co">#&gt; 115  chr6 121768896 121768897  frameshift            </span></span>
<span><span class="co">#&gt; 119  chr7  20725372  20725373  frameshift            </span></span>
<span><span class="co">#&gt; 121  chr7  21847599  21847600  frameshift            </span></span>
<span><span class="co">#&gt; 123  chr7  70880969  70880970  frameshift            </span></span>
<span><span class="co">#&gt; 124  chr7  91793446  91793447  frameshift            </span></span>
<span><span class="co">#&gt; 125  chr7  94293707  94293708  frameshift            </span></span>
<span><span class="co">#&gt; 128  chr7 100350058 100350059  frameshift            </span></span>
<span><span class="co">#&gt; 130  chr7 106509682 106509683  frameshift            </span></span>
<span><span class="co">#&gt; 133  chr7 113558721 113558722  frameshift            </span></span>
<span><span class="co">#&gt; 134  chr7 114619804 114619805  frameshift            </span></span>
<span><span class="co">#&gt; 135  chr7 128521730 128521731  frameshift            </span></span>
<span><span class="co">#&gt; 136  chr7 141629952 141629953  frameshift            </span></span>
<span><span class="co">#&gt; 139  chr8  10480111  10480112  frameshift            </span></span>
<span><span class="co">#&gt; 140  chr8  24171006  24171007  frameshift            </span></span>
<span><span class="co">#&gt; 142  chr8  38003841  38003842  frameshift            </span></span>
<span><span class="co">#&gt; 143  chr8  40389750  40389751  frameshift            </span></span>
<span><span class="co">#&gt; 145  chr8  72969169  72969170  frameshift            </span></span>
<span><span class="co">#&gt; 146  chr8  87679166  87679167  frameshift            </span></span>
<span><span class="co">#&gt; 148  chr8 145095790 145095791  frameshift            </span></span>
<span><span class="co">#&gt; 150  chr8 145577702 145577703  frameshift            </span></span>
<span><span class="co">#&gt; 153  chr9  14662289  14662290  frameshift            </span></span>
<span><span class="co">#&gt; 154  chr9  19049697  19049698  frameshift            </span></span>
<span><span class="co">#&gt; 155  chr9 135930484 135930485  frameshift            </span></span>
<span><span class="co">#&gt; 156 chr11   1250475   1250476  frameshift            </span></span>
<span><span class="co">#&gt; 158 chr11  59423175  59423176  frameshift            </span></span>
<span><span class="co">#&gt; 160 chr11  59829993  59829994  frameshift            </span></span>
<span><span class="co">#&gt; 161 chr11  60899918  60899919  frameshift            </span></span>
<span><span class="co">#&gt; 162 chr11  61300539  61300540  frameshift            </span></span>
<span><span class="co">#&gt; 164 chr11  62886736  62886737  frameshift            </span></span>
<span><span class="co">#&gt; 166 chr12  47629501  47629502  frameshift            </span></span>
<span><span class="co">#&gt; 169 chr12  54790107  54790108  frameshift            </span></span>
<span><span class="co">#&gt; 170 chr12  54790131  54790132  frameshift            </span></span>
<span><span class="co">#&gt; 171 chr12  58016861  58016862  frameshift            </span></span>
<span><span class="co">#&gt; 172 chr12 108589877 108589878  frameshift            </span></span>
<span><span class="co">#&gt; 174 chr12 117723962 117723963  frameshift            </span></span>
<span><span class="co">#&gt; 178 chr12 132636090 132636091  frameshift            </span></span>
<span><span class="co">#&gt; 179 chr14  22102490  22102491  frameshift            </span></span>
<span><span class="co">#&gt; 180 chr14  39772697  39772698  frameshift            </span></span>
<span><span class="co">#&gt; 192 chr14  61746872  61746873  frameshift            </span></span>
<span><span class="co">#&gt; 193 chr14 104216175 104216176  frameshift            </span></span>
<span><span class="co">#&gt; 195 chr14 105411929 105411930  frameshift            </span></span>
<span><span class="co">#&gt; 197 chr15  62456770  62456771  frameshift            </span></span>
<span><span class="co">#&gt; 199 chr16  11137915  11137916  frameshift            </span></span>
<span><span class="co">#&gt; 202 chr16  21213539  21213540  frameshift            </span></span>
<span><span class="co">#&gt; 204 chr16  30735316  30735317  frameshift            </span></span>
<span><span class="co">#&gt; 206 chr16  67646006  67646007  frameshift            </span></span>
<span><span class="co">#&gt; 207 chr17   4576719   4576720  frameshift            </span></span>
<span><span class="co">#&gt; 209 chr17   7577106   7577107  frameshift            </span></span>
<span><span class="co">#&gt; 218 chr17   8812440   8812441  frameshift            </span></span>
<span><span class="co">#&gt; 220 chr17  37903076  37903077  frameshift            </span></span>
<span><span class="co">#&gt; 224 chr19  14952247  14952248  frameshift            </span></span>
<span><span class="co">#&gt; 225 chr19  15292495  15292496  frameshift            </span></span>
<span><span class="co">#&gt; 227 chr19  49002070  49002071  frameshift            </span></span>
<span><span class="co">#&gt; 228 chr19  50397717  50397718  frameshift            </span></span>
<span><span class="co">#&gt; 233 chr19  50919900  50919901  frameshift            </span></span>
<span><span class="co">#&gt; 236 chr19  52001351  52001352  frameshift            </span></span>
<span><span class="co">#&gt; 239 chr19  52249683  52249684  frameshift            </span></span>
<span><span class="co">#&gt; 242 chr19  55711727  55711728  frameshift            </span></span>
<span><span class="co">#&gt; 245 chr21  32185475  32185476  frameshift            </span></span>
<span><span class="co">#&gt; 246  chrX   8555871   8555872  frameshift            </span></span>
<span><span class="co">#&gt; 247  chrX  19983163  19983164  frameshift            </span></span>
<span><span class="co">#&gt; 249  chrX 100240739 100240740  frameshift            </span></span>
<span><span class="co">#&gt; 252  chrX 100668298 100668299  frameshift            </span></span>
<span><span class="co">#&gt; 255  chrX 115304424 115304425  frameshift            </span></span>
<span><span class="co">#&gt; 257  chrX 124454501 124454502  frameshift            </span></span>
<span><span class="co">#&gt; 258  chrX 129349850 129349851  frameshift</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 5 × 10</span></span></span>
<span><span class="co">#&gt;   chr      from     to gene_symbol location consequence refAA varAA driver_label</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chr1   2.22<span style="color: #949494;">e</span>7 2.22<span style="color: #949494;">e</span>7 HSPG2       coding   frameshift  <span style="color: #949494;">""</span>    <span style="color: #949494;">""</span>    HSPG2_-&gt;    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chr3   1.69<span style="color: #949494;">e</span>8 1.69<span style="color: #949494;">e</span>8 MECOM       coding   frameshift  <span style="color: #949494;">""</span>    <span style="color: #949494;">""</span>    MECOM_-&gt;    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chr16  6.76<span style="color: #949494;">e</span>7 6.76<span style="color: #949494;">e</span>7 CTCF        coding   frameshift  <span style="color: #949494;">""</span>    <span style="color: #949494;">""</span>    CTCF_-&gt;     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chr17  7.58<span style="color: #949494;">e</span>6 7.58<span style="color: #949494;">e</span>6 TP53        coding:… frameshift  <span style="color: #949494;">""</span>    <span style="color: #949494;">""</span>    TP53_-&gt;     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> chr19  5.09<span style="color: #949494;">e</span>7 5.09<span style="color: #949494;">e</span>7 POLD1       coding   frameshift  <span style="color: #949494;">""</span>    <span style="color: #949494;">""</span>    POLD1_-&gt;    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: is_driver &lt;lgl&gt;</span></span></span>
<span></span>
<span><span class="va">x_new</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { <span style="color: #BBBB00;">CTCF_-&gt;, POLD1_-&gt;</span> }</span></span>
<span><span class="co">#&gt;    4:2  [n = 1893, L =  331 Mb] ■■■■■■■  { <span style="color: #BBBB00;">MECOM_-&gt;</span> }</span></span>
<span><span class="co">#&gt;    3:2  [n = 1625, L =  357 Mb] ■■■■■■  { <span style="color: #BBBB00;">HSPG2_-&gt;</span> }</span></span>
<span><span class="co">#&gt;    2:1  [n = 1563, L =  420 Mb] ■■■■■■</span></span>
<span><span class="co">#&gt;    3:0  [n =  312, L =  137 Mb] ■</span></span>
<span><span class="co">#&gt;    2:0  [n =   81, L =   39 Mb]   { <span style="color: #BBBB00;">TP53_-&gt;</span> }</span></span>
<span><span class="co">#&gt;   16:2  [n =    4, L =    0 Mb] </span></span>
<span><span class="co">#&gt;   25:2  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;    3:1  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;  106:1  [n =    1, L =    0 Mb]</span></span>
<span><span class="co">#&gt;          chr      from        to ref alt  DP NV       VAF driver_label is_driver</span></span>
<span><span class="co">#&gt;         chr1  22174512  22174513   G   A 168 94 0.5595238     HSPG2_-&gt;      TRUE</span></span>
<span><span class="co">#&gt;         chr3 168838908 168838909   C   T 192 24 0.1250000     MECOM_-&gt;      TRUE</span></span>
<span><span class="co">#&gt;        chr16  67646006  67646007   C   T 120 54 0.4500000      CTCF_-&gt;      TRUE</span></span>
<span><span class="co">#&gt;        chr17   7577106   7577107   G   C  84 78 0.9285714      TP53_-&gt;      TRUE</span></span>
<span><span class="co">#&gt;        chr19  50919900  50919901   C   T 145 61 0.4206897     POLD1_-&gt;      TRUE</span></span></code></pre></div>
<p>Note that there can be multiple locations and consequences for a
single variant. This happens as we try to annotate the mutations in a
transcript-agnostic manner, consequently we report all possible effects
and locations for any transcript (separated by <code>:</code>).
Comparison between known and newly-annotated drivers seems
consistent.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Reference driver mutations</span></span>
<span><span class="va">x</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] MySample</span> <span style="color: #00BB00;">12963</span> mutations in <span style="color: #00BB00;">267</span> segments (<span style="color: #00BB00;">267</span> clonal, <span style="color: #00BB00;">0</span> subclonal).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Clonal CNAs</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { <span style="color: #BBBB00;">CTCF</span> }</span></span>
<span><span class="co">#&gt;    4:2  [n = 1893, L =  331 Mb] ■■■■■■■</span></span>
<span><span class="co">#&gt;    3:2  [n = 1625, L =  357 Mb] ■■■■■■</span></span>
<span><span class="co">#&gt;    2:1  [n = 1563, L =  420 Mb] ■■■■■■  { <span style="color: #BBBB00;">TTN</span> }</span></span>
<span><span class="co">#&gt;    3:0  [n =  312, L =  137 Mb] ■</span></span>
<span><span class="co">#&gt;    2:0  [n =   81, L =   39 Mb]   { <span style="color: #BBBB00;">TP53</span> }</span></span>
<span><span class="co">#&gt;   16:2  [n =    4, L =    0 Mb] </span></span>
<span><span class="co">#&gt;   25:2  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;    3:1  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;  106:1  [n =    1, L =    0 Mb]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample Purity: 89% ~ Ploidy: 4.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> There are 3 annotated driver(s) mapped to clonal CNAs.</span></span>
<span><span class="co">#&gt;          chr      from        to ref alt  DP NV       VAF driver_label is_driver</span></span>
<span><span class="co">#&gt;         chr2 179431633 179431634   C   T 117 77 0.6581197          TTN      TRUE</span></span>
<span><span class="co">#&gt;        chr16  67646006  67646007   C   T 120 54 0.4500000         CTCF      TRUE</span></span>
<span><span class="co">#&gt;        chr17   7577106   7577107   G   C  84 78 0.9285714         TP53      TRUE</span></span>
<span></span>
<span><span class="co"># New driver mutations</span></span>
<span><span class="va">x_new</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] MySample</span> <span style="color: #00BB00;">12963</span> mutations in <span style="color: #00BB00;">267</span> segments (<span style="color: #00BB00;">267</span> clonal, <span style="color: #00BB00;">0</span> subclonal).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Clonal CNAs</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { <span style="color: #BBBB00;">CTCF_-&gt;, POLD1_-&gt;</span> }</span></span>
<span><span class="co">#&gt;    4:2  [n = 1893, L =  331 Mb] ■■■■■■■  { <span style="color: #BBBB00;">MECOM_-&gt;</span> }</span></span>
<span><span class="co">#&gt;    3:2  [n = 1625, L =  357 Mb] ■■■■■■  { <span style="color: #BBBB00;">HSPG2_-&gt;</span> }</span></span>
<span><span class="co">#&gt;    2:1  [n = 1563, L =  420 Mb] ■■■■■■</span></span>
<span><span class="co">#&gt;    3:0  [n =  312, L =  137 Mb] ■</span></span>
<span><span class="co">#&gt;    2:0  [n =   81, L =   39 Mb]   { <span style="color: #BBBB00;">TP53_-&gt;</span> }</span></span>
<span><span class="co">#&gt;   16:2  [n =    4, L =    0 Mb] </span></span>
<span><span class="co">#&gt;   25:2  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;    3:1  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;  106:1  [n =    1, L =    0 Mb]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample Purity: 89% ~ Ploidy: 4.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> There are 5 annotated driver(s) mapped to clonal CNAs.</span></span>
<span><span class="co">#&gt;          chr      from        to ref alt  DP NV       VAF driver_label is_driver</span></span>
<span><span class="co">#&gt;         chr1  22174512  22174513   G   A 168 94 0.5595238     HSPG2_-&gt;      TRUE</span></span>
<span><span class="co">#&gt;         chr3 168838908 168838909   C   T 192 24 0.1250000     MECOM_-&gt;      TRUE</span></span>
<span><span class="co">#&gt;        chr16  67646006  67646007   C   T 120 54 0.4500000      CTCF_-&gt;      TRUE</span></span>
<span><span class="co">#&gt;        chr17   7577106   7577107   G   C  84 78 0.9285714      TP53_-&gt;      TRUE</span></span>
<span><span class="co">#&gt;        chr19  50919900  50919901   C   T 145 61 0.4206897     POLD1_-&gt;      TRUE</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="type-specific-drivers-annotation">Type-specific drivers annotation<a class="anchor" aria-label="anchor" href="#type-specific-drivers-annotation"></a>
</h3>
<p>One can restrict the list of potential genes to use for drivers
detection. To this extent, it is convenient to use the available
database and the information regarding the input cancer type.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># We pretend to work with OV, ovarian cancer</span></span>
<span><span class="va">OV_drivers</span> <span class="op">=</span> <span class="va">intogen_drivers</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">tumour</span> <span class="op">==</span> <span class="st">'OV'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">OV_drivers</span><span class="op">$</span><span class="va">gene</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "ACKR3"   "ARID1A"  "ATRX"    "BRAF"    "BRCA1"   "BRCA2"   "CACNA1D"</span></span>
<span><span class="co">#&gt;  [8] "CDK12"   "ELL"     "EPHA7"   "ERBB2"   "ERG"     "FAT1"    "FGFR1"  </span></span>
<span><span class="co">#&gt; [15] "FOXL2"   "JAK1"    "KMT2A"   "KMT2C"   "KMT2D"   "KRAS"    "LATS1"  </span></span>
<span><span class="co">#&gt; [22] "LRP1B"   "MYH9"    "NF1"     "NF2"     "NOTCH1"  "NRAS"    "PIK3CA" </span></span>
<span><span class="co">#&gt; [29] "PPP2R1A" "PTEN"    "PTPRB"   "PTPRT"   "RB1"     "SETD1B"  "STAT5B" </span></span>
<span><span class="co">#&gt; [36] "TP53"    "UGT2B17"</span></span>
<span></span>
<span><span class="co"># Run annotation function</span></span>
<span><span class="va">x_new_ov</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotate_variants.html">annotate_variants</a></span><span class="op">(</span><span class="va">x</span>, drivers <span class="op">=</span> <span class="va">OV_drivers</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Preparing mutations</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Existing driver annotations in your data will be cancelled.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Preparing mutations</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 17</span></span></span>
<span><span class="co">#&gt;   chr        from       to ref   alt      NV    DP   VAF FILTER ANNOVAR_FUNCTION</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chr2  179<span style="text-decoration: underline;">431</span>633   1.79<span style="color: #949494;">e</span>8 C     T        77   117 0.658 PASS   exonic          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chr16  67<span style="text-decoration: underline;">646</span>006   6.76<span style="color: #949494;">e</span>7 C     T        54   120 0.45  PASS   exonic          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chr17   7<span style="text-decoration: underline;">577</span>106   7.58<span style="color: #949494;">e</span>6 G     C        78    84 0.929 PASS   exonic          </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 7 more variables: GENE &lt;chr&gt;, is_driver &lt;lgl&gt;, driver_label &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   type &lt;chr&gt;, karyotype &lt;chr&gt;, segment_id &lt;chr&gt;, cna &lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Preparing mutations ... done</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Locating variants with <span style="color: #BBBB00;">VariantAnnotation</span></span></span>
<span><span class="co">#&gt; Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 364 out-of-bound ranges located on sequences</span></span>
<span><span class="co">#&gt;   4529, 4631, 4632, 1653, 1654, 1655, 1656, 12192, 12362, 12363, 12364,</span></span>
<span><span class="co">#&gt;   12365, 12366, 12367, 12517, 12518, 12519, 12520, 13123, 13124, 13974,</span></span>
<span><span class="co">#&gt;   13975, 13976, 13994, 13995, 17007, 17008, 17009, 17010, 17011, 17012,</span></span>
<span><span class="co">#&gt;   17014, 17015, 17016, 17793, 18007, 20532, 22527, 22954, 22955, 22956,</span></span>
<span><span class="co">#&gt;   23347, 27275, 27276, 28075, 28085, 28893, 29046, 29047, 29242, 31362,</span></span>
<span><span class="co">#&gt;   32048, 32050, 33339, 33714, 33715, 32912, 32913, 40960, 40961, 44219,</span></span>
<span><span class="co">#&gt;   44220, 44242, 44262, 44263, 44356, 46050, 46054, 46062, 46063, 46064,</span></span>
<span><span class="co">#&gt;   46065, 46066, 53476, 53477, 53497, 53498, 59180, 57764, 57766, 61092,</span></span>
<span><span class="co">#&gt;   69815, 69930, 70330, 70331, 70332, 77129, 77130, 76239, 76240, 76241,</span></span>
<span><span class="co">#&gt;   77987, and 77988. Note that ranges located on a sequence whose length</span></span>
<span><span class="co">#&gt;   is unknown (NA) or on a circular sequence are not considered</span></span>
<span><span class="co">#&gt;   out-of-bound (use seqlengths() and isCircular() to get the lengths and</span></span>
<span><span class="co">#&gt;   circularity flags of the underlying sequences). You can use trim() to</span></span>
<span><span class="co">#&gt;   trim these ranges. See ?`trim,GenomicRanges-method` for more</span></span>
<span><span class="co">#&gt;   information.</span></span>
<span><span class="co">#&gt; 'select()' returned many:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned many:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned many:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned many:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Locating variants with <span style="color: #BBBB00;">VariantAnnotation</span> ... done</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Traslating Entrez ids</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Traslating Entrez ids ... done</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Transforming data</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Transforming data ... done</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Predicting coding</span></span>
<span><span class="co">#&gt; Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 315 out-of-bound ranges located on sequences</span></span>
<span><span class="co">#&gt;   4529, 4631, 4632, 1653, 1654, 1655, 1656, 12192, 12362, 12363, 12364,</span></span>
<span><span class="co">#&gt;   12365, 12366, 12367, 12517, 12518, 12519, 12520, 13123, 13124, 13974,</span></span>
<span><span class="co">#&gt;   13975, 13976, 13994, 13995, 17007, 17008, 17009, 17010, 17011, 17012,</span></span>
<span><span class="co">#&gt;   17014, 17015, 17016, 17793, 18007, 20532, 22527, 22954, 22955, 22956,</span></span>
<span><span class="co">#&gt;   23347, 27275, 27276, 28075, 28085, 28893, 29046, 29047, 29242, 31362,</span></span>
<span><span class="co">#&gt;   32048, 32050, 33339, 33714, 33715, 32912, 32913, 40960, 40961, 44219,</span></span>
<span><span class="co">#&gt;   44220, 44242, 44262, 44263, 44356, 53476, 53477, 53497, 53498, 59180,</span></span>
<span><span class="co">#&gt;   57764, 57766, 61092, 69815, 69930, 70330, 70331, 70332, 77129, 77130,</span></span>
<span><span class="co">#&gt;   76239, 76240, 76241, 77987, and 77988. Note that ranges located on a</span></span>
<span><span class="co">#&gt;   sequence whose length is unknown (NA) or on a circular sequence are not</span></span>
<span><span class="co">#&gt;   considered out-of-bound (use seqlengths() and isCircular() to get the</span></span>
<span><span class="co">#&gt;   lengths and circularity flags of the underlying sequences). You can use</span></span>
<span><span class="co">#&gt;   trim() to trim these ranges. See ?`trim,GenomicRanges-method` for more</span></span>
<span><span class="co">#&gt;   information.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Predicting coding── Coding substitutions found </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Predicting coding</span></span>
<span><span class="co">#&gt;       chr      from        to consequence refAA varAA</span></span>
<span><span class="co">#&gt; 1    chr1  19181096  19181097  frameshift            </span></span>
<span><span class="co">#&gt; 2    chr1  22174512  22174513  frameshift            </span></span>
<span><span class="co">#&gt; 4    chr1  53555518  53555519  frameshift            </span></span>
<span><span class="co">#&gt; 8    chr1  53905657  53905658  frameshift            </span></span>
<span><span class="co">#&gt; 9    chr1  60331554  60331555  frameshift            </span></span>
<span><span class="co">#&gt; 12   chr1  82447611  82447612  frameshift            </span></span>
<span><span class="co">#&gt; 18   chr1  92446883  92446884  frameshift            </span></span>
<span><span class="co">#&gt; 25   chr1 104160689 104160690  frameshift            </span></span>
<span><span class="co">#&gt; 27   chr1 216538391 216538392  frameshift            </span></span>
<span><span class="co">#&gt; 29   chr1 237729903 237729904  frameshift            </span></span>
<span><span class="co">#&gt; 30   chr1 248263060 248263061  frameshift            </span></span>
<span><span class="co">#&gt; 32   chr2 164466573 164466574  frameshift            </span></span>
<span><span class="co">#&gt; 33   chr2 166850776 166850777  frameshift            </span></span>
<span><span class="co">#&gt; 36   chr2 179431633 179431634  frameshift            </span></span>
<span><span class="co">#&gt; 42   chr2 200213658 200213659  frameshift            </span></span>
<span><span class="co">#&gt; 47   chr3   8787442   8787443  frameshift            </span></span>
<span><span class="co">#&gt; 49   chr3  36779518  36779519  frameshift            </span></span>
<span><span class="co">#&gt; 50   chr3  38357967  38357968  frameshift            </span></span>
<span><span class="co">#&gt; 52   chr3  57283510  57283511  frameshift            </span></span>
<span><span class="co">#&gt; 55   chr3  58154222  58154223  frameshift            </span></span>
<span><span class="co">#&gt; 61   chr3  58792166  58792167  frameshift            </span></span>
<span><span class="co">#&gt; 63   chr3 119316821 119316822  frameshift            </span></span>
<span><span class="co">#&gt; 65   chr3 124398309 124398310  frameshift            </span></span>
<span><span class="co">#&gt; 67   chr3 168838908 168838909  frameshift            </span></span>
<span><span class="co">#&gt; 77   chr4  40104548  40104549  frameshift            </span></span>
<span><span class="co">#&gt; 80   chr4  71275513  71275514  frameshift            </span></span>
<span><span class="co">#&gt; 81   chr4  74453626  74453627  frameshift            </span></span>
<span><span class="co">#&gt; 85   chr4 115544477 115544478  frameshift            </span></span>
<span><span class="co">#&gt; 87   chr4 188924096 188924097  frameshift            </span></span>
<span><span class="co">#&gt; 90   chr4 189061712 189061713  frameshift            </span></span>
<span><span class="co">#&gt; 91   chr5    169493    169494  frameshift            </span></span>
<span><span class="co">#&gt; 92   chr5  23523442  23523443  frameshift            </span></span>
<span><span class="co">#&gt; 93   chr5  52183723  52183724  frameshift            </span></span>
<span><span class="co">#&gt; 94   chr5  74441953  74441954  frameshift            </span></span>
<span><span class="co">#&gt; 95   chr5 106722968 106722969  frameshift            </span></span>
<span><span class="co">#&gt; 96   chr5 135692398 135692399  frameshift            </span></span>
<span><span class="co">#&gt; 100  chr5 140263104 140263105  frameshift            </span></span>
<span><span class="co">#&gt; 103  chr5 158745750 158745751  frameshift            </span></span>
<span><span class="co">#&gt; 104  chr6  26157225  26157226  frameshift            </span></span>
<span><span class="co">#&gt; 105  chr6  27861573  27861574  frameshift            </span></span>
<span><span class="co">#&gt; 106  chr6  36946316  36946317  frameshift            </span></span>
<span><span class="co">#&gt; 109  chr6 110763563 110763564  frameshift            </span></span>
<span><span class="co">#&gt; 111  chr6 111621324 111621325  frameshift            </span></span>
<span><span class="co">#&gt; 115  chr6 121768896 121768897  frameshift            </span></span>
<span><span class="co">#&gt; 119  chr7  20725372  20725373  frameshift            </span></span>
<span><span class="co">#&gt; 121  chr7  21847599  21847600  frameshift            </span></span>
<span><span class="co">#&gt; 123  chr7  70880969  70880970  frameshift            </span></span>
<span><span class="co">#&gt; 124  chr7  91793446  91793447  frameshift            </span></span>
<span><span class="co">#&gt; 125  chr7  94293707  94293708  frameshift            </span></span>
<span><span class="co">#&gt; 128  chr7 100350058 100350059  frameshift            </span></span>
<span><span class="co">#&gt; 130  chr7 106509682 106509683  frameshift            </span></span>
<span><span class="co">#&gt; 133  chr7 113558721 113558722  frameshift            </span></span>
<span><span class="co">#&gt; 134  chr7 114619804 114619805  frameshift            </span></span>
<span><span class="co">#&gt; 135  chr7 128521730 128521731  frameshift            </span></span>
<span><span class="co">#&gt; 136  chr7 141629952 141629953  frameshift            </span></span>
<span><span class="co">#&gt; 139  chr8  10480111  10480112  frameshift            </span></span>
<span><span class="co">#&gt; 140  chr8  24171006  24171007  frameshift            </span></span>
<span><span class="co">#&gt; 142  chr8  38003841  38003842  frameshift            </span></span>
<span><span class="co">#&gt; 143  chr8  40389750  40389751  frameshift            </span></span>
<span><span class="co">#&gt; 145  chr8  72969169  72969170  frameshift            </span></span>
<span><span class="co">#&gt; 146  chr8  87679166  87679167  frameshift            </span></span>
<span><span class="co">#&gt; 148  chr8 145095790 145095791  frameshift            </span></span>
<span><span class="co">#&gt; 150  chr8 145577702 145577703  frameshift            </span></span>
<span><span class="co">#&gt; 153  chr9  14662289  14662290  frameshift            </span></span>
<span><span class="co">#&gt; 154  chr9  19049697  19049698  frameshift            </span></span>
<span><span class="co">#&gt; 155  chr9 135930484 135930485  frameshift            </span></span>
<span><span class="co">#&gt; 156 chr11   1250475   1250476  frameshift            </span></span>
<span><span class="co">#&gt; 158 chr11  59423175  59423176  frameshift            </span></span>
<span><span class="co">#&gt; 160 chr11  59829993  59829994  frameshift            </span></span>
<span><span class="co">#&gt; 161 chr11  60899918  60899919  frameshift            </span></span>
<span><span class="co">#&gt; 162 chr11  61300539  61300540  frameshift            </span></span>
<span><span class="co">#&gt; 164 chr11  62886736  62886737  frameshift            </span></span>
<span><span class="co">#&gt; 166 chr12  47629501  47629502  frameshift            </span></span>
<span><span class="co">#&gt; 169 chr12  54790107  54790108  frameshift            </span></span>
<span><span class="co">#&gt; 170 chr12  54790131  54790132  frameshift            </span></span>
<span><span class="co">#&gt; 171 chr12  58016861  58016862  frameshift            </span></span>
<span><span class="co">#&gt; 172 chr12 108589877 108589878  frameshift            </span></span>
<span><span class="co">#&gt; 174 chr12 117723962 117723963  frameshift            </span></span>
<span><span class="co">#&gt; 178 chr12 132636090 132636091  frameshift            </span></span>
<span><span class="co">#&gt; 179 chr14  22102490  22102491  frameshift            </span></span>
<span><span class="co">#&gt; 180 chr14  39772697  39772698  frameshift            </span></span>
<span><span class="co">#&gt; 192 chr14  61746872  61746873  frameshift            </span></span>
<span><span class="co">#&gt; 193 chr14 104216175 104216176  frameshift            </span></span>
<span><span class="co">#&gt; 195 chr14 105411929 105411930  frameshift            </span></span>
<span><span class="co">#&gt; 197 chr15  62456770  62456771  frameshift            </span></span>
<span><span class="co">#&gt; 199 chr16  11137915  11137916  frameshift            </span></span>
<span><span class="co">#&gt; 202 chr16  21213539  21213540  frameshift            </span></span>
<span><span class="co">#&gt; 204 chr16  30735316  30735317  frameshift            </span></span>
<span><span class="co">#&gt; 206 chr16  67646006  67646007  frameshift            </span></span>
<span><span class="co">#&gt; 207 chr17   4576719   4576720  frameshift            </span></span>
<span><span class="co">#&gt; 209 chr17   7577106   7577107  frameshift            </span></span>
<span><span class="co">#&gt; 218 chr17   8812440   8812441  frameshift            </span></span>
<span><span class="co">#&gt; 220 chr17  37903076  37903077  frameshift            </span></span>
<span><span class="co">#&gt; 224 chr19  14952247  14952248  frameshift            </span></span>
<span><span class="co">#&gt; 225 chr19  15292495  15292496  frameshift            </span></span>
<span><span class="co">#&gt; 227 chr19  49002070  49002071  frameshift            </span></span>
<span><span class="co">#&gt; 228 chr19  50397717  50397718  frameshift            </span></span>
<span><span class="co">#&gt; 233 chr19  50919900  50919901  frameshift            </span></span>
<span><span class="co">#&gt; 236 chr19  52001351  52001352  frameshift            </span></span>
<span><span class="co">#&gt; 239 chr19  52249683  52249684  frameshift            </span></span>
<span><span class="co">#&gt; 242 chr19  55711727  55711728  frameshift            </span></span>
<span><span class="co">#&gt; 245 chr21  32185475  32185476  frameshift            </span></span>
<span><span class="co">#&gt; 246  chrX   8555871   8555872  frameshift            </span></span>
<span><span class="co">#&gt; 247  chrX  19983163  19983164  frameshift            </span></span>
<span><span class="co">#&gt; 249  chrX 100240739 100240740  frameshift            </span></span>
<span><span class="co">#&gt; 252  chrX 100668298 100668299  frameshift            </span></span>
<span><span class="co">#&gt; 255  chrX 115304424 115304425  frameshift            </span></span>
<span><span class="co">#&gt; 257  chrX 124454501 124454502  frameshift            </span></span>
<span><span class="co">#&gt; 258  chrX 129349850 129349851  frameshift</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Predicting coding ... done</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Drivers annotation</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Drivers annotation── Found 1 driver(s) </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Drivers annotation</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 10</span></span></span>
<span><span class="co">#&gt;   chr      from     to gene_symbol location consequence refAA varAA driver_label</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chr17 7<span style="text-decoration: underline;">577</span>106 7.58<span style="color: #949494;">e</span>6 TP53        coding:… frameshift  <span style="color: #949494;">""</span>    <span style="color: #949494;">""</span>    TP53_-&gt;     </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: is_driver &lt;lgl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Drivers annotation ... done</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">CNAqc - CNA Quality Check</span> <span style="color: #00BBBB;">───────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">hg19</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Found annotated driver mutations: <span style="color: #00BB00;">TP53_-&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">12963</span> somatic mutations: <span style="color: #00BB00;">12963</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">0</span> indels.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">267</span> segments: <span style="color: #00BB00;">267</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span></span>
<span><span class="co">#&gt; Warning in map_mutations_to_clonal_segments(mutations, cna_clonal): [CNAqc] a</span></span>
<span><span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">12963</span> mutations mapped to clonal CNAs.</span></span>
<span></span>
<span><span class="co"># Reference driver mutations</span></span>
<span><span class="va">x</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] MySample</span> <span style="color: #00BB00;">12963</span> mutations in <span style="color: #00BB00;">267</span> segments (<span style="color: #00BB00;">267</span> clonal, <span style="color: #00BB00;">0</span> subclonal).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Clonal CNAs</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { <span style="color: #BBBB00;">CTCF</span> }</span></span>
<span><span class="co">#&gt;    4:2  [n = 1893, L =  331 Mb] ■■■■■■■</span></span>
<span><span class="co">#&gt;    3:2  [n = 1625, L =  357 Mb] ■■■■■■</span></span>
<span><span class="co">#&gt;    2:1  [n = 1563, L =  420 Mb] ■■■■■■  { <span style="color: #BBBB00;">TTN</span> }</span></span>
<span><span class="co">#&gt;    3:0  [n =  312, L =  137 Mb] ■</span></span>
<span><span class="co">#&gt;    2:0  [n =   81, L =   39 Mb]   { <span style="color: #BBBB00;">TP53</span> }</span></span>
<span><span class="co">#&gt;   16:2  [n =    4, L =    0 Mb] </span></span>
<span><span class="co">#&gt;   25:2  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;    3:1  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;  106:1  [n =    1, L =    0 Mb]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample Purity: 89% ~ Ploidy: 4.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> There are 3 annotated driver(s) mapped to clonal CNAs.</span></span>
<span><span class="co">#&gt;          chr      from        to ref alt  DP NV       VAF driver_label is_driver</span></span>
<span><span class="co">#&gt;         chr2 179431633 179431634   C   T 117 77 0.6581197          TTN      TRUE</span></span>
<span><span class="co">#&gt;        chr16  67646006  67646007   C   T 120 54 0.4500000         CTCF      TRUE</span></span>
<span><span class="co">#&gt;        chr17   7577106   7577107   G   C  84 78 0.9285714         TP53      TRUE</span></span>
<span></span>
<span><span class="co"># New driver mutations, searching for OV drivers</span></span>
<span><span class="va">x_new_ov</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ CNAqc ] MySample</span> <span style="color: #00BB00;">12963</span> mutations in <span style="color: #00BB00;">267</span> segments (<span style="color: #00BB00;">267</span> clonal, <span style="color: #00BB00;">0</span> subclonal).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Clonal CNAs</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■</span></span>
<span><span class="co">#&gt;    4:2  [n = 1893, L =  331 Mb] ■■■■■■■</span></span>
<span><span class="co">#&gt;    3:2  [n = 1625, L =  357 Mb] ■■■■■■</span></span>
<span><span class="co">#&gt;    2:1  [n = 1563, L =  420 Mb] ■■■■■■</span></span>
<span><span class="co">#&gt;    3:0  [n =  312, L =  137 Mb] ■</span></span>
<span><span class="co">#&gt;    2:0  [n =   81, L =   39 Mb]   { <span style="color: #BBBB00;">TP53_-&gt;</span> }</span></span>
<span><span class="co">#&gt;   16:2  [n =    4, L =    0 Mb] </span></span>
<span><span class="co">#&gt;   25:2  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;    3:1  [n =    2, L =    1 Mb] </span></span>
<span><span class="co">#&gt;  106:1  [n =    1, L =    0 Mb]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample Purity: 89% ~ Ploidy: 4.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> There are 1 annotated driver(s) mapped to clonal CNAs.</span></span>
<span><span class="co">#&gt;          chr    from      to ref alt DP NV       VAF driver_label is_driver</span></span>
<span><span class="co">#&gt;        chr17 7577106 7577107   G   C 84 78 0.9285714      TP53_-&gt;      TRUE</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="example-annotations">Example annotations<a class="anchor" aria-label="anchor" href="#example-annotations"></a>
</h2>
<p>Annotated data can undergo the canonical analysis workflow.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x_new</span>, which <span class="op">=</span> <span class="st">'VAF'</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x_new</span>, which <span class="op">=</span> <span class="st">'DP'</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x_new</span>, which <span class="op">=</span> <span class="st">'NV'</span><span class="op">)</span>,</span>
<span>  ncol <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  nrow <span class="op">=</span> <span class="fl">1</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="a8_annotate_variants_files/figure-html/unnamed-chunk-9-1.png" width="960"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The actual plot function</span></span>
<span><span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="../reference/plot_gw_counts.html">plot_gw_counts</a></span><span class="op">(</span><span class="va">x_new</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/plot_gw_vaf.html">plot_gw_vaf</a></span><span class="op">(</span><span class="va">x_new</span>, N <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/plot_gw_depth.html">plot_gw_depth</a></span><span class="op">(</span><span class="va">x_new</span>, N <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x_new</span><span class="op">)</span>,</span>
<span>  align <span class="op">=</span> <span class="st">'v'</span>,</span>
<span>  nrow <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">.15</span>, <span class="fl">.15</span>, <span class="fl">.15</span>, <span class="fl">.8</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">fill</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">fill</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Warning: The dot-dot notation (`..count..`) was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `after_stat(count)` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">CNAqc</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/caravagnalab/CNAqc/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning: Removed 24 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_segment()`).</span></span>
<span><span class="co">#&gt; Warning: Removed 24 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_rect()`).</span></span>
<span><span class="co">#&gt; Warning: Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_hline()`).</span></span>
<span><span class="co">#&gt; Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_hline()`).</span></span></code></pre></div>
<p><img src="a8_annotate_variants_files/figure-html/unnamed-chunk-10-1.png" width="912"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.caravagnalab.org" class="external-link">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
