<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>9. Automatic CNA calling • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="9. Automatic CNA calling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
    <li>
      <a href="../articles/a9_automatic.html">9. Automatic CNA calling</a>
    </li>
    <li>
      <a href="../articles/b10_MultiCNAqc.html">10. multiCNAqc</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>9. Automatic CNA calling</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqc/vignettes/a9_automatic.Rmd" class="external-link"><code>vignettes/a9_automatic.Rmd</code></a></small>
      <div class="hidden name"><code>a9_automatic.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="iterative-cna-calling-and-qc">Iterative CNA calling and QC<a class="anchor" aria-label="anchor" href="#iterative-cna-calling-and-qc"></a>
</h2>
<p>We are happy to integrate any copy number caller with CNAqc; do <a href="gcaravagn@units.it">get in touch</a> to discuss how to integrate
your caller.</p>
<div class="section level3">
<h3 id="sequenza-tool">Sequenza tool<a class="anchor" aria-label="anchor" href="#sequenza-tool"></a>
</h3>
<p>CNAqc provides automatic integration with <a href="https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html" class="external-link">Sequenza</a></p>
<ul>
<li>Favero F, Joshi T, Marquard AM, Birkbak NJ, Krzystanek M, Li Q,
Szallasi Z, Eklund AC “Sequenza: allele-specific copy number and
mutation profiles from tumor sequencing data.” Annals of Oncology 2015
Jan;26(1):64-70. <a href="https://doi.org/10.1093/annonc/mdu479" class="external-link uri">https://doi.org/10.1093/annonc/mdu479</a>
</li>
</ul>
<p>Sequenza calls <em>clonal</em> CNAs in a multi-steps Python/R
pipeline:</p>
<ol style="list-style-type: decimal">
<li>process a FASTA file to produce a GC Wiggle track file;</li>
<li>process tumour and normal BAMs and Wiggle files to produce a
<code>seqz</code> file;</li>
<li>bins the original <code>seqz</code> file;</li>
<li>process the binned <code>seqz</code> for normalisation and
segmentation;</li>
<li>perform a grid-search approach to estimate <em>cellularity</em> and
<em>ploidy</em>;</li>
</ol>
<p>One advantage of Sequenza is that results are computed for both the
best and possible alternative solutions.</p>
<p>Here cellularity is the Sequenza terminology for tumour purity; steps
1-3 are implemented in Python, while steps 4-6 in R. In our experience
we have often found the steps 1-3 to be run just once (often with
default parameter values), while the last steps are iterated to optimise
CNA calling.</p>
<p>We have implemented a calling pipeline (function
<code>Sequenza_CNAqc</code>) that revolves around steps 4-6, using CNAqc
to drive the fitting steps of Sequenza - especially step 5 - to
determine better CNA segments and purity/ploidy values. To run this
pipeline steps 1-3 need to be executed beforehand; according to the <a href="https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html" class="external-link">Sequenza</a>
documentation the default execution is as follows.</p>
<pre><code># Process a FASTA file to produce a GC Wiggle track file:
sequenza−utils gc_wiggle −w 50 --fasta hg19.fa -o hg19.gc50Base.wig.gz

# Process BAM and Wiggle files to produce a seqz file:
sequenza−utils bam2seqz -n normal.bam -t tumor.bam --fasta hg19.fa \
    -gc hg19.gc50Base.wig.gz -o out.seqz.gz
    
# Post-process by binning the original seqz file:
sequenza−utils seqz_binning --seqz out.seqz.gz -w 50 -o out small.seqz.gz</code></pre>
</div>
<div class="section level3">
<h3 id="pipeline">Pipeline<a class="anchor" aria-label="anchor" href="#pipeline"></a>
</h3>
<div class="float">
<img src="sqz_pipe.png" title="Sequenza - CNAqc pipeline" alt="Sequenza - CNAqc pipeline flowchart"><div class="figcaption"><em>Sequenza - CNAqc pipeline
flowchart</em></div>
</div>
<p>The pipeline starts from input ranges for cellularity and ploidy
values, as canonically required by Sequenza.</p>
<p>Parameters for segmentation are also imputed following the standard
Sequenza convention, and step 4 is performed once before starting an
iterative procedure that uses a caching system to avoid useless
repetitions of computations. Default Sequenza parameter values we
release with CNAqc have been delected to work best with whole-genome
sequencing data, a type of data we often use to run this pipeline.</p>
<p>The pipeline fits (Sequenza steps 5-6) cellularity and ploidy values,
and dumps results, which are quality controlled by peak detection via
CNAqc. The analysis with CNAqc can be carried out using either:</p>
<ul>
<li>somatic mutations called by Sequenza,</li>
<li>or an external set of input calls. We often prefer this option, if
possibile, and take, e.g., <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037593851-Mutect2" class="external-link">Mutect2</a>
calls.</li>
</ul>
<p>Besides the best solution by Sequenza, at every run up to two
alternative solutions are generated:</p>
<ol style="list-style-type: decimal">
<li>One optional by Sequenza, which might propose new exact values for
cellularity and ploidy that differ from the current best ones;</li>
<li>One by CNAqc, adjusting the current cellularity of the best Sequenza
solution, leaving the ploidy unchanged.</li>
</ol>
<p>These alternative solutions are queued in a list
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>L</mi><annotation encoding="application/x-tex">L</annotation></semantics></math>
of cellularity and ploidy values that should be tested, based on their
presence in a cache (the cache avoids to re-test solutions already
visited). Until the list
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>L</mi><annotation encoding="application/x-tex">L</annotation></semantics></math>
is empty, the point values of cellularity and ploidy are tested
repeating steps 5-6 with small ranges built around the proposed values.
For example, if a solution with cellularity
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>34</mn><mi>%</mi></mrow><annotation encoding="application/x-tex">34\%</annotation></semantics></math>
and ploidy
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>2.5</mn><annotation encoding="application/x-tex">2.5</annotation></semantics></math>
is to be tested, CNAqc runs Sequenza by using ranges
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">[</mo><mn>34</mn><mi>%</mi><mo>−</mo><mi>d</mi><mo>;</mo><mn>34</mn><mi>%</mi><mo>+</mo><mi>d</mi><mo stretchy="true" form="postfix">]</mo></mrow><annotation encoding="application/x-tex">[34\%-d;34\%+d]</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">[</mo><mn>2.5</mn><mo>−</mo><mi>c</mi><mo>;</mo><mn>2.5</mn><mo>+</mo><mi>c</mi><mo stretchy="true" form="postfix">]</mo></mrow><annotation encoding="application/x-tex">[2.5-c;2.5+c]</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>c</mi><annotation encoding="application/x-tex">c</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>d</mi><annotation encoding="application/x-tex">d</annotation></semantics></math>
are parameters of the pipeline.</p>
<p>Therefore the pipeline follows two paths to optimise the calls:</p>
<ul>
<li>the gradient of correction suggested by CNAqc,</li>
<li>Sequenza’s alternative solutions.</li>
</ul>
<p>Note that when CNAqc flags as PASS a solution that is visited for the
first time, any further correction will not be evaluated since the
solution will be found in the cache system of the pipeline.</p>
<p>At the end of the runs all the solutions identified are scored by
CNAqc and reported with a PASS/FAIL status associated; the pipeline
always indicates the best among the tested ones, based on the assigned
quality score.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.caravagnalab.org" class="external-link">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
