<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>10. multiCNAqc • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="10. multiCNAqc">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
    <li>
      <a href="../articles/a9_automatic.html">9. Automatic CNA calling</a>
    </li>
    <li>
      <a href="../articles/b10_MultiCNAqc.html">10. multiCNAqc</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>10. multiCNAqc</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqc/vignettes/b10_MultiCNAqc.Rmd" class="external-link"><code>vignettes/b10_MultiCNAqc.Rmd</code></a></small>
      <div class="hidden name"><code>b10_MultiCNAqc.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: tidyverse</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/CNAqc" class="external-link">CNAqc</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://cli.r-lib.org" class="external-link">cli</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: cli</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: patchwork</span></span></code></pre></div>
<div class="section level2">
<h2 id="create-a-multisample-cnaqc-object">Create a multisample CNAqc object<a class="anchor" aria-label="anchor" href="#create-a-multisample-cnaqc-object"></a>
</h2>
<p>If treating data belonging to the same patient, but from different
samples (ie: longitudinal or multi-region sampling), it might be handful
searching for genomic regions harboring abnormal copy number states in
all of the samples. Creating a <code>mCNAqc</code> object allows to
modify the existing segments, by searching for regions with altered copy
number state in all samples and discarding the others. It defines new
segments and remaps existing mutations on them.</p>
<div class="section level3">
<h3 id="analysis-setup">Analysis setup<a class="anchor" aria-label="anchor" href="#analysis-setup"></a>
</h3>
<p>For this example we will use the same data presented in the <a href="https://caravagnalab.github.io/CNAqc/articles/a7_example_MSeq.html">example
data analysis</a> in order to create a multisample CNAqc object for the
patient <code>Set06</code> (samples 1-3 of the same patient).</p>
<p>After having extracted the information on mutations and CNA for each
sample, we create a list of CNAqc objects. List names must be sample
names.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">CNAqc_samples</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sample_mutations</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span>mutations <span class="op">=</span> <span class="va">sample_mutations</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span>, </span>
<span>              cna <span class="op">=</span> <span class="va">cna</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span>, </span>
<span>              purity <span class="op">=</span> <span class="va">purity</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span>, </span>
<span>              sample <span class="op">=</span> <span class="va">x</span>,</span>
<span>              ref <span class="op">=</span> <span class="st">"GRCh38"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">CNAqc_samples</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">CNAqc_samples</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span><span class="va">x</span><span class="op">$</span><span class="va">sample</span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p>Run the quality control on the created object.</p>
<p>All the samples are characterized by different segments among the
genome, as can be see in the plot below:</p>
<p><img src="b10_MultiCNAqc_files/figure-html/unnamed-chunk-4-1.png" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="create-the-mcnaqc-object">Create the mCNAqc object<a class="anchor" aria-label="anchor" href="#create-the-mcnaqc-object"></a>
</h3>
<p>Create a <code>m_CNAqc</code> object using the all the CNAs included
in the objects. This will define new segments according to which genomic
regions are affected by CNAs in all samples, and remap the mutations on
them. It is possible to create a <code>m_CNAqc</code> object either
including or not the results of the peak analysis: setting the argument
<code>QC_filter</code> to <code>TRUE</code> will use in the new
segmentation only the segments passing the QC among the different
samples.</p>
<p>All breakpoints (<code>from</code> and <code>to</code> of each CNA
table) from all samples are selected and reordered. The list is then
used to generate new segments intervals; only segments that are present
in all samples will be kept. Mutations are then remapped on the new
segments.</p>
<p>The results will be stored as a list of CNAqc objects (one per
sample) in the <code>cnaqc_obj_new_segmentation</code> attribute of the
mCNAqc object. If running <code>multisample_init</code> with the
<code>keep_original = TRUE</code>, the original CNAqc object will be
stored in the <code>original_cnaqc_objc</code> attribute; else, results
of any additional analysis performed on the original data (ie: peaks
analysis or CCF estimation) will be stored in the
<code>original_additional_info</code> attribute. Setting the
<code>discard_private</code> argument to <code>TRUE</code> will keep in
the new CNAqc object only those mutations falling in the same position
across all samples.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_multisample</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/multisample_init.html">multisample_init</a></span><span class="op">(</span>cnaqc_objs <span class="op">=</span> <span class="va">CNAqc_samples</span>, </span>
<span>                                              QC_filter <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># use only segments passing QC</span></span>
<span>                                              keep_original <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># keep the original CNAqc objects too</span></span>
<span>                                              discard_private <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="co"># keep also not private mutations among the samples</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">mCNAqcqc - Defining common segments</span> <span style="color: #00BBBB;">─────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">3</span> CNAqc objects:</span></span>
<span><span class="co">#&gt; • Set6_42</span></span>
<span><span class="co">#&gt; • Set6_44</span></span>
<span><span class="co">#&gt; • Set6_45</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Building a </span><span style="color: #0000BB; font-weight: bold;">&lt;mCNAqcqc&gt;</span><span style="font-weight: bold;"> object</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Selecting new segments ──────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr1</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr1</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr1</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 4</span></span>
<span><span class="co">#&gt; • Set6_44 = 5</span></span>
<span><span class="co">#&gt; • Set6_45 = 4</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">21</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr2</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr2</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr2</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 2</span></span>
<span><span class="co">#&gt; • Set6_44 = 2</span></span>
<span><span class="co">#&gt; • Set6_45 = 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">4</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr3</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr3</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr3</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 2</span></span>
<span><span class="co">#&gt; • Set6_44 = 2</span></span>
<span><span class="co">#&gt; • Set6_45 = 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">4</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr4</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr4</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr4</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 2</span></span>
<span><span class="co">#&gt; • Set6_44 = 2</span></span>
<span><span class="co">#&gt; • Set6_45 = 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">4</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr5</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr5</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr5</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 3</span></span>
<span><span class="co">#&gt; • Set6_44 = 3</span></span>
<span><span class="co">#&gt; • Set6_45 = 3</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">8</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr6</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr6</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr6</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 2</span></span>
<span><span class="co">#&gt; • Set6_44 = 3</span></span>
<span><span class="co">#&gt; • Set6_45 = 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">6</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr7</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr7</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr7</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 2</span></span>
<span><span class="co">#&gt; • Set6_44 = 2</span></span>
<span><span class="co">#&gt; • Set6_45 = 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">4</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr8</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr8</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr8</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 3</span></span>
<span><span class="co">#&gt; • Set6_44 = 3</span></span>
<span><span class="co">#&gt; • Set6_45 = 3</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">8</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr9</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr9</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr9</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 3</span></span>
<span><span class="co">#&gt; • Set6_44 = 3</span></span>
<span><span class="co">#&gt; • Set6_45 = 3</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">11</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr10</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr10</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr10</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 4</span></span>
<span><span class="co">#&gt; • Set6_44 = 4</span></span>
<span><span class="co">#&gt; • Set6_45 = 4</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">13</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr11</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr11</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr11</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 2</span></span>
<span><span class="co">#&gt; • Set6_44 = 2</span></span>
<span><span class="co">#&gt; • Set6_45 = 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">4</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr12</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr12</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr12</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 2</span></span>
<span><span class="co">#&gt; • Set6_44 = 2</span></span>
<span><span class="co">#&gt; • Set6_45 = 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">4</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr13</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr13</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr13</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 1</span></span>
<span><span class="co">#&gt; • Set6_44 = 1</span></span>
<span><span class="co">#&gt; • Set6_45 = 1</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">2</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr14</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr14</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr14</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 1</span></span>
<span><span class="co">#&gt; • Set6_44 = 1</span></span>
<span><span class="co">#&gt; • Set6_45 = 1</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">2</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr15</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr15</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr15</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 1</span></span>
<span><span class="co">#&gt; • Set6_44 = 1</span></span>
<span><span class="co">#&gt; • Set6_45 = 1</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">3</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr16</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr16</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr16</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 2</span></span>
<span><span class="co">#&gt; • Set6_44 = 2</span></span>
<span><span class="co">#&gt; • Set6_45 = 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">7</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr17</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr17</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr17</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 2</span></span>
<span><span class="co">#&gt; • Set6_44 = 2</span></span>
<span><span class="co">#&gt; • Set6_45 = 3</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">7</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr18</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr18</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr18</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 2</span></span>
<span><span class="co">#&gt; • Set6_44 = 2</span></span>
<span><span class="co">#&gt; • Set6_45 = 3</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">8</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr19</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr19</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr19</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 2</span></span>
<span><span class="co">#&gt; • Set6_44 = 2</span></span>
<span><span class="co">#&gt; • Set6_45 = 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">4</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr20</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr20</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr20</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 1</span></span>
<span><span class="co">#&gt; • Set6_44 = 1</span></span>
<span><span class="co">#&gt; • Set6_45 = 1</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">4</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr21</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr21</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr21</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 2</span></span>
<span><span class="co">#&gt; • Set6_44 = 2</span></span>
<span><span class="co">#&gt; • Set6_45 = 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">8</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chr22</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chr22</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chr22</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 1</span></span>
<span><span class="co">#&gt; • Set6_44 = 1</span></span>
<span><span class="co">#&gt; • Set6_45 = 1</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">2</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chrX</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chrX</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chrX</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 2</span></span>
<span><span class="co">#&gt; • Set6_44 = 0</span></span>
<span><span class="co">#&gt; • Set6_45 = 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">4</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Iterating on <span style="color: #00BB00;">chrY</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Iterating on </span><span style="color: #00BB00; font-weight: bold;">chrY</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of original segments in individual <span style="color: #0000BB;">&lt;CNAqc&gt;</span> objects in <span style="color: #00BB00;">chrY</span>:</span></span>
<span><span class="co">#&gt; • Set6_42 = 1</span></span>
<span><span class="co">#&gt; • Set6_44 = 1</span></span>
<span><span class="co">#&gt; • Set6_45 = 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #00BB00;">5</span> breakpoints</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Shared segments across samples: <span style="color: #0000BB;">58</span></span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> param `keep_original` is set to `TRUE`. Original CNAqc object will be kept.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Obtained updated CNA table with shared segments across samples</span></span>
<span><span class="co">#&gt; ── Mapping mutations on new segments ───────────────────────────────────────────</span></span>
<span><span class="co">#&gt; ── <span style="color: #BBBBBB; background-color: #00BBBB;">Set6_42</span> ─────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #0000BB;">8700</span> mutations in the original <span style="color: #0000BB;">&lt;CNAqc&gt;</span> object</span></span>
<span><span class="co">#&gt; ── Mapping mutations on shared segments ────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Detected indels mutation (substitutions with &gt;1 reference/alternative nucleotides).</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">8700</span> somatic mutations: <span style="color: #00BB00;">8684</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">16</span> indels.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Added segments length (in basepairs) to CNA segments.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">58</span> segments: <span style="color: #00BB00;">58</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span></span>
<span><span class="co">#&gt; Warning in map_mutations_to_clonal_segments(mutations, cna_clonal): [CNAqc] a</span></span>
<span><span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">8407</span> mutations mapped to clonal CNAs.</span></span>
<span><span class="co">#&gt; ── <span style="color: #BBBBBB; background-color: #00BBBB;">Set6_44</span> ─────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #0000BB;">8734</span> mutations in the original <span style="color: #0000BB;">&lt;CNAqc&gt;</span> object</span></span>
<span><span class="co">#&gt; ── Mapping mutations on shared segments ────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Detected indels mutation (substitutions with &gt;1 reference/alternative nucleotides).</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">8734</span> somatic mutations: <span style="color: #00BB00;">8719</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">15</span> indels.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Added segments length (in basepairs) to CNA segments.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">58</span> segments: <span style="color: #00BB00;">58</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span></span>
<span><span class="co">#&gt; Warning in map_mutations_to_clonal_segments(mutations, cna_clonal): [CNAqc] a</span></span>
<span><span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">8718</span> mutations mapped to clonal CNAs.</span></span>
<span><span class="co">#&gt; ── <span style="color: #BBBBBB; background-color: #00BBBB;">Set6_45</span> ─────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Found <span style="color: #0000BB;">9143</span> mutations in the original <span style="color: #0000BB;">&lt;CNAqc&gt;</span> object</span></span>
<span><span class="co">#&gt; ── Mapping mutations on shared segments ────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Detected indels mutation (substitutions with &gt;1 reference/alternative nucleotides).</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">9143</span> somatic mutations: <span style="color: #00BB00;">9125</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">18</span> indels.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Added segments length (in basepairs) to CNA segments.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">58</span> segments: <span style="color: #00BB00;">58</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span></span>
<span><span class="co">#&gt; Warning in map_mutations_to_clonal_segments(mutations, cna_clonal): [CNAqc] a</span></span>
<span><span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">8851</span> mutations mapped to clonal CNAs.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Defining the </span><span style="color: #0000BB; font-weight: bold;">&lt;mCNAqcqc&gt;</span><span style="font-weight: bold;"> object</span> <span style="color: #00BBBB;">──────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">CNAqc - CNA Quality Check</span> <span style="color: #00BBBB;">───────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">GRCh38</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Detected indels mutation (substitutions with &gt;1 reference/alternative nucleotides).</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">8407</span> somatic mutations: <span style="color: #00BB00;">8392</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">15</span> indels.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Added segments length (in basepairs) to CNA segments.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">58</span> segments: <span style="color: #00BB00;">58</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span></span>
<span><span class="co">#&gt; Warning in map_mutations_to_clonal_segments(mutations, cna_clonal): [CNAqc] a</span></span>
<span><span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">8407</span> mutations mapped to clonal CNAs.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">CNAqc - CNA Quality Check</span> <span style="color: #00BBBB;">───────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">GRCh38</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Detected indels mutation (substitutions with &gt;1 reference/alternative nucleotides).</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">8718</span> somatic mutations: <span style="color: #00BB00;">8703</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">15</span> indels.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Added segments length (in basepairs) to CNA segments.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">58</span> segments: <span style="color: #00BB00;">58</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span></span>
<span><span class="co">#&gt; Warning in map_mutations_to_clonal_segments(mutations, cna_clonal): [CNAqc] a</span></span>
<span><span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">8718</span> mutations mapped to clonal CNAs.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">CNAqc - CNA Quality Check</span> <span style="color: #00BBBB;">───────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">GRCh38</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Detected indels mutation (substitutions with &gt;1 reference/alternative nucleotides).</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">8851</span> somatic mutations: <span style="color: #00BB00;">8836</span> SNVs (<span style="color: #00BB00;">100%</span>) and <span style="color: #00BB00;">15</span> indels.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Added segments length (in basepairs) to CNA segments.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">58</span> segments: <span style="color: #00BB00;">58</span> clonal and <span style="color: #00BB00;">0</span> subclonal.</span></span>
<span><span class="co">#&gt; Warning in map_mutations_to_clonal_segments(mutations, cna_clonal): [CNAqc] a</span></span>
<span><span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">8851</span> mutations mapped to clonal CNAs.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Creating mCNAqc stats</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Ended</span> <span style="color: #00BBBB;">───────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">&lt;mCNAqc&gt;</span> object created including all samples</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">example_multisample</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="color: #BBBBBB; background-color: #00BBBB;">[ multi CNAqc ]</span> ─────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; • Rerefence genome: <span style="color: #00BB00;">GRCh38</span></span></span>
<span><span class="co">#&gt; • Samples used: <span style="color: #00BB00;">Set6_42, Set6_44, Set6_45</span></span></span>
<span><span class="co">#&gt; • <span style="color: #00BB00;">58</span> total shared segments across samples</span></span>
<span><span class="co">#&gt; • <span style="color: #00BB00;">8407</span>, <span style="color: #00BB00;">8718</span>, and <span style="color: #00BB00;">8851</span> mutations on shared segments across samples</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> All samples had good quality and were used to create the new segmentation</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Original CNAqc object avaiable for <span style="color: #00BB00;">Set6_42, Set6_44, Set6_45</span></span></span></code></pre></div>
<p>The function <code>plot_segments_multisample</code> allows
visualization of the effect of the new segmentation across all the
samples included in the mCNAqc object.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_segments_multisample.html">plot_segments_multisample</a></span><span class="op">(</span><span class="va">example_multisample</span>, which <span class="op">=</span> <span class="st">"shared"</span>, chromosomes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"New segmentation"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Retrieving </span><span style="color: #0000BB; font-weight: bold;">&lt;CNAqc&gt;</span><span style="font-weight: bold;"> objects with the new segmentation</span> <span style="color: #00BBBB;">────────────────────────</span></span></span></code></pre></div>
<p><img src="b10_MultiCNAqc_files/figure-html/unnamed-chunk-7-1.png" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="accessing-elements-inside-the-mcnaqc-object">Accessing elements inside the <code>mCNAqc</code> object<a class="anchor" aria-label="anchor" href="#accessing-elements-inside-the-mcnaqc-object"></a>
</h3>
<p>Sample names can be retrieved using the <code>get_sample_name</code>
function, which can also be applied on classical <code>CNAqc</code>
objects.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/get_sample_name.html">get_sample_name</a></span><span class="op">(</span><span class="va">example_multisample</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Set6_42" "Set6_44" "Set6_45"</span></span></code></pre></div>
<p>Elements inside the <code>mCNAqc</code> object (for desired samples)
can be accessed through the <code>get_sample</code> function. It returns
a list of CNAqc objects. In this way, it is possible to obtain a joint
table with all the mutations across samples:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">=</span> <span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/get_sample.html">get_sample</a></span><span class="op">(</span><span class="va">example_multisample</span>, </span>
<span>           sample <span class="op">=</span> <span class="fu"><a href="../reference/get_sample_name.html">get_sample_name</a></span><span class="op">(</span><span class="va">example_multisample</span><span class="op">)</span>, </span>
<span>           which_obj <span class="op">=</span> <span class="st">"shared"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Retrieving </span><span style="color: #0000BB; font-weight: bold;">&lt;CNAqc&gt;</span><span style="font-weight: bold;"> objects with the new segmentation</span> <span style="color: #00BBBB;">────────────────────────</span></span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">res</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span><span class="fu">CNAqc</span><span class="fu">::</span><span class="fu"><a href="../reference/Mutations.html">Mutations</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 25,976 × 51</span></span></span>
<span><span class="co">#&gt;    chr       from       to ref   alt      NV    DP   VAF Indiv   Key FR     MMLQ</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1   3<span style="text-decoration: underline;">634</span>121  3<span style="text-decoration: underline;">634</span>122 C     T        42    63 0.667 Set6…    72 0.42…    37</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1   3<span style="text-decoration: underline;">643</span>785  3<span style="text-decoration: underline;">643</span>786 G     A        34    59 0.576 Set6…    73 0.42…    37</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1   4<span style="text-decoration: underline;">161</span>648  4<span style="text-decoration: underline;">161</span>649 C     A        18    37 0.486 Set6…    81 0.42…    37</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1   5<span style="text-decoration: underline;">513</span>535  5<span style="text-decoration: underline;">513</span>536 A     T        10    44 0.227 Set6…    95 0.07…    37</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1   5<span style="text-decoration: underline;">882</span>139  5<span style="text-decoration: underline;">882</span>140 C     T        35    55 0.636 Set6…   110 0.42…    37</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1   6<span style="text-decoration: underline;">595</span>746  6<span style="text-decoration: underline;">595</span>747 G     A        23    53 0.434 Set6…   122 0.42…    37</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr1   7<span style="text-decoration: underline;">836</span>836  7<span style="text-decoration: underline;">836</span>837 C     T        18    35 0.514 Set6…   140 0.42…    37</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr1   8<span style="text-decoration: underline;">570</span>678  8<span style="text-decoration: underline;">570</span>679 C     T        32    55 0.582 Set6…   149 0.42…    37</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr1   9<span style="text-decoration: underline;">149</span>650  9<span style="text-decoration: underline;">149</span>651 T     G        45    59 0.763 Set6…   154 0.42…    37</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr1  10<span style="text-decoration: underline;">218</span>495 10<span style="text-decoration: underline;">218</span>496 T     G        20    41 0.488 Set6…   167 0.42…    37</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 25,966 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 39 more variables: TCR &lt;int&gt;, HP &lt;int&gt;, WE &lt;int&gt;, Source &lt;chr&gt;, FS &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   WS &lt;int&gt;, PP &lt;chr&gt;, TR &lt;chr&gt;, NF &lt;chr&gt;, TCF &lt;int&gt;, NR &lt;chr&gt;, TC &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   END &lt;chr&gt;, MGOF &lt;chr&gt;, SbPval &lt;chr&gt;, START &lt;chr&gt;, ReadPosRankSum &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   MQ &lt;chr&gt;, QD &lt;dbl&gt;, SC &lt;chr&gt;, BRF &lt;dbl&gt;, HapScore &lt;chr&gt;, Size &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ID &lt;chr&gt;, QUAL &lt;chr&gt;, FILTER &lt;chr&gt;, INFO &lt;chr&gt;, gt_GT &lt;chr&gt;, gt_GQ &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   gt_GOF &lt;chr&gt;, gt_NR &lt;chr&gt;, gt_GL &lt;chr&gt;, gt_NV &lt;chr&gt;, gt_GT_alleles &lt;chr&gt;, …</span></span></span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.caravagnalab.org" class="external-link">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer>
</div>






  </body>
</html>
